// inter/build/ar_km.js
function defaults(targetObj) {
    for (var i = 1; i < arguments.length; i += 1) {
        if (arguments[i] && typeof arguments[i] === "object") {
            for (var propertyName in arguments[i]) {
                if (typeof targetObj[propertyName] === "undefined") {
                    targetObj[propertyName] = arguments[i][propertyName];
                }
            }
        }
    }
    return targetObj;
}

var P = require("./ar.js"), inter = defaults({
    renderers: {}
}, P);

var _69dx1 = {
    displayName: "العصر"
}, _69dx2 = {
    displayName: "اليوم"
}, _69dx3 = {
    all: {
        leap: "{0}bis"
    }
}, _69dx4 = {
    displayName: "التوقيت"
}, _69dx5 = {
    a: "h a – h a",
    h: "h–h a"
}, _69dx6 = {
    a: "h a – h a v",
    h: "h–h a v"
}, _69dx7 = {
    d: "M/d – M/d",
    M: "M/d – M/d"
}, _69dx8 = [ "ح", "ن", "ث", "ر", "خ", "ج", "س" ], _69dx9 = {
    d: "MMM d–d",
    M: "d MMM – d MMM"
}, _69dx10 = {
    d: "MMM d–d",
    M: "MMM d – MMM d"
}, _69dx11 = {
    wide: [ "هـ" ],
    abbreviated: [ "هـ" ]
}, _69dx12 = [ "Before R.O.C.", "جمهورية الصي" ], _69dx13 = {
    H: "HH:mm–HH:mm",
    m: "HH:mm–HH:mm"
}, _69dx14 = {
    M: "y-MM – y-MM",
    y: "y-MM – y-MM"
}, _69dx15 = {
    M: "U MMM–MMM",
    y: "U MMM – U MMM"
}, _69dx16 = {
    H: "HH:mm–HH:mm v",
    m: "HH:mm–HH:mm v"
}, _69dx17 = {
    M: "M‏/y – M‏/y G",
    y: "M‏/y – M‏/y G"
}, _69dx18 = {
    d: "MM-dd – MM-dd",
    M: "MM-dd – MM-dd"
}, _69dx19 = {
    M: "U MMMM–MMMM",
    y: "U MMMM – U MMMM"
}, _69dx20 = {
    M: "MMM – MMM، y G",
    y: "MMM، y – MMM، y G"
}, _69dx21 = {
    d: "E، d – E، d MMM",
    M: "E، d MMM – E، d MMM"
}, _69dx22 = {
    d: "E، d/‏M –‏ E، d/‏M",
    M: "E، d/‏M – E، d/‏M"
}, _69dx23 = {
    M: "MMMM – MMMM، y G",
    y: "MMMM، y – MMMM، y G"
}, _69dx24 = [ "ي", "ف", "م", "أ", "و", "ن", "ل", "غ", "س", "ك", "ب", "د" ], _69dx25 = {
    d: "MM-dd, E – MM-dd, E",
    M: "MM-dd, E – MM-dd, E"
}, _69dx26 = {
    d: "MMM d, E – MMM d, E",
    M: "MMM d, E – MMM d, E"
}, _69dx27 = {
    wide: [ "ERA0", "ERA1" ],
    abbreviated: [ "ERA0", "ERA1" ]
}, _69dx28 = {
    "0": "هذا الربع",
    "1": "الربع القادم",
    "-1": "الربع الأخير"
}, _69dx29 = {
    "0": "هذا الشهر",
    "1": "الشهر التالي",
    "-1": "الشهر الماضي"
}, _69dx30 = [ "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12" ], _69dx31 = [ "١", "٢", "٣", "٤", "٥", "٦", "٧", "٨", "٩", "١٠", "١١", "١٢" ], _69dx32 = {
    a: "h:mm a – h:mm a",
    h: "h:mm–h:mm a",
    m: "h:mm–h:mm a"
}, _69dx33 = {
    "0": "الأحد الحالي",
    "1": "الأحد القادم",
    "-1": "الأحد الماضي"
}, _69dx34 = {
    "0": "السبت الحالي",
    "1": "السبت القادم",
    "-1": "السبت الماضي"
}, _69dx35 = {
    d: "U MMM d–d",
    M: "U MMM d – MMM d",
    y: "U MMM d – U MMM d"
}, _69dx36 = [ "١", "٢", "٣", "٤", "٥", "٦", "٧", "٨", "٩", "١٠", "١١", "١٢", "١٣" ], _69dx37 = [ "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13" ], _69dx38 = {
    "0": "السنة الحالية",
    "1": "السنة التالية",
    "-1": "السنة الماضية"
}, _69dx39 = {
    "0": "هذا الأسبوع",
    "1": "الأسبوع التالي",
    "-1": "الأسبوع الماضي"
}, _69dx40 = {
    "0": "الخميس الحالي",
    "1": "الخميس القادم",
    "-1": "الخميس الماضي"
}, _69dx41 = {
    "0": "الجمعة الحالي",
    "1": "الجمعة القادم",
    "-1": "الجمعة الماضي"
}, _69dx42 = {
    a: "h:mm a – h:mm a v",
    h: "h:mm–h:mm a v",
    m: "h:mm–h:mm a v"
}, _69dx43 = {
    "0": "اليوم",
    "1": "غدًا",
    "2": "بعد الغد",
    "-2": "أول أمس",
    "-1": "أمس"
}, _69dx44 = [ "الربع الأول", "الربع الثاني", "الربع الثالث", "الربع الرابع" ], _69dx45 = {
    "0": "الاثنين الحالي",
    "1": "الاثنين القادم",
    "-1": "الاثنين الماضي"
}, _69dx46 = {
    "0": "الثلاثاء الحالي",
    "1": "الثلاثاء القادم",
    "-1": "الثلاثاء الماضي"
}, _69dx47 = {
    "0": "الأربعاء الحالي",
    "1": "الأربعاء القادم",
    "-1": "الأربعاء الماضي"
}, _69dx48 = {
    relative: _69dx33
}, _69dx49 = {
    relative: _69dx34
}, _69dx50 = {
    d: "d–d MMM، y G",
    M: "d MMM – d MMM، y G",
    y: "d MMM، y – d MMM، y G"
}, _69dx51 = [ "الأحد", "الاثنين", "الثلاثاء", "الأربعاء", "الخميس", "الجمعة", "السبت" ], _69dx52 = {
    d: "y-MM-dd – y-MM-dd",
    M: "y-MM-dd – y-MM-dd",
    y: "y-MM-dd – y-MM-dd"
}, _69dx53 = {
    relative: _69dx40
}, _69dx54 = {
    relative: _69dx41
}, _69dx55 = {
    abbreviated: {
        leap: "{0}bis"
    },
    narrow: {
        leap: "{0}b"
    },
    wide: {
        leap: "{0}bis"
    }
}, _69dx56 = {
    relative: _69dx45
}, _69dx57 = {
    d: "d‏/M‏/y – d‏/M‏/y G",
    M: "d‏/M‏/y – d‏/M‏/y G",
    y: "d‏/M‏/y – d‏/M‏/y G"
}, _69dx58 = [ "M01", "M02", "M03", "M04", "M05", "M06", "M07", "M08", "M09", "M10", "M11", "M12" ], _69dx59 = {
    relative: _69dx46
}, _69dx60 = {
    relative: _69dx47
}, _69dx61 = [ undefined, "zi", "chou", "yin", "mao", "chen", "si", "wu", "wei", "shen", "you", "xu" ], _69dx62 = {
    d: "U MMM d, E – MMM d, E",
    M: "U MMM d, E – MMM d, E",
    y: "U MMM d, E – U MMM d, E"
}, _69dx63 = {
    d: "y-MM-dd, E – y-MM-dd, E",
    M: "y-MM-dd, E – y-MM-dd, E",
    y: "y-MM-dd, E – y-MM-dd, E"
}, _69dx64 = {
    d: "E، d – E، d MMM، y G",
    M: "E، d MMM – E، d MMM، y G",
    y: "E، d MMM، y – E، d MMM، y G"
}, _69dx65 = [ "فرفردن", "أذربيهشت", "خرداد", "تار", "مرداد", "شهرفار", "مهر", "آيان", "آذر", "دي", "بهمن", "اسفندار" ], _69dx66 = {
    d: "E، dd‏/MM‏/y – E، dd‏/MM‏/y G",
    M: "E، d‏/M‏/y – E، d‏/M‏/y G",
    y: "E، d‏/M‏/y – E، d‏/M‏/y G"
}, _69dx67 = [ "مسكريم", "تكمت", "هدار", "تهساس", "تر", "يكتت", "مجابيت", "ميازيا", "جنبت", "سين", "هامل", "نهاس", "باجمن" ], _69dx68 = [ "توت", "بابه", "هاتور", "كيهك", "طوبة", "أمشير", "برمهات", "برمودة", "بشنس", "بؤونة", "أبيب", "مسرى", "نسيئ" ], _69dx69 = [ "يناير", "فبراير", "مارس", "أبريل", "مايو", "يونيو", "يوليو", "أغسطس", "سبتمبر", "أكتوبر", "نوفمبر", "ديسمبر" ], _69dx70 = [ "تشري", "مرحشوان", "كيسلو", "طيفت", "شباط", "آذار الأول", "آذار", "نيسان", "أيار", "سيفان", "تموز", "آب", "أيلول" ], _69dx71 = {
    zero: "قبل {0} سنة",
    one: "قبل سنة واحدة",
    two: "قبل سنتين",
    few: "قبل {0} سنوات",
    many: "قبل {0} سنة",
    other: "قبل {0} سنة"
}, _69dx72 = {
    zero: "قبل {0} شهر",
    one: "قبل شهر واحد",
    two: "قبل شهرين",
    few: "قبل {0} أشهر",
    many: "قبل {0} شهرًا",
    other: "قبل {0} شهر"
}, _69dx73 = {
    zero: "قبل {0} يوم",
    one: "قبل يوم واحد",
    two: "قبل يومين",
    few: "قبل {0} أيام",
    many: "قبل {0} يومًا",
    other: "قبل {0} يوم"
}, _69dx74 = {
    zero: "قبل {0} شهر",
    one: "قبل شهر واحد",
    two: "قبل شهرين",
    few: "خلال {0} أشهر",
    many: "قبل {0} شهرًا",
    other: "قبل {0} شهر"
}, _69dx75 = {
    zero: "قبل {0} ساعة",
    one: "قبل ساعة واحدة",
    two: "قبل ساعتين",
    few: "قبل {0} ساعات",
    many: "قبل {0} ساعة",
    other: "قبل {0} ساعة"
}, _69dx76 = [ "Chaitra", "Vaisakha", "Jyaistha", "Asadha", "Sravana", "Bhadra", "Asvina", "Kartika", "Agrahayana", "Pausa", "Magha", "Phalguna" ], _69dx77 = {
    zero: "خلال {0} سنة",
    one: "خلال سنة واحدة",
    two: "خلال سنتين",
    few: "خلال {0} سنوات",
    many: "خلال {0} سنة",
    other: "خلال {0} سنة"
}, _69dx78 = {
    zero: "خلال {0} شهر",
    one: "خلال شهر واحد",
    two: "خلال شهرين",
    few: "خلال {0} أشهر",
    many: "خلال {0} شهرًا",
    other: "خلال {0} شهر"
}, _69dx79 = {
    zero: "خلال {0} يوم",
    one: "خلال يوم واحد",
    two: "خلال يومين",
    few: "خلال {0} أيام",
    many: "خلال {0} يومًا",
    other: "خلال {0} يوم"
}, _69dx80 = [ "محرم", "صفر", "ربيع الأول", "ربيع الآخر", "جمادى الأولى", "جمادى الآخرة", "رجب", "شعبان", "رمضان", "شوال", "ذو القعدة", "ذو الحجة" ], _69dx81 = {
    zero: "قبل {0} دقيقة",
    one: "قبل دقيقة واحدة",
    two: "قبل دقيقتين",
    few: "قبل {0} دقائق",
    many: "قبل {0} دقيقة",
    other: "قبل {0} دقيقة"
}, _69dx82 = {
    zero: "قبل {0} ثانية",
    one: "قبل ثانية واحدة",
    two: "قبل ثانيتين",
    few: "قبل {0} ثوانِ",
    many: "قبل {0} ثانية",
    other: "قبل {0} ثانية"
}, _69dx83 = {
    zero: "قبل {0} ثانية",
    one: "قبل ثانية واحدة",
    two: "قبل ثانيتين",
    few: "قبل {0} ثوانٍ",
    many: "قبل {0} ثانية",
    other: "قبل {0} ثانية"
}, _69dx84 = {
    zero: "خلال {0} ساعة",
    one: "خلال ساعة واحدة",
    two: "خلال ساعتين",
    few: "خلال {0} ساعات",
    many: "خلال {0} ساعة",
    other: "خلال {0} ساعة"
}, _69dx85 = {
    zero: "قبل {0} أسبوع",
    one: "قبل أسبوع واحد",
    two: "قبل أسبوعين",
    few: "قبل {0} أسابيع",
    many: "قبل {0} أسبوعًا",
    other: "قبل {0} أسبوع"
}, _69dx86 = {
    zero: "خلال {0} دقيقة",
    one: "خلال دقيقة واحدة",
    two: "خلال دقيقتين",
    few: "خلال {0} دقائق",
    many: "خلال {0} دقيقة",
    other: "خلال {0} دقيقة"
}, _69dx87 = {
    zero: "خلال {0} ثانية",
    one: "خلال ثانية واحدة",
    two: "خلال ثانيتين",
    few: "خلال {0} ثوانِ",
    many: "خلال {0} ثانية",
    other: "خلال {0} ثانية"
}, _69dx88 = {
    zero: "خلال {0} ثانية",
    one: "خلال ثانية واحدة",
    two: "خلال ثانيتين",
    few: "خلال {0} ثوانٍ",
    many: "خلال {0} ثانية",
    other: "خلال {0} ثانية"
}, _69dx89 = {
    zero: "خلال {0} أسبوع",
    one: "خلال أسبوع واحد",
    two: "خلال أسبوعين",
    few: "خلال {0} أسابيع",
    many: "خلال {0} أسبوعًا",
    other: "خلال {0} أسبوع"
}, _69dx90 = {
    zero: "قبل {0} ربع سنة",
    one: "قبل ربع سنة واحد",
    two: "قبل ربعي سنة",
    few: "قبل {0} أرباع سنة",
    many: "قبل {0} ربع سنة",
    other: "قبل {0} ربع سنة"
}, _69dx91 = {
    zero: "خلال {0} أسبوع",
    one: "خلال أسبوع واحد",
    two: "خلال {0} أسبوعين",
    few: "خلال {0} أسابيع",
    many: "خلال {0} أسبوعًا",
    other: "خلال {0} أسبوع"
}, _69dx92 = {
    zero: "خلال {0} ربع سنة",
    one: "خلال ربع سنة واحد",
    two: "خلال ربعي سنة",
    few: "خلال {0} أرباع سنة",
    many: "خلال {0} ربع سنة",
    other: "خلال {0} ربع سنة"
}, _69dx93 = {
    am: "ص",
    pm: "م",
    morning1: "فجرا",
    morning2: "صباحًا",
    afternoon1: "ظهرًا",
    afternoon2: "بعد الظهر",
    evening1: "مساءً",
    night1: "منتصف الليل",
    night2: "ليلاً"
}, _69dx94 = {
    am: "صباحًا",
    pm: "مساءً",
    morning1: "فجرا",
    morning2: "صباحًا",
    afternoon1: "ظهرًا",
    afternoon2: "بعد الظهر",
    evening1: "مساءً",
    night1: "منتصف الليل",
    night2: "ليلاً"
}, _69dx95 = {
    abbreviated: _69dx44,
    narrow: [ "١", "٢", "٣", "٤" ],
    wide: _69dx44
}, _69dx96 = {
    format: _69dx55,
    numeric: _69dx3,
    standAlone: _69dx55
}, _69dx97 = {
    abbreviated: _69dx58,
    narrow: _69dx30,
    wide: _69dx58
}, _69dx98 = {
    future: _69dx77,
    past: _69dx71
}, _69dx99 = {
    future: _69dx78,
    past: _69dx72
}, _69dx100 = {
    future: _69dx79,
    past: _69dx73
}, _69dx101 = {
    future: _69dx78,
    past: _69dx74
}, _69dx102 = {
    abbreviated: _69dx61,
    narrow: _69dx61,
    wide: _69dx61
}, _69dx103 = {
    future: _69dx84,
    past: _69dx75
}, _69dx104 = {
    format: _69dx102
}, _69dx105 = {
    abbreviated: _69dx65,
    narrow: _69dx30,
    wide: _69dx65
}, _69dx106 = {
    abbreviated: _69dx51,
    narrow: _69dx8,
    wide: _69dx51,
    "short": _69dx51
}, _69dx107 = {
    future: _69dx86,
    past: _69dx81
}, _69dx108 = {
    future: _69dx87,
    past: _69dx82
}, _69dx109 = {
    future: _69dx88,
    past: _69dx83
}, _69dx110 = {
    future: _69dx89,
    past: _69dx85
}, _69dx111 = {
    abbreviated: _69dx69,
    narrow: _69dx24,
    wide: _69dx69
}, _69dx112 = {
    future: _69dx91,
    past: _69dx85
}, _69dx113 = {
    abbreviated: _69dx67,
    narrow: _69dx37,
    wide: _69dx67
}, _69dx114 = {
    abbreviated: _69dx68,
    narrow: _69dx36,
    wide: _69dx68
}, _69dx115 = {
    displayName: "يوم",
    relativeTime: _69dx100
}, _69dx116 = {
    displayName: "السنة",
    relativeTime: _69dx98
}, _69dx117 = {
    displayName: "الشهر",
    relativeTime: _69dx99
}, _69dx118 = {
    displayName: "الشهر",
    relativeTime: _69dx101
}, _69dx119 = {
    abbreviated: _69dx70,
    narrow: _69dx37,
    wide: _69dx70
}, _69dx120 = {
    future: _69dx92,
    past: _69dx90
}, _69dx121 = {
    displayName: "الساعات",
    relativeTime: _69dx103
}, _69dx122 = {
    displayName: "الدقائق",
    relativeTime: _69dx107
}, _69dx123 = {
    displayName: "الثواني",
    relativeTime: _69dx109
}, _69dx124 = {
    displayName: "الأسبوع",
    relativeTime: _69dx110
}, _69dx125 = {
    displayName: "الأسبوع",
    relativeTime: _69dx112
}, _69dx126 = {
    abbreviated: _69dx76,
    narrow: _69dx30,
    wide: _69dx76
}, _69dx127 = {
    abbreviated: _69dx80,
    narrow: _69dx31,
    wide: _69dx80
}, _69dx128 = {
    displayName: "الثواني",
    relative: [ "الآن" ],
    relativeTime: _69dx108
}, _69dx129 = {
    fullDate: "r(U) MMMM d, EEEE",
    longDate: "r(U) MMMM d",
    mediumDate: "r MMM d",
    shortDate: "r-MM-dd",
    fullTime: "HH:mm:ss zzzz",
    longTime: "HH:mm:ss z",
    mediumTime: "HH:mm:ss",
    shortTime: "HH:mm",
    shortDateTime: "r-MM-dd HH:mm",
    mediumDateTime: "r MMM d HH:mm:ss",
    longDateTime: "r(U) MMMM d HH:mm:ss z",
    fullDateTime: "r(U) MMMM d, EEEE HH:mm:ss zzzz"
}, _69dx130 = {
    displayName: "الشهر",
    relative: _69dx29,
    relativeTime: _69dx99
}, _69dx131 = {
    fullDate: "EEEE، d MMMM، y G",
    longDate: "d MMMM، y G",
    mediumDate: "d MMM، y G",
    shortDate: "d‏/M‏/y GGGGG",
    fullTime: "HH:mm:ss zzzz",
    longTime: "HH:mm:ss z",
    mediumTime: "HH:mm:ss",
    shortTime: "HH:mm",
    shortDateTime: "d‏/M‏/y GGGGG HH:mm",
    mediumDateTime: "d MMM، y G HH:mm:ss",
    longDateTime: "d MMMM، y G HH:mm:ss z",
    fullDateTime: "EEEE، d MMMM، y G HH:mm:ss zzzz"
}, _69dx132 = {
    fullDate: "EEEE، d MMMM، y G",
    longDate: "d MMMM، y G",
    mediumDate: "dd‏/MM‏/y G",
    shortDate: "d‏/M‏/y GGGGG",
    fullTime: "HH:mm:ss zzzz",
    longTime: "HH:mm:ss z",
    mediumTime: "HH:mm:ss",
    shortTime: "HH:mm",
    shortDateTime: "d‏/M‏/y GGGGG HH:mm",
    mediumDateTime: "dd‏/MM‏/y G HH:mm:ss",
    longDateTime: "d MMMM، y G HH:mm:ss z",
    fullDateTime: "EEEE، d MMMM، y G HH:mm:ss zzzz"
}, _69dx133 = {
    format: _69dx95,
    standAlone: _69dx95
}, _69dx134 = {
    displayName: "يوم",
    relative: _69dx43,
    relativeTime: _69dx100
}, _69dx135 = {
    displayName: "السنة",
    relative: _69dx38,
    relativeTime: _69dx98
}, _69dx136 = {
    displayName: "الأسبوع",
    relative: _69dx39,
    relativeTime: _69dx110
}, _69dx137 = {
    basic: _69dx132,
    cldr: P.calendars.buddhist.dateFormats
}, _69dx138 = {
    displayName: "ربع السنة",
    relative: _69dx28,
    relativeTime: _69dx120
}, _69dx139 = {
    abbreviated: _69dx93,
    narrow: _69dx93,
    wide: _69dx93
}, _69dx140 = {
    abbreviated: _69dx93,
    narrow: _69dx93,
    wide: _69dx94
}, _69dx141 = {
    format: _69dx97,
    standAlone: _69dx97
}, _69dx142 = {
    d: "d",
    E: "ccc",
    Ed: "d, E",
    Gy: "r U",
    GyMMM: "r(U) MMM",
    GyMMMd: "r MMM d",
    GyMMMEd: "r(U) MMM d, E",
    h: "h a",
    H: "HH",
    hm: "h:mm a",
    Hm: "HH:mm",
    hms: "h:mm:ss a",
    Hms: "HH:mm:ss",
    M: "L",
    Md: "MM-dd",
    MEd: "MM-dd, E",
    MMM: "LLL",
    MMMd: "MMM d",
    MMMEd: "MMM d, E",
    MMMMd: "MMMM d",
    ms: "mm:ss",
    UM: "U MM",
    UMd: "U MM-d",
    UMMM: "U MMM",
    UMMMd: "U MMM d",
    y: "r(U)",
    yMd: "r-MM-dd",
    yyyy: "r(U)",
    yyyyM: "r-MM",
    yyyyMd: "r-MM-dd",
    yyyyMEd: "r-MM-dd, E",
    yyyyMMM: "r(U) MMM",
    yyyyMMMd: "r MMM d",
    yyyyMMMEd: "r(U) MMM d, E",
    yyyyMMMM: "r(U) MMMM",
    yyyyQQQ: "r(U) QQQ",
    yyyyQQQQ: "r(U) QQQQ"
}, _69dx143 = {
    d: "d",
    E: "ccc",
    Ed: "E، d",
    Gy: "y G",
    GyMMM: "MMM y G",
    GyMMMd: "d MMM، y G",
    GyMMMEd: "E، d MMM، y G",
    M: "L",
    Md: "d/‏M",
    MEd: "E، d/M",
    MMM: "LLL",
    MMMd: "d MMM",
    MMMEd: "E، d MMM",
    MMMMd: "d MMMM",
    y: "y G",
    yyyy: "y G",
    yyyyM: "M‏/y G",
    yyyyMd: "d‏/M‏/y G",
    yyyyMEd: "E، d/‏M/‏y G",
    yyyyMMM: "MMM، y G",
    yyyyMMMd: "d MMM، y G",
    yyyyMMMEd: "E، d MMM، y G",
    yyyyMMMM: "MMMM، y G",
    yyyyQQQ: "QQQ y G",
    yyyyQQQQ: "QQQQ y G",
    h: "h a",
    H: "HH",
    hm: "h:mm a",
    Hm: "HH:mm",
    hms: "h:mm:ss a",
    Hms: "HH:mm:ss",
    ms: "mm:ss",
    Ehm: "E h:mm a",
    EHm: "E HH:mm",
    Ehms: "E h:mm:ss a",
    EHms: "E HH:mm:ss"
}, _69dx144 = {
    format: _69dx106,
    standAlone: _69dx106
}, _69dx145 = {
    format: _69dx111,
    standAlone: _69dx111
}, _69dx146 = [ undefined, "jia-zi", "yi-chou", "bing-yin", "ding-mao", "wu-chen", "ji-si", "geng-wu", "xin-wei", "ren-shen", "gui-you", "jia-xu", "yi-hai", "bing-zi", "ding-chou", "wu-yin", "ji-mao", "geng-chen", "xin-si", "ren-wu", "gui-wei", "jia-shen", "yi-you", "bing-xu", "ding-hai", "wu-zi", "ji-chou", "geng-yin", "xin-mao", "ren-chen", "gui-si", "jia-wu", "yi-wei", "bing-shen", "ding-you", "wu-xu", "ji-hai", "geng-zi", "xin-chou", "ren-yin", "gui-mao", "jia-chen", "yi-si", "bing-wu", "ding-wei", "wu-shen", "ji-you", "geng-xu", "xin-hai", "ren-zi", "gui-chou", "jia-yin", "yi-mao", "bing-chen", "ding-si", "wu-wu", "ji-wei", "geng-shen", "xin-you", "ren-xu" ], _69dx147 = {
    format: _69dx127,
    standAlone: _69dx127
}, _69dx148 = {
    basic: _69dx129,
    cldr: _69dx142
}, _69dx149 = {
    basic: _69dx131,
    cldr: _69dx143
}, _69dx150 = {
    format: _69dx139,
    standAlone: _69dx140
}, _69dx151 = {
    d: {
        d: "d–d"
    },
    h: _69dx5,
    H: {
        H: "HH–HH"
    },
    hm: _69dx32,
    Hm: _69dx13,
    hmv: _69dx42,
    Hmv: _69dx16,
    hv: _69dx6,
    Hv: {
        H: "HH–HH v"
    },
    M: {
        M: "MM–MM"
    },
    Md: _69dx18,
    MEd: _69dx25,
    MMM: {
        M: "LLL–LLL"
    },
    MMMd: _69dx10,
    MMMEd: _69dx26,
    y: {
        y: "U–U"
    },
    yM: _69dx14,
    yMd: _69dx52,
    yMEd: _69dx63,
    yMMM: _69dx15,
    yMMMd: _69dx35,
    yMMMEd: _69dx62,
    yMMMM: _69dx19
}, _69dx152 = {
    d: {
        d: "d–d"
    },
    h: _69dx5,
    H: {
        H: "HH–HH"
    },
    hm: _69dx32,
    Hm: _69dx13,
    hmv: _69dx42,
    Hmv: _69dx16,
    hv: _69dx6,
    Hv: {
        H: "HH–HH v"
    },
    M: {
        M: "M–M"
    },
    Md: _69dx7,
    MEd: _69dx22,
    MMM: {
        M: "MMM–MMM"
    },
    MMMd: _69dx9,
    MMMEd: _69dx21,
    MMMM: {
        M: "LLLL–LLLL"
    },
    y: {
        y: "y–y G"
    },
    yM: _69dx17,
    yMd: _69dx57,
    yMEd: _69dx66,
    yMMM: _69dx20,
    yMMMd: _69dx50,
    yMMMEd: _69dx64,
    yMMMM: _69dx23
}, _69dx153 = {
    abbreviated: _69dx146,
    narrow: _69dx146,
    wide: _69dx146
}, _69dx154 = {
    format: _69dx153
}, _69dx155 = [ "تيكا", "هاكتشي", "هاكهو", "شتشو", "تيهو", "كيين", "وادو", "رييكي", "يورو", "جينكي", "تمبيو", "تمبيو-كامبو", "تمبيو-شوهو", "تمبيو-هوجي", "تمفو-جينجو", "جينجو-كيين", "هوكي", "تن-أو", "إنرياكو", "ديدو", "كونين", "تنتشو", "شووا (٨٣٤–٨٤٨)‏", "كاجو", "نينجو", "سيكو", "تنان", "جوجان", "جينكيي", "نينا", "كامبيو", "شوتاي", "انجي", "انتشو", "شوهيي", "تنجيو", "تنرياكو", "تنتوكو", "أووا", "كوهو", "آنا", "تينروكو", "تن-نن", "جوجن", "تنجن", "إيكان", "كانا", "اي-ان", "ايسو", "شورياكو (٩٩٠–٩٩٥)‏", "تشوتوكو", "تشوهو", "كانكو", "تشووا", "كانين", "جاين", "مانجو", "تشوجين", "تشورياكو", "تشوكيو (١٠٤٠–١٠٤٤)‏", "كانتوكو", "ايشو (١٠٤٦–١٠٥٣)‏", "تينجي", "كوهيي", "جيرياكو", "انكيو (١٠٦٩–١٠٧٤)‏", "شوهو (١٠٧٤–١٠٧٧)‏", "شورياكو (١٠٧٧–١٠٨١)‏", "ايهو", "أوتوكو", "كانجي", "كاهو", "ايتشو", "شوتوكو", "كووا (١٠٩٩–١١٠٤)‏", "تشوجي", "كاشو", "تنين", "تن-اي", "ايكيو (١١١٣–١١١٨)‏", "جن-اي", "هوان", "تنجي", "ديجي", "تنشو (١١٣١–١١٣٢)‏", "تشوشو", "هوين", "ايجي", "كوجي (١١٤٢–١١٤٤)‏", "تنيو", "كيوان", "نينبيي", "كيوجو", "هجين", "هيجي", "ايرياكو", "أوهو", "تشوكان", "ايمان", "نين-ان", "كاو", "شون", "أنجين", "جيشو", "يووا", "جيي", "جنريوكو", "بنجي", "كنكيو", "شوجي", "كنين", "جنكيو (١٢٠٤–١٢٠٦)‏", "كن-اي", "شوجن (١٢٠٧–١٢١١)‏", "كنرياكو", "كنبو (١٢١٣–١٢١٩)‏", "شوكيو", "جو", "جيننين", "كروكو", "أنتيي", "كنكي", "جويي", "تمبكو", "بنرياكو", "كاتيي", "رياكنين", "ان-أو", "نينجي", "كنجين", "هوجي", "كنتشو", "كوجن", "شوكا", "شوجن (١٢٥٩–١٢٦٠)‏", "بن-أو", "كوتشو", "بن-اي", "كنجي", "كوان", "شوو (١٢٨٨–١٢٩٣)‏", "اينين", "شوان", "كنجن", "كجن", "توكجي", "انكي", "أوتشو", "شووا (١٣١٢–١٣١٧)‏", "بنبو", "جنو", "جنكيو (١٣٢١–١٣٢٤)‏", "شوتشو (١٣٢٤–١٣٢٦)‏", "كريكي", "جنتكو", "جنكو", "كمو", "إنجن", "كوككو", "شوهي", "كنتكو", "بنتشو", "تنجو", "كورياكو", "كووا (١٣٨١–١٣٨٤)‏", "جنتشو", "مييتكو (١٣٨٤–١٣٨٧)‏", "كاكي", "كو", "مييتكو (١٣٩٠–١٣٩٤)‏", "أويي", "شوتشو (١٤٢٨–١٤٢٩)‏", "ايكيو (١٤٢٩–١٤٤١)‏", "ككيتسو", "بن-أن", "هوتكو", "كيوتكو", "كوشو", "تشوركو", "كنشو", "بنشو", "أونين", "بنمي", "تشوكيو (١٤٨٧–١٤٨٩)‏", "انتكو", "ميو", "بنكي", "ايشو (١٥٠٤–١٥٢١)‏", "تييي", "كيوركو", "تنمن", "كوجي (١٥٥٥–١٥٥٨)‏", "ايركو", "جنكي", "تنشو (١٥٧٣–١٥٩٢)‏", "بنركو", "كيتشو", "جنوا", "كان-اي", "شوهو (١٦٤٤–١٦٤٨)‏", "كيان", "شوو (١٦٥٢–١٦٥٥)‏", "ميرياكو", "منجي", "كنبن", "انبو", "تنوا", "جوكيو", "جنركو", "هويي", "شوتكو", "كيوهو", "جنبن", "كنبو (١٧٤١–١٧٤٤)‏", "انكيو (١٧٤٤–١٧٤٨)‏", "كان-ان", "هورياكو", "مييوا", "ان-اي", "تنمي", "كنسي", "كيووا", "بنكا", "بنسي", "تنبو", "كوكا", "كاي", "أنسي", "من-ان", "بنكيو", "جنجي", "كيو", "ميجي", "تيشو", "شووا", "هيسي" ], _69dx156 = {
    dayParts: _69dx104,
    days: _69dx154,
    months: _69dx154,
    years: _69dx154,
    zodiacs: _69dx104
}, _69dx157 = {
    era: _69dx1,
    year: _69dx135,
    "year-short": _69dx116,
    "year-narrow": _69dx116,
    quarter: _69dx138,
    "quarter-short": _69dx138,
    "quarter-narrow": _69dx138,
    month: _69dx130,
    "month-short": _69dx118,
    "month-narrow": _69dx117,
    week: _69dx136,
    "week-short": _69dx125,
    "week-narrow": _69dx124,
    day: _69dx134,
    "day-short": _69dx115,
    "day-narrow": _69dx115,
    weekday: _69dx2,
    dayperiod: {
        displayName: "ص/م"
    },
    hour: _69dx121,
    "hour-short": _69dx121,
    "hour-narrow": _69dx121,
    minute: _69dx122,
    "minute-short": _69dx122,
    "minute-narrow": _69dx122,
    second: _69dx128,
    "second-short": _69dx123,
    "second-narrow": _69dx123,
    zone: _69dx4,
    sun: _69dx48,
    "sun-short": _69dx48,
    "sun-narrow": _69dx48,
    mon: _69dx56,
    "mon-short": _69dx56,
    "mon-narrow": _69dx56,
    tue: _69dx59,
    "tue-short": _69dx59,
    "tue-narrow": _69dx59,
    wed: _69dx60,
    "wed-short": _69dx60,
    "wed-narrow": _69dx60,
    thu: _69dx53,
    "thu-short": _69dx53,
    "thu-narrow": _69dx53,
    fri: _69dx54,
    "fri-short": _69dx54,
    "fri-narrow": _69dx54,
    sat: _69dx49,
    "sat-short": _69dx49,
    "sat-narrow": _69dx49
}, _69dx158 = {
    dateFormats: {
        basic: {},
        cldr: {}
    },
    fields: _69dx157
}, _69dx159 = {
    dateFormats: _69dx149,
    datePartsOrder: "dmy",
    dateIntervalFormats: _69dx152,
    dateIntervalFallbackFormat: "{0} – {1}",
    dayNames: _69dx144,
    dayPeriods: _69dx150,
    monthNames: _69dx147,
    eraNames: _69dx11,
    quarterNames: _69dx133,
    fields: _69dx157
}, _69dx160 = {
    dateFormats: _69dx148,
    datePartsOrder: "md",
    dateIntervalFormats: _69dx151,
    dateIntervalFallbackFormat: "{0} – {1}",
    dayNames: _69dx144,
    dayPeriods: _69dx150,
    cyclicNames: _69dx156,
    monthNames: _69dx141,
    monthPatterns: _69dx96,
    quarterNames: _69dx133,
    fields: _69dx157
};

inter.id = "ar_km";

inter.calendars = {
    buddhist: {
        dateFormats: _69dx137,
        datePartsOrder: "dmy",
        dateIntervalFormats: _69dx152,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _69dx144,
        dayPeriods: _69dx150,
        monthNames: _69dx145,
        eraNames: {
            wide: [ "التقويم البوذي" ],
            abbreviated: [ "BE" ]
        },
        quarterNames: _69dx133,
        fields: _69dx157
    },
    chinese: _69dx160,
    coptic: {
        dateFormats: _69dx137,
        datePartsOrder: "dmy",
        dateIntervalFormats: _69dx152,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _69dx144,
        dayPeriods: _69dx150,
        monthNames: {
            format: _69dx114,
            standAlone: _69dx114
        },
        eraNames: _69dx27,
        quarterNames: _69dx133,
        fields: _69dx157
    },
    dangi: _69dx160,
    ethioaa: _69dx158,
    ethiopic: {
        dateFormats: _69dx137,
        datePartsOrder: "dmy",
        dateIntervalFormats: _69dx152,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _69dx144,
        dayPeriods: _69dx150,
        monthNames: {
            format: _69dx113,
            standAlone: _69dx113
        },
        eraNames: _69dx27,
        quarterNames: _69dx133,
        fields: _69dx157
    },
    gregorian: {
        dateFormats: {
            basic: {
                fullDate: "EEEE، d MMMM، y",
                longDate: "d MMMM، y",
                mediumDate: "dd‏/MM‏/y",
                shortDate: "d‏/M‏/y",
                fullTime: "HH:mm:ss zzzz",
                longTime: "HH:mm:ss z",
                mediumTime: "HH:mm:ss",
                shortTime: "HH:mm",
                shortDateTime: "d‏/M‏/y HH:mm",
                mediumDateTime: "dd‏/MM‏/y HH:mm:ss",
                longDateTime: "d MMMM، y HH:mm:ss z",
                fullDateTime: "EEEE، d MMMM، y HH:mm:ss zzzz"
            },
            cldr: {
                d: "d",
                E: "ccc",
                Ed: "E، d",
                Ehm: "E h:mm a",
                EHm: "E HH:mm",
                Ehms: "E h:mm:ss a",
                EHms: "E HH:mm:ss",
                Gy: "y G",
                GyMMM: "MMM y G",
                GyMMMd: "d MMM، y G",
                GyMMMEd: "E، d MMM، y G",
                h: "h a",
                H: "HH",
                hm: "h:mm a",
                Hm: "HH:mm",
                hms: "h:mm:ss a",
                Hms: "HH:mm:ss",
                hmsv: "h:mm:ss a v",
                Hmsv: "HH:mm:ss v",
                hmv: "h:mm a v",
                Hmv: "HH:mm v",
                M: "L",
                Md: "d/‏M",
                MEd: "E، d/M",
                MMdd: "dd‏/MM",
                MMM: "LLL",
                MMMd: "d MMM",
                MMMEd: "E، d MMM",
                MMMMd: "d MMMM",
                MMMMEd: "E، d MMMM",
                ms: "mm:ss",
                y: "y",
                yM: "M‏/y",
                yMd: "d‏/M‏/y",
                yMEd: "E، d/‏M/‏y",
                yMM: "MM‏/y",
                yMMM: "MMM y",
                yMMMd: "d MMM، y",
                yMMMEd: "E، d MMM، y",
                yMMMM: "MMMM y",
                yQQQ: "QQQ y",
                yQQQQ: "QQQQ y"
            }
        },
        datePartsOrder: "dmy",
        dateIntervalFormats: {
            d: {
                d: "d–d"
            },
            h: _69dx5,
            H: {
                H: "HH–HH"
            },
            hm: _69dx32,
            Hm: _69dx13,
            hmv: _69dx42,
            Hmv: _69dx16,
            hv: _69dx6,
            Hv: {
                H: "HH–HH v"
            },
            M: {
                M: "M–M"
            },
            Md: _69dx7,
            MEd: _69dx22,
            MMM: {
                M: "MMM–MMM"
            },
            MMMd: {
                d: "d–d MMM",
                M: "d MMM – d MMM"
            },
            MMMEd: _69dx21,
            MMMM: {
                M: "LLLL–LLLL"
            },
            y: {
                y: "y–y"
            },
            yM: {
                M: "M‏/y – M‏/y",
                y: "M‏/y – M‏/y"
            },
            yMd: {
                d: "d‏/M‏/y – d‏/M‏/y",
                M: "d‏/M‏/y – d‏/M‏/y",
                y: "d‏/M‏/y – d‏/M‏/y"
            },
            yMEd: {
                d: "E، dd‏/MM‏/y – E، dd‏/MM‏/y",
                M: "E، d‏/M‏/y – E، d‏/M‏/y",
                y: "E، d‏/M‏/y – E، d‏/M‏/y"
            },
            yMMM: {
                M: "MMM – MMM، y",
                y: "MMM، y – MMM، y"
            },
            yMMMd: {
                d: "d–d MMM، y",
                M: "d MMM – d MMM، y",
                y: "d MMM، y – d MMM، y"
            },
            yMMMEd: {
                d: "E، d – E، d MMM، y",
                M: "E، d MMM – E، d MMM، y",
                y: "E، d MMM، y – E، d MMM، y"
            },
            yMMMM: {
                M: "MMMM – MMMM، y",
                y: "MMMM، y – MMMM، y"
            }
        },
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _69dx144,
        dayPeriods: _69dx150,
        monthNames: _69dx145,
        eraNames: {
            wide: [ "قبل الميلاد", "ميلادي" ],
            abbreviated: [ "ق.م", "م" ]
        },
        quarterNames: _69dx133,
        fields: _69dx157
    },
    hebrew: {
        dateFormats: _69dx137,
        datePartsOrder: "dmy",
        dateIntervalFormats: _69dx152,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _69dx144,
        dayPeriods: _69dx150,
        monthNames: {
            format: _69dx119,
            standAlone: _69dx119
        },
        eraNames: {
            wide: [ "ص" ],
            abbreviated: [ "ص" ]
        },
        quarterNames: _69dx133,
        fields: _69dx157
    },
    indian: {
        dateFormats: _69dx137,
        datePartsOrder: "dmy",
        dateIntervalFormats: _69dx152,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _69dx144,
        dayPeriods: _69dx150,
        monthNames: {
            format: _69dx126,
            standAlone: _69dx126
        },
        eraNames: {
            wide: [ "Saka" ],
            abbreviated: [ "Saka" ]
        },
        quarterNames: _69dx133,
        fields: _69dx157
    },
    islamic: _69dx159,
    "islamic-umalqura": _69dx159,
    "islamic-tbla": _69dx159,
    "islamic-civil": _69dx159,
    "islamic-rgsa": _69dx159,
    iso8601: _69dx158,
    japanese: {
        dateFormats: _69dx137,
        datePartsOrder: "dmy",
        dateIntervalFormats: _69dx152,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _69dx144,
        dayPeriods: _69dx150,
        monthNames: _69dx145,
        eraNames: {
            wide: _69dx155,
            abbreviated: _69dx155
        },
        quarterNames: _69dx133,
        fields: _69dx157
    },
    persian: {
        dateFormats: _69dx137,
        datePartsOrder: "dmy",
        dateIntervalFormats: _69dx152,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _69dx144,
        dayPeriods: _69dx150,
        monthNames: {
            format: _69dx105,
            standAlone: _69dx105
        },
        eraNames: {
            wide: [ "ه‍.ش" ],
            abbreviated: [ "ه‍.ش" ]
        },
        quarterNames: _69dx133,
        fields: _69dx157
    },
    roc: {
        dateFormats: _69dx137,
        datePartsOrder: "dmy",
        dateIntervalFormats: _69dx152,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _69dx144,
        dayPeriods: _69dx150,
        monthNames: _69dx145,
        eraNames: {
            wide: _69dx12,
            abbreviated: _69dx12
        },
        quarterNames: _69dx133,
        fields: _69dx157
    },
    islamicc: _69dx158
};

inter.pluralRule = function(n) {
    if (typeof n === "string") n = parseInt(n, 10);
    if (n === 0) return "zero";
    if (n === 1) return "one";
    if (n === 2) return "two";
    if (n % 100 === Math.floor(n % 100) && n % 100 >= 3 && n % 100 <= 10) return "few";
    if (n % 100 === Math.floor(n % 100) && n % 100 >= 11 && n % 100 <= 99) return "many";
    return "other";
};

inter.renderDigitsOrdinal = function(n) {
    if (n < 0) return "−" + this.renderDigitsOrdinal(-n);
    if (n >= 0) return this.renderNumber(n, "#,##0") + ".";
};

inter.renderSpelloutNumbering = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "ناقص " + this.renderSpelloutNumbering(-n);
    if (isFractional && n > 1) return this.renderSpelloutNumbering(Math.floor(n)) + " فاصل " + this.renderSpelloutNumbering(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 2e15) return this.renderSpelloutNumberingM(Math.floor(n / 1e15)) + " بليار" + (n === 2e15 ? "" : " و " + this.renderSpelloutNumbering(n % 1e15));
    if (n >= 1e15) return "بليار" + (n === 1e15 ? "" : " و " + this.renderSpelloutNumbering(n % 1e15));
    if (n >= 2e12) return this.renderSpelloutNumberingM(Math.floor(n / 1e12)) + " بليون" + (n === 2e12 ? "" : " و " + this.renderSpelloutNumbering(n % 1e12));
    if (n >= 1e12) return "بليون" + (n === 1e12 ? "" : " و " + this.renderSpelloutNumbering(n % 1e12));
    if (n >= 2e9) return this.renderSpelloutNumberingM(Math.floor(n / 1e9)) + " مليار" + (n === 2e9 ? "" : " و " + this.renderSpelloutNumbering(n % 1e9));
    if (n >= 1e9) return "مليار" + (n === 1e9 ? "" : " و " + this.renderSpelloutNumbering(n % 1e9));
    if (n >= 2e6) return this.renderSpelloutNumberingM(Math.floor(n / 1e6)) + " مليون" + (n === 2e6 ? "" : " و " + this.renderSpelloutNumbering(n % 1e6));
    if (n >= 1e6) return "مليون" + (n === 1e6 ? "" : " و " + this.renderSpelloutNumbering(n % 1e6));
    if (n >= 11e3) return this.renderSpelloutNumberingM(Math.floor(n / 1e4)) + " ألف" + (n === 11e3 ? "" : " و " + this.renderSpelloutNumbering(n % 1e4));
    if (n >= 3e3) return this.renderSpelloutNumbering(Math.floor(n / 1e3)) + " آلاف" + (n === 3e3 ? "" : " و " + this.renderSpelloutNumbering(n % 1e3));
    if (n >= 2e3) return "ألفين" + (n === 2e3 ? "" : " و " + this.renderSpelloutNumbering(n % 1e3));
    if (n >= 1e3) return "ألف" + (n === 1e3 ? "" : " و " + this.renderSpelloutNumbering(n % 1e3));
    if (n >= 300) return this.renderSpelloutNumbering(Math.floor(n / 100)) + " مائة" + (n === 300 ? "" : " و " + this.renderSpelloutNumbering(n % 100));
    if (n >= 200) return "مائتان" + (n === 200 ? "" : " و " + this.renderSpelloutNumbering(n % 100));
    if (n >= 100) return "مائة" + (n === 100 ? "" : " و " + this.renderSpelloutNumbering(n % 100));
    if (n >= 90) return (n === 90 ? "" : this.renderSpelloutNumbering(n % 10) + " و ") + "تسعون";
    if (n >= 80) return (n === 80 ? "" : this.renderSpelloutNumbering(n % 10) + " و ") + "ثمانون";
    if (n >= 70) return (n === 70 ? "" : this.renderSpelloutNumbering(n % 10) + " و ") + "سبعون";
    if (n >= 60) return (n === 60 ? "" : this.renderSpelloutNumbering(n % 10) + " و ") + "ستون";
    if (n >= 50) return (n === 50 ? "" : this.renderSpelloutNumbering(n % 10) + " و ") + "خمسون";
    if (n >= 40) return (n === 40 ? "" : this.renderSpelloutNumbering(n % 10) + " و ") + "أربعون";
    if (n >= 30) return (n === 30 ? "" : this.renderSpelloutNumbering(n % 10) + " و ") + "ثلاثون";
    if (n >= 20) return (n === 20 ? "" : this.renderSpelloutNumbering(n % 10) + " و ") + "عشرون";
    if (n >= 13) return this.renderSpelloutNumbering(n % 10) + " عشر";
    if (n >= 12) return "إثنا عشر";
    if (n >= 11) return "إحدى عشر";
    if (n >= 10) return "عشرة";
    if (n >= 9) return "تسعة";
    if (n >= 8) return "ثمانية";
    if (n >= 7) return "سبعة";
    if (n >= 6) return "ستة";
    if (n >= 5) return "خمسة";
    if (n >= 4) return "أربعة";
    if (n >= 3) return "ثلاثة";
    if (n >= 2) return "إثنان";
    if (n >= 1) return "واحد";
    if (n >= 0) return "صفر";
};

inter.renderArmenianUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderArmenianUpper(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e4) return this.renderNumber(n, "#,##0");
    if (n >= 9e3) return "Ք" + (n === 9e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 8e3) return "Փ" + (n === 8e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 7e3) return "Ւ" + (n === 7e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 6e3) return "Ց" + (n === 6e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 5e3) return "Ր" + (n === 5e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 4e3) return "Տ" + (n === 4e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 3e3) return "Վ" + (n === 3e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 2e3) return "Ս" + (n === 2e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 1e3) return "Ռ" + (n === 1e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 900) return "Ջ" + (n === 900 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 800) return "Պ" + (n === 800 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 700) return "Չ" + (n === 700 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 600) return "Ո" + (n === 600 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 500) return "Շ" + (n === 500 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 400) return "Ն" + (n === 400 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 300) return "Յ" + (n === 300 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 200) return "Մ" + (n === 200 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 100) return "Ճ" + (n === 100 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 90) return "Ղ" + (n === 90 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 80) return "Ձ" + (n === 80 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 70) return "Հ" + (n === 70 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 60) return "Կ" + (n === 60 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 50) return "Ծ" + (n === 50 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 40) return "Խ" + (n === 40 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 30) return "Լ" + (n === 30 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 20) return "Ի" + (n === 20 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 10) return "Ժ" + (n === 10 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 9) return "Թ";
    if (n >= 8) return "Ը";
    if (n >= 7) return "Է";
    if (n >= 6) return "Զ";
    if (n >= 5) return "Ե";
    if (n >= 4) return "Դ";
    if (n >= 3) return "Գ";
    if (n >= 2) return "Բ";
    if (n >= 1) return "Ա";
    if (n >= 0) return "0";
};

inter.renderArmenianLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderArmenianLower(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e4) return this.renderNumber(n, "#,##0");
    if (n >= 9e3) return "ք" + (n === 9e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 8e3) return "փ" + (n === 8e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 7e3) return "ւ" + (n === 7e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 6e3) return "ց" + (n === 6e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 5e3) return "ր" + (n === 5e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 4e3) return "տ" + (n === 4e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 3e3) return "վ" + (n === 3e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 2e3) return "ս" + (n === 2e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 1e3) return "ռ" + (n === 1e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 900) return "ջ" + (n === 900 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 800) return "պ" + (n === 800 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 700) return "չ" + (n === 700 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 600) return "ո" + (n === 600 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 500) return "շ" + (n === 500 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 400) return "ն" + (n === 400 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 300) return "յ" + (n === 300 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 200) return "մ" + (n === 200 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 100) return "ճ" + (n === 100 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 90) return "ղ" + (n === 90 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 80) return "ձ" + (n === 80 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 70) return "հ" + (n === 70 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 60) return "կ" + (n === 60 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 50) return "ծ" + (n === 50 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 40) return "խ" + (n === 40 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 30) return "լ" + (n === 30 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 20) return "ի" + (n === 20 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 10) return "ժ" + (n === 10 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 9) return "թ";
    if (n >= 8) return "ը";
    if (n >= 7) return "է";
    if (n >= 6) return "զ";
    if (n >= 5) return "ե";
    if (n >= 4) return "դ";
    if (n >= 3) return "գ";
    if (n >= 2) return "բ";
    if (n >= 1) return "ա";
    if (n >= 0) return "0";
};

inter.renderCyrillicLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderCyrillicLower(-n);
    if (isFractional && n > 1) return this.renderCyrillicLower(Math.floor(n)) + "." + this.renderCyrillicLower(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e15) return "҂҂҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e15)) + (n === 1e15 ? "" : " " + this.renderCyrillicLower(n % 1e15));
    if (n >= 1e12) return "҂҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e12)) + (n === 1e12 ? "" : " " + this.renderCyrillicLower(n % 1e12));
    if (n >= 1e9) return "҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e9)) + (n === 1e9 ? "" : " " + this.renderCyrillicLower(n % 1e9));
    if (n >= 1e6) return "҂҂" + this.renderCyrillicLower(Math.floor(n / 1e6)) + (n === 1e6 ? "" : " " + this.renderCyrillicLower(n % 1e6));
    if (n >= 11e3) return this.renderCyrillicLowerThousands(Math.floor(n / 1e4)) + (n === 11e3 ? "" : " " + this.renderCyrillicLower(n % 1e4));
    if (n >= 1e4) return "҂" + this.renderCyrillicLower(Math.floor(n / 1e4)) + (n === 1e4 ? "" : " " + this.renderCyrillicLower(n % 1e4));
    if (n >= 1e3) return "҂" + this.renderCyrillicLower110(Math.floor(n / 1e3)) + this.renderCyrillicLowerPost(n % 1e3);
    if (n >= 900) return "ц" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 801) return "ѿ" + this.renderCyrillicLower(n % 100);
    if (n >= 800) return "ѿ҃";
    if (n >= 700) return "ѱ" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 600) return "х" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 500) return "ф" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 400) return "у" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 300) return "т" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 200) return "с" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 100) return "р" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 90) return "ч" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 80) return "п" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 70) return "ѻ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 60) return "ѯ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 50) return "н" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 40) return "м" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 30) return "л" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 20) return "к" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 19) return "ѳ҃і";
    if (n >= 18) return "и҃і";
    if (n >= 17) return "з҃і";
    if (n >= 16) return "ѕ҃і";
    if (n >= 15) return "є҃і";
    if (n >= 14) return "д҃і";
    if (n >= 13) return "г҃і";
    if (n >= 12) return "в҃і";
    if (n >= 11) return "а҃і";
    if (n >= 1) return this.renderCyrillicLower110(n) + "҃";
    if (n >= 0) return "0҃";
};

inter.renderEthiopic = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderEthiopic(-n);
    if (isFractional && n > 1) return this.renderEthiopic(Math.floor(n)) + "፡" + this.renderEthiopic(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 1e18) return this.renderNumber(n, "#,###0");
    if (n >= 2e16) return this.renderEthiopic(Math.floor(n / 1e16)) + "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 1e16) return "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 2e12) return this.renderEthiopic(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1e12) return "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 2e8) return this.renderEthiopic(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1e8) return "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 2e4) return this.renderEthiopic(Math.floor(n / 1e4)) + "፼" + (n === 2e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 1e4) return "፼" + (n === 1e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 200) return this.renderEthiopic(Math.floor(n / 100)) + "፻" + (n === 200 ? "" : this.renderEthiopic(n % 100));
    if (n >= 100) return "፻" + (n === 100 ? "" : this.renderEthiopic(n % 100));
    if (n >= 90) return "፺" + (n === 90 ? "" : this.renderEthiopic(n % 10));
    if (n >= 80) return "፹" + (n === 80 ? "" : this.renderEthiopic(n % 10));
    if (n >= 70) return "፸" + (n === 70 ? "" : this.renderEthiopic(n % 10));
    if (n >= 60) return "፷" + (n === 60 ? "" : this.renderEthiopic(n % 10));
    if (n >= 50) return "፶" + (n === 50 ? "" : this.renderEthiopic(n % 10));
    if (n >= 40) return "፵" + (n === 40 ? "" : this.renderEthiopic(n % 10));
    if (n >= 30) return "፴" + (n === 30 ? "" : this.renderEthiopic(n % 10));
    if (n >= 20) return "፳" + (n === 20 ? "" : this.renderEthiopic(n % 10));
    if (n >= 10) return "፲" + (n === 10 ? "" : this.renderEthiopic(n % 10));
    if (n >= 9) return "፱";
    if (n >= 8) return "፰";
    if (n >= 7) return "፯";
    if (n >= 6) return "፮";
    if (n >= 5) return "፭";
    if (n >= 4) return "፬";
    if (n >= 3) return "፫";
    if (n >= 2) return "፪";
    if (n >= 1) return "፩";
    if (n >= 0) return "ባዶ";
};

inter.renderGeorgian = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGeorgian(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 2e4) return this.renderNumber(n, "#,##0");
    if (n >= 1e4) return "ჯ" + (n === 1e4 ? "" : this.renderGeorgian(n % 1e4));
    if (n >= 9e3) return "ჵ" + (n === 9e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 8e3) return "ჴ" + (n === 8e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 7e3) return "ხ" + (n === 7e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 6e3) return "ჭ" + (n === 6e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 5e3) return "წ" + (n === 5e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 4e3) return "ძ" + (n === 4e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 3e3) return "ც" + (n === 3e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 2e3) return "ჩ" + (n === 2e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 1e3) return "შ" + (n === 1e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 900) return "ყ" + (n === 900 ? "" : this.renderGeorgian(n % 100));
    if (n >= 800) return "ღ" + (n === 800 ? "" : this.renderGeorgian(n % 100));
    if (n >= 700) return "ქ" + (n === 700 ? "" : this.renderGeorgian(n % 100));
    if (n >= 600) return "ფ" + (n === 600 ? "" : this.renderGeorgian(n % 100));
    if (n >= 500) return "ჳ" + (n === 500 ? "" : this.renderGeorgian(n % 100));
    if (n >= 400) return "უ" + (n === 400 ? "" : this.renderGeorgian(n % 100));
    if (n >= 300) return "ტ" + (n === 300 ? "" : this.renderGeorgian(n % 100));
    if (n >= 200) return "ს" + (n === 200 ? "" : this.renderGeorgian(n % 100));
    if (n >= 100) return "რ" + (n === 100 ? "" : this.renderGeorgian(n % 100));
    if (n >= 90) return "ჟ" + (n === 90 ? "" : this.renderGeorgian(n % 10));
    if (n >= 80) return "პ" + (n === 80 ? "" : this.renderGeorgian(n % 10));
    if (n >= 70) return "ო" + (n === 70 ? "" : this.renderGeorgian(n % 10));
    if (n >= 60) return "ჲ" + (n === 60 ? "" : this.renderGeorgian(n % 10));
    if (n >= 50) return "ნ" + (n === 50 ? "" : this.renderGeorgian(n % 10));
    if (n >= 40) return "მ" + (n === 40 ? "" : this.renderGeorgian(n % 10));
    if (n >= 30) return "ლ" + (n === 30 ? "" : this.renderGeorgian(n % 10));
    if (n >= 20) return "კ" + (n === 20 ? "" : this.renderGeorgian(n % 10));
    if (n >= 10) return "ი" + (n === 10 ? "" : this.renderGeorgian(n % 10));
    if (n >= 9) return "თ";
    if (n >= 8) return "ჱ";
    if (n >= 7) return "ზ";
    if (n >= 6) return "ვ";
    if (n >= 5) return "ე";
    if (n >= 4) return "დ";
    if (n >= 3) return "გ";
    if (n >= 2) return "ბ";
    if (n >= 1) return "ა";
    if (n >= 0) return this.renderNumber(n, "#,##0");
};

inter.renderGreekUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGreekUpper(-n);
    if (isFractional && n > 1) return this.renderGreekUpper(Math.floor(n)) + "." + this.renderGreekUpper(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 0) return this.renderGreekNumeralMajuscules(n) + "´";
};

inter.renderGreekLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGreekLower(-n);
    if (isFractional && n > 1) return this.renderGreekLower(Math.floor(n)) + "." + this.renderGreekLower(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 0) return this.renderGreekNumeralMinuscules(n) + "´";
};

inter.renderSpelloutCardinal = function(n) {
    if (n < 0) return "−" + this.renderSpelloutCardinal(-n);
    if (n >= 0) return this.renderNumber(n, "#,##0.#");
};

inter.renderHebrew = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderHebrew(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1000001) return this.renderNumber(n, "#,##0");
    if (n >= 1e6) return "אלף אלפים";
    if (n >= 3001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 3001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 3e3) return this.renderHebrew(Math.floor(n / 1e3)) + " אלפים";
    if (n >= 2001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 2001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 2e3) return "אלפיים";
    if (n >= 1001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 1001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 1e3) return "אלף";
    if (n >= 901) return "תתק" + this.renderHebrew099(n % 100);
    if (n >= 900) return "תת״ק";
    if (n >= 801) return "תת" + this.renderHebrew099(n % 100);
    if (n >= 800) return "ת״ת";
    if (n >= 745) return "תש" + this.renderHebrew099(n % 100);
    if (n >= 744) return "תשד״מ";
    if (n >= 701) return "תש" + this.renderHebrew099(n % 100);
    if (n >= 700) return "ת״ש";
    if (n >= 699) return "תר" + this.renderHebrew099(n % 100);
    if (n >= 698) return "תרח״צ";
    if (n >= 601) return "תר" + this.renderHebrew099(n % 100);
    if (n >= 600) return "ת״ר";
    if (n >= 501) return "תק" + this.renderHebrew099(n % 100);
    if (n >= 500) return "ת״ק";
    if (n >= 400) return "ת" + this.renderHebrew099(n % 100);
    if (n >= 345) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 344) return "שד״מ";
    if (n >= 305) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 304) return "ד״ש";
    if (n >= 300) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 299) return "ר" + this.renderHebrew099(n % 100);
    if (n >= 298) return "רח״צ";
    if (n >= 200) return "ר" + this.renderHebrew099(n % 100);
    if (n >= 100) return "ק" + this.renderHebrew099(n % 100);
    if (n >= 91) return "צ״" + this.renderHebrewItem(n % 10);
    if (n >= 90) return "צ׳";
    if (n >= 81) return "פ״" + this.renderHebrewItem(n % 10);
    if (n >= 80) return "פ׳";
    if (n >= 71) return "ע״" + this.renderHebrewItem(n % 10);
    if (n >= 70) return "ע׳";
    if (n >= 61) return "ס״" + this.renderHebrewItem(n % 10);
    if (n >= 60) return "ס׳";
    if (n >= 51) return "נ״" + this.renderHebrewItem(n % 10);
    if (n >= 50) return "נ׳";
    if (n >= 41) return "מ״" + this.renderHebrewItem(n % 10);
    if (n >= 40) return "מ׳";
    if (n >= 31) return "ל״" + this.renderHebrewItem(n % 10);
    if (n >= 30) return "ל׳";
    if (n >= 21) return "כ״" + this.renderHebrewItem(n % 10);
    if (n >= 20) return "כ׳";
    if (n >= 17) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "ט״ז";
    if (n >= 15) return "ט״ו";
    if (n >= 11) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 0) return this.renderHebrewItem(n) + "׳";
};

inter.renderRomanUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderRomanUpper(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 4e5) return this.renderNumber(n, "#,##0");
    if (n >= 3e5) return "ↈↈↈ" + (n === 3e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 2e5) return "ↈↈ" + (n === 2e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 1e5) return "ↈ" + (n === 1e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 9e4) return "ↂↈ" + (n === 9e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 8e4) return "ↇↂↂↂ" + (n === 8e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 7e4) return "ↇↂↂ" + (n === 7e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 6e4) return "ↇↂ" + (n === 6e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 5e4) return "ↇ" + (n === 5e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 4e4) return "ↂↇ" + (n === 4e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 3e4) return "ↂↂↂ" + (n === 3e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 2e4) return "ↂↂ" + (n === 2e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 1e4) return "ↂ" + (n === 1e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 9e3) return "Mↂ" + (n === 9e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 8e3) return "ↁMMM" + (n === 8e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 7e3) return "ↁMM" + (n === 7e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 6e3) return "ↁM" + (n === 6e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 5e3) return "ↁ" + (n === 5e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 4e3) return "Mↁ" + (n === 4e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 3e3) return "MMM" + (n === 3e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 2e3) return "MM" + (n === 2e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 1e3) return "M" + (n === 1e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 900) return "CM" + (n === 900 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 800) return "DCCC" + (n === 800 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 700) return "DCC" + (n === 700 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 600) return "DC" + (n === 600 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 500) return "D" + (n === 500 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 400) return "CD" + (n === 400 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 300) return "CCC" + (n === 300 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 200) return "CC" + (n === 200 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 100) return "C" + (n === 100 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 90) return "XC" + (n === 90 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 80) return "LXXX" + (n === 80 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 70) return "LXX" + (n === 70 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 60) return "LX" + (n === 60 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 50) return "L" + (n === 50 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 40) return "XL" + (n === 40 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 30) return "XXX" + (n === 30 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 20) return "XX" + (n === 20 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 10) return "X" + (n === 10 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 9) return "IX";
    if (n >= 8) return "VIII";
    if (n >= 7) return "VII";
    if (n >= 6) return "VI";
    if (n >= 5) return "V";
    if (n >= 4) return "IV";
    if (n >= 3) return "III";
    if (n >= 2) return "II";
    if (n >= 1) return "I";
    if (n >= 0) return "N";
};

inter.renderRomanLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderRomanLower(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 5e3) return this.renderNumber(n, "#,##0");
    if (n >= 4e3) return "mmmm" + (n === 4e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 3e3) return "mmm" + (n === 3e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 2e3) return "mm" + (n === 2e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 1e3) return "m" + (n === 1e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 900) return "cm" + (n === 900 ? "" : this.renderRomanLower(n % 100));
    if (n >= 800) return "dccc" + (n === 800 ? "" : this.renderRomanLower(n % 100));
    if (n >= 700) return "dcc" + (n === 700 ? "" : this.renderRomanLower(n % 100));
    if (n >= 600) return "dc" + (n === 600 ? "" : this.renderRomanLower(n % 100));
    if (n >= 500) return "d" + (n === 500 ? "" : this.renderRomanLower(n % 100));
    if (n >= 400) return "cd" + (n === 400 ? "" : this.renderRomanLower(n % 100));
    if (n >= 300) return "ccc" + (n === 300 ? "" : this.renderRomanLower(n % 100));
    if (n >= 200) return "cc" + (n === 200 ? "" : this.renderRomanLower(n % 100));
    if (n >= 100) return "c" + (n === 100 ? "" : this.renderRomanLower(n % 100));
    if (n >= 90) return "xc" + (n === 90 ? "" : this.renderRomanLower(n % 10));
    if (n >= 80) return "lxxx" + (n === 80 ? "" : this.renderRomanLower(n % 10));
    if (n >= 70) return "lxx" + (n === 70 ? "" : this.renderRomanLower(n % 10));
    if (n >= 60) return "lx" + (n === 60 ? "" : this.renderRomanLower(n % 10));
    if (n >= 50) return "l" + (n === 50 ? "" : this.renderRomanLower(n % 10));
    if (n >= 40) return "xl" + (n === 40 ? "" : this.renderRomanLower(n % 10));
    if (n >= 30) return "xxx" + (n === 30 ? "" : this.renderRomanLower(n % 10));
    if (n >= 20) return "xx" + (n === 20 ? "" : this.renderRomanLower(n % 10));
    if (n >= 10) return "x" + (n === 10 ? "" : this.renderRomanLower(n % 10));
    if (n >= 9) return "ix";
    if (n >= 8) return "viii";
    if (n >= 7) return "vii";
    if (n >= 6) return "vi";
    if (n >= 5) return "v";
    if (n >= 4) return "iv";
    if (n >= 3) return "iii";
    if (n >= 2) return "ii";
    if (n >= 1) return "i";
    if (n >= 0) return "n";
};

inter.renderTamil = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderTamil(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e8) return this.renderNumber(n, "#,##,##0");
    if (n >= 1e6) return this.renderTamil(Math.floor(n / 1e6)) + "௱௲" + (n === 1e6 ? "" : this.renderTamilThousands(n % 1e6));
    if (n >= 2e3) return this.renderTamil(Math.floor(n / 1e3)) + "௲" + (n === 2e3 ? "" : this.renderTamil(n % 1e3));
    if (n >= 1e3) return "௲" + (n === 1e3 ? "" : this.renderTamil(n % 1e3));
    if (n >= 200) return this.renderTamil(Math.floor(n / 100)) + "௱" + (n === 200 ? "" : this.renderTamil(n % 100));
    if (n >= 100) return "௱" + (n === 100 ? "" : this.renderTamil(n % 100));
    if (n >= 20) return this.renderTamil(Math.floor(n / 10)) + "௰" + (n === 20 ? "" : this.renderTamil(n % 10));
    if (n >= 10) return "௰" + (n === 10 ? "" : this.renderTamil(n % 10));
    if (n >= 9) return "௯";
    if (n >= 8) return "௮";
    if (n >= 7) return "௭";
    if (n >= 6) return "௬";
    if (n >= 5) return "௫";
    if (n >= 4) return "௪";
    if (n >= 3) return "௩";
    if (n >= 2) return "௨";
    if (n >= 1) return "௧";
    if (n >= 0) return "௦";
};

inter.renderSpelloutNumberingM = function(n) {
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 2e15) return this.renderSpelloutNumberingM(Math.floor(n / 1e15)) + " بليار" + (n === 2e15 ? "" : " و " + this.renderSpelloutNumberingM(n % 1e15));
    if (n >= 1e15) return "بليار" + (n === 1e15 ? "" : " و " + this.renderSpelloutNumberingM(n % 1e15));
    if (n >= 2e12) return this.renderSpelloutNumberingM(Math.floor(n / 1e12)) + " بليون" + (n === 2e12 ? "" : " و " + this.renderSpelloutNumberingM(n % 1e12));
    if (n >= 1e12) return "بليون" + (n === 1e12 ? "" : " و " + this.renderSpelloutNumberingM(n % 1e12));
    if (n >= 2e9) return this.renderSpelloutNumberingM(Math.floor(n / 1e9)) + " مليار" + (n === 2e9 ? "" : " و " + this.renderSpelloutNumberingM(n % 1e9));
    if (n >= 1e9) return "مليار" + (n === 1e9 ? "" : " و " + this.renderSpelloutNumberingM(n % 1e9));
    if (n >= 2e6) return this.renderSpelloutNumberingM(Math.floor(n / 1e6)) + " مليون" + (n === 2e6 ? "" : " و " + this.renderSpelloutNumberingM(n % 1e6));
    if (n >= 1e6) return "مليون" + (n === 1e6 ? "" : " و " + this.renderSpelloutNumberingM(n % 1e6));
    if (n >= 11e3) return this.renderSpelloutNumberingM(Math.floor(n / 1e4)) + " ألف" + (n === 11e3 ? "" : " و " + this.renderSpelloutNumberingM(n % 1e4));
    if (n >= 3e3) return this.renderSpelloutNumbering(Math.floor(n / 1e3)) + " آلاف" + (n === 3e3 ? "" : " و " + this.renderSpelloutNumberingM(n % 1e3));
    if (n >= 2e3) return "ألفي" + (n === 2e3 ? "" : " و " + this.renderSpelloutNumberingM(n % 1e3));
    if (n >= 1e3) return "ألف" + (n === 1e3 ? "" : " و " + this.renderSpelloutNumberingM(n % 1e3));
    if (n >= 300) return this.renderSpelloutNumbering(Math.floor(n / 100)) + " مائة" + (n === 300 ? "" : " و " + this.renderSpelloutNumberingM(n % 100));
    if (n >= 200) return "مائتان" + (n === 200 ? "" : " و " + this.renderSpelloutNumberingM(n % 100));
    if (n >= 100) return "مائة" + (n === 100 ? "" : " و " + this.renderSpelloutNumberingM(n % 100));
    if (n >= 90) return (n === 90 ? "" : this.renderSpelloutNumberingM(n % 10) + " و ") + "تسعون";
    if (n >= 80) return (n === 80 ? "" : this.renderSpelloutNumberingM(n % 10) + " و ") + "ثمانون";
    if (n >= 70) return (n === 70 ? "" : this.renderSpelloutNumberingM(n % 10) + " و ") + "سبعون";
    if (n >= 60) return (n === 60 ? "" : this.renderSpelloutNumberingM(n % 10) + " و ") + "ستون";
    if (n >= 50) return (n === 50 ? "" : this.renderSpelloutNumberingM(n % 10) + " و ") + "خمسون";
    if (n >= 40) return (n === 40 ? "" : this.renderSpelloutNumberingM(n % 10) + " و ") + "أربعون";
    if (n >= 30) return (n === 30 ? "" : this.renderSpelloutNumberingM(n % 10) + " و ") + "ثلاثون";
    if (n >= 20) return (n === 20 ? "" : this.renderSpelloutNumberingM(n % 10) + " و ") + "عشرون";
    if (n >= 13) return this.renderSpelloutNumberingM(n % 10) + " عشر";
    if (n >= 12) return "إثنا عشر";
    if (n >= 11) return "إحدى عشر";
    if (n >= 10) return "عشرة";
    if (n >= 9) return "تسعة";
    if (n >= 8) return "ثمانية";
    if (n >= 7) return "سبعة";
    if (n >= 6) return "ستة";
    if (n >= 5) return "خمسة";
    if (n >= 4) return "أربعة";
    if (n >= 3) return "ثلاثة";
    if (n >= 2) return "إثنان";
    if (n >= 1) return "واحد";
    if (n >= 0) return "صفر";
};

inter.renderCyrillicLowerThousands = function(n) {
    if (n >= 900) return "҂ц" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 800) return "҂ѿ" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 700) return "҂ѱ" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 600) return "҂х" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 500) return "҂ф" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 400) return "҂у" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 300) return "҂т" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 200) return "҂с" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 100) return "҂р" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 90) return "҂ч" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 80) return "҂п" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 70) return "҂ѻ" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 60) return "҂ѯ" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 50) return "҂н" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 40) return "҂м" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 30) return "҂л" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 20) return "҂к" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 19) return "҂ѳ҃҂і";
    if (n >= 18) return "҂и҃҂і";
    if (n >= 17) return "҂з҃҂і";
    if (n >= 16) return "҂ѕ҃҂і";
    if (n >= 15) return "҂є҃҂і";
    if (n >= 14) return "҂д҃҂і";
    if (n >= 13) return "҂г҃҂і";
    if (n >= 12) return "҂в҃҂і";
    if (n >= 11) return "҂а҃҂і";
    if (n >= 10) return "҃҂і";
    if (n >= 9) return "҃҂ѳ";
    if (n >= 8) return "҃҂и";
    if (n >= 7) return "҃҂з";
    if (n >= 6) return "҃҂ѕ";
    if (n >= 5) return "҃҂є";
    if (n >= 4) return "҃҂д";
    if (n >= 3) return "҃҂г";
    if (n >= 2) return "҃҂в";
    if (n >= 1) return "҃҂а";
    if (n >= 0) return "҃";
};

inter.renderCyrillicLower110 = function(n) {
    if (n >= 10) return "і";
    if (n >= 9) return "ѳ";
    if (n >= 8) return "и";
    if (n >= 7) return "з";
    if (n >= 6) return "ѕ";
    if (n >= 5) return "є";
    if (n >= 4) return "д";
    if (n >= 3) return "г";
    if (n >= 2) return "в";
    if (n >= 1) return "а";
};

inter.renderCyrillicLowerPost = function(n) {
    if (n >= 1) return this.renderCyrillicLower(n);
    if (n >= 0) return "҃";
};

inter.renderCyrillicLowerFinal = function(n) {
    if (n >= 91) return "ч" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 90) return "҃ч";
    if (n >= 81) return "п" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 80) return "҃п";
    if (n >= 71) return "ѻ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 70) return "҃ѻ";
    if (n >= 61) return "ѯ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 60) return "҃ѯ";
    if (n >= 51) return "н" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 50) return "҃н";
    if (n >= 41) return "м" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 40) return "҃м";
    if (n >= 31) return "л" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 30) return "҃л";
    if (n >= 21) return "к" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 20) return "҃к";
    if (n >= 19) return "ѳ҃і";
    if (n >= 18) return "и҃і";
    if (n >= 17) return "з҃і";
    if (n >= 16) return "ѕ҃і";
    if (n >= 15) return "є҃і";
    if (n >= 14) return "д҃і";
    if (n >= 13) return "г҃і";
    if (n >= 12) return "в҃і";
    if (n >= 11) return "а҃і";
    if (n >= 1) return "҃" + this.renderCyrillicLower110(n);
    if (n >= 0) return "҃";
};

inter.renderEthiopicP3 = function(n) {
    if (n >= 1e12) return this.renderEthiopic(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1) return "፼፼፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼፼፼";
};

inter.renderEthiopicP2 = function(n) {
    if (n >= 1e8) return this.renderEthiopic(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1) return "፼፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼፼";
};

inter.renderEthiopicP1 = function(n) {
    if (n >= 1e4) return this.renderEthiopic(Math.floor(n / 1e4)) + "፼" + (n === 1e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 1) return "፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼";
};

inter.renderGreekNumeralMajuscules = function(n) {
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e16) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e16)) + "ΜΜΜΜ" + (n === 1e16 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e16));
    if (n >= 1e12) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e12)) + "ΜΜΜ" + (n === 1e12 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e12));
    if (n >= 1e8) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e8)) + "ΜΜ" + (n === 1e8 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e8));
    if (n >= 1e4) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e4)) + "Μ" + (n === 1e4 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e4));
    if (n >= 1e3) return "͵" + this.renderGreekNumeralMajuscules(Math.floor(n / 1e3)) + (n === 1e3 ? "" : this.renderGreekNumeralMajuscules(n % 1e3));
    if (n >= 900) return "Ϡ" + (n === 900 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 800) return "Ω" + (n === 800 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 700) return "Ψ" + (n === 700 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 600) return "Χ" + (n === 600 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 500) return "Φ" + (n === 500 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 400) return "Υ" + (n === 400 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 300) return "Τ" + (n === 300 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 200) return "Σ" + (n === 200 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 100) return "Ρ" + (n === 100 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 90) return "Ϟ" + (n === 90 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 80) return "Π" + (n === 80 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 70) return "Ο" + (n === 70 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 60) return "Ξ" + (n === 60 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 50) return "Ν" + (n === 50 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 40) return "Μ" + (n === 40 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 30) return "Λ" + (n === 30 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 20) return "Κ" + (n === 20 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 10) return "Ι" + (n === 10 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 9) return "Θ";
    if (n >= 8) return "Η";
    if (n >= 7) return "Ζ";
    if (n >= 6) return "Ϝ";
    if (n >= 5) return "Ε";
    if (n >= 4) return "Δ";
    if (n >= 3) return "Γ";
    if (n >= 2) return "Β";
    if (n >= 1) return "Α";
    if (n >= 0) return "𐆊";
};

inter.renderGreekNumeralMinuscules = function(n) {
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e16) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e16)) + "μμμμ" + (n === 1e16 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e16));
    if (n >= 1e12) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e12)) + "μμμ" + (n === 1e12 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e12));
    if (n >= 1e8) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e8)) + "μμ" + (n === 1e8 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e8));
    if (n >= 1e4) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e4)) + "μ" + (n === 1e4 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e4));
    if (n >= 1e3) return "͵" + this.renderGreekNumeralMinuscules(Math.floor(n / 1e3)) + (n === 1e3 ? "" : this.renderGreekNumeralMinuscules(n % 1e3));
    if (n >= 900) return "ϡ" + (n === 900 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 800) return "ω" + (n === 800 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 700) return "ψ" + (n === 700 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 600) return "χ" + (n === 600 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 500) return "φ" + (n === 500 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 400) return "υ" + (n === 400 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 300) return "τ" + (n === 300 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 200) return "σ" + (n === 200 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 100) return "ρ" + (n === 100 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 90) return "ϟ" + (n === 90 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 80) return "π" + (n === 80 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 70) return "ο" + (n === 70 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 60) return "ξ" + (n === 60 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 50) return "ν" + (n === 50 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 40) return "μ" + (n === 40 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 30) return "λ" + (n === 30 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 20) return "κ" + (n === 20 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 10) return "ι" + (n === 10 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 9) return "θ";
    if (n >= 8) return "η";
    if (n >= 7) return "ζ";
    if (n >= 6) return "ϝ";
    if (n >= 5) return "ε";
    if (n >= 4) return "δ";
    if (n >= 3) return "γ";
    if (n >= 2) return "β";
    if (n >= 1) return "α";
    if (n >= 0) return "𐆊";
};

inter.renderHebrewThousands = function(n) {
    if (n >= 401) return this.renderHebrew(n) + "׳";
    if (n >= 100) return this.renderHebrew(n) + (n === 100 ? "" : "׳");
    if (n >= 10) return this.renderHebrew(n) + (n === 10 ? "" : "׳");
    if (n >= 0) return this.renderHebrew(n);
};

inter.renderHebrew099 = function(n) {
    if (n >= 91) return "צ״" + this.renderHebrewItem(n % 10);
    if (n >= 90) return "״צ";
    if (n >= 81) return "פ״" + this.renderHebrewItem(n % 10);
    if (n >= 80) return "״פ";
    if (n >= 71) return "ע״" + this.renderHebrewItem(n % 10);
    if (n >= 70) return "״ע";
    if (n >= 61) return "ס״" + this.renderHebrewItem(n % 10);
    if (n >= 60) return "״ס";
    if (n >= 51) return "נ״" + this.renderHebrewItem(n % 10);
    if (n >= 50) return "״נ";
    if (n >= 41) return "מ״" + this.renderHebrewItem(n % 10);
    if (n >= 40) return "״מ";
    if (n >= 31) return "ל״" + this.renderHebrewItem(n % 10);
    if (n >= 30) return "״ל";
    if (n >= 21) return "כ״" + this.renderHebrewItem(n % 10);
    if (n >= 20) return "״כ";
    if (n >= 17) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "ט״ז";
    if (n >= 15) return "ט״ו";
    if (n >= 11) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 1) return "״" + this.renderHebrewItem(n);
    if (n >= 0) return "׳";
};

inter.renderHebrewItem = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderHebrewItem(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 2100) return this.renderNumber(n, "#,##0");
    if (n >= 2e3) return "תתתתת" + (n === 2e3 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1900) return "תתתתש" + (n === 1900 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1800) return "תתתתר" + (n === 1800 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1700) return "תתתתק" + (n === 1700 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1600) return "תתתת" + (n === 1600 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1500) return "תתתש" + (n === 1500 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1400) return "תתתר" + (n === 1400 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1300) return "תתתק" + (n === 1300 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1200) return "תתת" + (n === 1200 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1100) return "תתש" + (n === 1100 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1e3) return "תתר" + (n === 1e3 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 900) return "תתק" + (n === 900 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 800) return "תת" + (n === 800 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 745) return "תש" + this.renderHebrewItem(n % 100);
    if (n >= 744) return "תשדמ";
    if (n >= 700) return "תש" + (n === 700 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 699) return "תר" + this.renderHebrewItem(n % 100);
    if (n >= 698) return "תרחצ";
    if (n >= 600) return "תר" + (n === 600 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 500) return "תק" + (n === 500 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 400) return "ת" + (n === 400 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 345) return "ש" + this.renderHebrewItem(n % 100);
    if (n >= 344) return "שדמ";
    if (n >= 305) return "ש" + this.renderHebrewItem(n % 100);
    if (n >= 304) return "דש";
    if (n >= 300) return "ש" + (n === 300 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 299) return "ר" + this.renderHebrewItem(n % 100);
    if (n >= 298) return "רחצ";
    if (n >= 200) return "ר" + (n === 200 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 100) return "ק" + (n === 100 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 90) return "צ" + (n === 90 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 80) return "פ" + (n === 80 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 70) return "ע" + (n === 70 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 60) return "ס" + (n === 60 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 50) return "נ" + (n === 50 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 40) return "מ" + (n === 40 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 30) return "ל" + (n === 30 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 20) return "כ" + (n === 20 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 17) return "י" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "טז";
    if (n >= 15) return "טו";
    if (n >= 10) return "י" + (n === 10 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 9) return "ט";
    if (n >= 8) return "ח";
    if (n >= 7) return "ז";
    if (n >= 6) return "ו";
    if (n >= 5) return "ה";
    if (n >= 4) return "ד";
    if (n >= 3) return "ג";
    if (n >= 2) return "ב";
    if (n >= 1) return "א";
    if (n >= 0) return "״";
};

inter.renderTamilThousands = function(n) {
    if (n >= 1e3) return this.renderTamilThousands(Math.floor(n / 1e3)) + "௲" + (n === 1e3 ? "" : this.renderTamilThousands(n % 1e3));
    if (n >= 0) return this.renderTamil(n);
};

inter.renderEthiopicP = function(n) {
    if (n >= 1e16) return this.renderEthiopicP(Math.floor(n / 1e16)) + "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 1e12) return this.renderEthiopicP(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1e8) return this.renderEthiopicP(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1e4) return this.renderEthiopicP(Math.floor(n / 1e4)) + "፼" + (n === 1e4 ? "" : this.renderEthiopicP(n % 1e4));
    if (n >= 1) return this.renderEthiopic(n);
};

module.exports = inter;
