// inter/build/ee_tg.js
function defaults(targetObj) {
    for (var i = 1; i < arguments.length; i += 1) {
        if (arguments[i] && typeof arguments[i] === "object") {
            for (var propertyName in arguments[i]) {
                if (typeof targetObj[propertyName] === "undefined") {
                    targetObj[propertyName] = arguments[i][propertyName];
                }
            }
        }
    }
    return targetObj;
}

var P = require("./ee.js"), inter = defaults({
    renderers: {}
}, P);

var _didi1 = [ "k1", "k2", "k3", "k4" ], _didi2 = {
    displayName: "ɣleti"
}, _didi3 = {
    all: {
        leap: "{0}bis"
    }
}, _didi4 = [ "hafi R.O.C.", "R.O.C." ], _didi5 = {
    a: "a h – a h",
    h: "a h–h"
}, _didi6 = {
    displayName: "ŋkekea me"
}, _didi7 = {
    a: "h a – h a",
    h: "h–h a"
}, _didi8 = {
    displayName: "nutomegaƒoƒo"
}, _didi9 = {
    d: "M/d – M/d",
    M: "M/d – M/d"
}, _didi10 = [ "k", "d", "b", "k", "y", "f", "m" ], _didi11 = {
    a: "h a – h a v",
    h: "h–h a v"
}, _didi12 = {
    displayName: "kɔsiɖa me ŋkeke"
}, _didi13 = {
    d: "MMM d–d",
    M: "MMM d – MMM d"
}, _didi14 = {
    wide: [ "AH" ],
    abbreviated: [ "AH" ]
}, _didi15 = {
    H: "HH:mm–HH:mm",
    m: "HH:mm–HH:mm"
}, _didi16 = {
    M: "M/y – M/y G",
    y: "M/y – M/y G"
}, _didi17 = {
    M: "y-MM – y-MM",
    y: "y-MM – y-MM"
}, _didi18 = {
    M: "U MMM–MMM",
    y: "U MMM – U MMM"
}, _didi19 = {
    a: "a h – a h v",
    h: "a 'ga' h–h v"
}, _didi20 = {
    H: "HH:mm–HH:mm v",
    m: "HH:mm–HH:mm v"
}, _didi21 = {
    M: "MMM–MMM y G",
    y: "MMM y – MMM y G"
}, _didi22 = {
    d: "MM-dd – MM-dd",
    M: "MM-dd – MM-dd"
}, _didi23 = {
    M: "U MMMM–MMMM",
    y: "U MMMM – U MMMM"
}, _didi24 = {
    M: "'ɣleti' M 'lia' – 'ɣleti' M 'lia'"
}, _didi25 = {
    d: "E, M/d – E, M/d",
    M: "E, M/d – E, M/d"
}, _didi26 = {
    M: "MMMM–MMMM y G",
    y: "MMMM y – MMMM y G"
}, _didi27 = {
    one: "le ƒe {0} me",
    other: "le ƒe {0} me"
}, _didi28 = [ "kɔs", "dzo", "bla", "kuɖ", "yaw", "fiɖ", "mem" ], _didi29 = {
    "0": "ƒe sia",
    "1": "ƒe si gbɔ na",
    "-1": "ƒe si va yi"
}, _didi30 = [ "d", "d", "t", "a", "d", "m", "s", "d", "a", "k", "a", "d" ], _didi31 = {
    d: "MM-dd, E – MM-dd, E",
    M: "MM-dd, E – MM-dd, E"
}, _didi32 = {
    d: "MMM d, E – MMM d, E",
    M: "MMM d, E – MMM d, E"
}, _didi33 = {
    one: "ƒe {0} si va yi",
    other: "ƒe {0} si wo va yi"
}, _didi34 = {
    one: "le ɣleti {0} me",
    other: "le ɣleti {0} wo me"
}, _didi35 = {
    one: "le ŋkeke {0} me",
    other: "le ŋkeke {0} wo me"
}, _didi36 = {
    wide: [ "ERA0", "ERA1" ],
    abbreviated: [ "ERA0", "ERA1" ]
}, _didi37 = {
    one: "le kɔsiɖa {0} me",
    other: "le kɔsiɖa {0} wo me"
}, _didi38 = {
    one: "le gaƒoƒo {0} me",
    other: "le gaƒoƒo {0} wo me"
}, _didi39 = {
    one: "le sekend {0} me",
    other: "le sekend {0} wo me"
}, _didi40 = [ "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12" ], _didi41 = {
    one: "ƒe {0} si va yi me",
    other: "ƒe {0} si va yi me"
}, _didi42 = {
    a: "h:mm a – h:mm a",
    h: "h:mm–h:mm a",
    m: "h:mm–h:mm a"
}, _didi43 = {
    "0": "kuɖa sia",
    "1": "kuɖa si gbɔ na",
    "-1": "kuɖa si va yi"
}, _didi44 = {
    "0": "fiɖa sia",
    "1": "fiɖa si gbɔ na",
    "-1": "fiɖa si va yi"
}, _didi45 = {
    "0": "this quarter",
    "1": "next quarter",
    "-1": "last quarter"
}, _didi46 = {
    one: "ɣleti {0} si va yi",
    other: "ɣleti {0} si wo va yi"
}, _didi47 = {
    one: "ŋkeke {0} si va yi",
    other: "ŋkeke {0} si wo va yi"
}, _didi48 = {
    d: "MMM d 'lia' – d 'lia'",
    M: "MMM d 'lia' – MMM d 'lia'"
}, _didi49 = [ "kɔta gbãtɔ", "kɔta evelia", "kɔta etɔ̃lia", "kɔta enelia" ], _didi50 = {
    one: "kɔta {0} si va yi me",
    other: "kɔta {0} si va yi me"
}, _didi51 = {
    "0": "ɣleti sia",
    "1": "ɣleti si gbɔ na",
    "-1": "ɣleti si va yi"
}, _didi52 = {
    "0": "dzoɖa sia",
    "1": "dzoɖa si gbɔ na",
    "-1": "dzoɖa si va yi"
}, _didi53 = {
    "0": "blaɖa sia",
    "1": "blaɖa si gbɔ na",
    "-1": "blaɖa si va yi"
}, _didi54 = {
    d: "U MMM d–d",
    M: "U MMM d – MMM d",
    y: "U MMM d – U MMM d"
}, _didi55 = [ "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13" ], _didi56 = {
    one: "kɔsiɖa {0} si va yi",
    other: "kɔsiɖa {0} si wo va yi"
}, _didi57 = {
    one: "gaƒoƒo {0} si va yi",
    other: "gaƒoƒo {0} si wo va yi"
}, _didi58 = {
    one: "le aɖabaƒoƒo {0} me",
    other: "le aɖabaƒoƒo {0} wo me"
}, _didi59 = {
    one: "sekend {0} si va yi",
    other: "sekend {0} si wo va yi"
}, _didi60 = [ "kɔsiɖa", "dzoɖa", "blaɖa", "kuɖa", "yawoɖa", "fiɖa", "memleɖa" ], _didi61 = {
    one: "le ƒe {0} si va yi me",
    other: "le ƒe {0} si va yi me"
}, _didi62 = {
    one: "le ƒe {0} si gbɔna me",
    other: "le ƒe {0} si gbɔna me"
}, _didi63 = {
    a: "h:mm a – h:mm a v",
    h: "h:mm–h:mm a v",
    m: "h:mm–h:mm a v"
}, _didi64 = {
    "0": "kɔsiɖa sia",
    "1": "kɔsiɖa si gbɔ na",
    "-1": "kɔsiɖa si va yi"
}, _didi65 = {
    "0": "yawoɖa sia",
    "1": "yawoɖa si gbɔ na",
    "-1": "yawoɖa si va yi"
}, _didi66 = {
    d: "M/d/y – M/d/y G",
    M: "M/d/y – M/d/y G",
    y: "M/d/y – M/d/y G"
}, _didi67 = {
    "0": "memleɖa sia",
    "1": "memleɖa si gbɔ na",
    "-1": "memleɖa si va yi"
}, _didi68 = {
    one: "le kɔta {0} si gbɔna me",
    other: "le kɔta {0} si gbɔ na me"
}, _didi69 = {
    one: "aɖabaƒoƒo {0} si va yi",
    other: "aɖabaƒoƒo {0} si wo va yi"
}, _didi70 = {
    one: "le kɔta {0} si gbɔ na me",
    other: "le kɔta {0} si gbɔ na me"
}, _didi71 = {
    relative: _didi43
}, _didi72 = {
    relative: _didi44
}, _didi73 = {
    d: "y-MM-dd – y-MM-dd",
    M: "y-MM-dd – y-MM-dd",
    y: "y-MM-dd – y-MM-dd"
}, _didi74 = {
    relative: _didi52
}, _didi75 = {
    relative: _didi53
}, _didi76 = {
    "0": "kɔsiɖa sia gbe",
    "1": "kɔsiɖagbe si gbɔ na",
    "-1": "kɔsiɖagbe si va yi"
}, _didi77 = {
    relative: _didi65
}, _didi78 = {
    abbreviated: {
        leap: "{0}bis"
    },
    narrow: {
        leap: "{0}b"
    },
    wide: {
        leap: "{0}bis"
    }
}, _didi79 = {
    d: "E, MMM d 'lia' – E, MMM d 'lia'",
    M: "E, MMM d 'lia' – E, MMM d 'lia'"
}, _didi80 = [ "dzv", "dzd", "ted", "afɔ", "dam", "mas", "sia", "dea", "any", "kel", "ade", "dzm" ], _didi81 = {
    a: "a 'ga' h:mm – a 'ga' h:mm v",
    h: "a 'ga' h:mm–h:mm v",
    m: "a h:mm–h:mm v"
}, _didi82 = {
    relative: _didi67
}, _didi83 = [ "dzv", "dzd", "ted", "afɔ", "dam", "mas", "sia", "dea", "any", "kele", "ade", "dzm" ], _didi84 = [ undefined, "zi", "chou", "yin", "mao", "chen", "si", "wu", "wei", "shen", "you", "xu" ], _didi85 = {
    d: "E, M/d/y – E, M/d/y G",
    M: "E, M/d/y – E, M/d/y G",
    y: "E, M/d/y – E, M/d/y G"
}, _didi86 = [ "dzv", "dzd", "ted", "afɔ", "dam", "mas", "sia", "dea", "any", "kel", "ade", "dzm", "foa" ], _didi87 = {
    relative: _didi76
}, _didi88 = {
    d: "U MMM d, E – MMM d, E",
    M: "U MMM d, E – MMM d, E",
    y: "U MMM d, E – U MMM d, E"
}, _didi89 = {
    d: "y-MM-dd, E – y-MM-dd, E",
    M: "y-MM-dd, E – y-MM-dd, E",
    y: "y-MM-dd, E – y-MM-dd, E"
}, _didi90 = {
    a: "a 'ga' h:mm – a 'ga' h:mm",
    h: "a 'ga' h:mm - 'ga' h:mm",
    m: "a 'ga' h:mm – 'ga' h:mm"
}, _didi91 = {
    "0": "egbe",
    "1": "etsɔ si gbɔna",
    "2": "nyitsɔ si gbɔna",
    "-2": "nyitsɔ si va yi",
    "-1": "etsɔ si va yi"
}, _didi92 = {
    future: _didi27,
    past: _didi33
}, _didi93 = {
    d: "MMM d 'lia' – d 'lia' , y G",
    M: "MMM d 'lia' – MMM d 'lia', y G",
    y: "MMM d 'lia' , y – MMM d 'lia', y G"
}, _didi94 = {
    am: "ŋ",
    pm: "ɣ",
    morning1: "fɔŋli",
    morning2: "ŋdi",
    afternoon1: "ŋdɔ",
    afternoon2: "ɣetrɔ",
    evening1: "fiẽ",
    night1: "zã"
}, _didi95 = [ "dzove", "dzodze", "tedoxe", "afɔfĩe", "dama", "masa", "siamlɔm", "deasiamime", "anyɔnyɔ", "kele", "adeɛmekpɔxe", "dzome" ], _didi96 = [ "dzove", "dzodze", "tedoxe", "afɔfiẽ", "damɛ", "masa", "siamlɔm", "deasiamime", "anyɔnyɔ", "kele", "adeɛmekpɔxe", "dzome" ], _didi97 = {
    future: _didi27,
    past: _didi61
}, _didi98 = {
    am: "ŋdi",
    pm: "ɣetrɔ",
    morning1: "fɔŋli",
    morning2: "ŋdi",
    afternoon1: "ŋdɔ",
    afternoon2: "ɣetrɔ",
    evening1: "fiẽ",
    night1: "zã"
}, _didi99 = [ "dzove", "dzodze", "tedoxe", "afɔfiẽ", "damɛ", "masa", "siamlɔm", "deasiamime", "anyɔnyɔ", "kele", "adeɛmekpɔxe", "dzome", "ƒoave" ], _didi100 = {
    future: _didi34,
    past: _didi46
}, _didi101 = {
    future: _didi35,
    past: _didi47
}, _didi102 = {
    abbreviated: _didi1,
    narrow: [ "1", "2", "3", "4" ],
    wide: _didi49
}, _didi103 = {
    future: _didi37,
    past: _didi56
}, _didi104 = {
    future: _didi38,
    past: _didi57
}, _didi105 = {
    future: _didi39,
    past: _didi59
}, _didi106 = {
    d: "E, MMM d 'lia' – E, MMM d 'lia', y G",
    M: "E, MMM d 'lia' – E, MMM d 'lia', y G",
    y: "E, MMM d 'lia', y – E, MMM d 'lia', y G"
}, _didi107 = {
    future: _didi62,
    past: _didi41
}, _didi108 = {
    future: _didi68,
    past: _didi50
}, _didi109 = {
    future: _didi70,
    past: _didi50
}, _didi110 = {
    future: _didi58,
    past: _didi69
}, _didi111 = {
    displayName: "ƒe",
    relativeTime: _didi97
}, _didi112 = {
    displayName: "ƒe",
    relativeTime: _didi107
}, _didi113 = {
    displayName: "gaƒoƒo",
    relativeTime: _didi104
}, _didi114 = {
    displayName: "kɔta",
    relativeTime: _didi108
}, _didi115 = {
    displayName: "kɔta",
    relativeTime: _didi109
}, _didi116 = {
    displayName: "aɖabaƒoƒo",
    relativeTime: _didi110
}, _didi117 = {
    displayName: "sekend",
    relative: [ "fifi" ],
    relativeTime: _didi105
}, _didi118 = {
    format: _didi78,
    numeric: _didi3,
    standAlone: _didi78
}, _didi119 = {
    displayName: "ƒe",
    relative: _didi29,
    relativeTime: _didi92
}, _didi120 = {
    abbreviated: _didi28,
    narrow: _didi10,
    wide: _didi60,
    "short": _didi28
}, _didi121 = {
    displayName: "ɣleti",
    relative: _didi51,
    relativeTime: _didi100
}, _didi122 = {
    displayName: "kɔta",
    relative: _didi45,
    relativeTime: _didi109
}, _didi123 = {
    displayName: "kɔsiɖa ɖeka",
    relative: _didi64,
    relativeTime: _didi103
}, _didi124 = {
    abbreviated: _didi84,
    narrow: _didi84,
    wide: _didi84
}, _didi125 = {
    displayName: "ŋkeke",
    relative: _didi91,
    relativeTime: _didi101
}, _didi126 = {
    abbreviated: _didi80,
    narrow: _didi30,
    wide: _didi95
}, _didi127 = {
    format: _didi124
}, _didi128 = {
    abbreviated: _didi80,
    narrow: _didi40,
    wide: _didi96
}, _didi129 = {
    abbreviated: _didi83,
    narrow: _didi40,
    wide: _didi96
}, _didi130 = {
    format: _didi102,
    standAlone: _didi102
}, _didi131 = {
    abbreviated: _didi86,
    narrow: _didi55,
    wide: _didi99
}, _didi132 = {
    abbreviated: _didi99,
    narrow: _didi55,
    wide: _didi99
}, _didi133 = {
    fullDate: "EEEE, U MMMM dd 'lia'",
    longDate: "U MMMM d 'lia'",
    mediumDate: "U MMM d 'lia'",
    shortDate: "r-MM-dd",
    fullTime: "HH:mm:ss zzzz",
    longTime: "HH:mm:ss z",
    mediumTime: "HH:mm:ss",
    shortTime: "HH:mm",
    shortDateTime: "r-MM-dd HH:mm",
    mediumDateTime: "U MMM d 'lia' HH:mm:ss",
    longDateTime: "U MMMM d 'lia' HH:mm:ss z",
    fullDateTime: "EEEE, U MMMM dd 'lia' HH:mm:ss zzzz"
}, _didi134 = {
    abbreviated: _didi98,
    narrow: _didi94,
    wide: _didi98
}, _didi135 = {
    fullDate: "EEEE, MMMM d 'lia' y G",
    longDate: "MMMM d 'lia' y G",
    mediumDate: "MMM d 'lia', y G",
    shortDate: "M/d/y GGGGG",
    fullTime: "HH:mm:ss zzzz",
    longTime: "HH:mm:ss z",
    mediumTime: "HH:mm:ss",
    shortTime: "HH:mm",
    shortDateTime: "HH:mm M/d/y GGGGG",
    mediumDateTime: "HH:mm:ss MMM d 'lia', y G",
    longDateTime: "HH:mm:ss z MMMM d 'lia' y G",
    fullDateTime: "HH:mm:ss zzzz EEEE, MMMM d 'lia' y G"
}, _didi136 = {
    format: _didi120,
    standAlone: _didi120
}, _didi137 = {
    d: "d",
    E: "ccc",
    Ed: "d, E",
    Gy: "r U",
    GyMMM: "r(U) MMM",
    GyMMMd: "r MMM d",
    GyMMMEd: "r(U) MMM d, E",
    h: "h a",
    H: "HH",
    hm: "h:mm a",
    Hm: "HH:mm",
    hms: "h:mm:ss a",
    Hms: "HH:mm:ss",
    M: "L",
    Md: "MM-dd",
    MEd: "MM-dd, E",
    MMM: "LLL",
    MMMd: "MMM d",
    MMMEd: "MMM d, E",
    MMMMd: "MMMM d",
    ms: "mm:ss",
    UM: "U MM",
    UMd: "U MM-d",
    UMMM: "U MMM",
    UMMMd: "U MMM d",
    y: "r(U)",
    yMd: "r-MM-dd",
    yyyy: "r(U)",
    yyyyM: "r-MM",
    yyyyMd: "r-MM-dd",
    yyyyMEd: "r-MM-dd, E",
    yyyyMMM: "r(U) MMM",
    yyyyMMMd: "r MMM d",
    yyyyMMMEd: "r(U) MMM d, E",
    yyyyMMMM: "r(U) MMMM",
    yyyyQQQ: "r(U) QQQ",
    yyyyQQQQ: "r(U) QQQQ"
}, _didi138 = {
    format: _didi126,
    standAlone: _didi126
}, _didi139 = {
    format: _didi128,
    standAlone: _didi128
}, _didi140 = {
    format: _didi128,
    standAlone: _didi129
}, _didi141 = [ undefined, "jia-zi", "yi-chou", "bing-yin", "ding-mao", "wu-chen", "ji-si", "geng-wu", "xin-wei", "ren-shen", "gui-you", "jia-xu", "yi-hai", "bing-zi", "ding-chou", "wu-yin", "ji-mao", "geng-chen", "xin-si", "ren-wu", "gui-wei", "jia-shen", "yi-you", "bing-xu", "ding-hai", "wu-zi", "ji-chou", "geng-yin", "xin-mao", "ren-chen", "gui-si", "jia-wu", "yi-wei", "bing-shen", "ding-you", "wu-xu", "ji-hai", "geng-zi", "xin-chou", "ren-yin", "gui-mao", "jia-chen", "yi-si", "bing-wu", "ding-wei", "wu-shen", "ji-you", "geng-xu", "xin-hai", "ren-zi", "gui-chou", "jia-yin", "yi-mao", "bing-chen", "ding-si", "wu-wu", "ji-wei", "geng-shen", "xin-you", "ren-xu" ], _didi142 = {
    d: "d",
    Ed: "E d",
    Gy: "y G",
    GyMMM: "MMM y G",
    GyMMMd: "MMM d 'lia', y G",
    GyMMMEd: "E, MMM d 'lia' y G",
    h: "a 'ga' h",
    H: "HH",
    hm: "a 'ga' h:mm",
    Hm: "HH:mm",
    hms: "a 'ga' h:mm:ss",
    Hms: "HH:mm:ss",
    M: "L",
    Md: "M/d",
    MEd: "E, M/d",
    MMM: "LLL",
    MMMd: "MMM d 'lia'",
    MMMEd: "E, MMM d 'lia'",
    MMMMd: "MMMM d 'lia'",
    MMMMEd: "E, MMMM d 'lia'",
    ms: "'aɖabaƒoƒo' mm:ss",
    y: "y G",
    yyyy: "y G",
    yyyyM: "M/y G",
    yyyyMd: "M/d/y G",
    yyyyMEd: "E, M/d/y G",
    yyyyMMM: "MMM y G",
    yyyyMMMd: "MMM d 'lia', y G",
    yyyyMMMEd: "E, MMM d 'lia' y G",
    yyyyMMMM: "MMMM y G",
    yyyyQQQ: "QQQ y G",
    yyyyQQQQ: "QQQQ y G",
    E: "ccc",
    Ehm: "E h:mm a",
    EHm: "E HH:mm",
    Ehms: "E h:mm:ss a",
    EHms: "E HH:mm:ss"
}, _didi143 = {
    format: _didi134,
    standAlone: _didi134
}, _didi144 = {
    basic: _didi133,
    cldr: _didi137
}, _didi145 = {
    d: {
        d: "d–d"
    },
    h: _didi7,
    H: {
        H: "HH–HH"
    },
    hm: _didi42,
    Hm: _didi15,
    hmv: _didi63,
    Hmv: _didi20,
    hv: _didi11,
    Hv: {
        H: "HH–HH v"
    },
    M: {
        M: "MM–MM"
    },
    Md: _didi22,
    MEd: _didi31,
    MMM: {
        M: "LLL–LLL"
    },
    MMMd: _didi13,
    MMMEd: _didi32,
    y: {
        y: "U–U"
    },
    yM: _didi17,
    yMd: _didi73,
    yMEd: _didi89,
    yMMM: _didi18,
    yMMMd: _didi54,
    yMMMEd: _didi88,
    yMMMM: _didi23
}, _didi146 = {
    basic: _didi135,
    cldr: _didi142
}, _didi147 = {
    d: {
        d: "d–d"
    },
    h: _didi5,
    H: {
        H: "HH–HH"
    },
    hm: _didi90,
    Hm: _didi15,
    hmv: _didi81,
    Hmv: _didi20,
    hv: _didi19,
    Hv: {
        H: "HH–HH v"
    },
    M: _didi24,
    Md: _didi9,
    MEd: _didi25,
    MMM: {
        M: "MMM–MMM"
    },
    MMMd: _didi48,
    MMMEd: _didi79,
    y: {
        y: "G y–y"
    },
    yM: _didi16,
    yMd: _didi66,
    yMEd: _didi85,
    yMMM: _didi21,
    yMMMd: _didi93,
    yMMMEd: _didi106,
    yMMMM: _didi26
}, _didi148 = {
    abbreviated: _didi141,
    narrow: _didi141,
    wide: _didi141
}, _didi149 = {
    format: _didi148
}, _didi150 = {
    era: {
        displayName: "ŋɔli"
    },
    year: _didi119,
    "year-short": _didi111,
    "year-narrow": _didi112,
    quarter: _didi122,
    "quarter-short": _didi115,
    "quarter-narrow": _didi114,
    month: _didi121,
    "month-short": _didi2,
    "month-narrow": _didi2,
    week: _didi123,
    day: _didi125,
    weekday: _didi12,
    dayperiod: _didi6,
    hour: _didi113,
    minute: _didi116,
    second: _didi117,
    zone: _didi8,
    sun: _didi87,
    mon: _didi74,
    tue: _didi75,
    wed: _didi71,
    thu: _didi77,
    fri: _didi72,
    sat: _didi82
}, _didi151 = {
    dateFormats: {
        basic: {},
        cldr: {}
    },
    fields: _didi150
}, _didi152 = [ "Taika (645–650)", "Hakuchi (650–671)", "Hakuhō (672–686)", "Shuchō (686–701)", "Taihō (701–704)", "Keiun (704–708)", "Wadō (708–715)", "Reiki (715–717)", "Yōrō (717–724)", "Jinki (724–729)", "Tenpyō (729–749)", "Tenpyō-kampō (749-749)", "Tenpyō-shōhō (749-757)", "Tenpyō-hōji (757-765)", "Tenpyō-jingo (765-767)", "Jingo-keiun (767-770)", "Hōki (770–780)", "Ten-ō (781-782)", "Enryaku (782–806)", "Daidō (806–810)", "Kōnin (810–824)", "Tenchō (824–834)", "Jōwa (834–848)", "Kajō (848–851)", "Ninju (851–854)", "Saikō (854–857)", "Ten-an (857-859)", "Jōgan (859–877)", "Gangyō (877–885)", "Ninna (885–889)", "Kanpyō (889–898)", "Shōtai (898–901)", "Engi (901–923)", "Enchō (923–931)", "Jōhei (931–938)", "Tengyō (938–947)", "Tenryaku (947–957)", "Tentoku (957–961)", "Ōwa (961–964)", "Kōhō (964–968)", "Anna (968–970)", "Tenroku (970–973)", "Ten’en (973–976)", "Jōgen (976–978)", "Tengen (978–983)", "Eikan (983–985)", "Kanna (985–987)", "Eien (987–989)", "Eiso (989–990)", "Shōryaku (990–995)", "Chōtoku (995–999)", "Chōhō (999–1004)", "Kankō (1004–1012)", "Chōwa (1012–1017)", "Kannin (1017–1021)", "Jian (1021–1024)", "Manju (1024–1028)", "Chōgen (1028–1037)", "Chōryaku (1037–1040)", "Chōkyū (1040–1044)", "Kantoku (1044–1046)", "Eishō (1046–1053)", "Tengi (1053–1058)", "Kōhei (1058–1065)", "Jiryaku (1065–1069)", "Enkyū (1069–1074)", "Shōho (1074–1077)", "Shōryaku (1077–1081)", "Eihō (1081–1084)", "Ōtoku (1084–1087)", "Kanji (1087–1094)", "Kahō (1094–1096)", "Eichō (1096–1097)", "Jōtoku (1097–1099)", "Kōwa (1099–1104)", "Chōji (1104–1106)", "Kashō (1106–1108)", "Tennin (1108–1110)", "Ten-ei (1110-1113)", "Eikyū (1113–1118)", "Gen’ei (1118–1120)", "Hōan (1120–1124)", "Tenji (1124–1126)", "Daiji (1126–1131)", "Tenshō (1131–1132)", "Chōshō (1132–1135)", "Hōen (1135–1141)", "Eiji (1141–1142)", "Kōji (1142–1144)", "Ten’yō (1144–1145)", "Kyūan (1145–1151)", "Ninpei (1151–1154)", "Kyūju (1154–1156)", "Hōgen (1156–1159)", "Heiji (1159–1160)", "Eiryaku (1160–1161)", "Ōho (1161–1163)", "Chōkan (1163–1165)", "Eiman (1165–1166)", "Nin’an (1166–1169)", "Kaō (1169–1171)", "Shōan (1171–1175)", "Angen (1175–1177)", "Jishō (1177–1181)", "Yōwa (1181–1182)", "Juei (1182–1184)", "Genryaku (1184–1185)", "Bunji (1185–1190)", "Kenkyū (1190–1199)", "Shōji (1199–1201)", "Kennin (1201–1204)", "Genkyū (1204–1206)", "Ken’ei (1206–1207)", "Jōgen (1207–1211)", "Kenryaku (1211–1213)", "Kenpō (1213–1219)", "Jōkyū (1219–1222)", "Jōō (1222–1224)", "Gennin (1224–1225)", "Karoku (1225–1227)", "Antei (1227–1229)", "Kanki (1229–1232)", "Jōei (1232–1233)", "Tenpuku (1233–1234)", "Bunryaku (1234–1235)", "Katei (1235–1238)", "Ryakunin (1238–1239)", "En’ō (1239–1240)", "Ninji (1240–1243)", "Kangen (1243–1247)", "Hōji (1247–1249)", "Kenchō (1249–1256)", "Kōgen (1256–1257)", "Shōka (1257–1259)", "Shōgen (1259–1260)", "Bun’ō (1260–1261)", "Kōchō (1261–1264)", "Bun’ei (1264–1275)", "Kenji (1275–1278)", "Kōan (1278–1288)", "Shōō (1288–1293)", "Einin (1293–1299)", "Shōan (1299–1302)", "Kengen (1302–1303)", "Kagen (1303–1306)", "Tokuji (1306–1308)", "Enkyō (1308–1311)", "Ōchō (1311–1312)", "Shōwa (1312–1317)", "Bunpō (1317–1319)", "Genō (1319–1321)", "Genkō (1321–1324)", "Shōchū (1324–1326)", "Karyaku (1326–1329)", "Gentoku (1329–1331)", "Genkō (1331–1334)", "Kenmu (1334–1336)", "Engen (1336–1340)", "Kōkoku (1340–1346)", "Shōhei (1346–1370)", "Kentoku (1370–1372)", "Bunchū (1372–1375)", "Tenju (1375–1379)", "Kōryaku (1379–1381)", "Kōwa (1381–1384)", "Genchū (1384–1392)", "Meitoku (1384–1387)", "Kakei (1387–1389)", "Kōō (1389–1390)", "Meitoku (1390–1394)", "Ōei (1394–1428)", "Shōchō (1428–1429)", "Eikyō (1429–1441)", "Kakitsu (1441–1444)", "Bun’an (1444–1449)", "Hōtoku (1449–1452)", "Kyōtoku (1452–1455)", "Kōshō (1455–1457)", "Chōroku (1457–1460)", "Kanshō (1460–1466)", "Bunshō (1466–1467)", "Ōnin (1467–1469)", "Bunmei (1469–1487)", "Chōkyō (1487–1489)", "Entoku (1489–1492)", "Meiō (1492–1501)", "Bunki (1501–1504)", "Eishō (1504–1521)", "Taiei (1521–1528)", "Kyōroku (1528–1532)", "Tenbun (1532–1555)", "Kōji (1555–1558)", "Eiroku (1558–1570)", "Genki (1570–1573)", "Tenshō (1573–1592)", "Bunroku (1592–1596)", "Keichō (1596–1615)", "Genna (1615–1624)", "Kan’ei (1624–1644)", "Shōho (1644–1648)", "Keian (1648–1652)", "Jōō (1652–1655)", "Meireki (1655–1658)", "Manji (1658–1661)", "Kanbun (1661–1673)", "Enpō (1673–1681)", "Tenna (1681–1684)", "Jōkyō (1684–1688)", "Genroku (1688–1704)", "Hōei (1704–1711)", "Shōtoku (1711–1716)", "Kyōhō (1716–1736)", "Genbun (1736–1741)", "Kanpō (1741–1744)", "Enkyō (1744–1748)", "Kan’en (1748–1751)", "Hōreki (1751–1764)", "Meiwa (1764–1772)", "An’ei (1772–1781)", "Tenmei (1781–1789)", "Kansei (1789–1801)", "Kyōwa (1801–1804)", "Bunka (1804–1818)", "Bunsei (1818–1830)", "Tenpō (1830–1844)", "Kōka (1844–1848)", "Kaei (1848–1854)", "Ansei (1854–1860)", "Man’en (1860–1861)", "Bunkyū (1861–1864)", "Genji (1864–1865)", "Keiō (1865–1868)", "Meiji", "Taishō", "Shōwa", "Heisei" ], _didi153 = {
    dayParts: _didi127,
    days: _didi149,
    months: _didi149,
    years: _didi149,
    zodiacs: _didi127
}, _didi154 = {
    dateFormats: _didi146,
    datePartsOrder: "mdy",
    dateIntervalFormats: _didi147,
    dateIntervalFallbackFormat: "{0} - {1}",
    dayNames: _didi136,
    dayPeriods: _didi143,
    monthNames: _didi139,
    eraNames: _didi14,
    quarterNames: _didi130,
    fields: _didi150
}, _didi155 = {
    dateFormats: _didi144,
    datePartsOrder: "md",
    dateIntervalFormats: _didi145,
    dateIntervalFallbackFormat: "{0} – {1}",
    dayNames: _didi136,
    dayPeriods: _didi143,
    cyclicNames: _didi153,
    monthNames: _didi140,
    monthPatterns: _didi118,
    quarterNames: _didi130,
    fields: _didi150
};

inter.id = "ee_tg";

inter.calendars = {
    buddhist: {
        dateFormats: {
            basic: _didi135,
            cldr: {
                h: "a 'ga' h",
                hm: "a 'ga' h: 'aɖabaƒoƒo' mm",
                hms: "a 'ga' h: 'aɖabaƒoƒo' mm:ss",
                ms: "'ga' mm:ss",
                d: "d",
                Ed: "E d",
                Gy: "y G",
                GyMMM: "MMM y G",
                GyMMMd: "MMM d 'lia', y G",
                GyMMMEd: "E, MMM d 'lia' y G",
                H: "HH",
                Hm: "HH:mm",
                Hms: "HH:mm:ss",
                M: "L",
                Md: "M/d",
                MEd: "E, M/d",
                MMM: "LLL",
                MMMd: "MMM d 'lia'",
                MMMEd: "E, MMM d 'lia'",
                MMMMd: "MMMM d 'lia'",
                MMMMEd: "E, MMMM d 'lia'",
                y: "y G",
                yyyy: "y G",
                yyyyM: "M/y G",
                yyyyMd: "M/d/y G",
                yyyyMEd: "E, M/d/y G",
                yyyyMMM: "MMM y G",
                yyyyMMMd: "MMM d 'lia', y G",
                yyyyMMMEd: "E, MMM d 'lia' y G",
                yyyyMMMM: "MMMM y G",
                yyyyQQQ: "QQQ y G",
                yyyyQQQQ: "QQQQ y G",
                E: "ccc",
                Ehm: "E h:mm a",
                EHm: "E HH:mm",
                Ehms: "E h:mm:ss a",
                EHms: "E HH:mm:ss"
            }
        },
        datePartsOrder: "mdy",
        dateIntervalFormats: _didi147,
        dateIntervalFallbackFormat: "{0} - {1}",
        dayNames: _didi136,
        dayPeriods: _didi143,
        monthNames: _didi138,
        eraNames: {
            wide: [ "BE" ],
            abbreviated: [ "BE" ]
        },
        quarterNames: _didi130,
        fields: _didi150
    },
    chinese: _didi155,
    coptic: {
        dateFormats: _didi146,
        datePartsOrder: "mdy",
        dateIntervalFormats: _didi147,
        dateIntervalFallbackFormat: "{0} - {1}",
        dayNames: _didi136,
        dayPeriods: _didi143,
        monthNames: {
            format: _didi131,
            standAlone: _didi131
        },
        eraNames: _didi36,
        quarterNames: _didi130,
        fields: _didi150
    },
    dangi: _didi155,
    ethioaa: _didi151,
    ethiopic: {
        dateFormats: _didi146,
        datePartsOrder: "mdy",
        dateIntervalFormats: _didi147,
        dateIntervalFallbackFormat: "{0} - {1}",
        dayNames: _didi136,
        dayPeriods: _didi143,
        monthNames: {
            format: _didi132,
            standAlone: {
                abbreviated: [ "dzv", "dzd", "ted", "afɔ", "dam", "mas", "sia", "dea", "any", "kel", "ade", "dzm", "ƒoa" ],
                narrow: _didi55,
                wide: [ "dzove", "dzodze", "tedoxe", "afɔfie", "damɛ", "masa", "siamlɔm", "deasiamime", "anyɔnyɔ", "kele", "adeɛmekpɔxe", "dzome", "ƒoave" ]
            }
        },
        eraNames: _didi36,
        quarterNames: _didi130,
        fields: _didi150
    },
    gregorian: {
        dateFormats: {
            basic: {
                fullDate: "EEEE, MMMM d 'lia' y",
                longDate: "MMMM d 'lia' y",
                mediumDate: "MMM d 'lia', y",
                shortDate: "M/d/yy",
                fullTime: "HH:mm:ss zzzz",
                longTime: "HH:mm:ss z",
                mediumTime: "HH:mm:ss",
                shortTime: "HH:mm",
                shortDateTime: "HH:mm M/d/yy",
                mediumDateTime: "HH:mm:ss MMM d 'lia', y",
                longDateTime: "HH:mm:ss z MMMM d 'lia' y",
                fullDateTime: "HH:mm:ss zzzz EEEE, MMMM d 'lia' y"
            },
            cldr: {
                d: "d",
                Ed: "E d",
                Ehm: "E a 'ga' h:mm",
                EHm: "E HH:mm",
                Ehms: "E a 'ga' h:mm:ss",
                EHms: "E HH:mm:ss",
                Gy: "y G",
                GyMMM: "MMM y G",
                GyMMMd: "MMM d 'lia', y G",
                GyMMMEd: "E, MMM d 'lia' y G",
                h: "a 'ga' h",
                H: "HH",
                hm: "a 'ga' h:mm",
                Hm: "HH:mm",
                hms: "a 'ga' h:mm:ss",
                Hms: "HH:mm:ss",
                M: "L",
                Md: "M/d",
                MEd: "E, M/d",
                MMM: "LLL",
                MMMd: "MMM d 'lia'",
                MMMEd: "E, MMM d 'lia'",
                MMMMd: "MMMM d 'lia'",
                MMMMEd: "E, MMMM d 'lia'",
                ms: "'aɖabaƒoƒo' mm:ss",
                y: "y",
                yM: "M/y",
                yMd: "M/d/y",
                yMEd: "E, M/d/y",
                yMMM: "MMM y",
                yMMMd: "MMM d 'lia', y",
                yMMMEd: "E, MMM d, y",
                yMMMM: "MMMM y",
                yQQQ: "QQQ y",
                yQQQQ: "QQQQ y",
                E: "ccc",
                hmsv: "h:mm:ss a v",
                Hmsv: "HH:mm:ss v",
                hmv: "h:mm a v",
                Hmv: "HH:mm v"
            }
        },
        datePartsOrder: "mdy",
        dateIntervalFormats: {
            d: {
                d: "d–d"
            },
            h: _didi5,
            H: {
                H: "HH–HH"
            },
            hm: _didi90,
            Hm: _didi15,
            hmv: _didi81,
            Hmv: _didi20,
            hv: _didi19,
            Hv: {
                H: "HH–HH v"
            },
            M: _didi24,
            Md: _didi9,
            MEd: _didi25,
            MMM: {
                M: "MMM–MMM"
            },
            MMMd: _didi48,
            MMMEd: _didi79,
            y: {
                y: "y–y"
            },
            yM: {
                M: "M/y – M/y",
                y: "M/y – M/y"
            },
            yMd: {
                d: "M/d/y – M/d/y",
                M: "M/d/y – M/d/y",
                y: "M/d/y – M/d/y"
            },
            yMEd: {
                d: "E, M/d/y – E, M/d/y",
                M: "E, M/d/y – E, M/d/y",
                y: "E, M/d/y – E, M/d/y"
            },
            yMMM: {
                M: "MMM–MMM y",
                y: "MMM y – MMM y"
            },
            yMMMd: {
                d: "MMM d 'lia' – d 'lia' , y",
                M: "MMM d 'lia' – MMM d 'lia', y",
                y: "MMM d 'lia' , y – MMM d 'lia', y"
            },
            yMMMEd: {
                d: "E, MMM d 'lia' – E, MMM d 'lia', y",
                M: "E, MMM d 'lia' – E, MMM d 'lia', y",
                y: "E, MMM d 'lia', y – E, MMM d 'lia', y"
            },
            yMMMM: {
                M: "MMMM–MMMM y",
                y: "MMMM y – MMMM y"
            }
        },
        dateIntervalFallbackFormat: "{0} - {1}",
        dayNames: _didi136,
        dayPeriods: _didi143,
        monthNames: _didi138,
        eraNames: {
            wide: [ "Hafi Yesu Va Do ŋgɔ", "Yesu Ŋɔli" ],
            abbreviated: [ "hY", "Yŋ" ]
        },
        quarterNames: _didi130,
        fields: _didi150
    },
    hebrew: {
        dateFormats: _didi146,
        datePartsOrder: "mdy",
        dateIntervalFormats: _didi147,
        dateIntervalFallbackFormat: "{0} - {1}",
        dayNames: _didi136,
        dayPeriods: _didi143,
        monthNames: {
            format: _didi132,
            standAlone: {
                abbreviated: _didi86,
                narrow: _didi55,
                wide: [ "dzove", "dzodze", "tedoxe", "afɔfiẽ", "damɛ", "masa", "siamlɔm", "dasiamime", "anyɔnyɔ", "kele", "adeɛmekpɔxe", "dzome", "ƒoave" ]
            }
        },
        eraNames: {
            wide: [ "ŋdi" ],
            abbreviated: [ "ŋdi" ]
        },
        quarterNames: _didi130,
        fields: _didi150
    },
    indian: {
        dateFormats: _didi146,
        datePartsOrder: "mdy",
        dateIntervalFormats: _didi147,
        dateIntervalFallbackFormat: "{0} - {1}",
        dayNames: _didi136,
        dayPeriods: _didi143,
        monthNames: {
            format: {
                abbreviated: _didi80,
                narrow: _didi40,
                wide: [ "dzove", "dzodze", "tedoxe", "afɔ", "damɛ", "masa", "siamlɔm", "deasiamime", "anyɔnyɔ", "kele", "adeɛmekpɔxe", "dzome" ]
            },
            standAlone: _didi128
        },
        eraNames: {
            wide: [ "Saka" ],
            abbreviated: [ "Saka" ]
        },
        quarterNames: _didi130,
        fields: _didi150
    },
    islamic: _didi154,
    "islamic-umalqura": _didi154,
    "islamic-tbla": _didi154,
    "islamic-civil": _didi154,
    "islamic-rgsa": _didi154,
    iso8601: _didi151,
    japanese: {
        dateFormats: {
            basic: {
                fullDate: "EEEE, MMMM dd 'lia', G y",
                longDate: "MMMM d 'lia', G y",
                mediumDate: "MMM d 'lia', G y",
                shortDate: "dd-MM-GGGGG yy",
                fullTime: "HH:mm:ss zzzz",
                longTime: "HH:mm:ss z",
                mediumTime: "HH:mm:ss",
                shortTime: "HH:mm",
                shortDateTime: "HH:mm dd-MM-GGGGG yy",
                mediumDateTime: "HH:mm:ss MMM d 'lia', G y",
                longDateTime: "HH:mm:ss z MMMM d 'lia', G y",
                fullDateTime: "HH:mm:ss zzzz EEEE, MMMM dd 'lia', G y"
            },
            cldr: {
                hms: "a 'ga' h:mm:ss",
                MMMEd: "E MMM d 'lia'",
                ms: "'aɖabaƒoƒo' mm:ss",
                d: "d",
                Ed: "E d",
                Gy: "y G",
                GyMMM: "MMM y G",
                GyMMMd: "MMM d 'lia', y G",
                GyMMMEd: "E, MMM d 'lia' y G",
                h: "a 'ga' h",
                H: "HH",
                hm: "a 'ga' h:mm",
                Hm: "HH:mm",
                Hms: "HH:mm:ss",
                M: "L",
                Md: "M/d",
                MEd: "E, M/d",
                MMM: "LLL",
                MMMd: "MMM d 'lia'",
                MMMMd: "MMMM d 'lia'",
                MMMMEd: "E, MMMM d 'lia'",
                y: "y G",
                yyyy: "y G",
                yyyyM: "M/y G",
                yyyyMd: "M/d/y G",
                yyyyMEd: "E, M/d/y G",
                yyyyMMM: "MMM y G",
                yyyyMMMd: "MMM d 'lia', y G",
                yyyyMMMEd: "E, MMM d 'lia' y G",
                yyyyMMMM: "MMMM y G",
                yyyyQQQ: "QQQ y G",
                yyyyQQQQ: "QQQQ y G",
                E: "ccc",
                Ehm: "E h:mm a",
                EHm: "E HH:mm",
                Ehms: "E h:mm:ss a",
                EHms: "E HH:mm:ss"
            }
        },
        datePartsOrder: "dmy",
        dateIntervalFormats: _didi147,
        dateIntervalFallbackFormat: "{0} - {1}",
        dayNames: _didi136,
        dayPeriods: _didi143,
        monthNames: _didi138,
        eraNames: {
            wide: _didi152,
            abbreviated: _didi152
        },
        quarterNames: _didi130,
        fields: _didi150
    },
    persian: {
        dateFormats: _didi146,
        datePartsOrder: "mdy",
        dateIntervalFormats: _didi147,
        dateIntervalFallbackFormat: "{0} - {1}",
        dayNames: _didi136,
        dayPeriods: _didi143,
        monthNames: _didi139,
        eraNames: {
            wide: [ "AP" ],
            abbreviated: [ "AP" ]
        },
        quarterNames: _didi130,
        fields: _didi150
    },
    roc: {
        dateFormats: {
            basic: {
                fullDate: "EEEE, MMMM dd 'lia', G y",
                longDate: "MMMM d 'lia', G y",
                mediumDate: "MMM d 'lia', G y",
                shortDate: "dd-MM-GGGGG y",
                fullTime: "HH:mm:ss zzzz",
                longTime: "HH:mm:ss z",
                mediumTime: "HH:mm:ss",
                shortTime: "HH:mm",
                shortDateTime: "HH:mm dd-MM-GGGGG y",
                mediumDateTime: "HH:mm:ss MMM d 'lia', G y",
                longDateTime: "HH:mm:ss z MMMM d 'lia', G y",
                fullDateTime: "HH:mm:ss zzzz EEEE, MMMM dd 'lia', G y"
            },
            cldr: {
                h: "a 'ga' h",
                hm: "a 'ga' h:mm",
                hms: "a 'ga' h:mm:ss",
                MMMEd: "E MMM d 'lia'",
                ms: "'aɖabaƒoƒo' mm:ss",
                d: "d",
                Ed: "E d",
                Gy: "y G",
                GyMMM: "MMM y G",
                GyMMMd: "MMM d 'lia', y G",
                GyMMMEd: "E, MMM d 'lia' y G",
                H: "HH",
                Hm: "HH:mm",
                Hms: "HH:mm:ss",
                M: "L",
                Md: "M/d",
                MEd: "E, M/d",
                MMM: "LLL",
                MMMd: "MMM d 'lia'",
                MMMMd: "MMMM d 'lia'",
                MMMMEd: "E, MMMM d 'lia'",
                y: "y G",
                yyyy: "y G",
                yyyyM: "M/y G",
                yyyyMd: "M/d/y G",
                yyyyMEd: "E, M/d/y G",
                yyyyMMM: "MMM y G",
                yyyyMMMd: "MMM d 'lia', y G",
                yyyyMMMEd: "E, MMM d 'lia' y G",
                yyyyMMMM: "MMMM y G",
                yyyyQQQ: "QQQ y G",
                yyyyQQQQ: "QQQQ y G",
                E: "ccc",
                Ehm: "E h:mm a",
                EHm: "E HH:mm",
                Ehms: "E h:mm:ss a",
                EHms: "E HH:mm:ss"
            }
        },
        datePartsOrder: "dmy",
        dateIntervalFormats: _didi147,
        dateIntervalFallbackFormat: "{0} - {1}",
        dayNames: _didi136,
        dayPeriods: _didi143,
        monthNames: _didi138,
        eraNames: {
            wide: _didi4,
            abbreviated: _didi4
        },
        quarterNames: _didi130,
        fields: _didi150
    },
    islamicc: _didi151
};

inter.pluralRule = function(n) {
    if (typeof n === "string") n = parseInt(n, 10);
    if (n === 1) return "one";
    return "other";
};

inter.renderDigitsOrdinal = function(n) {
    if (n < 0) return "−" + this.renderDigitsOrdinal(-n);
    if (n >= 2) return this.renderNumber(n, "#,##0") + " lia";
    if (n >= 1) return this.renderNumber(n, "#,##0") + " tɔ";
    if (n >= 0) return this.renderNumber(n, "#,##0") + " lia";
};

inter.renderSpelloutNumbering = function(n) {
    if (n < 0) return "−" + this.renderSpelloutNumbering(-n);
    if (n >= 0) return this.renderSpelloutCardinal(n);
};

inter.renderArmenianUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderArmenianUpper(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e4) return this.renderNumber(n, "#,##0");
    if (n >= 9e3) return "Ք" + (n === 9e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 8e3) return "Փ" + (n === 8e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 7e3) return "Ւ" + (n === 7e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 6e3) return "Ց" + (n === 6e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 5e3) return "Ր" + (n === 5e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 4e3) return "Տ" + (n === 4e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 3e3) return "Վ" + (n === 3e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 2e3) return "Ս" + (n === 2e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 1e3) return "Ռ" + (n === 1e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 900) return "Ջ" + (n === 900 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 800) return "Պ" + (n === 800 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 700) return "Չ" + (n === 700 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 600) return "Ո" + (n === 600 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 500) return "Շ" + (n === 500 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 400) return "Ն" + (n === 400 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 300) return "Յ" + (n === 300 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 200) return "Մ" + (n === 200 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 100) return "Ճ" + (n === 100 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 90) return "Ղ" + (n === 90 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 80) return "Ձ" + (n === 80 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 70) return "Հ" + (n === 70 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 60) return "Կ" + (n === 60 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 50) return "Ծ" + (n === 50 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 40) return "Խ" + (n === 40 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 30) return "Լ" + (n === 30 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 20) return "Ի" + (n === 20 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 10) return "Ժ" + (n === 10 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 9) return "Թ";
    if (n >= 8) return "Ը";
    if (n >= 7) return "Է";
    if (n >= 6) return "Զ";
    if (n >= 5) return "Ե";
    if (n >= 4) return "Դ";
    if (n >= 3) return "Գ";
    if (n >= 2) return "Բ";
    if (n >= 1) return "Ա";
    if (n >= 0) return "0";
};

inter.renderArmenianLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderArmenianLower(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e4) return this.renderNumber(n, "#,##0");
    if (n >= 9e3) return "ք" + (n === 9e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 8e3) return "փ" + (n === 8e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 7e3) return "ւ" + (n === 7e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 6e3) return "ց" + (n === 6e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 5e3) return "ր" + (n === 5e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 4e3) return "տ" + (n === 4e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 3e3) return "վ" + (n === 3e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 2e3) return "ս" + (n === 2e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 1e3) return "ռ" + (n === 1e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 900) return "ջ" + (n === 900 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 800) return "պ" + (n === 800 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 700) return "չ" + (n === 700 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 600) return "ո" + (n === 600 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 500) return "շ" + (n === 500 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 400) return "ն" + (n === 400 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 300) return "յ" + (n === 300 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 200) return "մ" + (n === 200 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 100) return "ճ" + (n === 100 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 90) return "ղ" + (n === 90 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 80) return "ձ" + (n === 80 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 70) return "հ" + (n === 70 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 60) return "կ" + (n === 60 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 50) return "ծ" + (n === 50 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 40) return "խ" + (n === 40 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 30) return "լ" + (n === 30 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 20) return "ի" + (n === 20 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 10) return "ժ" + (n === 10 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 9) return "թ";
    if (n >= 8) return "ը";
    if (n >= 7) return "է";
    if (n >= 6) return "զ";
    if (n >= 5) return "ե";
    if (n >= 4) return "դ";
    if (n >= 3) return "գ";
    if (n >= 2) return "բ";
    if (n >= 1) return "ա";
    if (n >= 0) return "0";
};

inter.renderCyrillicLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderCyrillicLower(-n);
    if (isFractional && n > 1) return this.renderCyrillicLower(Math.floor(n)) + "." + this.renderCyrillicLower(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e15) return "҂҂҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e15)) + (n === 1e15 ? "" : " " + this.renderCyrillicLower(n % 1e15));
    if (n >= 1e12) return "҂҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e12)) + (n === 1e12 ? "" : " " + this.renderCyrillicLower(n % 1e12));
    if (n >= 1e9) return "҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e9)) + (n === 1e9 ? "" : " " + this.renderCyrillicLower(n % 1e9));
    if (n >= 1e6) return "҂҂" + this.renderCyrillicLower(Math.floor(n / 1e6)) + (n === 1e6 ? "" : " " + this.renderCyrillicLower(n % 1e6));
    if (n >= 11e3) return this.renderCyrillicLowerThousands(Math.floor(n / 1e4)) + (n === 11e3 ? "" : " " + this.renderCyrillicLower(n % 1e4));
    if (n >= 1e4) return "҂" + this.renderCyrillicLower(Math.floor(n / 1e4)) + (n === 1e4 ? "" : " " + this.renderCyrillicLower(n % 1e4));
    if (n >= 1e3) return "҂" + this.renderCyrillicLower110(Math.floor(n / 1e3)) + this.renderCyrillicLowerPost(n % 1e3);
    if (n >= 900) return "ц" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 801) return "ѿ" + this.renderCyrillicLower(n % 100);
    if (n >= 800) return "ѿ҃";
    if (n >= 700) return "ѱ" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 600) return "х" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 500) return "ф" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 400) return "у" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 300) return "т" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 200) return "с" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 100) return "р" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 90) return "ч" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 80) return "п" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 70) return "ѻ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 60) return "ѯ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 50) return "н" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 40) return "м" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 30) return "л" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 20) return "к" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 19) return "ѳ҃і";
    if (n >= 18) return "и҃і";
    if (n >= 17) return "з҃і";
    if (n >= 16) return "ѕ҃і";
    if (n >= 15) return "є҃і";
    if (n >= 14) return "д҃і";
    if (n >= 13) return "г҃і";
    if (n >= 12) return "в҃і";
    if (n >= 11) return "а҃і";
    if (n >= 1) return this.renderCyrillicLower110(n) + "҃";
    if (n >= 0) return "0҃";
};

inter.renderEthiopic = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderEthiopic(-n);
    if (isFractional && n > 1) return this.renderEthiopic(Math.floor(n)) + "፡" + this.renderEthiopic(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 1e18) return this.renderNumber(n, "#,###0");
    if (n >= 2e16) return this.renderEthiopic(Math.floor(n / 1e16)) + "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 1e16) return "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 2e12) return this.renderEthiopic(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1e12) return "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 2e8) return this.renderEthiopic(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1e8) return "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 2e4) return this.renderEthiopic(Math.floor(n / 1e4)) + "፼" + (n === 2e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 1e4) return "፼" + (n === 1e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 200) return this.renderEthiopic(Math.floor(n / 100)) + "፻" + (n === 200 ? "" : this.renderEthiopic(n % 100));
    if (n >= 100) return "፻" + (n === 100 ? "" : this.renderEthiopic(n % 100));
    if (n >= 90) return "፺" + (n === 90 ? "" : this.renderEthiopic(n % 10));
    if (n >= 80) return "፹" + (n === 80 ? "" : this.renderEthiopic(n % 10));
    if (n >= 70) return "፸" + (n === 70 ? "" : this.renderEthiopic(n % 10));
    if (n >= 60) return "፷" + (n === 60 ? "" : this.renderEthiopic(n % 10));
    if (n >= 50) return "፶" + (n === 50 ? "" : this.renderEthiopic(n % 10));
    if (n >= 40) return "፵" + (n === 40 ? "" : this.renderEthiopic(n % 10));
    if (n >= 30) return "፴" + (n === 30 ? "" : this.renderEthiopic(n % 10));
    if (n >= 20) return "፳" + (n === 20 ? "" : this.renderEthiopic(n % 10));
    if (n >= 10) return "፲" + (n === 10 ? "" : this.renderEthiopic(n % 10));
    if (n >= 9) return "፱";
    if (n >= 8) return "፰";
    if (n >= 7) return "፯";
    if (n >= 6) return "፮";
    if (n >= 5) return "፭";
    if (n >= 4) return "፬";
    if (n >= 3) return "፫";
    if (n >= 2) return "፪";
    if (n >= 1) return "፩";
    if (n >= 0) return "ባዶ";
};

inter.renderGeorgian = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGeorgian(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 2e4) return this.renderNumber(n, "#,##0");
    if (n >= 1e4) return "ჯ" + (n === 1e4 ? "" : this.renderGeorgian(n % 1e4));
    if (n >= 9e3) return "ჵ" + (n === 9e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 8e3) return "ჴ" + (n === 8e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 7e3) return "ხ" + (n === 7e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 6e3) return "ჭ" + (n === 6e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 5e3) return "წ" + (n === 5e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 4e3) return "ძ" + (n === 4e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 3e3) return "ც" + (n === 3e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 2e3) return "ჩ" + (n === 2e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 1e3) return "შ" + (n === 1e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 900) return "ყ" + (n === 900 ? "" : this.renderGeorgian(n % 100));
    if (n >= 800) return "ღ" + (n === 800 ? "" : this.renderGeorgian(n % 100));
    if (n >= 700) return "ქ" + (n === 700 ? "" : this.renderGeorgian(n % 100));
    if (n >= 600) return "ფ" + (n === 600 ? "" : this.renderGeorgian(n % 100));
    if (n >= 500) return "ჳ" + (n === 500 ? "" : this.renderGeorgian(n % 100));
    if (n >= 400) return "უ" + (n === 400 ? "" : this.renderGeorgian(n % 100));
    if (n >= 300) return "ტ" + (n === 300 ? "" : this.renderGeorgian(n % 100));
    if (n >= 200) return "ს" + (n === 200 ? "" : this.renderGeorgian(n % 100));
    if (n >= 100) return "რ" + (n === 100 ? "" : this.renderGeorgian(n % 100));
    if (n >= 90) return "ჟ" + (n === 90 ? "" : this.renderGeorgian(n % 10));
    if (n >= 80) return "პ" + (n === 80 ? "" : this.renderGeorgian(n % 10));
    if (n >= 70) return "ო" + (n === 70 ? "" : this.renderGeorgian(n % 10));
    if (n >= 60) return "ჲ" + (n === 60 ? "" : this.renderGeorgian(n % 10));
    if (n >= 50) return "ნ" + (n === 50 ? "" : this.renderGeorgian(n % 10));
    if (n >= 40) return "მ" + (n === 40 ? "" : this.renderGeorgian(n % 10));
    if (n >= 30) return "ლ" + (n === 30 ? "" : this.renderGeorgian(n % 10));
    if (n >= 20) return "კ" + (n === 20 ? "" : this.renderGeorgian(n % 10));
    if (n >= 10) return "ი" + (n === 10 ? "" : this.renderGeorgian(n % 10));
    if (n >= 9) return "თ";
    if (n >= 8) return "ჱ";
    if (n >= 7) return "ზ";
    if (n >= 6) return "ვ";
    if (n >= 5) return "ე";
    if (n >= 4) return "დ";
    if (n >= 3) return "გ";
    if (n >= 2) return "ბ";
    if (n >= 1) return "ა";
    if (n >= 0) return this.renderNumber(n, "#,##0");
};

inter.renderGreekUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGreekUpper(-n);
    if (isFractional && n > 1) return this.renderGreekUpper(Math.floor(n)) + "." + this.renderGreekUpper(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 0) return this.renderGreekNumeralMajuscules(n) + "´";
};

inter.renderGreekLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGreekLower(-n);
    if (isFractional && n > 1) return this.renderGreekLower(Math.floor(n)) + "." + this.renderGreekLower(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 0) return this.renderGreekNumeralMinuscules(n) + "´";
};

inter.renderSpelloutCardinal = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return this.renderSpelloutCardinal(-n) + " xlẽyimegbee";
    if (isFractional && n > 1) return this.renderSpelloutCardinal(Math.floor(n)) + " kple kakɛ " + this.renderSpelloutCardinal(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n > 0 && n < 1) return "kakɛ " + this.renderSpelloutCardinal(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 2) return this.renderSpelloutBase(n);
    if (n >= 1) return "ɖeka";
    if (n >= 0) return "ɖekeo";
};

inter.renderHebrew = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderHebrew(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1000001) return this.renderNumber(n, "#,##0");
    if (n >= 1e6) return "אלף אלפים";
    if (n >= 3001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 3001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 3e3) return this.renderHebrew(Math.floor(n / 1e3)) + " אלפים";
    if (n >= 2001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 2001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 2e3) return "אלפיים";
    if (n >= 1001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 1001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 1e3) return "אלף";
    if (n >= 901) return "תתק" + this.renderHebrew099(n % 100);
    if (n >= 900) return "תת״ק";
    if (n >= 801) return "תת" + this.renderHebrew099(n % 100);
    if (n >= 800) return "ת״ת";
    if (n >= 745) return "תש" + this.renderHebrew099(n % 100);
    if (n >= 744) return "תשד״מ";
    if (n >= 701) return "תש" + this.renderHebrew099(n % 100);
    if (n >= 700) return "ת״ש";
    if (n >= 699) return "תר" + this.renderHebrew099(n % 100);
    if (n >= 698) return "תרח״צ";
    if (n >= 601) return "תר" + this.renderHebrew099(n % 100);
    if (n >= 600) return "ת״ר";
    if (n >= 501) return "תק" + this.renderHebrew099(n % 100);
    if (n >= 500) return "ת״ק";
    if (n >= 400) return "ת" + this.renderHebrew099(n % 100);
    if (n >= 345) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 344) return "שד״מ";
    if (n >= 305) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 304) return "ד״ש";
    if (n >= 300) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 299) return "ר" + this.renderHebrew099(n % 100);
    if (n >= 298) return "רח״צ";
    if (n >= 200) return "ר" + this.renderHebrew099(n % 100);
    if (n >= 100) return "ק" + this.renderHebrew099(n % 100);
    if (n >= 91) return "צ״" + this.renderHebrewItem(n % 10);
    if (n >= 90) return "צ׳";
    if (n >= 81) return "פ״" + this.renderHebrewItem(n % 10);
    if (n >= 80) return "פ׳";
    if (n >= 71) return "ע״" + this.renderHebrewItem(n % 10);
    if (n >= 70) return "ע׳";
    if (n >= 61) return "ס״" + this.renderHebrewItem(n % 10);
    if (n >= 60) return "ס׳";
    if (n >= 51) return "נ״" + this.renderHebrewItem(n % 10);
    if (n >= 50) return "נ׳";
    if (n >= 41) return "מ״" + this.renderHebrewItem(n % 10);
    if (n >= 40) return "מ׳";
    if (n >= 31) return "ל״" + this.renderHebrewItem(n % 10);
    if (n >= 30) return "ל׳";
    if (n >= 21) return "כ״" + this.renderHebrewItem(n % 10);
    if (n >= 20) return "כ׳";
    if (n >= 17) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "ט״ז";
    if (n >= 15) return "ט״ו";
    if (n >= 11) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 0) return this.renderHebrewItem(n) + "׳";
};

inter.renderRomanUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderRomanUpper(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 4e5) return this.renderNumber(n, "#,##0");
    if (n >= 3e5) return "ↈↈↈ" + (n === 3e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 2e5) return "ↈↈ" + (n === 2e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 1e5) return "ↈ" + (n === 1e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 9e4) return "ↂↈ" + (n === 9e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 8e4) return "ↇↂↂↂ" + (n === 8e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 7e4) return "ↇↂↂ" + (n === 7e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 6e4) return "ↇↂ" + (n === 6e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 5e4) return "ↇ" + (n === 5e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 4e4) return "ↂↇ" + (n === 4e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 3e4) return "ↂↂↂ" + (n === 3e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 2e4) return "ↂↂ" + (n === 2e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 1e4) return "ↂ" + (n === 1e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 9e3) return "Mↂ" + (n === 9e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 8e3) return "ↁMMM" + (n === 8e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 7e3) return "ↁMM" + (n === 7e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 6e3) return "ↁM" + (n === 6e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 5e3) return "ↁ" + (n === 5e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 4e3) return "Mↁ" + (n === 4e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 3e3) return "MMM" + (n === 3e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 2e3) return "MM" + (n === 2e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 1e3) return "M" + (n === 1e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 900) return "CM" + (n === 900 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 800) return "DCCC" + (n === 800 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 700) return "DCC" + (n === 700 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 600) return "DC" + (n === 600 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 500) return "D" + (n === 500 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 400) return "CD" + (n === 400 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 300) return "CCC" + (n === 300 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 200) return "CC" + (n === 200 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 100) return "C" + (n === 100 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 90) return "XC" + (n === 90 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 80) return "LXXX" + (n === 80 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 70) return "LXX" + (n === 70 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 60) return "LX" + (n === 60 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 50) return "L" + (n === 50 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 40) return "XL" + (n === 40 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 30) return "XXX" + (n === 30 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 20) return "XX" + (n === 20 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 10) return "X" + (n === 10 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 9) return "IX";
    if (n >= 8) return "VIII";
    if (n >= 7) return "VII";
    if (n >= 6) return "VI";
    if (n >= 5) return "V";
    if (n >= 4) return "IV";
    if (n >= 3) return "III";
    if (n >= 2) return "II";
    if (n >= 1) return "I";
    if (n >= 0) return "N";
};

inter.renderRomanLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderRomanLower(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 5e3) return this.renderNumber(n, "#,##0");
    if (n >= 4e3) return "mmmm" + (n === 4e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 3e3) return "mmm" + (n === 3e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 2e3) return "mm" + (n === 2e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 1e3) return "m" + (n === 1e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 900) return "cm" + (n === 900 ? "" : this.renderRomanLower(n % 100));
    if (n >= 800) return "dccc" + (n === 800 ? "" : this.renderRomanLower(n % 100));
    if (n >= 700) return "dcc" + (n === 700 ? "" : this.renderRomanLower(n % 100));
    if (n >= 600) return "dc" + (n === 600 ? "" : this.renderRomanLower(n % 100));
    if (n >= 500) return "d" + (n === 500 ? "" : this.renderRomanLower(n % 100));
    if (n >= 400) return "cd" + (n === 400 ? "" : this.renderRomanLower(n % 100));
    if (n >= 300) return "ccc" + (n === 300 ? "" : this.renderRomanLower(n % 100));
    if (n >= 200) return "cc" + (n === 200 ? "" : this.renderRomanLower(n % 100));
    if (n >= 100) return "c" + (n === 100 ? "" : this.renderRomanLower(n % 100));
    if (n >= 90) return "xc" + (n === 90 ? "" : this.renderRomanLower(n % 10));
    if (n >= 80) return "lxxx" + (n === 80 ? "" : this.renderRomanLower(n % 10));
    if (n >= 70) return "lxx" + (n === 70 ? "" : this.renderRomanLower(n % 10));
    if (n >= 60) return "lx" + (n === 60 ? "" : this.renderRomanLower(n % 10));
    if (n >= 50) return "l" + (n === 50 ? "" : this.renderRomanLower(n % 10));
    if (n >= 40) return "xl" + (n === 40 ? "" : this.renderRomanLower(n % 10));
    if (n >= 30) return "xxx" + (n === 30 ? "" : this.renderRomanLower(n % 10));
    if (n >= 20) return "xx" + (n === 20 ? "" : this.renderRomanLower(n % 10));
    if (n >= 10) return "x" + (n === 10 ? "" : this.renderRomanLower(n % 10));
    if (n >= 9) return "ix";
    if (n >= 8) return "viii";
    if (n >= 7) return "vii";
    if (n >= 6) return "vi";
    if (n >= 5) return "v";
    if (n >= 4) return "iv";
    if (n >= 3) return "iii";
    if (n >= 2) return "ii";
    if (n >= 1) return "i";
    if (n >= 0) return "n";
};

inter.renderTamil = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderTamil(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e8) return this.renderNumber(n, "#,##,##0");
    if (n >= 1e6) return this.renderTamil(Math.floor(n / 1e6)) + "௱௲" + (n === 1e6 ? "" : this.renderTamilThousands(n % 1e6));
    if (n >= 2e3) return this.renderTamil(Math.floor(n / 1e3)) + "௲" + (n === 2e3 ? "" : this.renderTamil(n % 1e3));
    if (n >= 1e3) return "௲" + (n === 1e3 ? "" : this.renderTamil(n % 1e3));
    if (n >= 200) return this.renderTamil(Math.floor(n / 100)) + "௱" + (n === 200 ? "" : this.renderTamil(n % 100));
    if (n >= 100) return "௱" + (n === 100 ? "" : this.renderTamil(n % 100));
    if (n >= 20) return this.renderTamil(Math.floor(n / 10)) + "௰" + (n === 20 ? "" : this.renderTamil(n % 10));
    if (n >= 10) return "௰" + (n === 10 ? "" : this.renderTamil(n % 10));
    if (n >= 9) return "௯";
    if (n >= 8) return "௮";
    if (n >= 7) return "௭";
    if (n >= 6) return "௬";
    if (n >= 5) return "௫";
    if (n >= 4) return "௪";
    if (n >= 3) return "௩";
    if (n >= 2) return "௨";
    if (n >= 1) return "௧";
    if (n >= 0) return "௦";
};

inter.renderCyrillicLowerThousands = function(n) {
    if (n >= 900) return "҂ц" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 800) return "҂ѿ" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 700) return "҂ѱ" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 600) return "҂х" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 500) return "҂ф" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 400) return "҂у" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 300) return "҂т" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 200) return "҂с" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 100) return "҂р" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 90) return "҂ч" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 80) return "҂п" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 70) return "҂ѻ" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 60) return "҂ѯ" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 50) return "҂н" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 40) return "҂м" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 30) return "҂л" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 20) return "҂к" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 19) return "҂ѳ҃҂і";
    if (n >= 18) return "҂и҃҂і";
    if (n >= 17) return "҂з҃҂і";
    if (n >= 16) return "҂ѕ҃҂і";
    if (n >= 15) return "҂є҃҂і";
    if (n >= 14) return "҂д҃҂і";
    if (n >= 13) return "҂г҃҂і";
    if (n >= 12) return "҂в҃҂і";
    if (n >= 11) return "҂а҃҂і";
    if (n >= 10) return "҃҂і";
    if (n >= 9) return "҃҂ѳ";
    if (n >= 8) return "҃҂и";
    if (n >= 7) return "҃҂з";
    if (n >= 6) return "҃҂ѕ";
    if (n >= 5) return "҃҂є";
    if (n >= 4) return "҃҂д";
    if (n >= 3) return "҃҂г";
    if (n >= 2) return "҃҂в";
    if (n >= 1) return "҃҂а";
    if (n >= 0) return "҃";
};

inter.renderCyrillicLower110 = function(n) {
    if (n >= 10) return "і";
    if (n >= 9) return "ѳ";
    if (n >= 8) return "и";
    if (n >= 7) return "з";
    if (n >= 6) return "ѕ";
    if (n >= 5) return "є";
    if (n >= 4) return "д";
    if (n >= 3) return "г";
    if (n >= 2) return "в";
    if (n >= 1) return "а";
};

inter.renderCyrillicLowerPost = function(n) {
    if (n >= 1) return this.renderCyrillicLower(n);
    if (n >= 0) return "҃";
};

inter.renderCyrillicLowerFinal = function(n) {
    if (n >= 91) return "ч" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 90) return "҃ч";
    if (n >= 81) return "п" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 80) return "҃п";
    if (n >= 71) return "ѻ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 70) return "҃ѻ";
    if (n >= 61) return "ѯ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 60) return "҃ѯ";
    if (n >= 51) return "н" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 50) return "҃н";
    if (n >= 41) return "м" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 40) return "҃м";
    if (n >= 31) return "л" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 30) return "҃л";
    if (n >= 21) return "к" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 20) return "҃к";
    if (n >= 19) return "ѳ҃і";
    if (n >= 18) return "и҃і";
    if (n >= 17) return "з҃і";
    if (n >= 16) return "ѕ҃і";
    if (n >= 15) return "є҃і";
    if (n >= 14) return "д҃і";
    if (n >= 13) return "г҃і";
    if (n >= 12) return "в҃і";
    if (n >= 11) return "а҃і";
    if (n >= 1) return "҃" + this.renderCyrillicLower110(n);
    if (n >= 0) return "҃";
};

inter.renderEthiopicP3 = function(n) {
    if (n >= 1e12) return this.renderEthiopic(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1) return "፼፼፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼፼፼";
};

inter.renderEthiopicP2 = function(n) {
    if (n >= 1e8) return this.renderEthiopic(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1) return "፼፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼፼";
};

inter.renderEthiopicP1 = function(n) {
    if (n >= 1e4) return this.renderEthiopic(Math.floor(n / 1e4)) + "፼" + (n === 1e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 1) return "፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼";
};

inter.renderGreekNumeralMajuscules = function(n) {
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e16) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e16)) + "ΜΜΜΜ" + (n === 1e16 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e16));
    if (n >= 1e12) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e12)) + "ΜΜΜ" + (n === 1e12 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e12));
    if (n >= 1e8) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e8)) + "ΜΜ" + (n === 1e8 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e8));
    if (n >= 1e4) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e4)) + "Μ" + (n === 1e4 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e4));
    if (n >= 1e3) return "͵" + this.renderGreekNumeralMajuscules(Math.floor(n / 1e3)) + (n === 1e3 ? "" : this.renderGreekNumeralMajuscules(n % 1e3));
    if (n >= 900) return "Ϡ" + (n === 900 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 800) return "Ω" + (n === 800 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 700) return "Ψ" + (n === 700 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 600) return "Χ" + (n === 600 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 500) return "Φ" + (n === 500 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 400) return "Υ" + (n === 400 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 300) return "Τ" + (n === 300 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 200) return "Σ" + (n === 200 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 100) return "Ρ" + (n === 100 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 90) return "Ϟ" + (n === 90 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 80) return "Π" + (n === 80 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 70) return "Ο" + (n === 70 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 60) return "Ξ" + (n === 60 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 50) return "Ν" + (n === 50 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 40) return "Μ" + (n === 40 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 30) return "Λ" + (n === 30 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 20) return "Κ" + (n === 20 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 10) return "Ι" + (n === 10 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 9) return "Θ";
    if (n >= 8) return "Η";
    if (n >= 7) return "Ζ";
    if (n >= 6) return "Ϝ";
    if (n >= 5) return "Ε";
    if (n >= 4) return "Δ";
    if (n >= 3) return "Γ";
    if (n >= 2) return "Β";
    if (n >= 1) return "Α";
    if (n >= 0) return "𐆊";
};

inter.renderGreekNumeralMinuscules = function(n) {
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e16) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e16)) + "μμμμ" + (n === 1e16 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e16));
    if (n >= 1e12) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e12)) + "μμμ" + (n === 1e12 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e12));
    if (n >= 1e8) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e8)) + "μμ" + (n === 1e8 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e8));
    if (n >= 1e4) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e4)) + "μ" + (n === 1e4 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e4));
    if (n >= 1e3) return "͵" + this.renderGreekNumeralMinuscules(Math.floor(n / 1e3)) + (n === 1e3 ? "" : this.renderGreekNumeralMinuscules(n % 1e3));
    if (n >= 900) return "ϡ" + (n === 900 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 800) return "ω" + (n === 800 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 700) return "ψ" + (n === 700 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 600) return "χ" + (n === 600 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 500) return "φ" + (n === 500 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 400) return "υ" + (n === 400 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 300) return "τ" + (n === 300 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 200) return "σ" + (n === 200 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 100) return "ρ" + (n === 100 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 90) return "ϟ" + (n === 90 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 80) return "π" + (n === 80 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 70) return "ο" + (n === 70 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 60) return "ξ" + (n === 60 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 50) return "ν" + (n === 50 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 40) return "μ" + (n === 40 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 30) return "λ" + (n === 30 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 20) return "κ" + (n === 20 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 10) return "ι" + (n === 10 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 9) return "θ";
    if (n >= 8) return "η";
    if (n >= 7) return "ζ";
    if (n >= 6) return "ϝ";
    if (n >= 5) return "ε";
    if (n >= 4) return "δ";
    if (n >= 3) return "γ";
    if (n >= 2) return "β";
    if (n >= 1) return "α";
    if (n >= 0) return "𐆊";
};

inter.renderSpelloutBase = function(n) {
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e12) return "biliɔn " + this.renderSpelloutCardinal(Math.floor(n / 1e12)) + (n === 1e12 ? "" : this.renderAfterBillions(n % 1e12));
    if (n >= 1e9) return "miliɔn akpe " + this.renderSpelloutCardinal(Math.floor(n / 1e9)) + (n === 1e9 ? "" : this.renderAfterMillions(n % 1e9));
    if (n >= 1e6) return "miliɔn " + this.renderSpelloutCardinal(Math.floor(n / 1e6)) + (n === 1e6 ? "" : this.renderAfterMillions(n % 1e6));
    if (n >= 1e5) return "akpe " + this.renderSpelloutCardinal(Math.floor(n / 1e5)) + (n === 1e5 ? "" : this.renderAfterHundredThousands(n % 1e5));
    if (n >= 1e3) return "akpe " + this.renderSpelloutCardinal(Math.floor(n / 1e3)) + (n === 1e3 ? "" : this.renderAfterThousands(n % 1e3));
    if (n >= 100) return "alafa " + this.renderSpelloutCardinal(Math.floor(n / 100)) + (n === 100 ? "" : " " + this.renderAfterHundreds(n % 100));
    if (n >= 20) return "bla" + this.renderSpelloutBase(Math.floor(n / 10)) + (n === 20 ? "" : " vɔ " + this.renderSpelloutBase(n % 10));
    if (n >= 11) return "wui" + this.renderSpelloutBase(n % 10);
    if (n >= 10) return "ewo";
    if (n >= 9) return "asieke";
    if (n >= 8) return "enyi";
    if (n >= 7) return "adre";
    if (n >= 6) return "ade";
    if (n >= 5) return "atɔ̃";
    if (n >= 4) return "ene";
    if (n >= 3) return "etɔ̃";
    if (n >= 2) return "eve";
    if (n >= 1) return "ɖekɛ";
    if (n >= 0) return "ɖekeo";
};

inter.renderHebrewThousands = function(n) {
    if (n >= 401) return this.renderHebrew(n) + "׳";
    if (n >= 100) return this.renderHebrew(n) + (n === 100 ? "" : "׳");
    if (n >= 10) return this.renderHebrew(n) + (n === 10 ? "" : "׳");
    if (n >= 0) return this.renderHebrew(n);
};

inter.renderHebrew099 = function(n) {
    if (n >= 91) return "צ״" + this.renderHebrewItem(n % 10);
    if (n >= 90) return "״צ";
    if (n >= 81) return "פ״" + this.renderHebrewItem(n % 10);
    if (n >= 80) return "״פ";
    if (n >= 71) return "ע״" + this.renderHebrewItem(n % 10);
    if (n >= 70) return "״ע";
    if (n >= 61) return "ס״" + this.renderHebrewItem(n % 10);
    if (n >= 60) return "״ס";
    if (n >= 51) return "נ״" + this.renderHebrewItem(n % 10);
    if (n >= 50) return "״נ";
    if (n >= 41) return "מ״" + this.renderHebrewItem(n % 10);
    if (n >= 40) return "״מ";
    if (n >= 31) return "ל״" + this.renderHebrewItem(n % 10);
    if (n >= 30) return "״ל";
    if (n >= 21) return "כ״" + this.renderHebrewItem(n % 10);
    if (n >= 20) return "״כ";
    if (n >= 17) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "ט״ז";
    if (n >= 15) return "ט״ו";
    if (n >= 11) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 1) return "״" + this.renderHebrewItem(n);
    if (n >= 0) return "׳";
};

inter.renderHebrewItem = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderHebrewItem(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 2100) return this.renderNumber(n, "#,##0");
    if (n >= 2e3) return "תתתתת" + (n === 2e3 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1900) return "תתתתש" + (n === 1900 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1800) return "תתתתר" + (n === 1800 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1700) return "תתתתק" + (n === 1700 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1600) return "תתתת" + (n === 1600 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1500) return "תתתש" + (n === 1500 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1400) return "תתתר" + (n === 1400 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1300) return "תתתק" + (n === 1300 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1200) return "תתת" + (n === 1200 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1100) return "תתש" + (n === 1100 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1e3) return "תתר" + (n === 1e3 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 900) return "תתק" + (n === 900 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 800) return "תת" + (n === 800 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 745) return "תש" + this.renderHebrewItem(n % 100);
    if (n >= 744) return "תשדמ";
    if (n >= 700) return "תש" + (n === 700 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 699) return "תר" + this.renderHebrewItem(n % 100);
    if (n >= 698) return "תרחצ";
    if (n >= 600) return "תר" + (n === 600 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 500) return "תק" + (n === 500 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 400) return "ת" + (n === 400 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 345) return "ש" + this.renderHebrewItem(n % 100);
    if (n >= 344) return "שדמ";
    if (n >= 305) return "ש" + this.renderHebrewItem(n % 100);
    if (n >= 304) return "דש";
    if (n >= 300) return "ש" + (n === 300 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 299) return "ר" + this.renderHebrewItem(n % 100);
    if (n >= 298) return "רחצ";
    if (n >= 200) return "ר" + (n === 200 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 100) return "ק" + (n === 100 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 90) return "צ" + (n === 90 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 80) return "פ" + (n === 80 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 70) return "ע" + (n === 70 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 60) return "ס" + (n === 60 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 50) return "נ" + (n === 50 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 40) return "מ" + (n === 40 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 30) return "ל" + (n === 30 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 20) return "כ" + (n === 20 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 17) return "י" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "טז";
    if (n >= 15) return "טו";
    if (n >= 10) return "י" + (n === 10 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 9) return "ט";
    if (n >= 8) return "ח";
    if (n >= 7) return "ז";
    if (n >= 6) return "ו";
    if (n >= 5) return "ה";
    if (n >= 4) return "ד";
    if (n >= 3) return "ג";
    if (n >= 2) return "ב";
    if (n >= 1) return "א";
    if (n >= 0) return "״";
};

inter.renderTamilThousands = function(n) {
    if (n >= 1e3) return this.renderTamilThousands(Math.floor(n / 1e3)) + "௲" + (n === 1e3 ? "" : this.renderTamilThousands(n % 1e3));
    if (n >= 0) return this.renderTamil(n);
};

inter.renderEthiopicP = function(n) {
    if (n >= 1e16) return this.renderEthiopicP(Math.floor(n / 1e16)) + "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 1e12) return this.renderEthiopicP(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1e8) return this.renderEthiopicP(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1e4) return this.renderEthiopicP(Math.floor(n / 1e4)) + "፼" + (n === 1e4 ? "" : this.renderEthiopicP(n % 1e4));
    if (n >= 1) return this.renderEthiopic(n);
};

inter.renderAfterBillions = function(n) {
    if (n >= 1e11) return " " + this.renderSpelloutCardinal(n);
    if (n >= 1e8) return " kple " + this.renderSpelloutCardinal(n);
    if (n >= 1e3) return " kple " + this.renderSpelloutCardinal(n);
    if (n >= 0) return " kpakple " + this.renderSpelloutCardinal(n);
};

inter.renderAfterMillions = function(n) {
    if (n >= 1e5) return " " + this.renderSpelloutCardinal(n);
    if (n >= 1e3) return " kple " + this.renderSpelloutCardinal(n);
    if (n >= 0) return " kpakple " + this.renderSpelloutCardinal(n);
};

inter.renderAfterHundredThousands = function(n) {
    if (n >= 100) return " " + this.renderSpelloutCardinal(n);
    if (n >= 0) return " kpakple " + this.renderSpelloutCardinal(n);
};

inter.renderAfterThousands = function(n) {
    if (n >= 100) return " " + this.renderSpelloutCardinal(n);
    if (n >= 0) return " kple " + this.renderSpelloutCardinal(n);
};

inter.renderAfterHundreds = function(n) {
    if (n >= 91) return this.renderSpelloutCardinal(n);
    if (n >= 90) return "kple " + this.renderSpelloutCardinal(n);
    if (n >= 81) return this.renderSpelloutCardinal(n);
    if (n >= 80) return "kple " + this.renderSpelloutCardinal(n);
    if (n >= 71) return this.renderSpelloutCardinal(n);
    if (n >= 70) return "kple " + this.renderSpelloutCardinal(n);
    if (n >= 61) return this.renderSpelloutCardinal(n);
    if (n >= 60) return "kple " + this.renderSpelloutCardinal(n);
    if (n >= 51) return this.renderSpelloutCardinal(n);
    if (n >= 50) return "kple " + this.renderSpelloutCardinal(n);
    if (n >= 41) return this.renderSpelloutCardinal(n);
    if (n >= 40) return "kple " + this.renderSpelloutCardinal(n);
    if (n >= 31) return this.renderSpelloutCardinal(n);
    if (n >= 30) return "kple " + this.renderSpelloutCardinal(n);
    if (n >= 21) return this.renderSpelloutCardinal(n);
    if (n >= 0) return "kple " + this.renderSpelloutCardinal(n);
};

module.exports = inter;
