// inter/build/en_il.js
function defaults(targetObj) {
    for (var i = 1; i < arguments.length; i += 1) {
        if (arguments[i] && typeof arguments[i] === "object") {
            for (var propertyName in arguments[i]) {
                if (typeof targetObj[propertyName] === "undefined") {
                    targetObj[propertyName] = arguments[i][propertyName];
                }
            }
        }
    }
    return targetObj;
}

var P = require("./en.js"), inter = defaults({
    renderers: {}
}, P);

var _42rr1 = [ "Q1", "Q2", "Q3", "Q4" ], _42rr2 = {
    displayName: "AM/PM"
}, _42rr3 = {
    all: {
        leap: "{0}bis"
    }
}, _42rr4 = {
    one: "{0}°",
    other: "{0}°"
}, _42rr5 = {
    one: "{0}′",
    other: "{0}′"
}, _42rr6 = {
    one: "{0}″",
    other: "{0}″"
}, _42rr7 = {
    one: "{0}m",
    other: "{0}m"
}, _42rr8 = {
    displayName: "time zone"
}, _42rr9 = [ "Before R.O.C.", "Minguo" ], _42rr10 = {
    one: "{0} c",
    other: "{0} c"
}, _42rr11 = {
    one: "{0}°F",
    other: "{0}°F"
}, _42rr12 = {
    a: "h a – h a",
    h: "h – h a"
}, _42rr13 = [ "S", "M", "T", "W", "T", "F", "S" ], _42rr14 = {
    a: "h a – h a v",
    h: "h – h a v"
}, _42rr15 = {
    displayName: "day of the week"
}, _42rr16 = {
    one: "{0} day",
    other: "{0} days"
}, _42rr17 = {
    wide: [ "AH" ],
    abbreviated: [ "AH" ]
}, _42rr18 = {
    d: "d – d MMM",
    M: "d MMM – d MMM"
}, _42rr19 = {
    M: "MM/y – MM/y",
    y: "MM/y – MM/y"
}, _42rr20 = {
    M: "MMM – MMM U",
    y: "MMM U – MMM U"
}, _42rr21 = {
    M: "MMM – MMM y",
    y: "MMM y – MMM y"
}, _42rr22 = [ "Su", "Mo", "Tu", "We", "Th", "Fr", "Sa" ], _42rr23 = {
    d: "dd/MM – dd/MM",
    M: "dd/MM – dd/MM"
}, _42rr24 = {
    H: "HH:mm – HH:mm",
    m: "HH:mm – HH:mm"
}, _42rr25 = {
    one: "in {0} yr.",
    other: "in {0} yr."
}, _42rr26 = {
    one: "in {0} mo.",
    other: "in {0} mo."
}, _42rr27 = {
    one: "in {0} wk.",
    other: "in {0} wk."
}, _42rr28 = {
    one: "in {0} hr.",
    other: "in {0} hr."
}, _42rr29 = {
    "0": "this M",
    "1": "next M",
    "-1": "last M"
}, _42rr30 = {
    "0": "this W",
    "1": "next W",
    "-1": "last W"
}, _42rr31 = {
    "0": "this F",
    "1": "next F",
    "-1": "last F"
}, _42rr32 = {
    one: "in {0} day",
    other: "in {0} days"
}, _42rr33 = {
    M: "MMM – MMM y G",
    y: "MMM y – MMM y G"
}, _42rr34 = {
    one: "{0} yr. ago",
    other: "{0} yr. ago"
}, _42rr35 = {
    one: "{0} mo. ago",
    other: "{0} mo. ago"
}, _42rr36 = {
    one: "{0} wk. ago",
    other: "{0} wk. ago"
}, _42rr37 = {
    one: "{0} hr. ago",
    other: "{0} hr. ago"
}, _42rr38 = {
    one: "in {0} min.",
    other: "in {0} min."
}, _42rr39 = {
    one: "in {0} sec.",
    other: "in {0} sec."
}, _42rr40 = {
    M: "MMMM – MMMM U",
    y: "MMMM U – MMMM U"
}, _42rr41 = {
    M: "MMMM – MMMM y",
    y: "MMMM y – MMMM y"
}, _42rr42 = {
    one: "in {0} year",
    other: "in {0} years"
}, _42rr43 = {
    one: "in {0} qtr.",
    other: "in {0} qtrs."
}, _42rr44 = {
    one: "in {0} week",
    other: "in {0} weeks"
}, _42rr45 = {
    one: "{0} day ago",
    other: "{0} days ago"
}, _42rr46 = {
    one: "in {0} hour",
    other: "in {0} hours"
}, _42rr47 = {
    "0": "this Su",
    "1": "next Su",
    "-1": "last Su"
}, _42rr48 = {
    "0": "this Tu",
    "1": "next Tu",
    "-1": "last Tu"
}, _42rr49 = {
    "0": "this Th",
    "1": "next Th",
    "-1": "last Th"
}, _42rr50 = {
    "0": "this Sa",
    "1": "next Sa",
    "-1": "last Sa"
}, _42rr51 = {
    d: "E d – E d MMM",
    M: "E d MMM – E d MMM"
}, _42rr52 = {
    H: "HH:mm – HH:mm v",
    m: "HH:mm – HH:mm v"
}, _42rr53 = {
    "0": "today",
    "1": "tomorrow",
    "-1": "yesterday"
}, _42rr54 = {
    one: "{0} min. ago",
    other: "{0} min. ago"
}, _42rr55 = {
    one: "{0} sec. ago",
    other: "{0} sec. ago"
}, _42rr56 = {
    one: "{0} year ago",
    other: "{0} years ago"
}, _42rr57 = {
    one: "{0} qtr. ago",
    other: "{0} qtrs. ago"
}, _42rr58 = {
    one: "in {0} month",
    other: "in {0} months"
}, _42rr59 = {
    one: "{0} week ago",
    other: "{0} weeks ago"
}, _42rr60 = {
    one: "{0} hour ago",
    other: "{0} hours ago"
}, _42rr61 = {
    M: "MMMM – MMMM y G",
    y: "MMMM y – MMMM y G"
}, _42rr62 = [ "Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat" ], _42rr63 = {
    "0": "this yr.",
    "1": "next yr.",
    "-1": "last yr."
}, _42rr64 = {
    "0": "this mo.",
    "1": "next mo.",
    "-1": "last mo."
}, _42rr65 = {
    "0": "this wk.",
    "1": "next wk.",
    "-1": "last wk."
}, _42rr66 = {
    one: "{0} month ago",
    other: "{0} months ago"
}, _42rr67 = {
    one: "in {0} minute",
    other: "in {0} minutes"
}, _42rr68 = {
    one: "in {0} second",
    other: "in {0} seconds"
}, _42rr69 = {
    d: "E dd/MM – E dd/MM",
    M: "E dd/MM – E dd/MM"
}, _42rr70 = {
    M: "MM/y – MM/y GGGGG",
    y: "MM/y – MM/y GGGGG"
}, _42rr71 = {
    "0": "this year",
    "1": "next year",
    "-1": "last year"
}, _42rr72 = {
    one: "in {0} quarter",
    other: "in {0} quarters"
}, _42rr73 = {
    "0": "this qtr.",
    "1": "next qtr.",
    "-1": "last qtr."
}, _42rr74 = {
    "0": "this week",
    "1": "next week",
    "-1": "last week"
}, _42rr75 = {
    one: "{0} minute ago",
    other: "{0} minutes ago"
}, _42rr76 = {
    one: "{0} second ago",
    other: "{0} seconds ago"
}, _42rr77 = {
    "0": "this Sun.",
    "1": "next Sun.",
    "-1": "last Sun."
}, _42rr78 = {
    "0": "this Mon.",
    "1": "next Mon.",
    "-1": "last Mon."
}, _42rr79 = {
    "0": "this Tue.",
    "1": "next Tue.",
    "-1": "last Tue."
}, _42rr80 = {
    "0": "this Wed.",
    "1": "next Wed.",
    "-1": "last Wed."
}, _42rr81 = {
    "0": "this Thu.",
    "1": "next Thu.",
    "-1": "last Thu."
}, _42rr82 = {
    "0": "this Fri.",
    "1": "next Fri.",
    "-1": "last Fri."
}, _42rr83 = {
    "0": "this Sat.",
    "1": "next Sat.",
    "-1": "last Sat."
}, _42rr84 = {
    one: "{0} quarter ago",
    other: "{0} quarters ago"
}, _42rr85 = {
    relative: _42rr29
}, _42rr86 = {
    relative: _42rr30
}, _42rr87 = {
    relative: _42rr31
}, _42rr88 = [ "J", "F", "M", "A", "M", "J", "J", "A", "S", "O", "N", "D" ], _42rr89 = {
    "0": "this month",
    "1": "next month",
    "-1": "last month"
}, _42rr90 = {
    d: "E, d MMM – E, d MMM",
    M: "E, d MMM – E, d MMM"
}, _42rr91 = {
    wide: [ "ERA0", "ERA1" ],
    abbreviated: [ "ERA0", "ERA1" ]
}, _42rr92 = {
    relative: _42rr47
}, _42rr93 = {
    relative: _42rr48
}, _42rr94 = {
    relative: _42rr49
}, _42rr95 = {
    relative: _42rr50
}, _42rr96 = {
    "0": "this Sunday",
    "1": "next Sunday",
    "-1": "last Sunday"
}, _42rr97 = {
    "0": "this Monday",
    "1": "next Monday",
    "-1": "last Monday"
}, _42rr98 = {
    "0": "this Friday",
    "1": "next Friday",
    "-1": "last Friday"
}, _42rr99 = [ "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12" ], _42rr100 = {
    "0": "this quarter",
    "1": "next quarter",
    "-1": "last quarter"
}, _42rr101 = {
    "0": "this Tuesday",
    "1": "next Tuesday",
    "-1": "last Tuesday"
}, _42rr102 = {
    a: "h:mm a – h:mm a",
    h: "h:mm – h:mm a",
    m: "h:mm – h:mm a"
}, _42rr103 = [ "1st quarter", "2nd quarter", "3rd quarter", "4th quarter" ], _42rr104 = {
    relative: _42rr77
}, _42rr105 = {
    relative: _42rr78
}, _42rr106 = {
    relative: _42rr79
}, _42rr107 = {
    relative: _42rr80
}, _42rr108 = {
    relative: _42rr81
}, _42rr109 = {
    relative: _42rr82
}, _42rr110 = {
    relative: _42rr83
}, _42rr111 = [ "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13" ], _42rr112 = {
    "0": "this Thursday",
    "1": "next Thursday",
    "-1": "last Thursday"
}, _42rr113 = {
    "0": "this Saturday",
    "1": "next Saturday",
    "-1": "last Saturday"
}, _42rr114 = {
    d: "d – d MMM U",
    M: "d MMM – d MMM U",
    y: "d MMM U – d MMM U"
}, _42rr115 = {
    d: "d – d MMM y",
    M: "d MMM – d MMM y",
    y: "d MMM y – d MMM y"
}, _42rr116 = {
    "0": "this Wednesday",
    "1": "next Wednesday",
    "-1": "last Wednesday"
}, _42rr117 = {
    a: "h:mm a – h:mm a v",
    h: "h:mm – h:mm a v",
    m: "h:mm – h:mm a v"
}, _42rr118 = {
    relative: _42rr96
}, _42rr119 = {
    relative: _42rr97
}, _42rr120 = {
    relative: _42rr98
}, _42rr121 = {
    d: "d – d MMM y G",
    M: "d MMM – d MMM y G",
    y: "d MMM y – d MMM y G"
}, _42rr122 = {
    relative: _42rr101
}, _42rr123 = {
    d: "dd/MM/y – dd/MM/y",
    M: "dd/MM/y – dd/MM/y",
    y: "dd/MM/y – dd/MM/y"
}, _42rr124 = {
    relative: _42rr112
}, _42rr125 = {
    relative: _42rr113
}, _42rr126 = {
    abbreviated: {
        leap: "{0}bis"
    },
    narrow: {
        leap: "{0}b"
    },
    wide: {
        leap: "{0}bis"
    }
}, _42rr127 = [ "Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday" ], _42rr128 = {
    relative: _42rr116
}, _42rr129 = [ "Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec" ], _42rr130 = [ undefined, "zi", "chou", "yin", "mao", "chen", "si", "wu", "wei", "shen", "you", "xu" ], _42rr131 = [ "Mo1", "Mo2", "Mo3", "Mo4", "Mo5", "Mo6", "Mo7", "Mo8", "Mo9", "Mo10", "Mo11", "Mo12" ], _42rr132 = {
    d: "E, d – E, d MMM U",
    M: "E, d MMM – E, d MMM U",
    y: "E, d MMM U – E, d MMM U"
}, _42rr133 = {
    d: "E, d MMM – E, d MMM y",
    M: "E, d MMM – E, d MMM y",
    y: "E, d MMM y – E, d MMM y"
}, _42rr134 = {
    d: "E, d – E, d MMM y G",
    M: "E, d MMM – E, d MMM y G",
    y: "E, d MMM y – E, d MMM y G"
}, _42rr135 = {
    d: "dd/MM/y – dd/MM/y GGGGG",
    M: "dd/MM/y – dd/MM/y GGGGG",
    y: "dd/MM/y – dd/MM/y GGGGG"
}, _42rr136 = {
    d: "E, dd/MM/y – E, dd/MM/y",
    M: "E, dd/MM/y – E, dd/MM/y",
    y: "E, dd/MM/y – E, dd/MM/y"
}, _42rr137 = {
    future: _42rr25,
    past: _42rr34
}, _42rr138 = {
    future: _42rr26,
    past: _42rr35
}, _42rr139 = {
    future: _42rr27,
    past: _42rr36
}, _42rr140 = {
    future: _42rr28,
    past: _42rr37
}, _42rr141 = {
    future: _42rr32,
    past: _42rr45
}, _42rr142 = {
    future: _42rr38,
    past: _42rr54
}, _42rr143 = {
    future: _42rr39,
    past: _42rr55
}, _42rr144 = [ undefined, "Rat", "Ox", "Tiger", "Rabbit", "Dragon", "Snake", "Horse", "Goat", "Monkey", "Rooster", "Dog" ], _42rr145 = {
    future: _42rr42,
    past: _42rr56
}, _42rr146 = {
    future: _42rr43,
    past: _42rr57
}, _42rr147 = {
    future: _42rr44,
    past: _42rr59
}, _42rr148 = {
    future: _42rr46,
    past: _42rr60
}, _42rr149 = {
    future: _42rr58,
    past: _42rr66
}, _42rr150 = {
    d: "E, dd/MM/y – E, dd/MM/y GGGGG",
    M: "E, dd/MM/y – E, dd/MM/y GGGGG",
    y: "E, dd/MM/y – E, dd/MM/y GGGGG"
}, _42rr151 = [ "Muh.", "Saf.", "Rab. I", "Rab. II", "Jum. I", "Jum. II", "Raj.", "Sha.", "Ram.", "Shaw.", "Dhuʻl-Q.", "Dhuʻl-H." ], _42rr152 = {
    future: _42rr67,
    past: _42rr75
}, _42rr153 = {
    future: _42rr68,
    past: _42rr76
}, _42rr154 = [ "Tishri", "Heshvan", "Kislev", "Tevet", "Shevat", "Adar I", "Adar", "Nisan", "Iyar", "Sivan", "Tamuz", "Av", "Elul" ], _42rr155 = {
    future: _42rr72,
    past: _42rr84
}, _42rr156 = [ "Farvardin", "Ordibehesht", "Khordad", "Tir", "Mordad", "Shahrivar", "Mehr", "Aban", "Azar", "Dey", "Bahman", "Esfand" ], _42rr157 = [ "January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December" ], _42rr158 = [ "Tout", "Baba", "Hator", "Kiahk", "Toba", "Amshir", "Baramhat", "Baramouda", "Bashans", "Paona", "Epep", "Mesra", "Nasie" ], _42rr159 = [ "Meskerem", "Tekemt", "Hedar", "Tahsas", "Ter", "Yekatit", "Megabit", "Miazia", "Genbot", "Sene", "Hamle", "Nehasse", "Pagumen" ], _42rr160 = [ "Chaitra", "Vaisakha", "Jyaistha", "Asadha", "Sravana", "Bhadra", "Asvina", "Kartika", "Agrahayana", "Pausa", "Magha", "Phalguna" ], _42rr161 = {
    abbreviated: _42rr1,
    narrow: [ "1", "2", "3", "4" ],
    wide: _42rr103
}, _42rr162 = {
    displayName: "hr.",
    relativeTime: _42rr140
}, _42rr163 = {
    midnight: "midnight",
    am: "AM",
    noon: "noon",
    pm: "PM",
    morning1: "morning",
    afternoon1: "afternoon",
    evening1: "evening",
    night1: "night"
}, _42rr164 = {
    displayName: "day",
    relativeTime: _42rr141
}, _42rr165 = {
    displayName: "min.",
    relativeTime: _42rr142
}, _42rr166 = {
    displayName: "hour",
    relativeTime: _42rr148
}, _42rr167 = [ "Muharram", "Safar", "Rabiʻ I", "Rabiʻ II", "Jumada I", "Jumada II", "Rajab", "Shaʻban", "Ramadan", "Shawwal", "Dhuʻl-Qiʻdah", "Dhuʻl-Hijjah" ], _42rr168 = {
    midnight: "mi",
    am: "a",
    noon: "n",
    pm: "p",
    morning1: "in the morning",
    afternoon1: "in the afternoon",
    evening1: "in the evening",
    night1: "at night"
}, _42rr169 = {
    displayName: "minute",
    relativeTime: _42rr152
}, _42rr170 = {
    displayName: "sec.",
    relative: [ "now" ],
    relativeTime: _42rr143
}, _42rr171 = {
    midnight: "midnight",
    am: "AM",
    noon: "noon",
    pm: "PM",
    morning1: "in the morning",
    afternoon1: "in the afternoon",
    evening1: "in the evening",
    night1: "at night"
}, _42rr172 = {
    displayName: "second",
    relative: [ "now" ],
    relativeTime: _42rr153
}, _42rr173 = {
    displayName: "yr.",
    relative: _42rr63,
    relativeTime: _42rr137
}, _42rr174 = {
    displayName: "mo.",
    relative: _42rr64,
    relativeTime: _42rr138
}, _42rr175 = {
    displayName: "wk.",
    relative: _42rr65,
    relativeTime: _42rr139
}, _42rr176 = {
    displayName: "day",
    relative: _42rr53,
    relativeTime: _42rr141
}, _42rr177 = [ "First Month", "Second Month", "Third Month", "Fourth Month", "Fifth Month", "Sixth Month", "Seventh Month", "Eighth Month", "Ninth Month", "Tenth Month", "Eleventh Month", "Twelfth Month" ], _42rr178 = {
    displayName: "year",
    relative: _42rr71,
    relativeTime: _42rr145
}, _42rr179 = {
    displayName: "qtr.",
    relative: _42rr73,
    relativeTime: _42rr146
}, _42rr180 = {
    displayName: "week",
    relative: _42rr74,
    relativeTime: _42rr147
}, _42rr181 = {
    format: _42rr126,
    numeric: _42rr3,
    standAlone: _42rr126
}, _42rr182 = {
    displayName: "month",
    relative: _42rr89,
    relativeTime: _42rr149
}, _42rr183 = {
    displayName: "quarter",
    relative: _42rr100,
    relativeTime: _42rr155
}, _42rr184 = {
    abbreviated: _42rr62,
    narrow: _42rr13,
    wide: _42rr127,
    "short": _42rr22
}, _42rr185 = {
    abbreviated: _42rr130,
    narrow: _42rr130,
    wide: _42rr130
}, _42rr186 = {
    abbreviated: _42rr129,
    narrow: _42rr88,
    wide: _42rr157
}, _42rr187 = {
    format: _42rr185
}, _42rr188 = {
    format: _42rr161,
    standAlone: _42rr161
}, _42rr189 = {
    abbreviated: _42rr154,
    narrow: _42rr111,
    wide: _42rr154
}, _42rr190 = {
    abbreviated: _42rr156,
    narrow: _42rr99,
    wide: _42rr156
}, _42rr191 = {
    abbreviated: _42rr158,
    narrow: _42rr111,
    wide: _42rr158
}, _42rr192 = {
    abbreviated: _42rr144,
    narrow: _42rr144,
    wide: _42rr144
}, _42rr193 = {
    abbreviated: _42rr151,
    narrow: _42rr99,
    wide: _42rr167
}, _42rr194 = {
    abbreviated: _42rr159,
    narrow: _42rr111,
    wide: _42rr159
}, _42rr195 = {
    abbreviated: _42rr160,
    narrow: _42rr99,
    wide: _42rr160
}, _42rr196 = {
    format: _42rr192
}, _42rr197 = {
    fullDate: "EEEE, d MMMM r(U)",
    longDate: "d MMMM r(U)",
    mediumDate: "d MMM r",
    shortDate: "dd/MM/r",
    fullTime: "H:mm:ss zzzz",
    longTime: "H:mm:ss z",
    mediumTime: "H:mm:ss",
    shortTime: "H:mm",
    shortDateTime: "dd/MM/r, H:mm",
    mediumDateTime: "d MMM r, H:mm:ss",
    longDateTime: "d MMMM r(U) 'at' H:mm:ss z",
    fullDateTime: "EEEE, d MMMM r(U) 'at' H:mm:ss zzzz"
}, _42rr198 = {
    abbreviated: _42rr131,
    narrow: _42rr99,
    wide: _42rr177
}, _42rr199 = {
    fullDate: "EEEE, d MMMM y G",
    longDate: "d MMMM y G",
    mediumDate: "d MMM y G",
    shortDate: "dd/MM/y GGGGG",
    fullTime: "H:mm:ss zzzz",
    longTime: "H:mm:ss z",
    mediumTime: "H:mm:ss",
    shortTime: "H:mm",
    shortDateTime: "dd/MM/y GGGGG, H:mm",
    mediumDateTime: "d MMM y G, H:mm:ss",
    longDateTime: "d MMMM y G 'at' H:mm:ss z",
    fullDateTime: "EEEE, d MMMM y G 'at' H:mm:ss zzzz"
}, _42rr200 = {
    abbreviated: _42rr171,
    narrow: _42rr171,
    wide: _42rr163
}, _42rr201 = {
    format: _42rr184,
    standAlone: _42rr184
}, _42rr202 = {
    abbreviated: _42rr171,
    narrow: _42rr168,
    wide: _42rr171
}, _42rr203 = {
    EHm: "E H:mm",
    EHms: "E H:mm:ss",
    H: "H",
    Hm: "H:mm",
    Hms: "H:mm:ss",
    Ed: "E d",
    GyMMMd: "d MMM y G",
    GyMMMEd: "E, d MMM y G",
    M: "LL",
    Md: "dd/MM",
    MEd: "E, dd/MM",
    MMMd: "d MMM",
    MMMEd: "E, d MMM",
    MMMMd: "d MMMM",
    yyyyM: "MM/y GGGGG",
    yyyyMd: "dd/MM/y GGGGG",
    yyyyMEd: "E, dd/MM/y GGGGG",
    yyyyMMMd: "d MMM y G",
    yyyyMMMEd: "E, d MMM y G",
    d: "d",
    E: "ccc",
    Ehm: "E h:mm a",
    Ehms: "E h:mm:ss a",
    Gy: "y G",
    GyMMM: "MMM y G",
    h: "h a",
    hm: "h:mm a",
    hms: "h:mm:ss a",
    MMM: "LLL",
    ms: "mm:ss",
    y: "y G",
    yyyy: "y G",
    yyyyMMM: "MMM y G",
    yyyyMMMM: "MMMM y G",
    yyyyQQQ: "QQQ y G",
    yyyyQQQQ: "QQQQ y G"
}, _42rr204 = {
    format: _42rr186,
    standAlone: _42rr186
}, _42rr205 = {
    Ed: "E d",
    GyMMMd: "d MMM r",
    GyMMMEd: "E, d MMM r(U)",
    M: "LL",
    Md: "dd/MM",
    MEd: "E, dd/MM",
    MMMd: "d MMM",
    MMMEd: "E, d MMM",
    MMMMd: "d MMMM",
    UMd: "dd/MM/U",
    UMMMd: "d MMM U",
    yMd: "dd/MM/r",
    yyyyM: "MM/r",
    yyyyMd: "dd/MM/r",
    yyyyMEd: "E, dd/MM/r",
    yyyyMMMd: "d MMM r",
    yyyyMMMEd: "E, d MMM r(U)",
    d: "d",
    E: "ccc",
    Ehm: "E h:mm a",
    EHm: "E HH:mm",
    Ehms: "E h:mm:ss a",
    EHms: "E HH:mm:ss",
    Gy: "r(U)",
    GyMMM: "MMM r(U)",
    h: "h a",
    H: "HH",
    hm: "h:mm a",
    Hm: "HH:mm",
    hms: "h:mm:ss a",
    Hms: "HH:mm:ss",
    MMM: "LLL",
    ms: "mm:ss",
    UM: "M/U",
    UMMM: "MMM U",
    y: "r(U)",
    yyyy: "r(U)",
    yyyyMMM: "MMM r(U)",
    yyyyMMMM: "MMMM r(U)",
    yyyyQQQ: "QQQ r(U)",
    yyyyQQQQ: "QQQQ r(U)"
}, _42rr206 = [ undefined, "jia-zi", "yi-chou", "bing-yin", "ding-mao", "wu-chen", "ji-si", "geng-wu", "xin-wei", "ren-shen", "gui-you", "jia-xu", "yi-hai", "bing-zi", "ding-chou", "wu-yin", "ji-mao", "geng-chen", "xin-si", "ren-wu", "gui-wei", "jia-shen", "yi-you", "bing-xu", "ding-hai", "wu-zi", "ji-chou", "geng-yin", "xin-mao", "ren-chen", "gui-si", "jia-wu", "yi-wei", "bing-shen", "ding-you", "wu-xu", "ji-hai", "geng-zi", "xin-chou", "ren-yin", "gui-mao", "jia-chen", "yi-si", "bing-wu", "ding-wei", "wu-shen", "ji-you", "geng-xu", "xin-hai", "ren-zi", "gui-chou", "jia-yin", "yi-mao", "bing-chen", "ding-si", "wu-wu", "ji-wei", "geng-shen", "xin-you", "ren-xu" ], _42rr207 = {
    format: _42rr193,
    standAlone: _42rr193
}, _42rr208 = {
    format: _42rr198,
    standAlone: _42rr198
}, _42rr209 = {
    basic: _42rr199,
    cldr: _42rr203
}, _42rr210 = {
    format: _42rr202,
    standAlone: _42rr200
}, _42rr211 = {
    basic: _42rr197,
    cldr: _42rr205
}, _42rr212 = {
    Md: _42rr23,
    MEd: _42rr69,
    MMMd: _42rr18,
    MMMEd: _42rr51,
    yM: _42rr19,
    yMd: _42rr123,
    yMEd: _42rr136,
    yMMMd: _42rr114,
    yMMMEd: _42rr132,
    d: {
        d: "d – d"
    },
    h: _42rr12,
    H: {
        H: "HH – HH"
    },
    hm: _42rr102,
    Hm: _42rr24,
    hmv: _42rr117,
    Hmv: _42rr52,
    hv: _42rr14,
    Hv: {
        H: "HH – HH v"
    },
    M: {
        M: "M – M"
    },
    MMM: {
        M: "MMM – MMM"
    },
    y: {
        y: "U – U"
    },
    yMMM: _42rr20,
    yMMMM: _42rr40
}, _42rr213 = {
    Md: _42rr23,
    MEd: _42rr69,
    MMMd: _42rr18,
    MMMEd: _42rr51,
    yM: _42rr70,
    yMd: _42rr135,
    yMEd: _42rr150,
    yMMMd: _42rr121,
    yMMMEd: _42rr134,
    d: {
        d: "d – d"
    },
    h: _42rr12,
    H: {
        H: "HH – HH"
    },
    hm: _42rr102,
    Hm: _42rr24,
    hmv: _42rr117,
    Hmv: _42rr52,
    hv: _42rr14,
    Hv: {
        H: "HH – HH v"
    },
    M: {
        M: "M – M"
    },
    MMM: {
        M: "MMM – MMM"
    },
    y: {
        y: "y – y G"
    },
    yMMM: _42rr33,
    yMMMM: _42rr61
}, _42rr214 = {
    abbreviated: _42rr206,
    narrow: _42rr206,
    wide: _42rr206
}, _42rr215 = {
    format: _42rr214
}, _42rr216 = {
    era: {
        displayName: "era"
    },
    year: _42rr178,
    "year-short": _42rr173,
    quarter: _42rr183,
    "quarter-short": _42rr179,
    month: _42rr182,
    "month-short": _42rr174,
    week: _42rr180,
    "week-short": _42rr175,
    day: _42rr176,
    "day-short": _42rr164,
    weekday: _42rr15,
    dayperiod: _42rr2,
    hour: _42rr166,
    "hour-short": _42rr162,
    minute: _42rr169,
    "minute-short": _42rr165,
    second: _42rr172,
    "second-short": _42rr170,
    zone: _42rr8,
    sun: _42rr118,
    "sun-short": _42rr104,
    "sun-narrow": _42rr92,
    mon: _42rr119,
    "mon-short": _42rr105,
    "mon-narrow": _42rr85,
    tue: _42rr122,
    "tue-short": _42rr106,
    "tue-narrow": _42rr93,
    wed: _42rr128,
    "wed-short": _42rr107,
    "wed-narrow": _42rr86,
    thu: _42rr124,
    "thu-short": _42rr108,
    "thu-narrow": _42rr94,
    fri: _42rr120,
    "fri-short": _42rr109,
    "fri-narrow": _42rr87,
    sat: _42rr125,
    "sat-short": _42rr110,
    "sat-narrow": _42rr95
}, _42rr217 = {
    dateFormats: {
        basic: {},
        cldr: {}
    },
    fields: _42rr216
}, _42rr218 = [ "Taika (645–650)", "Hakuchi (650–671)", "Hakuhō (672–686)", "Shuchō (686–701)", "Taihō (701–704)", "Keiun (704–708)", "Wadō (708–715)", "Reiki (715–717)", "Yōrō (717–724)", "Jinki (724–729)", "Tenpyō (729–749)", "Tenpyō-kampō (749-749)", "Tenpyō-shōhō (749-757)", "Tenpyō-hōji (757-765)", "Tenpyō-jingo (765-767)", "Jingo-keiun (767-770)", "Hōki (770–780)", "Ten-ō (781-782)", "Enryaku (782–806)", "Daidō (806–810)", "Kōnin (810–824)", "Tenchō (824–834)", "Jōwa (834–848)", "Kajō (848–851)", "Ninju (851–854)", "Saikō (854–857)", "Ten-an (857-859)", "Jōgan (859–877)", "Gangyō (877–885)", "Ninna (885–889)", "Kanpyō (889–898)", "Shōtai (898–901)", "Engi (901–923)", "Enchō (923–931)", "Jōhei (931–938)", "Tengyō (938–947)", "Tenryaku (947–957)", "Tentoku (957–961)", "Ōwa (961–964)", "Kōhō (964–968)", "Anna (968–970)", "Tenroku (970–973)", "Ten’en (973–976)", "Jōgen (976–978)", "Tengen (978–983)", "Eikan (983–985)", "Kanna (985–987)", "Eien (987–989)", "Eiso (989–990)", "Shōryaku (990–995)", "Chōtoku (995–999)", "Chōhō (999–1004)", "Kankō (1004–1012)", "Chōwa (1012–1017)", "Kannin (1017–1021)", "Jian (1021–1024)", "Manju (1024–1028)", "Chōgen (1028–1037)", "Chōryaku (1037–1040)", "Chōkyū (1040–1044)", "Kantoku (1044–1046)", "Eishō (1046–1053)", "Tengi (1053–1058)", "Kōhei (1058–1065)", "Jiryaku (1065–1069)", "Enkyū (1069–1074)", "Shōho (1074–1077)", "Shōryaku (1077–1081)", "Eihō (1081–1084)", "Ōtoku (1084–1087)", "Kanji (1087–1094)", "Kahō (1094–1096)", "Eichō (1096–1097)", "Jōtoku (1097–1099)", "Kōwa (1099–1104)", "Chōji (1104–1106)", "Kashō (1106–1108)", "Tennin (1108–1110)", "Ten-ei (1110-1113)", "Eikyū (1113–1118)", "Gen’ei (1118–1120)", "Hōan (1120–1124)", "Tenji (1124–1126)", "Daiji (1126–1131)", "Tenshō (1131–1132)", "Chōshō (1132–1135)", "Hōen (1135–1141)", "Eiji (1141–1142)", "Kōji (1142–1144)", "Ten’yō (1144–1145)", "Kyūan (1145–1151)", "Ninpei (1151–1154)", "Kyūju (1154–1156)", "Hōgen (1156–1159)", "Heiji (1159–1160)", "Eiryaku (1160–1161)", "Ōho (1161–1163)", "Chōkan (1163–1165)", "Eiman (1165–1166)", "Nin’an (1166–1169)", "Kaō (1169–1171)", "Shōan (1171–1175)", "Angen (1175–1177)", "Jishō (1177–1181)", "Yōwa (1181–1182)", "Juei (1182–1184)", "Genryaku (1184–1185)", "Bunji (1185–1190)", "Kenkyū (1190–1199)", "Shōji (1199–1201)", "Kennin (1201–1204)", "Genkyū (1204–1206)", "Ken’ei (1206–1207)", "Jōgen (1207–1211)", "Kenryaku (1211–1213)", "Kenpō (1213–1219)", "Jōkyū (1219–1222)", "Jōō (1222–1224)", "Gennin (1224–1225)", "Karoku (1225–1227)", "Antei (1227–1229)", "Kanki (1229–1232)", "Jōei (1232–1233)", "Tenpuku (1233–1234)", "Bunryaku (1234–1235)", "Katei (1235–1238)", "Ryakunin (1238–1239)", "En’ō (1239–1240)", "Ninji (1240–1243)", "Kangen (1243–1247)", "Hōji (1247–1249)", "Kenchō (1249–1256)", "Kōgen (1256–1257)", "Shōka (1257–1259)", "Shōgen (1259–1260)", "Bun’ō (1260–1261)", "Kōchō (1261–1264)", "Bun’ei (1264–1275)", "Kenji (1275–1278)", "Kōan (1278–1288)", "Shōō (1288–1293)", "Einin (1293–1299)", "Shōan (1299–1302)", "Kengen (1302–1303)", "Kagen (1303–1306)", "Tokuji (1306–1308)", "Enkyō (1308–1311)", "Ōchō (1311–1312)", "Shōwa (1312–1317)", "Bunpō (1317–1319)", "Genō (1319–1321)", "Genkō (1321–1324)", "Shōchū (1324–1326)", "Karyaku (1326–1329)", "Gentoku (1329–1331)", "Genkō (1331–1334)", "Kenmu (1334–1336)", "Engen (1336–1340)", "Kōkoku (1340–1346)", "Shōhei (1346–1370)", "Kentoku (1370–1372)", "Bunchū (1372–1375)", "Tenju (1375–1379)", "Kōryaku (1379–1381)", "Kōwa (1381–1384)", "Genchū (1384–1392)", "Meitoku (1384–1387)", "Kakei (1387–1389)", "Kōō (1389–1390)", "Meitoku (1390–1394)", "Ōei (1394–1428)", "Shōchō (1428–1429)", "Eikyō (1429–1441)", "Kakitsu (1441–1444)", "Bun’an (1444–1449)", "Hōtoku (1449–1452)", "Kyōtoku (1452–1455)", "Kōshō (1455–1457)", "Chōroku (1457–1460)", "Kanshō (1460–1466)", "Bunshō (1466–1467)", "Ōnin (1467–1469)", "Bunmei (1469–1487)", "Chōkyō (1487–1489)", "Entoku (1489–1492)", "Meiō (1492–1501)", "Bunki (1501–1504)", "Eishō (1504–1521)", "Taiei (1521–1528)", "Kyōroku (1528–1532)", "Tenbun (1532–1555)", "Kōji (1555–1558)", "Eiroku (1558–1570)", "Genki (1570–1573)", "Tenshō (1573–1592)", "Bunroku (1592–1596)", "Keichō (1596–1615)", "Genna (1615–1624)", "Kan’ei (1624–1644)", "Shōho (1644–1648)", "Keian (1648–1652)", "Jōō (1652–1655)", "Meireki (1655–1658)", "Manji (1658–1661)", "Kanbun (1661–1673)", "Enpō (1673–1681)", "Tenna (1681–1684)", "Jōkyō (1684–1688)", "Genroku (1688–1704)", "Hōei (1704–1711)", "Shōtoku (1711–1716)", "Kyōhō (1716–1736)", "Genbun (1736–1741)", "Kanpō (1741–1744)", "Enkyō (1744–1748)", "Kan’en (1748–1751)", "Hōreki (1751–1764)", "Meiwa (1764–1772)", "An’ei (1772–1781)", "Tenmei (1781–1789)", "Kansei (1789–1801)", "Kyōwa (1801–1804)", "Bunka (1804–1818)", "Bunsei (1818–1830)", "Tenpō (1830–1844)", "Kōka (1844–1848)", "Kaei (1848–1854)", "Ansei (1854–1860)", "Man’en (1860–1861)", "Bunkyū (1861–1864)", "Genji (1864–1865)", "Keiō (1865–1868)", "Meiji", "Taishō", "Shōwa", "Heisei" ], _42rr219 = {
    dayParts: _42rr187,
    days: _42rr215,
    months: _42rr215,
    years: _42rr215,
    zodiacs: _42rr196
}, _42rr220 = {
    dateFormats: _42rr209,
    datePartsOrder: "dmy",
    dateIntervalFormats: _42rr213,
    dateIntervalFallbackFormat: "{0} – {1}",
    dayNames: _42rr201,
    dayPeriods: _42rr210,
    monthNames: _42rr207,
    eraNames: _42rr17,
    quarterNames: _42rr188,
    fields: _42rr216
}, _42rr221 = {
    dateFormats: _42rr211,
    datePartsOrder: "dm",
    dateIntervalFormats: _42rr212,
    dateIntervalFallbackFormat: "{0} – {1}",
    dayNames: _42rr201,
    dayPeriods: _42rr210,
    cyclicNames: _42rr219,
    monthNames: _42rr208,
    monthPatterns: _42rr181,
    quarterNames: _42rr188,
    fields: _42rr216
};

inter.id = "en_il";

inter.unitPatterns = {
    "long": {
        unit: {
            accelerationMeterPerSecondSquared: {
                one: "{0} metre per second squared",
                other: "{0} metres per second squared"
            },
            areaSquareKilometer: {
                one: "{0} square kilometre",
                other: "{0} square kilometres"
            },
            areaSquareMeter: {
                one: "{0} square metre",
                other: "{0} square metres"
            },
            areaSquareCentimeter: {
                one: "{0} square centimetre",
                other: "{0} square centimetres"
            },
            concentrMilligramPerDeciliter: {
                one: "{0} milligram per decilitre",
                other: "{0} milligrams per decilitre"
            },
            concentrMillimolePerLiter: {
                one: "{0} millimole per litre",
                other: "{0} millimoles per litre"
            },
            consumptionLiterPerKilometer: {
                one: "{0} litre per kilometre",
                other: "{0} litres per kilometre"
            },
            "consumptionLiterPer-100kilometers": {
                one: "{0} litre per 100 kilometres",
                other: "{0} litres per 100 kilometres"
            },
            consumptionMilePerGallon: {
                one: "{0} mile per US gallon",
                other: "{0} miles per US gallon"
            },
            consumptionMilePerGallonImperial: {
                one: "{0} mile per gallon",
                other: "{0} miles per gallon"
            },
            lengthKilometer: {
                one: "{0} kilometre",
                other: "{0} kilometres"
            },
            lengthMeter: {
                one: "{0} metre",
                other: "{0} metres"
            },
            lengthDecimeter: {
                one: "{0} decimetre",
                other: "{0} decimetres"
            },
            lengthCentimeter: {
                one: "{0} centimetre",
                other: "{0} centimetres"
            },
            lengthMillimeter: {
                one: "{0} millimetre",
                other: "{0} millimetres"
            },
            lengthMicrometer: {
                one: "{0} micrometre",
                other: "{0} micrometres"
            },
            lengthNanometer: {
                one: "{0} nanometre",
                other: "{0} nanometres"
            },
            lengthPicometer: {
                one: "{0} picometre",
                other: "{0} picometres"
            },
            pressureMillimeterOfMercury: {
                one: "{0} millimetre of mercury",
                other: "{0} millimetres of mercury"
            },
            speedKilometerPerHour: {
                one: "{0} kilometre per hour",
                other: "{0} kilometres per hour"
            },
            speedMeterPerSecond: {
                one: "{0} metre per second",
                other: "{0} metres per second"
            },
            volumeCubicKilometer: {
                one: "{0} cubic kilometre",
                other: "{0} cubic kilometres"
            },
            volumeCubicMeter: {
                one: "{0} cubic metre",
                other: "{0} cubic metres"
            },
            volumeCubicCentimeter: {
                one: "{0} cubic centimetre",
                other: "{0} cubic centimetres"
            },
            volumeMegaliter: {
                one: "{0} megalitre",
                other: "{0} megalitres"
            },
            volumeHectoliter: {
                one: "{0} hectolitre",
                other: "{0} hectolitres"
            },
            volumeLiter: {
                one: "{0} litre",
                other: "{0} litres"
            },
            volumeDeciliter: {
                one: "{0} decilitre",
                other: "{0} decilitres"
            },
            volumeCentiliter: {
                one: "{0} centilitre",
                other: "{0} centilitres"
            },
            volumeMilliliter: {
                one: "{0} millilitre",
                other: "{0} millilitres"
            },
            volumeGallon: {
                one: "{0} US gallon",
                other: "{0} US gallons"
            },
            volumeGallonImperial: {
                one: "{0} gallon",
                other: "{0} gallons"
            },
            accelerationGForce: {
                one: "{0} g-force",
                other: "{0} g-force"
            },
            angleRevolution: {
                one: "{0} revolution",
                other: "{0} revolutions"
            },
            angleRadian: {
                one: "{0} radian",
                other: "{0} radians"
            },
            angleDegree: {
                one: "{0} degree",
                other: "{0} degrees"
            },
            angleArcMinute: {
                one: "{0} arcminute",
                other: "{0} arcminutes"
            },
            angleArcSecond: {
                one: "{0} arcsecond",
                other: "{0} arcseconds"
            },
            areaHectare: {
                one: "{0} hectare",
                other: "{0} hectares"
            },
            areaSquareMile: {
                one: "{0} square mile",
                other: "{0} square miles"
            },
            areaAcre: {
                one: "{0} acre",
                other: "{0} acres"
            },
            areaSquareYard: {
                one: "{0} square yard",
                other: "{0} square yards"
            },
            areaSquareFoot: {
                one: "{0} square foot",
                other: "{0} square feet"
            },
            areaSquareInch: {
                one: "{0} square inch",
                other: "{0} square inches"
            },
            concentrKarat: {
                one: "{0} karat",
                other: "{0} karats"
            },
            concentrPartPerMillion: {
                one: "{0} part per million",
                other: "{0} parts per million"
            },
            digitalTerabyte: {
                one: "{0} terabyte",
                other: "{0} terabytes"
            },
            digitalTerabit: {
                one: "{0} terabit",
                other: "{0} terabits"
            },
            digitalGigabyte: {
                one: "{0} gigabyte",
                other: "{0} gigabytes"
            },
            digitalGigabit: {
                one: "{0} gigabit",
                other: "{0} gigabits"
            },
            digitalMegabyte: {
                one: "{0} megabyte",
                other: "{0} megabytes"
            },
            digitalMegabit: {
                one: "{0} megabit",
                other: "{0} megabits"
            },
            digitalKilobyte: {
                one: "{0} kilobyte",
                other: "{0} kilobytes"
            },
            digitalKilobit: {
                one: "{0} kilobit",
                other: "{0} kilobits"
            },
            digitalByte: {
                one: "{0} byte",
                other: "{0} bytes"
            },
            digitalBit: {
                one: "{0} bit",
                other: "{0} bits"
            },
            durationCentury: {
                one: "{0} century",
                other: "{0} centuries"
            },
            durationYear: {
                one: "{0} year",
                other: "{0} years"
            },
            durationMonth: {
                one: "{0} month",
                other: "{0} months"
            },
            durationWeek: {
                one: "{0} week",
                other: "{0} weeks"
            },
            durationDay: _42rr16,
            durationHour: {
                one: "{0} hour",
                other: "{0} hours"
            },
            durationMinute: {
                one: "{0} minute",
                other: "{0} minutes"
            },
            durationSecond: {
                one: "{0} second",
                other: "{0} seconds"
            },
            durationMillisecond: {
                one: "{0} millisecond",
                other: "{0} milliseconds"
            },
            durationMicrosecond: {
                one: "{0} microsecond",
                other: "{0} microseconds"
            },
            durationNanosecond: {
                one: "{0} nanosecond",
                other: "{0} nanoseconds"
            },
            electricAmpere: {
                one: "{0} ampere",
                other: "{0} amperes"
            },
            electricMilliampere: {
                one: "{0} milliampere",
                other: "{0} milliamperes"
            },
            electricOhm: {
                one: "{0} ohm",
                other: "{0} ohms"
            },
            electricVolt: {
                one: "{0} volt",
                other: "{0} volts"
            },
            energyKilocalorie: {
                one: "{0} kilocalorie",
                other: "{0} kilocalories"
            },
            energyCalorie: {
                one: "{0} calorie",
                other: "{0} calories"
            },
            energyFoodcalorie: {
                one: "{0} Calorie",
                other: "{0} Calories"
            },
            energyKilojoule: {
                one: "{0} kilojoule",
                other: "{0} kilojoules"
            },
            energyJoule: {
                one: "{0} joule",
                other: "{0} joules"
            },
            energyKilowattHour: {
                one: "{0} kilowatt hour",
                other: "{0} kilowatt-hours"
            },
            frequencyGigahertz: {
                one: "{0} gigahertz",
                other: "{0} gigahertz"
            },
            frequencyMegahertz: {
                one: "{0} megahertz",
                other: "{0} megahertz"
            },
            frequencyKilohertz: {
                one: "{0} kilohertz",
                other: "{0} kilohertz"
            },
            frequencyHertz: {
                one: "{0} hertz",
                other: "{0} hertz"
            },
            lengthMile: {
                one: "{0} mile",
                other: "{0} miles"
            },
            lengthYard: {
                one: "{0} yard",
                other: "{0} yards"
            },
            lengthFoot: {
                one: "{0} foot",
                other: "{0} feet"
            },
            lengthInch: {
                one: "{0} inch",
                other: "{0} inches"
            },
            lengthParsec: {
                one: "{0} parsec",
                other: "{0} parsecs"
            },
            lengthLightYear: {
                one: "{0} light year",
                other: "{0} light years"
            },
            lengthAstronomicalUnit: {
                one: "{0} astronomical unit",
                other: "{0} astronomical units"
            },
            lengthFurlong: {
                one: "{0} furlong",
                other: "{0} furlongs"
            },
            lengthFathom: {
                one: "{0} fathom",
                other: "{0} fathoms"
            },
            lengthNauticalMile: {
                one: "{0} nautical mile",
                other: "{0} nautical miles"
            },
            lengthMileScandinavian: {
                one: "{0} mile-scandinavian",
                other: "{0} miles-scandinavian"
            },
            lightLux: {
                one: "{0} lux",
                other: "{0} lux"
            },
            massMetricTon: {
                one: "{0} metric ton",
                other: "{0} metric tons"
            },
            massKilogram: {
                one: "{0} kilogram",
                other: "{0} kilograms"
            },
            massGram: {
                one: "{0} gram",
                other: "{0} grams"
            },
            massMilligram: {
                one: "{0} milligram",
                other: "{0} milligrams"
            },
            massMicrogram: {
                one: "{0} microgram",
                other: "{0} micrograms"
            },
            massTon: {
                one: "{0} ton",
                other: "{0} tons"
            },
            massStone: {
                one: "{0} stone",
                other: "{0} stones"
            },
            massPound: {
                one: "{0} pound",
                other: "{0} pounds"
            },
            massOunce: {
                one: "{0} ounce",
                other: "{0} ounces"
            },
            massOunceTroy: {
                one: "{0} troy ounce",
                other: "{0} troy ounces"
            },
            massCarat: {
                one: "{0} carat",
                other: "{0} carats"
            },
            powerGigawatt: {
                one: "{0} gigawatt",
                other: "{0} gigawatts"
            },
            powerMegawatt: {
                one: "{0} megawatt",
                other: "{0} megawatts"
            },
            powerKilowatt: {
                one: "{0} kilowatt",
                other: "{0} kilowatts"
            },
            powerWatt: {
                one: "{0} watt",
                other: "{0} watts"
            },
            powerMilliwatt: {
                one: "{0} milliwatt",
                other: "{0} milliwatts"
            },
            powerHorsepower: {
                one: "{0} horsepower",
                other: "{0} horsepower"
            },
            pressureHectopascal: {
                one: "{0} hectopascal",
                other: "{0} hectopascals"
            },
            pressurePoundPerSquareInch: {
                one: "{0} pound per square inch",
                other: "{0} pounds per square inch"
            },
            pressureInchHg: {
                one: "{0} inch of mercury",
                other: "{0} inches of mercury"
            },
            pressureMillibar: {
                one: "{0} millibar",
                other: "{0} millibars"
            },
            speedMilePerHour: {
                one: "{0} mile per hour",
                other: "{0} miles per hour"
            },
            speedKnot: {
                one: "{0} knot",
                other: "{0} knots"
            },
            temperatureCelsius: {
                one: "{0} degree Celsius",
                other: "{0} degrees Celsius"
            },
            temperatureFahrenheit: {
                one: "{0} degree Fahrenheit",
                other: "{0} degrees Fahrenheit"
            },
            temperatureKelvin: {
                one: "{0} kelvin",
                other: "{0} kelvins"
            },
            volumeCubicMile: {
                one: "{0} cubic mile",
                other: "{0} cubic miles"
            },
            volumeCubicYard: {
                one: "{0} cubic yard",
                other: "{0} cubic yards"
            },
            volumeCubicFoot: {
                one: "{0} cubic foot",
                other: "{0} cubic feet"
            },
            volumeCubicInch: {
                one: "{0} cubic inch",
                other: "{0} cubic inches"
            },
            volumePintMetric: {
                one: "{0} metric pint",
                other: "{0} metric pints"
            },
            volumeCupMetric: {
                one: "{0} metric cup",
                other: "{0} metric cups"
            },
            volumeAcreFoot: {
                one: "{0} acre-foot",
                other: "{0} acre-feet"
            },
            volumeBushel: {
                one: "{0} bushel",
                other: "{0} bushels"
            },
            volumeQuart: {
                one: "{0} quart",
                other: "{0} quarts"
            },
            volumePint: {
                one: "{0} pint",
                other: "{0} pints"
            },
            volumeCup: {
                one: "{0} cup",
                other: "{0} cups"
            },
            volumeFluidOunce: {
                one: "{0} fluid ounce",
                other: "{0} fluid ounces"
            },
            volumeTablespoon: {
                one: "{0} tablespoon",
                other: "{0} tablespoons"
            },
            volumeTeaspoon: {
                one: "{0} teaspoon",
                other: "{0} teaspoons"
            }
        },
        compoundUnit: {
            per: "{0} per {1}"
        }
    },
    "short": {
        unit: {
            concentrMilligramPerDeciliter: {
                one: "{0} mg/dl",
                other: "{0} mg/dl"
            },
            concentrMillimolePerLiter: {
                one: "{0} mmol/l",
                other: "{0} mmol/l"
            },
            consumptionLiterPerKilometer: {
                one: "{0} l/km",
                other: "{0} l/km"
            },
            "consumptionLiterPer-100kilometers": {
                one: "{0} l/100 km",
                other: "{0} l/100 km"
            },
            consumptionMilePerGallon: {
                one: "{0} mpg US",
                other: "{0} mpg US"
            },
            consumptionMilePerGallonImperial: {
                one: "{0} mpg",
                other: "{0} mpg"
            },
            durationHour: {
                one: "{0} hr",
                other: "{0} hrs"
            },
            durationMinute: {
                one: "{0} min",
                other: "{0} mins"
            },
            durationSecond: {
                one: "{0} sec",
                other: "{0} secs"
            },
            volumeMegaliter: {
                one: "{0} Ml",
                other: "{0} Ml"
            },
            volumeHectoliter: {
                one: "{0} hl",
                other: "{0} hl"
            },
            volumeLiter: {
                one: "{0} l",
                other: "{0} l"
            },
            volumeDeciliter: {
                one: "{0} dl",
                other: "{0} dl"
            },
            volumeCentiliter: {
                one: "{0} cl",
                other: "{0} cl"
            },
            volumeMilliliter: {
                one: "{0} ml",
                other: "{0} ml"
            },
            volumeGallon: {
                one: "{0} gal US",
                other: "{0} gal US"
            },
            volumeGallonImperial: {
                one: "{0} gal",
                other: "{0} gal"
            },
            accelerationGForce: {
                one: "{0} G",
                other: "{0} G"
            },
            accelerationMeterPerSecondSquared: {
                one: "{0} m/s²",
                other: "{0} m/s²"
            },
            angleRevolution: {
                one: "{0} rev",
                other: "{0} rev"
            },
            angleRadian: {
                one: "{0} rad",
                other: "{0} rad"
            },
            angleDegree: {
                one: "{0} deg",
                other: "{0} deg"
            },
            angleArcMinute: {
                one: "{0} arcmin",
                other: "{0} arcmins"
            },
            angleArcSecond: {
                one: "{0} arcsec",
                other: "{0} arcsecs"
            },
            areaSquareKilometer: {
                one: "{0} km²",
                other: "{0} km²"
            },
            areaHectare: {
                one: "{0} ha",
                other: "{0} ha"
            },
            areaSquareMeter: {
                one: "{0} m²",
                other: "{0} m²"
            },
            areaSquareCentimeter: {
                one: "{0} cm²",
                other: "{0} cm²"
            },
            areaSquareMile: {
                one: "{0} sq mi",
                other: "{0} sq mi"
            },
            areaAcre: {
                one: "{0} ac",
                other: "{0} ac"
            },
            areaSquareYard: {
                one: "{0} yd²",
                other: "{0} yd²"
            },
            areaSquareFoot: {
                one: "{0} sq ft",
                other: "{0} sq ft"
            },
            areaSquareInch: {
                one: "{0} in²",
                other: "{0} in²"
            },
            concentrKarat: {
                one: "{0} kt",
                other: "{0} kt"
            },
            concentrPartPerMillion: {
                one: "{0} ppm",
                other: "{0} ppm"
            },
            digitalTerabyte: {
                one: "{0} TB",
                other: "{0} TB"
            },
            digitalTerabit: {
                one: "{0} Tb",
                other: "{0} Tb"
            },
            digitalGigabyte: {
                one: "{0} GB",
                other: "{0} GB"
            },
            digitalGigabit: {
                one: "{0} Gb",
                other: "{0} Gb"
            },
            digitalMegabyte: {
                one: "{0} MB",
                other: "{0} MB"
            },
            digitalMegabit: {
                one: "{0} Mb",
                other: "{0} Mb"
            },
            digitalKilobyte: {
                one: "{0} kB",
                other: "{0} kB"
            },
            digitalKilobit: {
                one: "{0} kb",
                other: "{0} kb"
            },
            digitalByte: {
                one: "{0} byte",
                other: "{0} byte"
            },
            digitalBit: {
                one: "{0} bit",
                other: "{0} bit"
            },
            durationCentury: _42rr10,
            durationYear: {
                one: "{0} yr",
                other: "{0} yrs"
            },
            durationMonth: {
                one: "{0} mth",
                other: "{0} mths"
            },
            durationWeek: {
                one: "{0} wk",
                other: "{0} wks"
            },
            durationDay: _42rr16,
            durationMillisecond: {
                one: "{0} ms",
                other: "{0} ms"
            },
            durationMicrosecond: {
                one: "{0} μs",
                other: "{0} μs"
            },
            durationNanosecond: {
                one: "{0} ns",
                other: "{0} ns"
            },
            electricAmpere: {
                one: "{0} A",
                other: "{0} A"
            },
            electricMilliampere: {
                one: "{0} mA",
                other: "{0} mA"
            },
            electricOhm: {
                one: "{0} Ω",
                other: "{0} Ω"
            },
            electricVolt: {
                one: "{0} V",
                other: "{0} V"
            },
            energyKilocalorie: {
                one: "{0} kcal",
                other: "{0} kcal"
            },
            energyCalorie: {
                one: "{0} cal",
                other: "{0} cal"
            },
            energyFoodcalorie: {
                one: "{0} Cal",
                other: "{0} Cal"
            },
            energyKilojoule: {
                one: "{0} kJ",
                other: "{0} kJ"
            },
            energyJoule: {
                one: "{0} J",
                other: "{0} J"
            },
            energyKilowattHour: {
                one: "{0} kWh",
                other: "{0} kWh"
            },
            frequencyGigahertz: {
                one: "{0} GHz",
                other: "{0} GHz"
            },
            frequencyMegahertz: {
                one: "{0} MHz",
                other: "{0} MHz"
            },
            frequencyKilohertz: {
                one: "{0} kHz",
                other: "{0} kHz"
            },
            frequencyHertz: {
                one: "{0} Hz",
                other: "{0} Hz"
            },
            lengthKilometer: {
                one: "{0} km",
                other: "{0} km"
            },
            lengthMeter: {
                one: "{0} m",
                other: "{0} m"
            },
            lengthDecimeter: {
                one: "{0} dm",
                other: "{0} dm"
            },
            lengthCentimeter: {
                one: "{0} cm",
                other: "{0} cm"
            },
            lengthMillimeter: {
                one: "{0} mm",
                other: "{0} mm"
            },
            lengthMicrometer: {
                one: "{0} µm",
                other: "{0} µm"
            },
            lengthNanometer: {
                one: "{0} nm",
                other: "{0} nm"
            },
            lengthPicometer: {
                one: "{0} pm",
                other: "{0} pm"
            },
            lengthMile: {
                one: "{0} mi",
                other: "{0} mi"
            },
            lengthYard: {
                one: "{0} yd",
                other: "{0} yd"
            },
            lengthFoot: {
                one: "{0} ft",
                other: "{0} ft"
            },
            lengthInch: {
                one: "{0} in",
                other: "{0} in"
            },
            lengthParsec: {
                one: "{0} pc",
                other: "{0} pc"
            },
            lengthLightYear: {
                one: "{0} ly",
                other: "{0} ly"
            },
            lengthAstronomicalUnit: {
                one: "{0} au",
                other: "{0} au"
            },
            lengthFurlong: {
                one: "{0} fur",
                other: "{0} fur"
            },
            lengthFathom: {
                one: "{0} ftm",
                other: "{0} ftm"
            },
            lengthNauticalMile: {
                one: "{0} nmi",
                other: "{0} nmi"
            },
            lengthMileScandinavian: {
                one: "{0} smi",
                other: "{0} smi"
            },
            lightLux: {
                one: "{0} lx",
                other: "{0} lx"
            },
            massMetricTon: {
                one: "{0} t",
                other: "{0} t"
            },
            massKilogram: {
                one: "{0} kg",
                other: "{0} kg"
            },
            massGram: {
                one: "{0} g",
                other: "{0} g"
            },
            massMilligram: {
                one: "{0} mg",
                other: "{0} mg"
            },
            massMicrogram: {
                one: "{0} µg",
                other: "{0} µg"
            },
            massTon: {
                one: "{0} tn",
                other: "{0} tn"
            },
            massStone: {
                one: "{0} st",
                other: "{0} st"
            },
            massPound: {
                one: "{0} lb",
                other: "{0} lb"
            },
            massOunce: {
                one: "{0} oz",
                other: "{0} oz"
            },
            massOunceTroy: {
                one: "{0} oz t",
                other: "{0} oz t"
            },
            massCarat: {
                one: "{0} CD",
                other: "{0} CD"
            },
            powerGigawatt: {
                one: "{0} GW",
                other: "{0} GW"
            },
            powerMegawatt: {
                one: "{0} MW",
                other: "{0} MW"
            },
            powerKilowatt: {
                one: "{0} kW",
                other: "{0} kW"
            },
            powerWatt: {
                one: "{0} W",
                other: "{0} W"
            },
            powerMilliwatt: {
                one: "{0} mW",
                other: "{0} mW"
            },
            powerHorsepower: {
                one: "{0} hp",
                other: "{0} hp"
            },
            pressureHectopascal: {
                one: "{0} hPa",
                other: "{0} hPa"
            },
            pressureMillimeterOfMercury: {
                one: "{0} mm Hg",
                other: "{0} mm Hg"
            },
            pressurePoundPerSquareInch: {
                one: "{0} psi",
                other: "{0} psi"
            },
            pressureInchHg: {
                one: "{0} inHg",
                other: "{0} inHg"
            },
            pressureMillibar: {
                one: "{0} mbar",
                other: "{0} mbar"
            },
            speedKilometerPerHour: {
                one: "{0} kph",
                other: "{0} kph"
            },
            speedMeterPerSecond: {
                one: "{0} m/s",
                other: "{0} m/s"
            },
            speedMilePerHour: {
                one: "{0} mph",
                other: "{0} mph"
            },
            speedKnot: {
                one: "{0} kn",
                other: "{0} kn"
            },
            temperatureCelsius: {
                one: "{0}°C",
                other: "{0}°C"
            },
            temperatureFahrenheit: _42rr11,
            temperatureKelvin: {
                one: "{0} K",
                other: "{0} K"
            },
            volumeCubicKilometer: {
                one: "{0} km³",
                other: "{0} km³"
            },
            volumeCubicMeter: {
                one: "{0} m³",
                other: "{0} m³"
            },
            volumeCubicCentimeter: {
                one: "{0} cm³",
                other: "{0} cm³"
            },
            volumeCubicMile: {
                one: "{0} mi³",
                other: "{0} mi³"
            },
            volumeCubicYard: {
                one: "{0} yd³",
                other: "{0} yd³"
            },
            volumeCubicFoot: {
                one: "{0} ft³",
                other: "{0} ft³"
            },
            volumeCubicInch: {
                one: "{0} in³",
                other: "{0} in³"
            },
            volumePintMetric: {
                one: "{0} mpt",
                other: "{0} mpt"
            },
            volumeCupMetric: {
                one: "{0} mc",
                other: "{0} mc"
            },
            volumeAcreFoot: {
                one: "{0} ac ft",
                other: "{0} ac ft"
            },
            volumeBushel: {
                one: "{0} bu",
                other: "{0} bu"
            },
            volumeQuart: {
                one: "{0} qt",
                other: "{0} qt"
            },
            volumePint: {
                one: "{0} pt",
                other: "{0} pt"
            },
            volumeCup: _42rr10,
            volumeFluidOunce: {
                one: "{0} fl oz",
                other: "{0} fl oz"
            },
            volumeTablespoon: {
                one: "{0} tbsp",
                other: "{0} tbsp"
            },
            volumeTeaspoon: {
                one: "{0} tsp",
                other: "{0} tsp"
            },
            temperatureGeneric: {
                other: "{0}°"
            }
        },
        compoundUnit: {
            per: "{0}/{1}"
        }
    },
    narrow: {
        unit: {
            concentrMilligramPerDeciliter: {
                one: "{0}mg/dl",
                other: "{0}mg/dl"
            },
            concentrMillimolePerLiter: {
                one: "{0}mmol/l",
                other: "{0}mmol/l"
            },
            consumptionLiterPerKilometer: {
                one: "{0}l/km",
                other: "{0}l/km"
            },
            "consumptionLiterPer-100kilometers": {
                one: "{0}l/100km",
                other: "{0}l/100km"
            },
            consumptionMilePerGallon: {
                one: "{0}mpgUS",
                other: "{0}mpgUS"
            },
            consumptionMilePerGallonImperial: {
                one: "{0}mpg",
                other: "{0}mpg"
            },
            massPound: {
                one: "{0}lb",
                other: "{0}lb"
            },
            temperatureCelsius: _42rr4,
            temperatureFahrenheit: _42rr11,
            volumeMegaliter: {
                one: "{0}Ml",
                other: "{0}Ml"
            },
            volumeHectoliter: {
                one: "{0}hl",
                other: "{0}hl"
            },
            volumeLiter: {
                one: "{0}l",
                other: "{0}l"
            },
            volumeDeciliter: {
                one: "{0}dl",
                other: "{0}dl"
            },
            volumeCentiliter: {
                one: "{0}cl",
                other: "{0}cl"
            },
            volumeMilliliter: {
                one: "{0}ml",
                other: "{0}ml"
            },
            volumeGallon: {
                one: "{0}galUS",
                other: "{0}galUS"
            },
            volumeGallonImperial: {
                one: "{0}gal",
                other: "{0}gal"
            },
            accelerationGForce: {
                one: "{0}G",
                other: "{0}Gs"
            },
            accelerationMeterPerSecondSquared: {
                one: "{0}m/s²",
                other: "{0}m/s²"
            },
            angleRevolution: {
                one: "{0}rev",
                other: "{0}rev"
            },
            angleRadian: {
                one: "{0}rad",
                other: "{0}rad"
            },
            angleDegree: _42rr4,
            angleArcMinute: _42rr5,
            angleArcSecond: _42rr6,
            areaHectare: {
                one: "{0}ha",
                other: "{0}ha"
            },
            areaSquareCentimeter: {
                one: "{0}cm²",
                other: "{0}cm²"
            },
            areaSquareMile: {
                one: "{0}mi²",
                other: "{0}mi²"
            },
            areaAcre: {
                one: "{0}ac",
                other: "{0}ac"
            },
            areaSquareYard: {
                one: "{0}yd²",
                other: "{0}yd²"
            },
            areaSquareFoot: {
                one: "{0}ft²",
                other: "{0}ft²"
            },
            areaSquareInch: {
                one: "{0}in²",
                other: "{0}in²"
            },
            concentrKarat: {
                one: "{0}kt",
                other: "{0}kt"
            },
            concentrPartPerMillion: {
                one: "{0}ppm",
                other: "{0}ppm"
            },
            digitalTerabyte: {
                one: "{0}TB",
                other: "{0}TB"
            },
            digitalTerabit: {
                one: "{0}Tb",
                other: "{0}Tb"
            },
            digitalGigabyte: {
                one: "{0}GB",
                other: "{0}GB"
            },
            digitalGigabit: {
                one: "{0}Gb",
                other: "{0}Gb"
            },
            digitalMegabyte: {
                one: "{0}MB",
                other: "{0}MB"
            },
            digitalMegabit: {
                one: "{0}Mb",
                other: "{0}Mb"
            },
            digitalKilobyte: {
                one: "{0}kB",
                other: "{0}kB"
            },
            digitalKilobit: {
                one: "{0}kb",
                other: "{0}kb"
            },
            digitalByte: {
                one: "{0}byte",
                other: "{0}byte"
            },
            digitalBit: {
                one: "{0}bit",
                other: "{0}bit"
            },
            durationYear: {
                one: "{0}y",
                other: "{0}y"
            },
            durationMonth: _42rr7,
            durationWeek: {
                one: "{0}w",
                other: "{0}w"
            },
            durationDay: {
                one: "{0}d",
                other: "{0}d"
            },
            durationHour: {
                one: "{0}h",
                other: "{0}h"
            },
            durationMinute: _42rr7,
            durationSecond: {
                one: "{0}s",
                other: "{0}s"
            },
            durationMillisecond: {
                one: "{0}ms",
                other: "{0}ms"
            },
            durationMicrosecond: {
                one: "{0}μs",
                other: "{0}μs"
            },
            durationNanosecond: {
                one: "{0}ns",
                other: "{0}ns"
            },
            electricAmpere: {
                one: "{0}A",
                other: "{0}A"
            },
            electricMilliampere: {
                one: "{0}mA",
                other: "{0}mA"
            },
            electricOhm: {
                one: "{0}Ω",
                other: "{0}Ω"
            },
            electricVolt: {
                one: "{0}V",
                other: "{0}V"
            },
            energyKilocalorie: {
                one: "{0}kcal",
                other: "{0}kcal"
            },
            energyCalorie: {
                one: "{0}cal",
                other: "{0}cal"
            },
            energyFoodcalorie: {
                one: "{0}Cal",
                other: "{0}Cal"
            },
            energyKilojoule: {
                one: "{0}kJ",
                other: "{0}kJ"
            },
            energyJoule: {
                one: "{0}J",
                other: "{0}J"
            },
            energyKilowattHour: {
                one: "{0}kWh",
                other: "{0}kWh"
            },
            frequencyGigahertz: {
                one: "{0}GHz",
                other: "{0}GHz"
            },
            frequencyMegahertz: {
                one: "{0}MHz",
                other: "{0}MHz"
            },
            frequencyKilohertz: {
                one: "{0}kHz",
                other: "{0}kHz"
            },
            frequencyHertz: {
                one: "{0}Hz",
                other: "{0}Hz"
            },
            lengthKilometer: {
                one: "{0}km",
                other: "{0}km"
            },
            lengthMeter: _42rr7,
            lengthDecimeter: {
                one: "{0}dm",
                other: "{0}dm"
            },
            lengthCentimeter: {
                one: "{0}cm",
                other: "{0}cm"
            },
            lengthMillimeter: {
                one: "{0}mm",
                other: "{0}mm"
            },
            lengthMicrometer: {
                one: "{0}µm",
                other: "{0}µm"
            },
            lengthNanometer: {
                one: "{0}nm",
                other: "{0}nm"
            },
            lengthPicometer: {
                one: "{0}pm",
                other: "{0}pm"
            },
            lengthMile: {
                one: "{0}mi",
                other: "{0}mi"
            },
            lengthYard: {
                one: "{0}yd",
                other: "{0}yd"
            },
            lengthFoot: _42rr5,
            lengthInch: _42rr6,
            lengthParsec: {
                one: "{0}pc",
                other: "{0}pc"
            },
            lengthLightYear: {
                one: "{0}ly",
                other: "{0}ly"
            },
            lengthAstronomicalUnit: {
                one: "{0}au",
                other: "{0}au"
            },
            lengthFurlong: {
                one: "{0}fur",
                other: "{0}fur"
            },
            lengthFathom: {
                one: "{0}fm",
                other: "{0}fm"
            },
            lengthNauticalMile: {
                one: "{0}nmi",
                other: "{0}nmi"
            },
            lengthMileScandinavian: {
                one: "{0}smi",
                other: "{0}smi"
            },
            lightLux: {
                one: "{0}lx",
                other: "{0}lx"
            },
            massMetricTon: {
                one: "{0}t",
                other: "{0}t"
            },
            massKilogram: {
                one: "{0}kg",
                other: "{0}kg"
            },
            massGram: {
                one: "{0}g",
                other: "{0}g"
            },
            massMilligram: {
                one: "{0}mg",
                other: "{0}mg"
            },
            massMicrogram: {
                one: "{0}µg",
                other: "{0}µg"
            },
            massTon: {
                one: "{0}tn",
                other: "{0}tn"
            },
            massStone: {
                one: "{0}st",
                other: "{0}st"
            },
            massOunce: {
                one: "{0}oz",
                other: "{0}oz"
            },
            massOunceTroy: {
                one: "{0}oz t",
                other: "{0}oz t"
            },
            massCarat: {
                one: "{0}CD",
                other: "{0}CD"
            },
            powerGigawatt: {
                one: "{0}GW",
                other: "{0}GW"
            },
            powerMegawatt: {
                one: "{0}MW",
                other: "{0}MW"
            },
            powerKilowatt: {
                one: "{0}kW",
                other: "{0}kW"
            },
            powerWatt: {
                one: "{0}W",
                other: "{0}W"
            },
            powerMilliwatt: {
                one: "{0}mW",
                other: "{0}mW"
            },
            powerHorsepower: {
                one: "{0}hp",
                other: "{0}hp"
            },
            pressureHectopascal: {
                one: "{0}hPa",
                other: "{0}hPa"
            },
            pressureMillimeterOfMercury: {
                one: "{0}mm Hg",
                other: "{0}mm Hg"
            },
            pressurePoundPerSquareInch: {
                one: "{0}psi",
                other: "{0}psi"
            },
            pressureInchHg: {
                one: "{0}″ Hg",
                other: "{0}″ Hg"
            },
            pressureMillibar: {
                one: "{0}mb",
                other: "{0}mb"
            },
            speedKilometerPerHour: {
                one: "{0}kph",
                other: "{0}kph"
            },
            speedMeterPerSecond: {
                one: "{0}m/s",
                other: "{0}m/s"
            },
            speedMilePerHour: {
                one: "{0}mph",
                other: "{0}mph"
            },
            speedKnot: {
                one: "{0}kn",
                other: "{0}kn"
            },
            temperatureKelvin: {
                one: "{0}K",
                other: "{0}K"
            },
            volumeCubicKilometer: {
                one: "{0}km³",
                other: "{0}km³"
            },
            volumeCubicMeter: {
                one: "{0}m³",
                other: "{0}m³"
            },
            volumeCubicCentimeter: {
                one: "{0}cm³",
                other: "{0}cm³"
            },
            volumeCubicMile: {
                one: "{0}mi³",
                other: "{0}mi³"
            },
            volumeCubicYard: {
                one: "{0}yd³",
                other: "{0}yd³"
            },
            volumeCubicFoot: {
                one: "{0}ft³",
                other: "{0}ft³"
            },
            volumeCubicInch: {
                one: "{0}in³",
                other: "{0}in³"
            },
            volumePintMetric: {
                one: "{0}mpt",
                other: "{0}mpt"
            },
            volumeCupMetric: {
                one: "{0}mc",
                other: "{0}mc"
            },
            volumeAcreFoot: {
                one: "{0}ac ft",
                other: "{0}ac ft"
            },
            volumeBushel: {
                one: "{0}bu",
                other: "{0}bu"
            },
            volumeQuart: {
                one: "{0}qt",
                other: "{0}qt"
            },
            volumePint: {
                one: "{0}pt",
                other: "{0}pt"
            },
            volumeCup: {
                one: "{0}c",
                other: "{0}c"
            },
            volumeFluidOunce: {
                one: "{0}fl oz",
                other: "{0}fl oz"
            },
            volumeTablespoon: {
                one: "{0}tbsp",
                other: "{0}tbsp"
            },
            volumeTeaspoon: {
                one: "{0}tsp",
                other: "{0}tsp"
            }
        }
    }
};

inter.calendars = {
    buddhist: {
        dateFormats: _42rr209,
        datePartsOrder: "dmy",
        dateIntervalFormats: _42rr213,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _42rr201,
        dayPeriods: _42rr210,
        monthNames: _42rr204,
        eraNames: {
            wide: [ "BE" ],
            abbreviated: [ "BE" ]
        },
        quarterNames: _42rr188,
        fields: _42rr216
    },
    chinese: _42rr221,
    coptic: {
        dateFormats: _42rr209,
        datePartsOrder: "dmy",
        dateIntervalFormats: _42rr213,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _42rr201,
        dayPeriods: _42rr210,
        monthNames: {
            format: _42rr191,
            standAlone: _42rr191
        },
        eraNames: _42rr91,
        quarterNames: _42rr188,
        fields: _42rr216
    },
    dangi: _42rr221,
    ethioaa: _42rr217,
    ethiopic: {
        dateFormats: _42rr209,
        datePartsOrder: "dmy",
        dateIntervalFormats: _42rr213,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _42rr201,
        dayPeriods: _42rr210,
        monthNames: {
            format: _42rr194,
            standAlone: _42rr194
        },
        eraNames: _42rr91,
        quarterNames: _42rr188,
        fields: _42rr216
    },
    gregorian: {
        dateFormats: {
            basic: {
                fullDate: "EEEE, d MMMM y",
                longDate: "d MMMM y",
                mediumDate: "d MMM y",
                shortDate: "dd/MM/y",
                fullTime: "H:mm:ss zzzz",
                longTime: "H:mm:ss z",
                mediumTime: "H:mm:ss",
                shortTime: "H:mm",
                shortDateTime: "dd/MM/y, H:mm",
                mediumDateTime: "d MMM y, H:mm:ss",
                longDateTime: "d MMMM y 'at' H:mm:ss z",
                fullDateTime: "EEEE, d MMMM y 'at' H:mm:ss zzzz"
            },
            cldr: {
                EHm: "E H:mm",
                EHms: "E H:mm:ss",
                H: "H",
                Hm: "H:mm",
                Hms: "H:mm:ss",
                Ed: "E d",
                GyMMMd: "d MMM y G",
                GyMMMEd: "E, d MMM y G",
                Md: "dd/MM",
                MEd: "E, dd/MM",
                MMdd: "dd/MM",
                MMMd: "d MMM",
                MMMEd: "E, d MMM",
                MMMMd: "d MMMM",
                yM: "MM/y",
                yMd: "dd/MM/y",
                yMEd: "E, dd/MM/y",
                yMMMd: "d MMM y",
                yMMMEd: "E, d MMM y",
                d: "d",
                E: "ccc",
                Ehm: "E h:mm a",
                Ehms: "E h:mm:ss a",
                Gy: "y G",
                GyMMM: "MMM y G",
                h: "h a",
                hm: "h:mm a",
                hms: "h:mm:ss a",
                hmsv: "h:mm:ss a v",
                Hmsv: "HH:mm:ss v",
                hmv: "h:mm a v",
                Hmv: "HH:mm v",
                M: "L",
                MMM: "LLL",
                ms: "mm:ss",
                y: "y",
                yMMM: "MMM y",
                yMMMM: "MMMM y",
                yQQQ: "QQQ y",
                yQQQQ: "QQQQ y"
            }
        },
        datePartsOrder: "dmy",
        dateIntervalFormats: {
            Md: _42rr23,
            MEd: _42rr69,
            MMMd: _42rr18,
            MMMEd: _42rr51,
            yM: _42rr19,
            yMd: _42rr123,
            yMEd: _42rr136,
            yMMMd: _42rr115,
            yMMMEd: {
                d: "E, d – E, d MMM y",
                M: "E, d MMM – E, d MMM y",
                y: "E, d MMM y – E, d MMM y"
            },
            d: {
                d: "d – d"
            },
            h: _42rr12,
            H: {
                H: "HH – HH"
            },
            hm: _42rr102,
            Hm: _42rr24,
            hmv: _42rr117,
            Hmv: _42rr52,
            hv: _42rr14,
            Hv: {
                H: "HH – HH v"
            },
            M: {
                M: "M – M"
            },
            MMM: {
                M: "MMM – MMM"
            },
            y: {
                y: "y – y"
            },
            yMMM: _42rr21,
            yMMMM: _42rr41
        },
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _42rr201,
        dayPeriods: _42rr210,
        monthNames: _42rr204,
        eraNames: {
            wide: [ "Before Christ", "Anno Domini" ],
            abbreviated: [ "BC", "AD" ]
        },
        quarterNames: _42rr188,
        fields: _42rr216
    },
    hebrew: {
        dateFormats: {
            basic: {
                fullDate: "EEEE, d MMMM y",
                longDate: "d MMMM y",
                mediumDate: "d MMM y",
                shortDate: "d MMM y",
                fullTime: "H:mm:ss zzzz",
                longTime: "H:mm:ss z",
                mediumTime: "H:mm:ss",
                shortTime: "H:mm",
                shortDateTime: "d MMM y, H:mm",
                mediumDateTime: "d MMM y, H:mm:ss",
                longDateTime: "d MMMM y 'at' H:mm:ss z",
                fullDateTime: "EEEE, d MMMM y 'at' H:mm:ss zzzz"
            },
            cldr: {
                Gy: "y G",
                GyMMM: "MMM y G",
                GyMMMd: "d MMM y G",
                GyMMMEd: "E, d MMM y G",
                M: "L",
                Md: "d MMM",
                MEd: "E, d MMM",
                MMM: "LLL",
                MMMd: "d MMM",
                MMMEd: "E, d MMM",
                MMMMd: "d MMMM",
                y: "y",
                yM: "MMM y",
                yMd: "d MMM y",
                yMEd: "E, d MMM y",
                yMMM: "MMM y",
                yMMMd: "d MMM y",
                yMMMEd: "E, d MMM y",
                yMMMM: "MMMM y",
                yQQQ: "QQQ y",
                yQQQQ: "QQQQ y",
                EHm: "E H:mm",
                EHms: "E H:mm:ss",
                H: "H",
                Hm: "H:mm",
                Hms: "H:mm:ss",
                Ed: "E d",
                yyyyM: "MM/y GGGGG",
                yyyyMd: "dd/MM/y GGGGG",
                yyyyMEd: "E, dd/MM/y GGGGG",
                yyyyMMMd: "d MMM y G",
                yyyyMMMEd: "E, d MMM y G",
                d: "d",
                E: "ccc",
                Ehm: "E h:mm a",
                Ehms: "E h:mm:ss a",
                h: "h a",
                hm: "h:mm a",
                hms: "h:mm:ss a",
                ms: "mm:ss",
                yyyy: "y G",
                yyyyMMM: "MMM y G",
                yyyyMMMM: "MMMM y G",
                yyyyQQQ: "QQQ y G",
                yyyyQQQQ: "QQQQ y G"
            }
        },
        datePartsOrder: "dmy",
        dateIntervalFormats: {
            M: {
                M: "MMM – MMM"
            },
            Md: _42rr18,
            MEd: _42rr90,
            MMM: {
                M: "MMM – MMM"
            },
            MMMd: _42rr18,
            MMMEd: _42rr90,
            y: {
                y: "y – y G"
            },
            yM: _42rr21,
            yMd: _42rr115,
            yMEd: _42rr133,
            yMMM: _42rr21,
            yMMMd: _42rr115,
            yMMMEd: _42rr133,
            yMMMM: _42rr41,
            d: {
                d: "d – d"
            },
            h: _42rr12,
            H: {
                H: "HH – HH"
            },
            hm: _42rr102,
            Hm: _42rr24,
            hmv: _42rr117,
            Hmv: _42rr52,
            hv: _42rr14,
            Hv: {
                H: "HH – HH v"
            }
        },
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _42rr201,
        dayPeriods: _42rr210,
        monthNames: {
            format: _42rr189,
            standAlone: _42rr189
        },
        eraNames: {
            wide: [ "AM" ],
            abbreviated: [ "AM" ]
        },
        quarterNames: _42rr188,
        fields: _42rr216
    },
    indian: {
        dateFormats: _42rr209,
        datePartsOrder: "dmy",
        dateIntervalFormats: _42rr213,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _42rr201,
        dayPeriods: _42rr210,
        monthNames: {
            format: _42rr195,
            standAlone: _42rr195
        },
        eraNames: {
            wide: [ "Saka" ],
            abbreviated: [ "Saka" ]
        },
        quarterNames: _42rr188,
        fields: _42rr216
    },
    islamic: _42rr220,
    "islamic-umalqura": _42rr220,
    "islamic-tbla": _42rr220,
    "islamic-civil": _42rr220,
    "islamic-rgsa": _42rr220,
    iso8601: _42rr217,
    japanese: {
        dateFormats: {
            basic: {
                fullDate: "EEEE, MMMM d, y G",
                longDate: "MMMM d, y G",
                mediumDate: "MMM d, y G",
                shortDate: "M/d/y GGGGG",
                fullTime: "H:mm:ss zzzz",
                longTime: "H:mm:ss z",
                mediumTime: "H:mm:ss",
                shortTime: "H:mm",
                shortDateTime: "M/d/y GGGGG, H:mm",
                mediumDateTime: "MMM d, y G, H:mm:ss",
                longDateTime: "MMMM d, y G 'at' H:mm:ss z",
                fullDateTime: "EEEE, MMMM d, y G 'at' H:mm:ss zzzz"
            },
            cldr: _42rr203
        },
        datePartsOrder: "mdy",
        dateIntervalFormats: _42rr213,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _42rr201,
        dayPeriods: _42rr210,
        monthNames: _42rr204,
        eraNames: {
            wide: _42rr218,
            abbreviated: _42rr218
        },
        quarterNames: _42rr188,
        fields: _42rr216
    },
    persian: {
        dateFormats: _42rr209,
        datePartsOrder: "dmy",
        dateIntervalFormats: _42rr213,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _42rr201,
        dayPeriods: _42rr210,
        monthNames: {
            format: _42rr190,
            standAlone: _42rr190
        },
        eraNames: {
            wide: [ "AP" ],
            abbreviated: [ "AP" ]
        },
        quarterNames: _42rr188,
        fields: _42rr216
    },
    roc: {
        dateFormats: _42rr209,
        datePartsOrder: "dmy",
        dateIntervalFormats: _42rr213,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _42rr201,
        dayPeriods: _42rr210,
        monthNames: _42rr204,
        eraNames: {
            wide: _42rr9,
            abbreviated: _42rr9
        },
        quarterNames: _42rr188,
        fields: _42rr216
    },
    islamicc: _42rr217
};

inter.pluralRule = function(n) {
    var i = Math.floor(Math.abs(n)), v = n.toString().replace(/^[^.]*\.?/, "").length;
    if (typeof n === "string") n = parseInt(n, 10);
    if (i === 1 && v === 0) return "one";
    return "other";
};

inter.renderDigitsOrdinal = function(n) {
    if (n < 0) return "−" + this.renderDigitsOrdinal(-n);
    if (n >= 0) return this.renderNumber(n, "#,##0") + "ordinal" + "one" + "st" + "two" + "nd" + "few" + "rd" + "other" + "th";
};

inter.renderSpelloutNumbering = function(n) {
    if (n < 0) return "minus " + this.renderSpelloutNumbering(-n);
    if (n >= 0) return this.renderSpelloutCardinal(n);
};

inter.renderArmenianUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderArmenianUpper(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e4) return this.renderNumber(n, "#,##0");
    if (n >= 9e3) return "Ք" + (n === 9e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 8e3) return "Փ" + (n === 8e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 7e3) return "Ւ" + (n === 7e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 6e3) return "Ց" + (n === 6e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 5e3) return "Ր" + (n === 5e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 4e3) return "Տ" + (n === 4e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 3e3) return "Վ" + (n === 3e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 2e3) return "Ս" + (n === 2e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 1e3) return "Ռ" + (n === 1e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 900) return "Ջ" + (n === 900 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 800) return "Պ" + (n === 800 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 700) return "Չ" + (n === 700 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 600) return "Ո" + (n === 600 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 500) return "Շ" + (n === 500 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 400) return "Ն" + (n === 400 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 300) return "Յ" + (n === 300 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 200) return "Մ" + (n === 200 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 100) return "Ճ" + (n === 100 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 90) return "Ղ" + (n === 90 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 80) return "Ձ" + (n === 80 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 70) return "Հ" + (n === 70 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 60) return "Կ" + (n === 60 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 50) return "Ծ" + (n === 50 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 40) return "Խ" + (n === 40 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 30) return "Լ" + (n === 30 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 20) return "Ի" + (n === 20 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 10) return "Ժ" + (n === 10 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 9) return "Թ";
    if (n >= 8) return "Ը";
    if (n >= 7) return "Է";
    if (n >= 6) return "Զ";
    if (n >= 5) return "Ե";
    if (n >= 4) return "Դ";
    if (n >= 3) return "Գ";
    if (n >= 2) return "Բ";
    if (n >= 1) return "Ա";
    if (n >= 0) return "0";
};

inter.renderArmenianLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderArmenianLower(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e4) return this.renderNumber(n, "#,##0");
    if (n >= 9e3) return "ք" + (n === 9e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 8e3) return "փ" + (n === 8e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 7e3) return "ւ" + (n === 7e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 6e3) return "ց" + (n === 6e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 5e3) return "ր" + (n === 5e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 4e3) return "տ" + (n === 4e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 3e3) return "վ" + (n === 3e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 2e3) return "ս" + (n === 2e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 1e3) return "ռ" + (n === 1e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 900) return "ջ" + (n === 900 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 800) return "պ" + (n === 800 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 700) return "չ" + (n === 700 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 600) return "ո" + (n === 600 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 500) return "շ" + (n === 500 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 400) return "ն" + (n === 400 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 300) return "յ" + (n === 300 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 200) return "մ" + (n === 200 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 100) return "ճ" + (n === 100 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 90) return "ղ" + (n === 90 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 80) return "ձ" + (n === 80 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 70) return "հ" + (n === 70 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 60) return "կ" + (n === 60 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 50) return "ծ" + (n === 50 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 40) return "խ" + (n === 40 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 30) return "լ" + (n === 30 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 20) return "ի" + (n === 20 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 10) return "ժ" + (n === 10 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 9) return "թ";
    if (n >= 8) return "ը";
    if (n >= 7) return "է";
    if (n >= 6) return "զ";
    if (n >= 5) return "ե";
    if (n >= 4) return "դ";
    if (n >= 3) return "գ";
    if (n >= 2) return "բ";
    if (n >= 1) return "ա";
    if (n >= 0) return "0";
};

inter.renderCyrillicLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderCyrillicLower(-n);
    if (isFractional && n > 1) return this.renderCyrillicLower(Math.floor(n)) + "." + this.renderCyrillicLower(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e15) return "҂҂҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e15)) + (n === 1e15 ? "" : " " + this.renderCyrillicLower(n % 1e15));
    if (n >= 1e12) return "҂҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e12)) + (n === 1e12 ? "" : " " + this.renderCyrillicLower(n % 1e12));
    if (n >= 1e9) return "҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e9)) + (n === 1e9 ? "" : " " + this.renderCyrillicLower(n % 1e9));
    if (n >= 1e6) return "҂҂" + this.renderCyrillicLower(Math.floor(n / 1e6)) + (n === 1e6 ? "" : " " + this.renderCyrillicLower(n % 1e6));
    if (n >= 11e3) return this.renderCyrillicLowerThousands(Math.floor(n / 1e4)) + (n === 11e3 ? "" : " " + this.renderCyrillicLower(n % 1e4));
    if (n >= 1e4) return "҂" + this.renderCyrillicLower(Math.floor(n / 1e4)) + (n === 1e4 ? "" : " " + this.renderCyrillicLower(n % 1e4));
    if (n >= 1e3) return "҂" + this.renderCyrillicLower110(Math.floor(n / 1e3)) + this.renderCyrillicLowerPost(n % 1e3);
    if (n >= 900) return "ц" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 801) return "ѿ" + this.renderCyrillicLower(n % 100);
    if (n >= 800) return "ѿ҃";
    if (n >= 700) return "ѱ" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 600) return "х" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 500) return "ф" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 400) return "у" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 300) return "т" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 200) return "с" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 100) return "р" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 90) return "ч" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 80) return "п" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 70) return "ѻ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 60) return "ѯ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 50) return "н" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 40) return "м" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 30) return "л" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 20) return "к" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 19) return "ѳ҃і";
    if (n >= 18) return "и҃і";
    if (n >= 17) return "з҃і";
    if (n >= 16) return "ѕ҃і";
    if (n >= 15) return "є҃і";
    if (n >= 14) return "д҃і";
    if (n >= 13) return "г҃і";
    if (n >= 12) return "в҃і";
    if (n >= 11) return "а҃і";
    if (n >= 1) return this.renderCyrillicLower110(n) + "҃";
    if (n >= 0) return "0҃";
};

inter.renderEthiopic = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderEthiopic(-n);
    if (isFractional && n > 1) return this.renderEthiopic(Math.floor(n)) + "፡" + this.renderEthiopic(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 1e18) return this.renderNumber(n, "#,###0");
    if (n >= 2e16) return this.renderEthiopic(Math.floor(n / 1e16)) + "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 1e16) return "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 2e12) return this.renderEthiopic(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1e12) return "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 2e8) return this.renderEthiopic(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1e8) return "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 2e4) return this.renderEthiopic(Math.floor(n / 1e4)) + "፼" + (n === 2e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 1e4) return "፼" + (n === 1e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 200) return this.renderEthiopic(Math.floor(n / 100)) + "፻" + (n === 200 ? "" : this.renderEthiopic(n % 100));
    if (n >= 100) return "፻" + (n === 100 ? "" : this.renderEthiopic(n % 100));
    if (n >= 90) return "፺" + (n === 90 ? "" : this.renderEthiopic(n % 10));
    if (n >= 80) return "፹" + (n === 80 ? "" : this.renderEthiopic(n % 10));
    if (n >= 70) return "፸" + (n === 70 ? "" : this.renderEthiopic(n % 10));
    if (n >= 60) return "፷" + (n === 60 ? "" : this.renderEthiopic(n % 10));
    if (n >= 50) return "፶" + (n === 50 ? "" : this.renderEthiopic(n % 10));
    if (n >= 40) return "፵" + (n === 40 ? "" : this.renderEthiopic(n % 10));
    if (n >= 30) return "፴" + (n === 30 ? "" : this.renderEthiopic(n % 10));
    if (n >= 20) return "፳" + (n === 20 ? "" : this.renderEthiopic(n % 10));
    if (n >= 10) return "፲" + (n === 10 ? "" : this.renderEthiopic(n % 10));
    if (n >= 9) return "፱";
    if (n >= 8) return "፰";
    if (n >= 7) return "፯";
    if (n >= 6) return "፮";
    if (n >= 5) return "፭";
    if (n >= 4) return "፬";
    if (n >= 3) return "፫";
    if (n >= 2) return "፪";
    if (n >= 1) return "፩";
    if (n >= 0) return "ባዶ";
};

inter.renderGeorgian = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGeorgian(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 2e4) return this.renderNumber(n, "#,##0");
    if (n >= 1e4) return "ჯ" + (n === 1e4 ? "" : this.renderGeorgian(n % 1e4));
    if (n >= 9e3) return "ჵ" + (n === 9e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 8e3) return "ჴ" + (n === 8e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 7e3) return "ხ" + (n === 7e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 6e3) return "ჭ" + (n === 6e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 5e3) return "წ" + (n === 5e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 4e3) return "ძ" + (n === 4e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 3e3) return "ც" + (n === 3e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 2e3) return "ჩ" + (n === 2e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 1e3) return "შ" + (n === 1e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 900) return "ყ" + (n === 900 ? "" : this.renderGeorgian(n % 100));
    if (n >= 800) return "ღ" + (n === 800 ? "" : this.renderGeorgian(n % 100));
    if (n >= 700) return "ქ" + (n === 700 ? "" : this.renderGeorgian(n % 100));
    if (n >= 600) return "ფ" + (n === 600 ? "" : this.renderGeorgian(n % 100));
    if (n >= 500) return "ჳ" + (n === 500 ? "" : this.renderGeorgian(n % 100));
    if (n >= 400) return "უ" + (n === 400 ? "" : this.renderGeorgian(n % 100));
    if (n >= 300) return "ტ" + (n === 300 ? "" : this.renderGeorgian(n % 100));
    if (n >= 200) return "ს" + (n === 200 ? "" : this.renderGeorgian(n % 100));
    if (n >= 100) return "რ" + (n === 100 ? "" : this.renderGeorgian(n % 100));
    if (n >= 90) return "ჟ" + (n === 90 ? "" : this.renderGeorgian(n % 10));
    if (n >= 80) return "პ" + (n === 80 ? "" : this.renderGeorgian(n % 10));
    if (n >= 70) return "ო" + (n === 70 ? "" : this.renderGeorgian(n % 10));
    if (n >= 60) return "ჲ" + (n === 60 ? "" : this.renderGeorgian(n % 10));
    if (n >= 50) return "ნ" + (n === 50 ? "" : this.renderGeorgian(n % 10));
    if (n >= 40) return "მ" + (n === 40 ? "" : this.renderGeorgian(n % 10));
    if (n >= 30) return "ლ" + (n === 30 ? "" : this.renderGeorgian(n % 10));
    if (n >= 20) return "კ" + (n === 20 ? "" : this.renderGeorgian(n % 10));
    if (n >= 10) return "ი" + (n === 10 ? "" : this.renderGeorgian(n % 10));
    if (n >= 9) return "თ";
    if (n >= 8) return "ჱ";
    if (n >= 7) return "ზ";
    if (n >= 6) return "ვ";
    if (n >= 5) return "ე";
    if (n >= 4) return "დ";
    if (n >= 3) return "გ";
    if (n >= 2) return "ბ";
    if (n >= 1) return "ა";
    if (n >= 0) return this.renderNumber(n, "#,##0");
};

inter.renderGreekUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGreekUpper(-n);
    if (isFractional && n > 1) return this.renderGreekUpper(Math.floor(n)) + "." + this.renderGreekUpper(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 0) return this.renderGreekNumeralMajuscules(n) + "´";
};

inter.renderGreekLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGreekLower(-n);
    if (isFractional && n > 1) return this.renderGreekLower(Math.floor(n)) + "." + this.renderGreekLower(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 0) return this.renderGreekNumeralMinuscules(n) + "´";
};

inter.renderSpelloutCardinal = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "minus " + this.renderSpelloutCardinal(-n);
    if (isFractional && n > 1) return this.renderSpelloutCardinal(Math.floor(n)) + " point " + this.renderSpelloutCardinal(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e15) return this.renderSpelloutCardinal(Math.floor(n / 1e15)) + " quadrillion" + (n === 1e15 ? "" : " " + this.renderSpelloutCardinal(n % 1e15));
    if (n >= 1e12) return this.renderSpelloutCardinal(Math.floor(n / 1e12)) + " trillion" + (n === 1e12 ? "" : " " + this.renderSpelloutCardinal(n % 1e12));
    if (n >= 1e9) return this.renderSpelloutCardinal(Math.floor(n / 1e9)) + " billion" + (n === 1e9 ? "" : " " + this.renderSpelloutCardinal(n % 1e9));
    if (n >= 1e6) return this.renderSpelloutCardinal(Math.floor(n / 1e6)) + " million" + (n === 1e6 ? "" : " " + this.renderSpelloutCardinal(n % 1e6));
    if (n >= 1e3) return this.renderSpelloutCardinal(Math.floor(n / 1e3)) + " thousand" + (n === 1e3 ? "" : " " + this.renderSpelloutCardinal(n % 1e3));
    if (n >= 100) return this.renderSpelloutCardinal(Math.floor(n / 100)) + " hundred" + (n === 100 ? "" : " " + this.renderSpelloutCardinal(n % 100));
    if (n >= 90) return "ninety" + (n === 90 ? "" : "-" + this.renderSpelloutCardinal(n % 10));
    if (n >= 80) return "eighty" + (n === 80 ? "" : "-" + this.renderSpelloutCardinal(n % 10));
    if (n >= 70) return "seventy" + (n === 70 ? "" : "-" + this.renderSpelloutCardinal(n % 10));
    if (n >= 60) return "sixty" + (n === 60 ? "" : "-" + this.renderSpelloutCardinal(n % 10));
    if (n >= 50) return "fifty" + (n === 50 ? "" : "-" + this.renderSpelloutCardinal(n % 10));
    if (n >= 40) return "forty" + (n === 40 ? "" : "-" + this.renderSpelloutCardinal(n % 10));
    if (n >= 30) return "thirty" + (n === 30 ? "" : "-" + this.renderSpelloutCardinal(n % 10));
    if (n >= 20) return "twenty" + (n === 20 ? "" : "-" + this.renderSpelloutCardinal(n % 10));
    if (n >= 19) return "nineteen";
    if (n >= 18) return "eighteen";
    if (n >= 17) return "seventeen";
    if (n >= 16) return "sixteen";
    if (n >= 15) return "fifteen";
    if (n >= 14) return "fourteen";
    if (n >= 13) return "thirteen";
    if (n >= 12) return "twelve";
    if (n >= 11) return "eleven";
    if (n >= 10) return "ten";
    if (n >= 9) return "nine";
    if (n >= 8) return "eight";
    if (n >= 7) return "seven";
    if (n >= 6) return "six";
    if (n >= 5) return "five";
    if (n >= 4) return "four";
    if (n >= 3) return "three";
    if (n >= 2) return "two";
    if (n >= 1) return "one";
    if (n >= 0) return "zero";
};

inter.renderHebrew = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderHebrew(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1000001) return this.renderNumber(n, "#,##0");
    if (n >= 1e6) return "אלף אלפים";
    if (n >= 3001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 3001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 3e3) return this.renderHebrew(Math.floor(n / 1e3)) + " אלפים";
    if (n >= 2001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 2001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 2e3) return "אלפיים";
    if (n >= 1001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 1001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 1e3) return "אלף";
    if (n >= 901) return "תתק" + this.renderHebrew099(n % 100);
    if (n >= 900) return "תת״ק";
    if (n >= 801) return "תת" + this.renderHebrew099(n % 100);
    if (n >= 800) return "ת״ת";
    if (n >= 745) return "תש" + this.renderHebrew099(n % 100);
    if (n >= 744) return "תשד״מ";
    if (n >= 701) return "תש" + this.renderHebrew099(n % 100);
    if (n >= 700) return "ת״ש";
    if (n >= 699) return "תר" + this.renderHebrew099(n % 100);
    if (n >= 698) return "תרח״צ";
    if (n >= 601) return "תר" + this.renderHebrew099(n % 100);
    if (n >= 600) return "ת״ר";
    if (n >= 501) return "תק" + this.renderHebrew099(n % 100);
    if (n >= 500) return "ת״ק";
    if (n >= 400) return "ת" + this.renderHebrew099(n % 100);
    if (n >= 345) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 344) return "שד״מ";
    if (n >= 305) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 304) return "ד״ש";
    if (n >= 300) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 299) return "ר" + this.renderHebrew099(n % 100);
    if (n >= 298) return "רח״צ";
    if (n >= 200) return "ר" + this.renderHebrew099(n % 100);
    if (n >= 100) return "ק" + this.renderHebrew099(n % 100);
    if (n >= 91) return "צ״" + this.renderHebrewItem(n % 10);
    if (n >= 90) return "צ׳";
    if (n >= 81) return "פ״" + this.renderHebrewItem(n % 10);
    if (n >= 80) return "פ׳";
    if (n >= 71) return "ע״" + this.renderHebrewItem(n % 10);
    if (n >= 70) return "ע׳";
    if (n >= 61) return "ס״" + this.renderHebrewItem(n % 10);
    if (n >= 60) return "ס׳";
    if (n >= 51) return "נ״" + this.renderHebrewItem(n % 10);
    if (n >= 50) return "נ׳";
    if (n >= 41) return "מ״" + this.renderHebrewItem(n % 10);
    if (n >= 40) return "מ׳";
    if (n >= 31) return "ל״" + this.renderHebrewItem(n % 10);
    if (n >= 30) return "ל׳";
    if (n >= 21) return "כ״" + this.renderHebrewItem(n % 10);
    if (n >= 20) return "כ׳";
    if (n >= 17) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "ט״ז";
    if (n >= 15) return "ט״ו";
    if (n >= 11) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 0) return this.renderHebrewItem(n) + "׳";
};

inter.renderRomanUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderRomanUpper(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 4e5) return this.renderNumber(n, "#,##0");
    if (n >= 3e5) return "ↈↈↈ" + (n === 3e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 2e5) return "ↈↈ" + (n === 2e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 1e5) return "ↈ" + (n === 1e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 9e4) return "ↂↈ" + (n === 9e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 8e4) return "ↇↂↂↂ" + (n === 8e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 7e4) return "ↇↂↂ" + (n === 7e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 6e4) return "ↇↂ" + (n === 6e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 5e4) return "ↇ" + (n === 5e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 4e4) return "ↂↇ" + (n === 4e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 3e4) return "ↂↂↂ" + (n === 3e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 2e4) return "ↂↂ" + (n === 2e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 1e4) return "ↂ" + (n === 1e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 9e3) return "Mↂ" + (n === 9e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 8e3) return "ↁMMM" + (n === 8e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 7e3) return "ↁMM" + (n === 7e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 6e3) return "ↁM" + (n === 6e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 5e3) return "ↁ" + (n === 5e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 4e3) return "Mↁ" + (n === 4e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 3e3) return "MMM" + (n === 3e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 2e3) return "MM" + (n === 2e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 1e3) return "M" + (n === 1e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 900) return "CM" + (n === 900 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 800) return "DCCC" + (n === 800 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 700) return "DCC" + (n === 700 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 600) return "DC" + (n === 600 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 500) return "D" + (n === 500 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 400) return "CD" + (n === 400 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 300) return "CCC" + (n === 300 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 200) return "CC" + (n === 200 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 100) return "C" + (n === 100 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 90) return "XC" + (n === 90 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 80) return "LXXX" + (n === 80 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 70) return "LXX" + (n === 70 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 60) return "LX" + (n === 60 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 50) return "L" + (n === 50 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 40) return "XL" + (n === 40 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 30) return "XXX" + (n === 30 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 20) return "XX" + (n === 20 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 10) return "X" + (n === 10 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 9) return "IX";
    if (n >= 8) return "VIII";
    if (n >= 7) return "VII";
    if (n >= 6) return "VI";
    if (n >= 5) return "V";
    if (n >= 4) return "IV";
    if (n >= 3) return "III";
    if (n >= 2) return "II";
    if (n >= 1) return "I";
    if (n >= 0) return "N";
};

inter.renderRomanLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderRomanLower(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 5e3) return this.renderNumber(n, "#,##0");
    if (n >= 4e3) return "mmmm" + (n === 4e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 3e3) return "mmm" + (n === 3e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 2e3) return "mm" + (n === 2e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 1e3) return "m" + (n === 1e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 900) return "cm" + (n === 900 ? "" : this.renderRomanLower(n % 100));
    if (n >= 800) return "dccc" + (n === 800 ? "" : this.renderRomanLower(n % 100));
    if (n >= 700) return "dcc" + (n === 700 ? "" : this.renderRomanLower(n % 100));
    if (n >= 600) return "dc" + (n === 600 ? "" : this.renderRomanLower(n % 100));
    if (n >= 500) return "d" + (n === 500 ? "" : this.renderRomanLower(n % 100));
    if (n >= 400) return "cd" + (n === 400 ? "" : this.renderRomanLower(n % 100));
    if (n >= 300) return "ccc" + (n === 300 ? "" : this.renderRomanLower(n % 100));
    if (n >= 200) return "cc" + (n === 200 ? "" : this.renderRomanLower(n % 100));
    if (n >= 100) return "c" + (n === 100 ? "" : this.renderRomanLower(n % 100));
    if (n >= 90) return "xc" + (n === 90 ? "" : this.renderRomanLower(n % 10));
    if (n >= 80) return "lxxx" + (n === 80 ? "" : this.renderRomanLower(n % 10));
    if (n >= 70) return "lxx" + (n === 70 ? "" : this.renderRomanLower(n % 10));
    if (n >= 60) return "lx" + (n === 60 ? "" : this.renderRomanLower(n % 10));
    if (n >= 50) return "l" + (n === 50 ? "" : this.renderRomanLower(n % 10));
    if (n >= 40) return "xl" + (n === 40 ? "" : this.renderRomanLower(n % 10));
    if (n >= 30) return "xxx" + (n === 30 ? "" : this.renderRomanLower(n % 10));
    if (n >= 20) return "xx" + (n === 20 ? "" : this.renderRomanLower(n % 10));
    if (n >= 10) return "x" + (n === 10 ? "" : this.renderRomanLower(n % 10));
    if (n >= 9) return "ix";
    if (n >= 8) return "viii";
    if (n >= 7) return "vii";
    if (n >= 6) return "vi";
    if (n >= 5) return "v";
    if (n >= 4) return "iv";
    if (n >= 3) return "iii";
    if (n >= 2) return "ii";
    if (n >= 1) return "i";
    if (n >= 0) return "n";
};

inter.renderTamil = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderTamil(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e8) return this.renderNumber(n, "#,##,##0");
    if (n >= 1e6) return this.renderTamil(Math.floor(n / 1e6)) + "௱௲" + (n === 1e6 ? "" : this.renderTamilThousands(n % 1e6));
    if (n >= 2e3) return this.renderTamil(Math.floor(n / 1e3)) + "௲" + (n === 2e3 ? "" : this.renderTamil(n % 1e3));
    if (n >= 1e3) return "௲" + (n === 1e3 ? "" : this.renderTamil(n % 1e3));
    if (n >= 200) return this.renderTamil(Math.floor(n / 100)) + "௱" + (n === 200 ? "" : this.renderTamil(n % 100));
    if (n >= 100) return "௱" + (n === 100 ? "" : this.renderTamil(n % 100));
    if (n >= 20) return this.renderTamil(Math.floor(n / 10)) + "௰" + (n === 20 ? "" : this.renderTamil(n % 10));
    if (n >= 10) return "௰" + (n === 10 ? "" : this.renderTamil(n % 10));
    if (n >= 9) return "௯";
    if (n >= 8) return "௮";
    if (n >= 7) return "௭";
    if (n >= 6) return "௬";
    if (n >= 5) return "௫";
    if (n >= 4) return "௪";
    if (n >= 3) return "௩";
    if (n >= 2) return "௨";
    if (n >= 1) return "௧";
    if (n >= 0) return "௦";
};

inter.renderCyrillicLowerThousands = function(n) {
    if (n >= 900) return "҂ц" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 800) return "҂ѿ" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 700) return "҂ѱ" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 600) return "҂х" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 500) return "҂ф" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 400) return "҂у" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 300) return "҂т" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 200) return "҂с" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 100) return "҂р" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 90) return "҂ч" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 80) return "҂п" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 70) return "҂ѻ" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 60) return "҂ѯ" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 50) return "҂н" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 40) return "҂м" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 30) return "҂л" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 20) return "҂к" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 19) return "҂ѳ҃҂і";
    if (n >= 18) return "҂и҃҂і";
    if (n >= 17) return "҂з҃҂і";
    if (n >= 16) return "҂ѕ҃҂і";
    if (n >= 15) return "҂є҃҂і";
    if (n >= 14) return "҂д҃҂і";
    if (n >= 13) return "҂г҃҂і";
    if (n >= 12) return "҂в҃҂і";
    if (n >= 11) return "҂а҃҂і";
    if (n >= 10) return "҃҂і";
    if (n >= 9) return "҃҂ѳ";
    if (n >= 8) return "҃҂и";
    if (n >= 7) return "҃҂з";
    if (n >= 6) return "҃҂ѕ";
    if (n >= 5) return "҃҂є";
    if (n >= 4) return "҃҂д";
    if (n >= 3) return "҃҂г";
    if (n >= 2) return "҃҂в";
    if (n >= 1) return "҃҂а";
    if (n >= 0) return "҃";
};

inter.renderCyrillicLower110 = function(n) {
    if (n >= 10) return "і";
    if (n >= 9) return "ѳ";
    if (n >= 8) return "и";
    if (n >= 7) return "з";
    if (n >= 6) return "ѕ";
    if (n >= 5) return "є";
    if (n >= 4) return "д";
    if (n >= 3) return "г";
    if (n >= 2) return "в";
    if (n >= 1) return "а";
};

inter.renderCyrillicLowerPost = function(n) {
    if (n >= 1) return this.renderCyrillicLower(n);
    if (n >= 0) return "҃";
};

inter.renderCyrillicLowerFinal = function(n) {
    if (n >= 91) return "ч" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 90) return "҃ч";
    if (n >= 81) return "п" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 80) return "҃п";
    if (n >= 71) return "ѻ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 70) return "҃ѻ";
    if (n >= 61) return "ѯ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 60) return "҃ѯ";
    if (n >= 51) return "н" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 50) return "҃н";
    if (n >= 41) return "м" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 40) return "҃м";
    if (n >= 31) return "л" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 30) return "҃л";
    if (n >= 21) return "к" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 20) return "҃к";
    if (n >= 19) return "ѳ҃і";
    if (n >= 18) return "и҃і";
    if (n >= 17) return "з҃і";
    if (n >= 16) return "ѕ҃і";
    if (n >= 15) return "є҃і";
    if (n >= 14) return "д҃і";
    if (n >= 13) return "г҃і";
    if (n >= 12) return "в҃і";
    if (n >= 11) return "а҃і";
    if (n >= 1) return "҃" + this.renderCyrillicLower110(n);
    if (n >= 0) return "҃";
};

inter.renderEthiopicP3 = function(n) {
    if (n >= 1e12) return this.renderEthiopic(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1) return "፼፼፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼፼፼";
};

inter.renderEthiopicP2 = function(n) {
    if (n >= 1e8) return this.renderEthiopic(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1) return "፼፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼፼";
};

inter.renderEthiopicP1 = function(n) {
    if (n >= 1e4) return this.renderEthiopic(Math.floor(n / 1e4)) + "፼" + (n === 1e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 1) return "፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼";
};

inter.renderGreekNumeralMajuscules = function(n) {
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e16) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e16)) + "ΜΜΜΜ" + (n === 1e16 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e16));
    if (n >= 1e12) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e12)) + "ΜΜΜ" + (n === 1e12 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e12));
    if (n >= 1e8) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e8)) + "ΜΜ" + (n === 1e8 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e8));
    if (n >= 1e4) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e4)) + "Μ" + (n === 1e4 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e4));
    if (n >= 1e3) return "͵" + this.renderGreekNumeralMajuscules(Math.floor(n / 1e3)) + (n === 1e3 ? "" : this.renderGreekNumeralMajuscules(n % 1e3));
    if (n >= 900) return "Ϡ" + (n === 900 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 800) return "Ω" + (n === 800 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 700) return "Ψ" + (n === 700 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 600) return "Χ" + (n === 600 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 500) return "Φ" + (n === 500 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 400) return "Υ" + (n === 400 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 300) return "Τ" + (n === 300 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 200) return "Σ" + (n === 200 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 100) return "Ρ" + (n === 100 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 90) return "Ϟ" + (n === 90 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 80) return "Π" + (n === 80 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 70) return "Ο" + (n === 70 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 60) return "Ξ" + (n === 60 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 50) return "Ν" + (n === 50 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 40) return "Μ" + (n === 40 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 30) return "Λ" + (n === 30 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 20) return "Κ" + (n === 20 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 10) return "Ι" + (n === 10 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 9) return "Θ";
    if (n >= 8) return "Η";
    if (n >= 7) return "Ζ";
    if (n >= 6) return "Ϝ";
    if (n >= 5) return "Ε";
    if (n >= 4) return "Δ";
    if (n >= 3) return "Γ";
    if (n >= 2) return "Β";
    if (n >= 1) return "Α";
    if (n >= 0) return "𐆊";
};

inter.renderGreekNumeralMinuscules = function(n) {
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e16) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e16)) + "μμμμ" + (n === 1e16 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e16));
    if (n >= 1e12) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e12)) + "μμμ" + (n === 1e12 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e12));
    if (n >= 1e8) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e8)) + "μμ" + (n === 1e8 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e8));
    if (n >= 1e4) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e4)) + "μ" + (n === 1e4 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e4));
    if (n >= 1e3) return "͵" + this.renderGreekNumeralMinuscules(Math.floor(n / 1e3)) + (n === 1e3 ? "" : this.renderGreekNumeralMinuscules(n % 1e3));
    if (n >= 900) return "ϡ" + (n === 900 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 800) return "ω" + (n === 800 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 700) return "ψ" + (n === 700 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 600) return "χ" + (n === 600 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 500) return "φ" + (n === 500 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 400) return "υ" + (n === 400 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 300) return "τ" + (n === 300 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 200) return "σ" + (n === 200 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 100) return "ρ" + (n === 100 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 90) return "ϟ" + (n === 90 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 80) return "π" + (n === 80 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 70) return "ο" + (n === 70 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 60) return "ξ" + (n === 60 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 50) return "ν" + (n === 50 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 40) return "μ" + (n === 40 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 30) return "λ" + (n === 30 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 20) return "κ" + (n === 20 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 10) return "ι" + (n === 10 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 9) return "θ";
    if (n >= 8) return "η";
    if (n >= 7) return "ζ";
    if (n >= 6) return "ϝ";
    if (n >= 5) return "ε";
    if (n >= 4) return "δ";
    if (n >= 3) return "γ";
    if (n >= 2) return "β";
    if (n >= 1) return "α";
    if (n >= 0) return "𐆊";
};

inter.renderHebrewThousands = function(n) {
    if (n >= 401) return this.renderHebrew(n) + "׳";
    if (n >= 100) return this.renderHebrew(n) + (n === 100 ? "" : "׳");
    if (n >= 10) return this.renderHebrew(n) + (n === 10 ? "" : "׳");
    if (n >= 0) return this.renderHebrew(n);
};

inter.renderHebrew099 = function(n) {
    if (n >= 91) return "צ״" + this.renderHebrewItem(n % 10);
    if (n >= 90) return "״צ";
    if (n >= 81) return "פ״" + this.renderHebrewItem(n % 10);
    if (n >= 80) return "״פ";
    if (n >= 71) return "ע״" + this.renderHebrewItem(n % 10);
    if (n >= 70) return "״ע";
    if (n >= 61) return "ס״" + this.renderHebrewItem(n % 10);
    if (n >= 60) return "״ס";
    if (n >= 51) return "נ״" + this.renderHebrewItem(n % 10);
    if (n >= 50) return "״נ";
    if (n >= 41) return "מ״" + this.renderHebrewItem(n % 10);
    if (n >= 40) return "״מ";
    if (n >= 31) return "ל״" + this.renderHebrewItem(n % 10);
    if (n >= 30) return "״ל";
    if (n >= 21) return "כ״" + this.renderHebrewItem(n % 10);
    if (n >= 20) return "״כ";
    if (n >= 17) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "ט״ז";
    if (n >= 15) return "ט״ו";
    if (n >= 11) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 1) return "״" + this.renderHebrewItem(n);
    if (n >= 0) return "׳";
};

inter.renderHebrewItem = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderHebrewItem(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 2100) return this.renderNumber(n, "#,##0");
    if (n >= 2e3) return "תתתתת" + (n === 2e3 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1900) return "תתתתש" + (n === 1900 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1800) return "תתתתר" + (n === 1800 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1700) return "תתתתק" + (n === 1700 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1600) return "תתתת" + (n === 1600 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1500) return "תתתש" + (n === 1500 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1400) return "תתתר" + (n === 1400 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1300) return "תתתק" + (n === 1300 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1200) return "תתת" + (n === 1200 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1100) return "תתש" + (n === 1100 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1e3) return "תתר" + (n === 1e3 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 900) return "תתק" + (n === 900 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 800) return "תת" + (n === 800 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 745) return "תש" + this.renderHebrewItem(n % 100);
    if (n >= 744) return "תשדמ";
    if (n >= 700) return "תש" + (n === 700 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 699) return "תר" + this.renderHebrewItem(n % 100);
    if (n >= 698) return "תרחצ";
    if (n >= 600) return "תר" + (n === 600 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 500) return "תק" + (n === 500 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 400) return "ת" + (n === 400 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 345) return "ש" + this.renderHebrewItem(n % 100);
    if (n >= 344) return "שדמ";
    if (n >= 305) return "ש" + this.renderHebrewItem(n % 100);
    if (n >= 304) return "דש";
    if (n >= 300) return "ש" + (n === 300 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 299) return "ר" + this.renderHebrewItem(n % 100);
    if (n >= 298) return "רחצ";
    if (n >= 200) return "ר" + (n === 200 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 100) return "ק" + (n === 100 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 90) return "צ" + (n === 90 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 80) return "פ" + (n === 80 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 70) return "ע" + (n === 70 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 60) return "ס" + (n === 60 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 50) return "נ" + (n === 50 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 40) return "מ" + (n === 40 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 30) return "ל" + (n === 30 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 20) return "כ" + (n === 20 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 17) return "י" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "טז";
    if (n >= 15) return "טו";
    if (n >= 10) return "י" + (n === 10 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 9) return "ט";
    if (n >= 8) return "ח";
    if (n >= 7) return "ז";
    if (n >= 6) return "ו";
    if (n >= 5) return "ה";
    if (n >= 4) return "ד";
    if (n >= 3) return "ג";
    if (n >= 2) return "ב";
    if (n >= 1) return "א";
    if (n >= 0) return "״";
};

inter.renderTamilThousands = function(n) {
    if (n >= 1e3) return this.renderTamilThousands(Math.floor(n / 1e3)) + "௲" + (n === 1e3 ? "" : this.renderTamilThousands(n % 1e3));
    if (n >= 0) return this.renderTamil(n);
};

inter.renderEthiopicP = function(n) {
    if (n >= 1e16) return this.renderEthiopicP(Math.floor(n / 1e16)) + "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 1e12) return this.renderEthiopicP(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1e8) return this.renderEthiopicP(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1e4) return this.renderEthiopicP(Math.floor(n / 1e4)) + "፼" + (n === 1e4 ? "" : this.renderEthiopicP(n % 1e4));
    if (n >= 1) return this.renderEthiopic(n);
};

inter.currencies = [].concat(P.currencies.slice(0, 26), {
    id: "BYB",
    displayName: "Belarusian New Rouble (1994–1999)",
    one: "Belarusian new ruble (1994–1999)",
    other: "Belarusian new rubles (1994–1999)"
}, {
    id: "BYR",
    displayName: "Belarusian Rouble",
    symbol: "р.",
    one: "Belarusian ruble",
    other: "Belarusian rubles"
}, P.currencies.slice(28, 135), {
    id: "JPY",
    displayName: "Japanese Yen",
    symbol: "JP¥",
    one: "Japanese yen",
    other: "Japanese yen"
}, P.currencies.slice(136, 143), {
    id: "LVR",
    displayName: "Latvian Rouble",
    one: "Latvian ruble",
    other: "Latvian rubles"
}, P.currencies.slice(144, 210), {
    id: "RUB",
    displayName: "Russian Rouble",
    symbol: "₽",
    one: "Russian ruble",
    other: "Russian rubles"
}, {
    id: "RUR",
    displayName: "Russian Rouble (1991–1998)",
    symbol: "р.",
    one: "Russian ruble (1991–1998)",
    other: "Russian rubles (1991–1998)"
}, P.currencies.slice(212, 250), {
    id: "TJR",
    displayName: "Tajikistani Rouble",
    one: "Tajikistani ruble",
    other: "Tajikistani rubles"
}, P.currencies.slice(251, 263), {
    id: "USD",
    displayName: "US Dollar",
    symbol: "US$",
    one: "US dollar",
    other: "US dollars"
}, P.currencies.slice(264, 297));

module.exports = inter;
