// inter/build/shi_latn.js
function defaults(targetObj) {
    for (var i = 1; i < arguments.length; i += 1) {
        if (arguments[i] && typeof arguments[i] === "object") {
            for (var propertyName in arguments[i]) {
                if (typeof targetObj[propertyName] === "undefined") {
                    targetObj[propertyName] = arguments[i][propertyName];
                }
            }
        }
    }
    return targetObj;
}

var P = require("./shi.js"), inter = defaults({
    renderers: {}
}, P);

var _i1a71 = {
    displayName: "tasut"
}, _i1a72 = {
    all: {
        leap: "{0}bis"
    }
}, _i1a73 = {
    a: "h a – h a",
    h: "h–h a"
}, _i1a74 = [ "Before R.O.C.", "R.O.C." ], _i1a75 = {
    am: "tifawt",
    pm: "tadggʷat"
}, _i1a76 = {
    a: "h a – h a v",
    h: "h–h a v"
}, _i1a77 = [ "S", "M", "T", "W", "T", "F", "S" ], _i1a78 = [ "ak 1", "ak 2", "ak 3", "ak 4" ], _i1a79 = {
    displayName: "ass g imalass"
}, _i1a710 = {
    displayName: "akud n ugmmaḍ"
}, _i1a711 = {
    d: "MMM d–d",
    M: "MMM d – MMM d"
}, _i1a712 = {
    wide: [ "AH" ],
    abbreviated: [ "AH" ]
}, _i1a713 = {
    H: "HH:mm–HH:mm",
    m: "HH:mm–HH:mm"
}, _i1a714 = {
    "0": "assa",
    "1": "askka",
    "-1": "iḍlli"
}, _i1a715 = {
    M: "y-MM – y-MM",
    y: "y-MM – y-MM"
}, _i1a716 = {
    M: "U MMM–MMM",
    y: "U MMM – U MMM"
}, _i1a717 = {
    H: "HH:mm–HH:mm v",
    m: "HH:mm–HH:mm v"
}, _i1a718 = {
    d: "MM-dd – MM-dd",
    M: "MM-dd – MM-dd"
}, _i1a719 = {
    M: "G y MMM–MMM",
    y: "G y MMM – y MMM"
}, _i1a720 = {
    M: "U MMMM–MMMM",
    y: "U MMMM – U MMMM"
}, _i1a721 = {
    M: "G y MMMM–MMMM",
    y: "G y MMMM – y MMMM"
}, _i1a722 = {
    M: "GGGGG y-MM – y-MM",
    y: "GGGGG y-MM – y-MM"
}, _i1a723 = [ "asa", "ayn", "asi", "akṛ", "akw", "asim", "asiḍ" ], _i1a724 = {
    "0": "this year",
    "1": "next year",
    "-1": "last year"
}, _i1a725 = {
    "0": "this week",
    "1": "next week",
    "-1": "last week"
}, _i1a726 = {
    displayName: "tizi g wass: tifawt / tadggʷat"
}, _i1a727 = {
    future: {
        other: "+{0} y"
    },
    past: {
        other: "-{0} y"
    }
}, _i1a728 = {
    future: {
        other: "+{0} m"
    },
    past: {
        other: "-{0} m"
    }
}, _i1a729 = {
    future: {
        other: "+{0} w"
    },
    past: {
        other: "-{0} w"
    }
}, _i1a730 = {
    future: {
        other: "+{0} d"
    },
    past: {
        other: "-{0} d"
    }
}, _i1a731 = {
    future: {
        other: "+{0} h"
    },
    past: {
        other: "-{0} h"
    }
}, _i1a732 = {
    future: {
        other: "+{0} s"
    },
    past: {
        other: "-{0} s"
    }
}, _i1a733 = {
    future: {
        other: "+{0} Q"
    },
    past: {
        other: "-{0} Q"
    }
}, _i1a734 = {
    d: "MM-dd, E – MM-dd, E",
    M: "MM-dd, E – MM-dd, E"
}, _i1a735 = {
    d: "MMM d, E – MMM d, E",
    M: "MMM d, E – MMM d, E"
}, _i1a736 = [ "i", "b", "m", "i", "m", "y", "y", "ɣ", "c", "k", "n", "d" ], _i1a737 = {
    "0": "this month",
    "1": "next month",
    "-1": "last month"
}, _i1a738 = {
    wide: [ "ERA0", "ERA1" ],
    abbreviated: [ "ERA0", "ERA1" ]
}, _i1a739 = {
    future: {
        other: "+{0} min"
    },
    past: {
        other: "-{0} min"
    }
}, _i1a740 = {
    "0": "this Sunday",
    "1": "next Sunday",
    "-1": "last Sunday"
}, _i1a741 = {
    "0": "this Monday",
    "1": "next Monday",
    "-1": "last Monday"
}, _i1a742 = {
    "0": "this Friday",
    "1": "next Friday",
    "-1": "last Friday"
}, _i1a743 = [ "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12" ], _i1a744 = {
    a: "h:mm a – h:mm a",
    h: "h:mm–h:mm a",
    m: "h:mm–h:mm a"
}, _i1a745 = [ "akṛaḍyur 1", "akṛaḍyur 2", "akṛaḍyur 3", "akṛaḍyur 4" ], _i1a746 = {
    "0": "this quarter",
    "1": "next quarter",
    "-1": "last quarter"
}, _i1a747 = {
    "0": "this Tuesday",
    "1": "next Tuesday",
    "-1": "last Tuesday"
}, _i1a748 = {
    d: "U MMM d–d",
    M: "U MMM d – MMM d",
    y: "U MMM d – U MMM d"
}, _i1a749 = [ "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13" ], _i1a750 = {
    "0": "this Thursday",
    "1": "next Thursday",
    "-1": "last Thursday"
}, _i1a751 = {
    "0": "this Saturday",
    "1": "next Saturday",
    "-1": "last Saturday"
}, _i1a752 = {
    a: "h:mm a – h:mm a v",
    h: "h:mm–h:mm a v",
    m: "h:mm–h:mm a v"
}, _i1a753 = {
    "0": "this Wednesday",
    "1": "next Wednesday",
    "-1": "last Wednesday"
}, _i1a754 = {
    id: "PE",
    displayName: "biru",
    regionId: "005",
    hasTimeZones: true
}, _i1a755 = {
    id: "GU",
    displayName: "gwam",
    regionId: "057",
    hasTimeZones: true
}, _i1a756 = {
    id: "IR",
    displayName: "iran",
    regionId: "034",
    hasTimeZones: true
}, _i1a757 = {
    id: "CU",
    displayName: "kuba",
    regionId: "029",
    hasTimeZones: true
}, _i1a758 = {
    id: "LA",
    displayName: "laws",
    regionId: "035",
    hasTimeZones: true
}, _i1a759 = {
    id: "ML",
    displayName: "mali",
    regionId: "011",
    hasTimeZones: true
}, _i1a760 = {
    id: "EG",
    displayName: "miṣṛ",
    regionId: "015",
    hasTimeZones: true
}, _i1a761 = {
    id: "NU",
    displayName: "niwi",
    regionId: "061",
    hasTimeZones: true
}, _i1a762 = {
    id: "TD",
    displayName: "tcad",
    regionId: "017",
    hasTimeZones: true
}, _i1a763 = {
    id: "TN",
    displayName: "tuns",
    regionId: "015",
    hasTimeZones: true
}, _i1a764 = {
    id: "GH",
    displayName: "ɣana",
    regionId: "011",
    hasTimeZones: true
}, _i1a765 = {
    id: "TG",
    displayName: "ṭugu",
    regionId: "011",
    hasTimeZones: true
}, _i1a766 = {
    id: "AW",
    displayName: "aruba",
    regionId: "029",
    hasTimeZones: true
}, _i1a767 = {
    id: "PW",
    displayName: "balaw",
    regionId: "057",
    hasTimeZones: true
}, _i1a768 = {
    id: "BZ",
    displayName: "biliz",
    regionId: "013",
    hasTimeZones: true
}, _i1a769 = {
    id: "BJ",
    displayName: "binin",
    regionId: "011",
    hasTimeZones: true
}, _i1a770 = {
    id: "BN",
    displayName: "bruni",
    regionId: "035",
    hasTimeZones: true
}, _i1a771 = {
    id: "CL",
    displayName: "ccili",
    regionId: "005",
    hasTimeZones: true
}, _i1a772 = {
    id: "KM",
    displayName: "cumur",
    regionId: "014",
    hasTimeZones: true
}, _i1a773 = {
    id: "DZ",
    displayName: "dzayr",
    regionId: "015",
    hasTimeZones: true
}, _i1a774 = {
    id: "FJ",
    displayName: "fidji",
    regionId: "054",
    hasTimeZones: true
}, _i1a775 = {
    id: "GA",
    displayName: "gabun",
    regionId: "017",
    hasTimeZones: true
}, _i1a776 = {
    id: "HT",
    displayName: "hayti",
    regionId: "029",
    hasTimeZones: true
}, _i1a777 = {
    id: "KE",
    displayName: "kinya",
    regionId: "014",
    hasTimeZones: true
}, _i1a778 = {
    id: "CG",
    displayName: "kungu",
    regionId: "017",
    hasTimeZones: true
}, _i1a779 = {
    id: "IN",
    displayName: "lhind",
    regionId: "034",
    hasTimeZones: true
}, _i1a780 = {
    id: "LY",
    displayName: "libya",
    regionId: "015",
    hasTimeZones: true
}, _i1a781 = {
    id: "KW",
    displayName: "lkwit",
    regionId: "145",
    hasTimeZones: true
}, _i1a782 = {
    id: "MT",
    displayName: "malṭa",
    regionId: "039",
    hasTimeZones: true
}, _i1a783 = {
    id: "YT",
    displayName: "mayuṭ",
    regionId: "014",
    hasTimeZones: true
}, _i1a784 = {
    id: "MU",
    displayName: "muris",
    regionId: "014",
    hasTimeZones: true
}, _i1a785 = {
    id: "NR",
    displayName: "nawru",
    regionId: "057",
    hasTimeZones: true
}, _i1a786 = {
    id: "NP",
    displayName: "nibal",
    regionId: "034",
    hasTimeZones: true
}, _i1a787 = {
    id: "AT",
    displayName: "nnmsa",
    regionId: "155",
    hasTimeZones: true
}, _i1a788 = {
    id: "QA",
    displayName: "qatar",
    regionId: "145",
    hasTimeZones: true
}, _i1a789 = {
    id: "RU",
    displayName: "rusya",
    regionId: "151",
    hasTimeZones: true
}, _i1a790 = {
    id: "WS",
    displayName: "samwa",
    regionId: "061",
    hasTimeZones: true
}, _i1a791 = {
    id: "SE",
    displayName: "sswid",
    regionId: "154",
    hasTimeZones: true
}, _i1a792 = {
    id: "SY",
    displayName: "surya",
    regionId: "145",
    hasTimeZones: true
}, _i1a793 = {
    id: "YE",
    displayName: "yaman",
    regionId: "145",
    hasTimeZones: true
}, _i1a794 = {
    id: "OM",
    displayName: "ɛuman",
    regionId: "145",
    hasTimeZones: true
}, _i1a795 = {
    id: "GN",
    displayName: "ɣinya",
    regionId: "011",
    hasTimeZones: true
}, _i1a796 = {
    id: "TO",
    displayName: "ṭunga",
    regionId: "061",
    hasTimeZones: true
}, _i1a797 = {
    d: "G y MMM d–d",
    M: "G y MMM d – MMM d",
    y: "G y MMM d – y MMM d"
}, _i1a798 = [ "asamas", "aynas", "asinas", "akṛas", "akwas", "asimwas", "asiḍyas" ], _i1a799 = {
    relative: _i1a740
}, _i1a7100 = {
    relative: _i1a741
}, _i1a7101 = {
    relative: _i1a742
}, _i1a7102 = {
    id: "AD",
    displayName: "andura",
    regionId: "039",
    hasTimeZones: true
}, _i1a7103 = {
    id: "AI",
    displayName: "angila",
    regionId: "029",
    hasTimeZones: true
}, _i1a7104 = {
    id: "AO",
    displayName: "angula",
    regionId: "017",
    hasTimeZones: true
}, _i1a7105 = {
    id: "PA",
    displayName: "banama",
    regionId: "013",
    hasTimeZones: true
}, _i1a7106 = {
    id: "BB",
    displayName: "barbad",
    regionId: "029",
    hasTimeZones: true
}, _i1a7107 = {
    id: "BT",
    displayName: "bhutan",
    regionId: "034",
    hasTimeZones: true
}, _i1a7108 = {
    id: "BE",
    displayName: "bljika",
    regionId: "155",
    hasTimeZones: true
}, _i1a7109 = {
    id: "BG",
    displayName: "blɣara",
    regionId: "151",
    hasTimeZones: true
}, _i1a7110 = {
    id: "BR",
    displayName: "brazil",
    regionId: "005",
    hasTimeZones: true
}, _i1a7111 = {
    id: "BM",
    displayName: "brmuda",
    regionId: "021",
    hasTimeZones: true
}, _i1a7112 = {
    id: "BH",
    displayName: "bḥrayn",
    regionId: "145",
    hasTimeZones: true
}, _i1a7113 = {
    id: "PT",
    displayName: "bṛṭqiz",
    regionId: "039",
    hasTimeZones: true
}, _i1a7114 = {
    id: "CN",
    displayName: "ccinwa",
    regionId: "030",
    hasTimeZones: true
}, _i1a7115 = {
    id: "VN",
    displayName: "fitnam",
    regionId: "035",
    hasTimeZones: true
}, _i1a7116 = {
    id: "FR",
    displayName: "fransa",
    regionId: "155",
    hasTimeZones: true
}, _i1a7117 = {
    id: "GM",
    displayName: "gambya",
    regionId: "011",
    hasTimeZones: true
}, _i1a7118 = {
    id: "IS",
    displayName: "island",
    regionId: "154",
    hasTimeZones: true
}, _i1a7119 = {
    id: "IT",
    displayName: "iṭalya",
    regionId: "039",
    hasTimeZones: true
}, _i1a7120 = {
    id: "GE",
    displayName: "jurjya",
    regionId: "145",
    hasTimeZones: true
}, _i1a7121 = {
    id: "CA",
    displayName: "kanada",
    regionId: "021",
    hasTimeZones: true
}, _i1a7122 = {
    id: "LV",
    displayName: "latfya",
    regionId: "154",
    hasTimeZones: true
}, _i1a7123 = {
    id: "LS",
    displayName: "liṣuṭu",
    regionId: "018",
    hasTimeZones: true
}, _i1a7124 = {
    id: "MA",
    displayName: "lmɣrib",
    regionId: "015",
    hasTimeZones: true
}, _i1a7125 = {
    id: "LB",
    displayName: "lubnan",
    regionId: "145",
    hasTimeZones: true
}, _i1a7126 = {
    id: "JO",
    displayName: "lurdun",
    regionId: "145",
    hasTimeZones: true
}, _i1a7127 = {
    id: "JP",
    displayName: "lyaban",
    regionId: "030",
    hasTimeZones: true
}, _i1a7128 = {
    id: "GR",
    displayName: "lyunan",
    regionId: "039",
    hasTimeZones: true
}, _i1a7129 = {
    id: "IQ",
    displayName: "lɛiraq",
    regionId: "145",
    hasTimeZones: true
}, _i1a7130 = {
    id: "MW",
    displayName: "malawi",
    regionId: "014",
    hasTimeZones: true
}, _i1a7131 = {
    id: "MV",
    displayName: "maldif",
    regionId: "034",
    hasTimeZones: true
}, _i1a7132 = {
    id: "MX",
    displayName: "miksik",
    regionId: "013",
    hasTimeZones: true
}, _i1a7133 = {
    id: "MC",
    displayName: "munaku",
    regionId: "155",
    hasTimeZones: true
}, _i1a7134 = {
    id: "NE",
    displayName: "nnijir",
    regionId: "011",
    hasTimeZones: true
}, _i1a7135 = {
    id: "NO",
    displayName: "nnrwij",
    regionId: "154",
    hasTimeZones: true
}, _i1a7136 = {
    id: "CY",
    displayName: "qubrus",
    regionId: "145",
    hasTimeZones: true
}, _i1a7137 = {
    id: "RW",
    displayName: "rwanda",
    regionId: "014",
    hasTimeZones: true
}, _i1a7138 = {
    id: "ES",
    displayName: "sbanya",
    regionId: "039",
    hasTimeZones: true
}, _i1a7139 = {
    id: "SC",
    displayName: "ssicil",
    regionId: "014",
    hasTimeZones: true
}, _i1a7140 = {
    id: "SD",
    displayName: "ssudan",
    regionId: "015",
    hasTimeZones: true
}, _i1a7141 = {
    id: "CH",
    displayName: "swisra",
    regionId: "155",
    hasTimeZones: true
}, _i1a7142 = {
    id: "TV",
    displayName: "tufalu",
    regionId: "061",
    hasTimeZones: true
}, _i1a7143 = {
    id: "TR",
    displayName: "turkya",
    regionId: "145",
    hasTimeZones: true
}, _i1a7144 = {
    id: "UG",
    displayName: "uɣanda",
    regionId: "014",
    hasTimeZones: true
}, _i1a7145 = {
    id: "ZM",
    displayName: "zambya",
    regionId: "014",
    hasTimeZones: true
}, _i1a7146 = {
    id: "GD",
    displayName: "ɣrnaṭa",
    regionId: "029",
    hasTimeZones: true
}, _i1a7147 = {
    id: "SO",
    displayName: "ṣṣumal",
    regionId: "014",
    hasTimeZones: true
}, _i1a7148 = {
    id: "TW",
    displayName: "ṭaywan",
    regionId: "030",
    hasTimeZones: true
}, _i1a7149 = {
    id: "TK",
    displayName: "ṭuklaw",
    regionId: "061",
    hasTimeZones: true
}, _i1a7150 = {
    id: "AL",
    displayName: "albanya",
    regionId: "039",
    hasTimeZones: true
}, _i1a7151 = {
    id: "DE",
    displayName: "almanya",
    regionId: "155",
    hasTimeZones: true
}, _i1a7152 = {
    id: "AM",
    displayName: "arminya",
    regionId: "145",
    hasTimeZones: true
}, _i1a7153 = {
    id: "BS",
    displayName: "bahamas",
    regionId: "029",
    hasTimeZones: true
}, _i1a7154 = {
    id: "BO",
    displayName: "bulibya",
    regionId: "005",
    hasTimeZones: true
}, _i1a7155 = {
    id: "PL",
    displayName: "bulunya",
    regionId: "151",
    hasTimeZones: true
}, _i1a7156 = {
    id: "BI",
    displayName: "burundi",
    regionId: "014",
    hasTimeZones: true
}, _i1a7157 = {
    id: "DK",
    displayName: "danmark",
    regionId: "154",
    hasTimeZones: true
}, _i1a7158 = {
    id: "DJ",
    displayName: "djibuti",
    regionId: "014",
    hasTimeZones: true
}, _i1a7159 = {
    id: "DM",
    displayName: "duminik",
    regionId: "029",
    hasTimeZones: true
}, _i1a7160 = {
    id: "VU",
    displayName: "fanwaṭu",
    regionId: "054",
    hasTimeZones: true
}, _i1a7161 = {
    id: "GL",
    displayName: "griland",
    regionId: "021",
    hasTimeZones: true
}, _i1a7162 = {
    id: "GY",
    displayName: "gwiyana",
    regionId: "005",
    hasTimeZones: true
}, _i1a7163 = {
    id: "HU",
    displayName: "hnɣarya",
    regionId: "151",
    hasTimeZones: true
}, _i1a7164 = {
    id: "NL",
    displayName: "hulanda",
    regionId: "155",
    hasTimeZones: true
}, _i1a7165 = {
    id: "EC",
    displayName: "ikwadur",
    regionId: "005",
    hasTimeZones: true
}, _i1a7166 = {
    id: "IE",
    displayName: "irlanda",
    regionId: "154",
    hasTimeZones: true
}, _i1a7167 = {
    id: "IL",
    displayName: "israyil",
    regionId: "145",
    hasTimeZones: true
}, _i1a7168 = {
    id: "EE",
    displayName: "istunya",
    regionId: "154",
    hasTimeZones: true
}, _i1a7169 = {
    id: "ET",
    displayName: "ityubya",
    regionId: "014",
    hasTimeZones: true
}, _i1a7170 = {
    id: "JM",
    displayName: "jamayka",
    regionId: "029",
    hasTimeZones: true
}, _i1a7171 = {
    id: "CM",
    displayName: "kamirun",
    regionId: "017",
    hasTimeZones: true
}, _i1a7172 = {
    id: "HR",
    displayName: "krwatya",
    regionId: "039",
    hasTimeZones: true
}, _i1a7173 = {
    id: "LR",
    displayName: "libirya",
    regionId: "011",
    hasTimeZones: true
}, _i1a7174 = {
    id: "AE",
    displayName: "limarat",
    regionId: "145",
    hasTimeZones: true
}, _i1a7175 = {
    id: "MY",
    displayName: "malizya",
    regionId: "035",
    hasTimeZones: true
}, _i1a7176 = {
    id: "MN",
    displayName: "mnɣulya",
    regionId: "030",
    hasTimeZones: true
}, _i1a7177 = {
    id: "MZ",
    displayName: "muznbiq",
    regionId: "014",
    hasTimeZones: true
}, _i1a7178 = {
    id: "MM",
    displayName: "myanmar",
    regionId: "035",
    hasTimeZones: true
}, _i1a7179 = {
    id: "NA",
    displayName: "namibya",
    regionId: "018",
    hasTimeZones: true
}, _i1a7180 = {
    id: "NG",
    displayName: "nijirya",
    regionId: "011",
    hasTimeZones: true
}, _i1a7181 = {
    id: "RO",
    displayName: "rumanya",
    regionId: "151",
    hasTimeZones: true
}, _i1a7182 = {
    id: "SR",
    displayName: "surinam",
    regionId: "005",
    hasTimeZones: true
}, _i1a7183 = {
    id: "UA",
    displayName: "ukranya",
    regionId: "151",
    hasTimeZones: true
}, _i1a7184 = {
    id: "UY",
    displayName: "urugway",
    regionId: "005",
    hasTimeZones: true
}, _i1a7185 = {
    id: "TH",
    displayName: "ṭayland",
    regionId: "035",
    hasTimeZones: true
}, _i1a7186 = {
    id: "AR",
    displayName: "arjantin",
    regionId: "005",
    hasTimeZones: true
}, _i1a7187 = {
    id: "PK",
    displayName: "bakistan",
    regionId: "034",
    hasTimeZones: true
}, _i1a7188 = {
    id: "PY",
    displayName: "baragway",
    regionId: "005",
    hasTimeZones: true
}, _i1a7189 = {
    id: "PN",
    displayName: "bitkayrn",
    regionId: "061",
    hasTimeZones: true
}, _i1a7190 = {
    id: "BW",
    displayName: "butswana",
    regionId: "018",
    hasTimeZones: true
}, _i1a7191 = {
    id: "CO",
    displayName: "culumbya",
    regionId: "005",
    hasTimeZones: true
}, _i1a7192 = {
    id: "PH",
    displayName: "filibbin",
    regionId: "035",
    hasTimeZones: true
}, _i1a7193 = {
    id: "FI",
    displayName: "fillanda",
    regionId: "154",
    hasTimeZones: true
}, _i1a7194 = {
    id: "VE",
    displayName: "finzwila",
    regionId: "005",
    hasTimeZones: true
}, _i1a7195 = {
    id: "GP",
    displayName: "gwadalub",
    regionId: "029",
    hasTimeZones: true
}, _i1a7196 = {
    id: "HN",
    displayName: "hunduras",
    regionId: "013",
    hasTimeZones: true
}, _i1a7197 = {
    id: "ER",
    displayName: "iritirya",
    regionId: "014",
    hasTimeZones: true
}, _i1a7198 = {
    id: "KH",
    displayName: "kambudya",
    regionId: "035",
    hasTimeZones: true
}, _i1a7199 = {
    id: "KI",
    displayName: "kiribati",
    regionId: "057",
    hasTimeZones: true
}, _i1a7200 = {
    id: "LT",
    displayName: "litwanya",
    regionId: "154",
    hasTimeZones: true
}, _i1a7201 = {
    id: "MQ",
    displayName: "martinik",
    regionId: "029",
    hasTimeZones: true
}, _i1a7202 = {
    id: "MD",
    displayName: "muldufya",
    regionId: "151",
    hasTimeZones: true
}, _i1a7203 = {
    id: "MS",
    displayName: "munsirat",
    regionId: "029",
    hasTimeZones: true
}, _i1a7204 = {
    id: "RE",
    displayName: "riyunyun",
    regionId: "014",
    hasTimeZones: true
}, _i1a7205 = {
    id: "SV",
    displayName: "salfadur",
    regionId: "013",
    hasTimeZones: true
}, _i1a7206 = {
    id: "SH",
    displayName: "santilin",
    regionId: "011",
    hasTimeZones: true
}, _i1a7207 = {
    id: "LC",
    displayName: "santlusi",
    regionId: "029",
    hasTimeZones: true
}, _i1a7208 = {
    id: "SK",
    displayName: "slufakya",
    regionId: "151",
    hasTimeZones: true
}, _i1a7209 = {
    id: "SI",
    displayName: "slufinya",
    regionId: "039",
    hasTimeZones: true
}, _i1a7210 = {
    id: "SG",
    displayName: "snɣafura",
    regionId: "035",
    hasTimeZones: true
}, _i1a7211 = {
    id: "LK",
    displayName: "srilanka",
    regionId: "034",
    hasTimeZones: true
}, _i1a7212 = {
    id: "SN",
    displayName: "ssinigal",
    regionId: "011",
    hasTimeZones: true
}, _i1a7213 = {
    id: "AU",
    displayName: "ustralya",
    regionId: "053",
    hasTimeZones: true
}, _i1a7214 = {
    id: "ZW",
    displayName: "zimbabwi",
    regionId: "014",
    hasTimeZones: true
}, _i1a7215 = {
    id: "TZ",
    displayName: "ṭanẓanya",
    regionId: "014",
    hasTimeZones: true
}, _i1a7216 = {
    relative: _i1a747
}, _i1a7217 = {
    id: "AZ",
    displayName: "adrabijan",
    regionId: "145",
    hasTimeZones: true
}, _i1a7218 = {
    id: "ID",
    displayName: "andunisya",
    regionId: "035",
    hasTimeZones: true
}, _i1a7219 = {
    id: "BD",
    displayName: "bangladic",
    regionId: "034",
    hasTimeZones: true
}, _i1a7220 = {
    id: "BY",
    displayName: "bilarusya",
    regionId: "151",
    hasTimeZones: true
}, _i1a7221 = {
    id: "GT",
    displayName: "gwatimala",
    regionId: "013",
    hasTimeZones: true
}, _i1a7222 = {
    id: "KZ",
    displayName: "kazaxstan",
    regionId: "143",
    hasTimeZones: true
}, _i1a7223 = {
    id: "MK",
    displayName: "masidunya",
    regionId: "039",
    hasTimeZones: true
}, _i1a7224 = {
    id: "MR",
    displayName: "muṛiṭanya",
    regionId: "011",
    hasTimeZones: true
}, _i1a7225 = {
    id: "NI",
    displayName: "nikaragwa",
    regionId: "013",
    hasTimeZones: true
}, _i1a7226 = {
    id: "SM",
    displayName: "sanmarinu",
    regionId: "039",
    hasTimeZones: true
}, _i1a7227 = {
    id: "SA",
    displayName: "ssaɛudiya",
    regionId: "145",
    hasTimeZones: true
}, _i1a7228 = {
    id: "SL",
    displayName: "ssiralyun",
    regionId: "011",
    hasTimeZones: true
}, _i1a7229 = {
    d: "y-MM-dd – y-MM-dd",
    M: "y-MM-dd – y-MM-dd",
    y: "y-MM-dd – y-MM-dd"
}, _i1a7230 = {
    id: "AF",
    displayName: "afɣanistan",
    regionId: "034",
    hasTimeZones: true
}, _i1a7231 = {
    id: "PR",
    displayName: "burtu riku",
    regionId: "029",
    hasTimeZones: true
}, _i1a7232 = {
    id: "CR",
    displayName: "kusta rika",
    regionId: "013",
    hasTimeZones: true
}, _i1a7233 = {
    id: "CI",
    displayName: "kut difwar",
    regionId: "011",
    hasTimeZones: true
}, _i1a7234 = {
    id: "LI",
    displayName: "likinctayn",
    regionId: "155",
    hasTimeZones: true
}, _i1a7235 = {
    id: "LU",
    displayName: "luksanburg",
    regionId: "155",
    hasTimeZones: true
}, _i1a7236 = {
    id: "MG",
    displayName: "madaɣacqar",
    regionId: "014",
    hasTimeZones: true
}, _i1a7237 = {
    id: "FM",
    displayName: "mikrunizya",
    regionId: "057",
    hasTimeZones: true
}, _i1a7238 = {
    id: "NZ",
    displayName: "nyuzilanda",
    regionId: "053",
    hasTimeZones: true
}, _i1a7239 = {
    id: "SZ",
    displayName: "swazilanda",
    regionId: "018",
    hasTimeZones: true
}, _i1a7240 = {
    id: "UZ",
    displayName: "uzbakistan",
    regionId: "143",
    hasTimeZones: true
}, _i1a7241 = {
    id: "KG",
    displayName: "kirɣizistan",
    regionId: "143",
    hasTimeZones: true
}, _i1a7242 = {
    id: "TJ",
    displayName: "tadjakistan",
    regionId: "143",
    hasTimeZones: true
}, _i1a7243 = {
    id: "TM",
    displayName: "turkmanstan",
    regionId: "143",
    hasTimeZones: true
}, _i1a7244 = {
    id: "GW",
    displayName: "ɣinya bisaw",
    regionId: "011",
    hasTimeZones: true
}, _i1a7245 = {
    relative: _i1a750
}, _i1a7246 = {
    relative: _i1a751
}, _i1a7247 = {
    id: "BF",
    displayName: "burkina fasu",
    regionId: "011",
    hasTimeZones: true
}, _i1a7248 = {
    id: "GI",
    displayName: "adrar n ṭaṛiq",
    regionId: "039",
    hasTimeZones: true
}, _i1a7249 = {
    id: "KR",
    displayName: "kurya n iffus",
    regionId: "030",
    hasTimeZones: true
}, _i1a7250 = {
    id: "GB",
    displayName: "tagldit imunn",
    regionId: "154",
    hasTimeZones: true
}, _i1a7251 = {
    id: "TL",
    displayName: "timur n lqblt",
    regionId: "035",
    hasTimeZones: true
}, _i1a7252 = {
    abbreviated: {
        leap: "{0}bis"
    },
    narrow: {
        leap: "{0}b"
    },
    wide: {
        leap: "{0}bis"
    }
}, _i1a7253 = {
    id: "BA",
    displayName: "busna d hirsik",
    regionId: "039",
    hasTimeZones: true
}, _i1a7254 = {
    id: "KP",
    displayName: "kurya n iẓẓlmḍ",
    regionId: "030",
    hasTimeZones: true
}, _i1a7255 = {
    id: "CK",
    displayName: "tigzirin n kuk",
    regionId: "061",
    hasTimeZones: true
}, _i1a7256 = {
    id: "WF",
    displayName: "walis d futuna",
    regionId: "061",
    hasTimeZones: true
}, _i1a7257 = {
    relative: _i1a753
}, _i1a7258 = {
    id: "ZA",
    displayName: "afriqya n iffus",
    regionId: "018",
    hasTimeZones: true
}, _i1a7259 = {
    id: "AG",
    displayName: "antiga d brbuda",
    regionId: "029",
    hasTimeZones: true
}, _i1a7260 = {
    id: "VA",
    displayName: "awank n fatikan",
    regionId: "039",
    hasTimeZones: true
}, _i1a7261 = {
    id: "KN",
    displayName: "sankris d nifis",
    regionId: "029",
    hasTimeZones: true
}, _i1a7262 = {
    id: "GQ",
    displayName: "ɣinya n ikwadur",
    regionId: "017",
    hasTimeZones: true
}, _i1a7263 = [ "inn", "bṛa", "maṛ", "ibr", "may", "yun", "yul", "ɣuc", "cut", "ktu", "nuw", "duj" ], _i1a7264 = [ "M01", "M02", "M03", "M04", "M05", "M06", "M07", "M08", "M09", "M10", "M11", "M12" ], _i1a7265 = {
    id: "PM",
    displayName: "sanbyir d miklun",
    regionId: "021",
    hasTimeZones: true
}, _i1a7266 = {
    id: "GF",
    displayName: "gwiyan tafransist",
    regionId: "005",
    hasTimeZones: true
}, _i1a7267 = {
    id: "AS",
    displayName: "samwa tamirikanit",
    regionId: "061",
    hasTimeZones: true
}, _i1a7268 = {
    id: "ST",
    displayName: "sawṭumi d bransib",
    regionId: "017",
    hasTimeZones: true
}, _i1a7269 = {
    id: "IO",
    displayName: "tamnaḍt tanglizit n ugaru ahindi",
    hasTimeZones: true
}, _i1a7270 = {
    id: "KY",
    displayName: "tigzirin n kayman",
    regionId: "029",
    hasTimeZones: true
}, _i1a7271 = {
    id: "FK",
    displayName: "tigzirin n malawi",
    regionId: "005",
    hasTimeZones: true
}, _i1a7272 = {
    id: "MH",
    displayName: "tigzirin n marcal",
    regionId: "057",
    hasTimeZones: true
}, _i1a7273 = {
    id: "TT",
    displayName: "trinidad d ṭubagu",
    regionId: "029",
    hasTimeZones: true
}, _i1a7274 = {
    id: "NC",
    displayName: "kalidunya tamaynut",
    regionId: "054",
    hasTimeZones: true
}, _i1a7275 = {
    id: "CZ",
    displayName: "tagdudant tatcikit",
    regionId: "151",
    hasTimeZones: true
}, _i1a7276 = {
    id: "NF",
    displayName: "tigzirin n nurfulk",
    regionId: "053",
    hasTimeZones: true
}, _i1a7277 = {
    id: "SB",
    displayName: "tigzirin n saluman",
    regionId: "054",
    hasTimeZones: true
}, _i1a7278 = [ undefined, "zi", "chou", "yin", "mao", "chen", "si", "wu", "wei", "shen", "you", "xu" ], _i1a7279 = {
    id: "CV",
    displayName: "tigzirin n kabbirdi",
    regionId: "011",
    hasTimeZones: true
}, _i1a7280 = {
    id: "PG",
    displayName: "babwa ɣinya tamaynut",
    regionId: "054",
    hasTimeZones: true
}, _i1a7281 = {
    id: "PF",
    displayName: "bulinizya tafransist",
    regionId: "061",
    hasTimeZones: true
}, _i1a7282 = {
    id: "VC",
    displayName: "sanfansan d grinadin",
    regionId: "029",
    hasTimeZones: true
}, _i1a7283 = {
    id: "DO",
    displayName: "tagdudant taduminikt",
    regionId: "029",
    hasTimeZones: true
}, _i1a7284 = {
    id: "PS",
    displayName: "agmmaḍ n tagut d ɣzza",
    regionId: "145",
    hasTimeZones: true
}, _i1a7285 = {
    id: "US",
    displayName: "iwunak munnin n mirikan",
    regionId: "021",
    hasTimeZones: true
}, _i1a7286 = {
    d: "U MMM d, E – MMM d, E",
    M: "U MMM d, E – MMM d, E",
    y: "U MMM d, E – U MMM d, E"
}, _i1a7287 = {
    id: "TC",
    displayName: "tigzirin n turkya d kayk",
    regionId: "029",
    hasTimeZones: true
}, _i1a7288 = {
    id: "VG",
    displayName: "tigzirin timgad n nngliz",
    regionId: "029",
    hasTimeZones: true
}, _i1a7289 = {
    id: "MP",
    displayName: "tigzirin n maryan n iẓẓlmḍ",
    regionId: "057",
    hasTimeZones: true
}, _i1a7290 = {
    displayName: "tasragt",
    relativeTime: _i1a731
}, _i1a7291 = {
    d: "GGGGG y-MM-dd – y-MM-dd",
    M: "GGGGG y-MM-dd – y-MM-dd",
    y: "GGGGG y-MM-dd – y-MM-dd"
}, _i1a7292 = {
    d: "y-MM-dd, E – y-MM-dd, E",
    M: "y-MM-dd, E – y-MM-dd, E",
    y: "y-MM-dd, E – y-MM-dd, E"
}, _i1a7293 = {
    id: "CF",
    displayName: "tagdudant tanammast n ifriqya",
    regionId: "017",
    hasTimeZones: true
}, _i1a7294 = {
    d: "G y MMM d, E – MMM d, E",
    M: "G y MMM d, E – MMM d, E",
    y: "G y MMM d, E – y MMM d, E"
}, _i1a7295 = {
    id: "CD",
    displayName: "tagdudant tadimukratit n Kongo",
    regionId: "017",
    hasTimeZones: true
}, _i1a7296 = {
    displayName: "tusdidt",
    relativeTime: _i1a739
}, _i1a7297 = {
    id: "VI",
    displayName: "tigzirin timgad n iwunak munnin",
    regionId: "029",
    hasTimeZones: true
}, _i1a7298 = {
    displayName: "tasint",
    relative: [ "now" ],
    relativeTime: _i1a732
}, _i1a7299 = {
    d: "GGGGG y-MM-dd, E – y-MM-dd, E",
    M: "GGGGG y-MM-dd, E – y-MM-dd, E",
    y: "GGGGG y-MM-dd, E – y-MM-dd, E"
}, _i1a7300 = [ "Muh.", "Saf.", "Rab. I", "Rab. II", "Jum. I", "Jum. II", "Raj.", "Sha.", "Ram.", "Shaw.", "Dhuʻl-Q.", "Dhuʻl-H." ], _i1a7301 = [ "Tishri", "Heshvan", "Kislev", "Tevet", "Shevat", "Adar I", "Adar", "Nisan", "Iyar", "Sivan", "Tamuz", "Av", "Elul" ], _i1a7302 = [ "innayr", "bṛayṛ", "maṛṣ", "ibrir", "mayyu", "yunyu", "yulyuz", "ɣuct", "cutanbir", "ktubr", "nuwanbir", "dujanbir" ], _i1a7303 = {
    abbreviated: _i1a75,
    narrow: _i1a75,
    wide: _i1a75
}, _i1a7304 = [ "Farvardin", "Ordibehesht", "Khordad", "Tir", "Mordad", "Shahrivar", "Mehr", "Aban", "Azar", "Dey", "Bahman", "Esfand" ], _i1a7305 = [ "Tout", "Baba", "Hator", "Kiahk", "Toba", "Amshir", "Baramhat", "Baramouda", "Bashans", "Paona", "Epep", "Mesra", "Nasie" ], _i1a7306 = [ "Meskerem", "Tekemt", "Hedar", "Tahsas", "Ter", "Yekatit", "Megabit", "Miazia", "Genbot", "Sene", "Hamle", "Nehasse", "Pagumen" ], _i1a7307 = [ "Chaitra", "Vaisakha", "Jyaistha", "Asadha", "Sravana", "Bhadra", "Asvina", "Kartika", "Agrahayana", "Pausa", "Magha", "Phalguna" ], _i1a7308 = {
    displayName: "ass",
    relative: _i1a714,
    relativeTime: _i1a730
}, _i1a7309 = {
    abbreviated: _i1a78,
    narrow: [ "1", "2", "3", "4" ],
    wide: _i1a745
}, _i1a7310 = [ "Muharram", "Safar", "Rabiʻ I", "Rabiʻ II", "Jumada I", "Jumada II", "Rajab", "Shaʻban", "Ramadan", "Shawwal", "Dhuʻl-Qiʻdah", "Dhuʻl-Hijjah" ], _i1a7311 = {
    displayName: "asggʷas",
    relative: _i1a724,
    relativeTime: _i1a727
}, _i1a7312 = {
    displayName: "imalass",
    relative: _i1a725,
    relativeTime: _i1a729
}, _i1a7313 = {
    displayName: "ayyur",
    relative: _i1a737,
    relativeTime: _i1a728
}, _i1a7314 = {
    displayName: "Quarter",
    relative: _i1a746,
    relativeTime: _i1a733
}, _i1a7315 = {
    format: _i1a7252,
    numeric: _i1a72,
    standAlone: _i1a7252
}, _i1a7316 = {
    abbreviated: _i1a723,
    narrow: _i1a77,
    wide: _i1a798,
    "short": _i1a723
}, _i1a7317 = {
    abbreviated: _i1a7264,
    narrow: _i1a743,
    wide: _i1a7264
}, _i1a7318 = {
    format: _i1a7303,
    standAlone: _i1a7303
}, _i1a7319 = {
    abbreviated: _i1a7263,
    narrow: _i1a736,
    wide: _i1a7302
}, _i1a7320 = {
    abbreviated: _i1a7278,
    narrow: _i1a7278,
    wide: _i1a7278
}, _i1a7321 = {
    format: _i1a7320
}, _i1a7322 = {
    format: _i1a7309,
    standAlone: _i1a7309
}, _i1a7323 = {
    abbreviated: _i1a7301,
    narrow: _i1a749,
    wide: _i1a7301
}, _i1a7324 = {
    abbreviated: _i1a7304,
    narrow: _i1a743,
    wide: _i1a7304
}, _i1a7325 = {
    abbreviated: _i1a7305,
    narrow: _i1a749,
    wide: _i1a7305
}, _i1a7326 = {
    abbreviated: _i1a7300,
    narrow: _i1a743,
    wide: _i1a7310
}, _i1a7327 = {
    abbreviated: _i1a7306,
    narrow: _i1a749,
    wide: _i1a7306
}, _i1a7328 = {
    abbreviated: _i1a7307,
    narrow: _i1a743,
    wide: _i1a7307
}, _i1a7329 = {
    fullDate: "r(U) MMMM d, EEEE",
    longDate: "r(U) MMMM d",
    mediumDate: "r MMM d",
    shortDate: "r-MM-dd",
    fullTime: "HH:mm:ss zzzz",
    longTime: "HH:mm:ss z",
    mediumTime: "HH:mm:ss",
    shortTime: "HH:mm",
    shortDateTime: "r-MM-dd HH:mm",
    mediumDateTime: "r MMM d HH:mm:ss",
    longDateTime: "r(U) MMMM d HH:mm:ss z",
    fullDateTime: "r(U) MMMM d, EEEE HH:mm:ss zzzz"
}, _i1a7330 = {
    format: _i1a7316,
    standAlone: _i1a7316
}, _i1a7331 = {
    format: _i1a7317,
    standAlone: _i1a7317
}, _i1a7332 = {
    d: "d",
    E: "ccc",
    Ed: "d, E",
    Gy: "r U",
    GyMMM: "r(U) MMM",
    GyMMMd: "r MMM d",
    GyMMMEd: "r(U) MMM d, E",
    h: "h a",
    H: "HH",
    hm: "h:mm a",
    Hm: "HH:mm",
    hms: "h:mm:ss a",
    Hms: "HH:mm:ss",
    M: "L",
    Md: "MM-dd",
    MEd: "MM-dd, E",
    MMM: "LLL",
    MMMd: "MMM d",
    MMMEd: "MMM d, E",
    MMMMd: "MMMM d",
    ms: "mm:ss",
    UM: "U MM",
    UMd: "U MM-d",
    UMMM: "U MMM",
    UMMMd: "U MMM d",
    y: "r(U)",
    yMd: "r-MM-dd",
    yyyy: "r(U)",
    yyyyM: "r-MM",
    yyyyMd: "r-MM-dd",
    yyyyMEd: "r-MM-dd, E",
    yyyyMMM: "r(U) MMM",
    yyyyMMMd: "r MMM d",
    yyyyMMMEd: "r(U) MMM d, E",
    yyyyMMMM: "r(U) MMMM",
    yyyyQQQ: "r(U) QQQ",
    yyyyQQQQ: "r(U) QQQQ"
}, _i1a7333 = {
    format: _i1a7319,
    standAlone: _i1a7319
}, _i1a7334 = [ undefined, "jia-zi", "yi-chou", "bing-yin", "ding-mao", "wu-chen", "ji-si", "geng-wu", "xin-wei", "ren-shen", "gui-you", "jia-xu", "yi-hai", "bing-zi", "ding-chou", "wu-yin", "ji-mao", "geng-chen", "xin-si", "ren-wu", "gui-wei", "jia-shen", "yi-you", "bing-xu", "ding-hai", "wu-zi", "ji-chou", "geng-yin", "xin-mao", "ren-chen", "gui-si", "jia-wu", "yi-wei", "bing-shen", "ding-you", "wu-xu", "ji-hai", "geng-zi", "xin-chou", "ren-yin", "gui-mao", "jia-chen", "yi-si", "bing-wu", "ding-wei", "wu-shen", "ji-you", "geng-xu", "xin-hai", "ren-zi", "gui-chou", "jia-yin", "yi-mao", "bing-chen", "ding-si", "wu-wu", "ji-wei", "geng-shen", "xin-you", "ren-xu" ], _i1a7335 = {
    format: _i1a7326,
    standAlone: _i1a7326
}, _i1a7336 = {
    basic: _i1a7329,
    cldr: _i1a7332
}, _i1a7337 = {
    d: {
        d: "d–d"
    },
    h: _i1a73,
    H: {
        H: "HH–HH"
    },
    hm: _i1a744,
    Hm: _i1a713,
    hmv: _i1a752,
    Hmv: _i1a717,
    hv: _i1a76,
    Hv: {
        H: "HH–HH v"
    },
    M: {
        M: "MM–MM"
    },
    Md: _i1a718,
    MEd: _i1a734,
    MMM: {
        M: "LLL–LLL"
    },
    MMMd: _i1a711,
    MMMEd: _i1a735,
    y: {
        y: "U–U"
    },
    yM: _i1a715,
    yMd: _i1a7229,
    yMEd: _i1a7292,
    yMMM: _i1a716,
    yMMMd: _i1a748,
    yMMMEd: _i1a7286,
    yMMMM: _i1a720
}, _i1a7338 = {
    d: {
        d: "d–d"
    },
    h: _i1a73,
    H: {
        H: "HH–HH"
    },
    hm: _i1a744,
    Hm: _i1a713,
    hmv: _i1a752,
    Hmv: _i1a717,
    hv: _i1a76,
    Hv: {
        H: "HH–HH v"
    },
    M: {
        M: "MM–MM"
    },
    Md: _i1a718,
    MEd: _i1a734,
    MMM: {
        M: "LLL–LLL"
    },
    MMMd: _i1a711,
    MMMEd: _i1a735,
    y: {
        y: "G y–y"
    },
    yM: _i1a722,
    yMd: _i1a7291,
    yMEd: _i1a7299,
    yMMM: _i1a719,
    yMMMd: _i1a797,
    yMMMEd: _i1a7294,
    yMMMM: _i1a721
}, _i1a7339 = {
    era: _i1a71,
    year: _i1a7311,
    month: _i1a7313,
    week: _i1a7312,
    day: _i1a7308,
    weekday: _i1a79,
    dayperiod: _i1a726,
    hour: _i1a7290,
    minute: _i1a7296,
    second: _i1a7298,
    zone: _i1a710,
    quarter: _i1a7314,
    sun: _i1a799,
    mon: _i1a7100,
    tue: _i1a7216,
    wed: _i1a7257,
    thu: _i1a7245,
    fri: _i1a7101,
    sat: _i1a7246
}, _i1a7340 = {
    dateFormats: {
        basic: {},
        cldr: {}
    },
    fields: _i1a7339
}, _i1a7341 = {
    abbreviated: _i1a7334,
    narrow: _i1a7334,
    wide: _i1a7334
}, _i1a7342 = {
    format: _i1a7341
}, _i1a7343 = {
    dateFormats: P.calendars.buddhist.dateFormats,
    datePartsOrder: "dmy",
    dateIntervalFormats: _i1a7338,
    dateIntervalFallbackFormat: "{0} – {1}",
    dayNames: _i1a7330,
    dayPeriods: _i1a7318,
    monthNames: _i1a7335,
    eraNames: _i1a712,
    quarterNames: _i1a7322,
    fields: _i1a7339
}, _i1a7344 = [ "Taika (645–650)", "Hakuchi (650–671)", "Hakuhō (672–686)", "Shuchō (686–701)", "Taihō (701–704)", "Keiun (704–708)", "Wadō (708–715)", "Reiki (715–717)", "Yōrō (717–724)", "Jinki (724–729)", "Tenpyō (729–749)", "Tenpyō-kampō (749-749)", "Tenpyō-shōhō (749-757)", "Tenpyō-hōji (757-765)", "Tenpyō-jingo (765-767)", "Jingo-keiun (767-770)", "Hōki (770–780)", "Ten-ō (781-782)", "Enryaku (782–806)", "Daidō (806–810)", "Kōnin (810–824)", "Tenchō (824–834)", "Jōwa (834–848)", "Kajō (848–851)", "Ninju (851–854)", "Saikō (854–857)", "Ten-an (857-859)", "Jōgan (859–877)", "Gangyō (877–885)", "Ninna (885–889)", "Kanpyō (889–898)", "Shōtai (898–901)", "Engi (901–923)", "Enchō (923–931)", "Jōhei (931–938)", "Tengyō (938–947)", "Tenryaku (947–957)", "Tentoku (957–961)", "Ōwa (961–964)", "Kōhō (964–968)", "Anna (968–970)", "Tenroku (970–973)", "Ten’en (973–976)", "Jōgen (976–978)", "Tengen (978–983)", "Eikan (983–985)", "Kanna (985–987)", "Eien (987–989)", "Eiso (989–990)", "Shōryaku (990–995)", "Chōtoku (995–999)", "Chōhō (999–1004)", "Kankō (1004–1012)", "Chōwa (1012–1017)", "Kannin (1017–1021)", "Jian (1021–1024)", "Manju (1024–1028)", "Chōgen (1028–1037)", "Chōryaku (1037–1040)", "Chōkyū (1040–1044)", "Kantoku (1044–1046)", "Eishō (1046–1053)", "Tengi (1053–1058)", "Kōhei (1058–1065)", "Jiryaku (1065–1069)", "Enkyū (1069–1074)", "Shōho (1074–1077)", "Shōryaku (1077–1081)", "Eihō (1081–1084)", "Ōtoku (1084–1087)", "Kanji (1087–1094)", "Kahō (1094–1096)", "Eichō (1096–1097)", "Jōtoku (1097–1099)", "Kōwa (1099–1104)", "Chōji (1104–1106)", "Kashō (1106–1108)", "Tennin (1108–1110)", "Ten-ei (1110-1113)", "Eikyū (1113–1118)", "Gen’ei (1118–1120)", "Hōan (1120–1124)", "Tenji (1124–1126)", "Daiji (1126–1131)", "Tenshō (1131–1132)", "Chōshō (1132–1135)", "Hōen (1135–1141)", "Eiji (1141–1142)", "Kōji (1142–1144)", "Ten’yō (1144–1145)", "Kyūan (1145–1151)", "Ninpei (1151–1154)", "Kyūju (1154–1156)", "Hōgen (1156–1159)", "Heiji (1159–1160)", "Eiryaku (1160–1161)", "Ōho (1161–1163)", "Chōkan (1163–1165)", "Eiman (1165–1166)", "Nin’an (1166–1169)", "Kaō (1169–1171)", "Shōan (1171–1175)", "Angen (1175–1177)", "Jishō (1177–1181)", "Yōwa (1181–1182)", "Juei (1182–1184)", "Genryaku (1184–1185)", "Bunji (1185–1190)", "Kenkyū (1190–1199)", "Shōji (1199–1201)", "Kennin (1201–1204)", "Genkyū (1204–1206)", "Ken’ei (1206–1207)", "Jōgen (1207–1211)", "Kenryaku (1211–1213)", "Kenpō (1213–1219)", "Jōkyū (1219–1222)", "Jōō (1222–1224)", "Gennin (1224–1225)", "Karoku (1225–1227)", "Antei (1227–1229)", "Kanki (1229–1232)", "Jōei (1232–1233)", "Tenpuku (1233–1234)", "Bunryaku (1234–1235)", "Katei (1235–1238)", "Ryakunin (1238–1239)", "En’ō (1239–1240)", "Ninji (1240–1243)", "Kangen (1243–1247)", "Hōji (1247–1249)", "Kenchō (1249–1256)", "Kōgen (1256–1257)", "Shōka (1257–1259)", "Shōgen (1259–1260)", "Bun’ō (1260–1261)", "Kōchō (1261–1264)", "Bun’ei (1264–1275)", "Kenji (1275–1278)", "Kōan (1278–1288)", "Shōō (1288–1293)", "Einin (1293–1299)", "Shōan (1299–1302)", "Kengen (1302–1303)", "Kagen (1303–1306)", "Tokuji (1306–1308)", "Enkyō (1308–1311)", "Ōchō (1311–1312)", "Shōwa (1312–1317)", "Bunpō (1317–1319)", "Genō (1319–1321)", "Genkō (1321–1324)", "Shōchū (1324–1326)", "Karyaku (1326–1329)", "Gentoku (1329–1331)", "Genkō (1331–1334)", "Kenmu (1334–1336)", "Engen (1336–1340)", "Kōkoku (1340–1346)", "Shōhei (1346–1370)", "Kentoku (1370–1372)", "Bunchū (1372–1375)", "Tenju (1375–1379)", "Kōryaku (1379–1381)", "Kōwa (1381–1384)", "Genchū (1384–1392)", "Meitoku (1384–1387)", "Kakei (1387–1389)", "Kōō (1389–1390)", "Meitoku (1390–1394)", "Ōei (1394–1428)", "Shōchō (1428–1429)", "Eikyō (1429–1441)", "Kakitsu (1441–1444)", "Bun’an (1444–1449)", "Hōtoku (1449–1452)", "Kyōtoku (1452–1455)", "Kōshō (1455–1457)", "Chōroku (1457–1460)", "Kanshō (1460–1466)", "Bunshō (1466–1467)", "Ōnin (1467–1469)", "Bunmei (1469–1487)", "Chōkyō (1487–1489)", "Entoku (1489–1492)", "Meiō (1492–1501)", "Bunki (1501–1504)", "Eishō (1504–1521)", "Taiei (1521–1528)", "Kyōroku (1528–1532)", "Tenbun (1532–1555)", "Kōji (1555–1558)", "Eiroku (1558–1570)", "Genki (1570–1573)", "Tenshō (1573–1592)", "Bunroku (1592–1596)", "Keichō (1596–1615)", "Genna (1615–1624)", "Kan’ei (1624–1644)", "Shōho (1644–1648)", "Keian (1648–1652)", "Jōō (1652–1655)", "Meireki (1655–1658)", "Manji (1658–1661)", "Kanbun (1661–1673)", "Enpō (1673–1681)", "Tenna (1681–1684)", "Jōkyō (1684–1688)", "Genroku (1688–1704)", "Hōei (1704–1711)", "Shōtoku (1711–1716)", "Kyōhō (1716–1736)", "Genbun (1736–1741)", "Kanpō (1741–1744)", "Enkyō (1744–1748)", "Kan’en (1748–1751)", "Hōreki (1751–1764)", "Meiwa (1764–1772)", "An’ei (1772–1781)", "Tenmei (1781–1789)", "Kansei (1789–1801)", "Kyōwa (1801–1804)", "Bunka (1804–1818)", "Bunsei (1818–1830)", "Tenpō (1830–1844)", "Kōka (1844–1848)", "Kaei (1848–1854)", "Ansei (1854–1860)", "Man’en (1860–1861)", "Bunkyū (1861–1864)", "Genji (1864–1865)", "Keiō (1865–1868)", "Meiji", "Taishō", "Shōwa", "Heisei" ], _i1a7345 = {
    dayParts: _i1a7321,
    days: _i1a7342,
    months: _i1a7342,
    years: _i1a7342,
    zodiacs: _i1a7321
}, _i1a7346 = {
    dateFormats: _i1a7336,
    datePartsOrder: "md",
    dateIntervalFormats: _i1a7337,
    dateIntervalFallbackFormat: "{0} – {1}",
    dayNames: _i1a7330,
    dayPeriods: _i1a7318,
    cyclicNames: _i1a7345,
    monthNames: _i1a7331,
    monthPatterns: _i1a7315,
    quarterNames: _i1a7322,
    fields: _i1a7339
}, _i1a7347 = [ _i1a7217, _i1a7248, _i1a7258, _i1a7230, _i1a7284, _i1a7150, _i1a7151, _i1a7218, _i1a7102, _i1a7103, _i1a7104, _i1a7259, _i1a7186, _i1a7152, _i1a766, _i1a7260, _i1a7280, _i1a7153, _i1a7187, _i1a767, _i1a7105, _i1a7219, _i1a7188, _i1a7106, _i1a7107, _i1a7220, _i1a768, _i1a769, _i1a754, _i1a7189, _i1a7108, _i1a7109, _i1a7110, _i1a7111, _i1a770, _i1a7154, _i1a7281, _i1a7155, _i1a7247, _i1a7231, _i1a7156, _i1a7253, _i1a7190, _i1a7112, _i1a7113, _i1a771, _i1a7114, _i1a7191, _i1a772, _i1a7157, _i1a7158, _i1a7159, _i1a773, _i1a7160, _i1a774, _i1a7192, _i1a7193, _i1a7194, _i1a7115, _i1a7116, _i1a775, _i1a7117, _i1a7161, _i1a7195, _i1a755, _i1a7221, _i1a7266, _i1a7162, _i1a776, _i1a7163, _i1a7164, _i1a7196, _i1a7165, _i1a756, _i1a7197, _i1a7166, _i1a7118, _i1a7167, _i1a7168, _i1a7169, _i1a7285, _i1a7119, _i1a7170, _i1a7120, _i1a7274, _i1a7198, _i1a7171, _i1a7121, _i1a7222, _i1a777, _i1a7199, _i1a7241, _i1a7172, _i1a757, _i1a778, _i1a7249, _i1a7254, _i1a7232, _i1a7233, _i1a7122, _i1a758, _i1a779, _i1a7173, _i1a780, _i1a7234, _i1a7174, _i1a7200, _i1a7123, _i1a781, _i1a7124, _i1a7125, _i1a7235, _i1a7126, _i1a7127, _i1a7128, _i1a7129, _i1a7236, _i1a7130, _i1a7131, _i1a759, _i1a7175, _i1a782, _i1a7201, _i1a7223, _i1a783, _i1a7237, _i1a7132, _i1a760, _i1a7176, _i1a7202, _i1a7133, _i1a7203, _i1a784, _i1a7177, _i1a7224, _i1a7178, _i1a7179, _i1a785, _i1a786, _i1a7180, _i1a7225, _i1a761, _i1a7134, _i1a787, _i1a7135, _i1a7238, _i1a788, _i1a7136, _i1a7204, _i1a7181, _i1a789, _i1a7137, _i1a7205, _i1a790, _i1a7267, _i1a7265, _i1a7282, _i1a7261, _i1a7226, _i1a7206, _i1a7207, _i1a7268, _i1a7138, _i1a7208, _i1a7209, _i1a7210, _i1a7211, _i1a7227, _i1a7139, _i1a7212, _i1a7228, _i1a7140, _i1a791, _i1a7182, _i1a792, _i1a7239, _i1a7141, _i1a7242, _i1a7295, _i1a7283, _i1a7293, _i1a7275, _i1a7250, _i1a7269, _i1a762, _i1a7279, _i1a7270, _i1a7255, _i1a7271, _i1a7272, _i1a7289, _i1a7276, _i1a7277, _i1a7287, _i1a7297, _i1a7288, _i1a7251, _i1a7273, _i1a7142, _i1a763, _i1a7243, _i1a7143, _i1a7183, _i1a7184, _i1a7213, _i1a7240, _i1a7144, _i1a7256, _i1a793, _i1a7145, _i1a7214, _i1a794, _i1a764, _i1a795, _i1a7244, _i1a7262, _i1a7146, _i1a7147, _i1a7215, _i1a7185, _i1a7148, _i1a765, _i1a7149, _i1a796 ];

inter.id = "shi_latn";

inter.characters = {
    exemplar: {
        "default": [ "a", "b", "c", "d", "ḍ", "e", "ɛ", "f", "g", "{gʷ}", "ɣ", "h", "ḥ", "i", "j", "k", "{kʷ}", "l", "m", "n", "q", "r", "ṛ", "s", "ṣ", "t", "ṭ", "u", "w", "x", "y", "z" ],
        auxiliary: [ "o", "p", "v" ],
        index: [ "A", "B", "C", "D", "Ḍ", "E", "Ɛ", "F", "G", "{Gʷ}", "Ɣ", "H", "Ḥ", "I", "J", "K", "{Kʷ}", "L", "M", "N", "Q", "R", "Ṛ", "S", "Ṣ", "T", "Ṭ", "U", "W", "X", "Y", "Z" ],
        punctuation: [ "\\-", ",", ";", "\\:", "!", "?", ".", "(", ")", "\\[", "\\]", "\\{", "\\}" ]
    },
    ellipsis: {
        "final": "{0}…",
        initial: "…{0}",
        medial: "{0}…{1}",
        "word-final": "{0} …",
        "word-initial": "… {0}",
        "word-medial": "{0} … {1}"
    },
    moreInformation: "?"
};

inter.calendars = {
    buddhist: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _i1a7338,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _i1a7330,
        dayPeriods: _i1a7318,
        monthNames: _i1a7333,
        eraNames: {
            wide: [ "BE" ],
            abbreviated: [ "BE" ]
        },
        quarterNames: _i1a7322,
        fields: _i1a7339
    },
    chinese: _i1a7346,
    coptic: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _i1a7338,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _i1a7330,
        dayPeriods: _i1a7318,
        monthNames: {
            format: _i1a7325,
            standAlone: _i1a7325
        },
        eraNames: _i1a738,
        quarterNames: _i1a7322,
        fields: _i1a7339
    },
    dangi: _i1a7346,
    ethioaa: _i1a7340,
    ethiopic: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _i1a7338,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _i1a7330,
        dayPeriods: _i1a7318,
        monthNames: {
            format: _i1a7327,
            standAlone: _i1a7327
        },
        eraNames: _i1a738,
        quarterNames: _i1a7322,
        fields: _i1a7339
    },
    gregorian: {
        dateFormats: {
            basic: {
                fullDate: "EEEE d MMMM y",
                longDate: "d MMMM y",
                mediumDate: "d MMM, y",
                shortDate: "d/M/y",
                fullTime: "HH:mm:ss zzzz",
                longTime: "HH:mm:ss z",
                mediumTime: "HH:mm:ss",
                shortTime: "HH:mm",
                shortDateTime: "d/M/y HH:mm",
                mediumDateTime: "d MMM, y HH:mm:ss",
                longDateTime: "d MMMM y HH:mm:ss z",
                fullDateTime: "EEEE d MMMM y HH:mm:ss zzzz"
            },
            cldr: {
                M: "M",
                Md: "d/M",
                MMd: "d/MM",
                MMdd: "dd/MM",
                MMM: "MMM",
                MMMd: "d MMM",
                MMMEd: "E d MMM",
                MMMMd: "d MMMM",
                MMMMEd: "E d MMMM",
                ms: "m:ss",
                y: "y",
                yM: "M/y",
                yMEd: "E d/M/y",
                yMMM: "MMM y",
                yMMMEd: "E d MMM y",
                yMMMM: "MMMM y",
                yQQQ: "QQQ y",
                yQQQQ: "QQQQ y",
                d: "d",
                E: "ccc",
                Ed: "d, E",
                Ehm: "E h:mm a",
                EHm: "E HH:mm",
                Ehms: "E h:mm:ss a",
                EHms: "E HH:mm:ss",
                Gy: "G y",
                GyMMM: "G y MMM",
                GyMMMd: "G y MMM d",
                GyMMMEd: "G y MMM d, E",
                h: "h a",
                H: "HH",
                hm: "h:mm a",
                Hm: "HH:mm",
                hms: "h:mm:ss a",
                Hms: "HH:mm:ss",
                hmsv: "h:mm:ss a v",
                Hmsv: "HH:mm:ss v",
                hmv: "h:mm a v",
                Hmv: "HH:mm v",
                MEd: "MM-dd, E",
                yMd: "y-MM-dd",
                yMMMd: "y MMM d"
            }
        },
        datePartsOrder: "dmy",
        dateIntervalFormats: {
            d: {
                d: "d–d"
            },
            h: _i1a73,
            H: {
                H: "HH–HH"
            },
            hm: _i1a744,
            Hm: _i1a713,
            hmv: _i1a752,
            Hmv: _i1a717,
            hv: _i1a76,
            Hv: {
                H: "HH–HH v"
            },
            M: {
                M: "MM–MM"
            },
            Md: _i1a718,
            MEd: _i1a734,
            MMM: {
                M: "LLL–LLL"
            },
            MMMd: _i1a711,
            MMMEd: _i1a735,
            y: {
                y: "y–y"
            },
            yM: _i1a715,
            yMd: _i1a7229,
            yMEd: _i1a7292,
            yMMM: {
                M: "y MMM–MMM",
                y: "y MMM – y MMM"
            },
            yMMMd: {
                d: "y MMM d–d",
                M: "y MMM d – MMM d",
                y: "y MMM d – y MMM d"
            },
            yMMMEd: {
                d: "y MMM d, E – MMM d, E",
                M: "y MMM d, E – MMM d, E",
                y: "y MMM d, E – y MMM d, E"
            },
            yMMMM: {
                M: "y MMMM–MMMM",
                y: "y MMMM – y MMMM"
            }
        },
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _i1a7330,
        dayPeriods: _i1a7318,
        monthNames: _i1a7333,
        eraNames: {
            wide: [ "dat n ɛisa", "dffir n ɛisa" ],
            abbreviated: [ "daɛ", "dfɛ" ]
        },
        quarterNames: _i1a7322,
        fields: _i1a7339
    },
    hebrew: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _i1a7338,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _i1a7330,
        dayPeriods: _i1a7318,
        monthNames: {
            format: _i1a7323,
            standAlone: _i1a7323
        },
        eraNames: {
            wide: [ "AM" ],
            abbreviated: [ "AM" ]
        },
        quarterNames: _i1a7322,
        fields: _i1a7339
    },
    indian: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _i1a7338,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _i1a7330,
        dayPeriods: _i1a7318,
        monthNames: {
            format: _i1a7328,
            standAlone: _i1a7328
        },
        eraNames: {
            wide: [ "Saka" ],
            abbreviated: [ "Saka" ]
        },
        quarterNames: _i1a7322,
        fields: _i1a7339
    },
    islamic: _i1a7343,
    "islamic-umalqura": _i1a7343,
    "islamic-tbla": _i1a7343,
    "islamic-civil": _i1a7343,
    "islamic-rgsa": _i1a7343,
    iso8601: _i1a7340,
    japanese: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _i1a7338,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _i1a7330,
        dayPeriods: _i1a7318,
        monthNames: _i1a7333,
        eraNames: {
            wide: _i1a7344,
            abbreviated: _i1a7344
        },
        quarterNames: _i1a7322,
        fields: _i1a7339
    },
    persian: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _i1a7338,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _i1a7330,
        dayPeriods: _i1a7318,
        monthNames: {
            format: _i1a7324,
            standAlone: _i1a7324
        },
        eraNames: {
            wide: [ "AP" ],
            abbreviated: [ "AP" ]
        },
        quarterNames: _i1a7322,
        fields: _i1a7339
    },
    roc: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _i1a7338,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _i1a7330,
        dayPeriods: _i1a7318,
        monthNames: _i1a7333,
        eraNames: {
            wide: _i1a74,
            abbreviated: _i1a74
        },
        quarterNames: _i1a7322,
        fields: _i1a7339
    },
    islamicc: _i1a7340
};

inter.pluralRule = function(n) {
    return "other";
};

inter.renderDigitsOrdinal = function(n) {
    if (n < 0) return "−" + this.renderDigitsOrdinal(-n);
    if (n >= 0) return this.renderNumber(n, "#,##0") + ".";
};

inter.renderSpelloutNumbering = function(n) {
    if (n < 0) return "−" + this.renderSpelloutNumbering(-n);
    if (n >= 0) return this.renderNumber(n, "#,##0.#");
};

inter.renderArmenianUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderArmenianUpper(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e4) return this.renderNumber(n, "#,##0");
    if (n >= 9e3) return "Ք" + (n === 9e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 8e3) return "Փ" + (n === 8e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 7e3) return "Ւ" + (n === 7e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 6e3) return "Ց" + (n === 6e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 5e3) return "Ր" + (n === 5e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 4e3) return "Տ" + (n === 4e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 3e3) return "Վ" + (n === 3e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 2e3) return "Ս" + (n === 2e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 1e3) return "Ռ" + (n === 1e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 900) return "Ջ" + (n === 900 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 800) return "Պ" + (n === 800 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 700) return "Չ" + (n === 700 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 600) return "Ո" + (n === 600 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 500) return "Շ" + (n === 500 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 400) return "Ն" + (n === 400 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 300) return "Յ" + (n === 300 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 200) return "Մ" + (n === 200 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 100) return "Ճ" + (n === 100 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 90) return "Ղ" + (n === 90 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 80) return "Ձ" + (n === 80 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 70) return "Հ" + (n === 70 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 60) return "Կ" + (n === 60 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 50) return "Ծ" + (n === 50 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 40) return "Խ" + (n === 40 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 30) return "Լ" + (n === 30 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 20) return "Ի" + (n === 20 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 10) return "Ժ" + (n === 10 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 9) return "Թ";
    if (n >= 8) return "Ը";
    if (n >= 7) return "Է";
    if (n >= 6) return "Զ";
    if (n >= 5) return "Ե";
    if (n >= 4) return "Դ";
    if (n >= 3) return "Գ";
    if (n >= 2) return "Բ";
    if (n >= 1) return "Ա";
    if (n >= 0) return "0";
};

inter.renderArmenianLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderArmenianLower(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e4) return this.renderNumber(n, "#,##0");
    if (n >= 9e3) return "ք" + (n === 9e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 8e3) return "փ" + (n === 8e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 7e3) return "ւ" + (n === 7e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 6e3) return "ց" + (n === 6e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 5e3) return "ր" + (n === 5e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 4e3) return "տ" + (n === 4e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 3e3) return "վ" + (n === 3e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 2e3) return "ս" + (n === 2e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 1e3) return "ռ" + (n === 1e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 900) return "ջ" + (n === 900 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 800) return "պ" + (n === 800 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 700) return "չ" + (n === 700 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 600) return "ո" + (n === 600 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 500) return "շ" + (n === 500 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 400) return "ն" + (n === 400 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 300) return "յ" + (n === 300 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 200) return "մ" + (n === 200 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 100) return "ճ" + (n === 100 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 90) return "ղ" + (n === 90 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 80) return "ձ" + (n === 80 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 70) return "հ" + (n === 70 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 60) return "կ" + (n === 60 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 50) return "ծ" + (n === 50 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 40) return "խ" + (n === 40 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 30) return "լ" + (n === 30 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 20) return "ի" + (n === 20 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 10) return "ժ" + (n === 10 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 9) return "թ";
    if (n >= 8) return "ը";
    if (n >= 7) return "է";
    if (n >= 6) return "զ";
    if (n >= 5) return "ե";
    if (n >= 4) return "դ";
    if (n >= 3) return "գ";
    if (n >= 2) return "բ";
    if (n >= 1) return "ա";
    if (n >= 0) return "0";
};

inter.renderCyrillicLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderCyrillicLower(-n);
    if (isFractional && n > 1) return this.renderCyrillicLower(Math.floor(n)) + "." + this.renderCyrillicLower(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e15) return "҂҂҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e15)) + (n === 1e15 ? "" : " " + this.renderCyrillicLower(n % 1e15));
    if (n >= 1e12) return "҂҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e12)) + (n === 1e12 ? "" : " " + this.renderCyrillicLower(n % 1e12));
    if (n >= 1e9) return "҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e9)) + (n === 1e9 ? "" : " " + this.renderCyrillicLower(n % 1e9));
    if (n >= 1e6) return "҂҂" + this.renderCyrillicLower(Math.floor(n / 1e6)) + (n === 1e6 ? "" : " " + this.renderCyrillicLower(n % 1e6));
    if (n >= 11e3) return this.renderCyrillicLowerThousands(Math.floor(n / 1e4)) + (n === 11e3 ? "" : " " + this.renderCyrillicLower(n % 1e4));
    if (n >= 1e4) return "҂" + this.renderCyrillicLower(Math.floor(n / 1e4)) + (n === 1e4 ? "" : " " + this.renderCyrillicLower(n % 1e4));
    if (n >= 1e3) return "҂" + this.renderCyrillicLower110(Math.floor(n / 1e3)) + this.renderCyrillicLowerPost(n % 1e3);
    if (n >= 900) return "ц" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 801) return "ѿ" + this.renderCyrillicLower(n % 100);
    if (n >= 800) return "ѿ҃";
    if (n >= 700) return "ѱ" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 600) return "х" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 500) return "ф" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 400) return "у" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 300) return "т" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 200) return "с" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 100) return "р" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 90) return "ч" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 80) return "п" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 70) return "ѻ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 60) return "ѯ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 50) return "н" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 40) return "м" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 30) return "л" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 20) return "к" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 19) return "ѳ҃і";
    if (n >= 18) return "и҃і";
    if (n >= 17) return "з҃і";
    if (n >= 16) return "ѕ҃і";
    if (n >= 15) return "є҃і";
    if (n >= 14) return "д҃і";
    if (n >= 13) return "г҃і";
    if (n >= 12) return "в҃і";
    if (n >= 11) return "а҃і";
    if (n >= 1) return this.renderCyrillicLower110(n) + "҃";
    if (n >= 0) return "0҃";
};

inter.renderEthiopic = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderEthiopic(-n);
    if (isFractional && n > 1) return this.renderEthiopic(Math.floor(n)) + "፡" + this.renderEthiopic(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 1e18) return this.renderNumber(n, "#,###0");
    if (n >= 2e16) return this.renderEthiopic(Math.floor(n / 1e16)) + "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 1e16) return "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 2e12) return this.renderEthiopic(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1e12) return "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 2e8) return this.renderEthiopic(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1e8) return "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 2e4) return this.renderEthiopic(Math.floor(n / 1e4)) + "፼" + (n === 2e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 1e4) return "፼" + (n === 1e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 200) return this.renderEthiopic(Math.floor(n / 100)) + "፻" + (n === 200 ? "" : this.renderEthiopic(n % 100));
    if (n >= 100) return "፻" + (n === 100 ? "" : this.renderEthiopic(n % 100));
    if (n >= 90) return "፺" + (n === 90 ? "" : this.renderEthiopic(n % 10));
    if (n >= 80) return "፹" + (n === 80 ? "" : this.renderEthiopic(n % 10));
    if (n >= 70) return "፸" + (n === 70 ? "" : this.renderEthiopic(n % 10));
    if (n >= 60) return "፷" + (n === 60 ? "" : this.renderEthiopic(n % 10));
    if (n >= 50) return "፶" + (n === 50 ? "" : this.renderEthiopic(n % 10));
    if (n >= 40) return "፵" + (n === 40 ? "" : this.renderEthiopic(n % 10));
    if (n >= 30) return "፴" + (n === 30 ? "" : this.renderEthiopic(n % 10));
    if (n >= 20) return "፳" + (n === 20 ? "" : this.renderEthiopic(n % 10));
    if (n >= 10) return "፲" + (n === 10 ? "" : this.renderEthiopic(n % 10));
    if (n >= 9) return "፱";
    if (n >= 8) return "፰";
    if (n >= 7) return "፯";
    if (n >= 6) return "፮";
    if (n >= 5) return "፭";
    if (n >= 4) return "፬";
    if (n >= 3) return "፫";
    if (n >= 2) return "፪";
    if (n >= 1) return "፩";
    if (n >= 0) return "ባዶ";
};

inter.renderGeorgian = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGeorgian(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 2e4) return this.renderNumber(n, "#,##0");
    if (n >= 1e4) return "ჯ" + (n === 1e4 ? "" : this.renderGeorgian(n % 1e4));
    if (n >= 9e3) return "ჵ" + (n === 9e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 8e3) return "ჴ" + (n === 8e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 7e3) return "ხ" + (n === 7e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 6e3) return "ჭ" + (n === 6e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 5e3) return "წ" + (n === 5e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 4e3) return "ძ" + (n === 4e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 3e3) return "ც" + (n === 3e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 2e3) return "ჩ" + (n === 2e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 1e3) return "შ" + (n === 1e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 900) return "ყ" + (n === 900 ? "" : this.renderGeorgian(n % 100));
    if (n >= 800) return "ღ" + (n === 800 ? "" : this.renderGeorgian(n % 100));
    if (n >= 700) return "ქ" + (n === 700 ? "" : this.renderGeorgian(n % 100));
    if (n >= 600) return "ფ" + (n === 600 ? "" : this.renderGeorgian(n % 100));
    if (n >= 500) return "ჳ" + (n === 500 ? "" : this.renderGeorgian(n % 100));
    if (n >= 400) return "უ" + (n === 400 ? "" : this.renderGeorgian(n % 100));
    if (n >= 300) return "ტ" + (n === 300 ? "" : this.renderGeorgian(n % 100));
    if (n >= 200) return "ს" + (n === 200 ? "" : this.renderGeorgian(n % 100));
    if (n >= 100) return "რ" + (n === 100 ? "" : this.renderGeorgian(n % 100));
    if (n >= 90) return "ჟ" + (n === 90 ? "" : this.renderGeorgian(n % 10));
    if (n >= 80) return "პ" + (n === 80 ? "" : this.renderGeorgian(n % 10));
    if (n >= 70) return "ო" + (n === 70 ? "" : this.renderGeorgian(n % 10));
    if (n >= 60) return "ჲ" + (n === 60 ? "" : this.renderGeorgian(n % 10));
    if (n >= 50) return "ნ" + (n === 50 ? "" : this.renderGeorgian(n % 10));
    if (n >= 40) return "მ" + (n === 40 ? "" : this.renderGeorgian(n % 10));
    if (n >= 30) return "ლ" + (n === 30 ? "" : this.renderGeorgian(n % 10));
    if (n >= 20) return "კ" + (n === 20 ? "" : this.renderGeorgian(n % 10));
    if (n >= 10) return "ი" + (n === 10 ? "" : this.renderGeorgian(n % 10));
    if (n >= 9) return "თ";
    if (n >= 8) return "ჱ";
    if (n >= 7) return "ზ";
    if (n >= 6) return "ვ";
    if (n >= 5) return "ე";
    if (n >= 4) return "დ";
    if (n >= 3) return "გ";
    if (n >= 2) return "ბ";
    if (n >= 1) return "ა";
    if (n >= 0) return this.renderNumber(n, "#,##0");
};

inter.renderGreekUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGreekUpper(-n);
    if (isFractional && n > 1) return this.renderGreekUpper(Math.floor(n)) + "." + this.renderGreekUpper(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 0) return this.renderGreekNumeralMajuscules(n) + "´";
};

inter.renderGreekLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGreekLower(-n);
    if (isFractional && n > 1) return this.renderGreekLower(Math.floor(n)) + "." + this.renderGreekLower(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 0) return this.renderGreekNumeralMinuscules(n) + "´";
};

inter.renderSpelloutCardinal = function(n) {
    if (n < 0) return "−" + this.renderSpelloutCardinal(-n);
    if (n >= 0) return this.renderNumber(n, "#,##0.#");
};

inter.renderHebrew = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderHebrew(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1000001) return this.renderNumber(n, "#,##0");
    if (n >= 1e6) return "אלף אלפים";
    if (n >= 3001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 3001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 3e3) return this.renderHebrew(Math.floor(n / 1e3)) + " אלפים";
    if (n >= 2001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 2001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 2e3) return "אלפיים";
    if (n >= 1001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 1001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 1e3) return "אלף";
    if (n >= 901) return "תתק" + this.renderHebrew099(n % 100);
    if (n >= 900) return "תת״ק";
    if (n >= 801) return "תת" + this.renderHebrew099(n % 100);
    if (n >= 800) return "ת״ת";
    if (n >= 745) return "תש" + this.renderHebrew099(n % 100);
    if (n >= 744) return "תשד״מ";
    if (n >= 701) return "תש" + this.renderHebrew099(n % 100);
    if (n >= 700) return "ת״ש";
    if (n >= 699) return "תר" + this.renderHebrew099(n % 100);
    if (n >= 698) return "תרח״צ";
    if (n >= 601) return "תר" + this.renderHebrew099(n % 100);
    if (n >= 600) return "ת״ר";
    if (n >= 501) return "תק" + this.renderHebrew099(n % 100);
    if (n >= 500) return "ת״ק";
    if (n >= 400) return "ת" + this.renderHebrew099(n % 100);
    if (n >= 345) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 344) return "שד״מ";
    if (n >= 305) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 304) return "ד״ש";
    if (n >= 300) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 299) return "ר" + this.renderHebrew099(n % 100);
    if (n >= 298) return "רח״צ";
    if (n >= 200) return "ר" + this.renderHebrew099(n % 100);
    if (n >= 100) return "ק" + this.renderHebrew099(n % 100);
    if (n >= 91) return "צ״" + this.renderHebrewItem(n % 10);
    if (n >= 90) return "צ׳";
    if (n >= 81) return "פ״" + this.renderHebrewItem(n % 10);
    if (n >= 80) return "פ׳";
    if (n >= 71) return "ע״" + this.renderHebrewItem(n % 10);
    if (n >= 70) return "ע׳";
    if (n >= 61) return "ס״" + this.renderHebrewItem(n % 10);
    if (n >= 60) return "ס׳";
    if (n >= 51) return "נ״" + this.renderHebrewItem(n % 10);
    if (n >= 50) return "נ׳";
    if (n >= 41) return "מ״" + this.renderHebrewItem(n % 10);
    if (n >= 40) return "מ׳";
    if (n >= 31) return "ל״" + this.renderHebrewItem(n % 10);
    if (n >= 30) return "ל׳";
    if (n >= 21) return "כ״" + this.renderHebrewItem(n % 10);
    if (n >= 20) return "כ׳";
    if (n >= 17) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "ט״ז";
    if (n >= 15) return "ט״ו";
    if (n >= 11) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 0) return this.renderHebrewItem(n) + "׳";
};

inter.renderRomanUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderRomanUpper(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 4e5) return this.renderNumber(n, "#,##0");
    if (n >= 3e5) return "ↈↈↈ" + (n === 3e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 2e5) return "ↈↈ" + (n === 2e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 1e5) return "ↈ" + (n === 1e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 9e4) return "ↂↈ" + (n === 9e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 8e4) return "ↇↂↂↂ" + (n === 8e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 7e4) return "ↇↂↂ" + (n === 7e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 6e4) return "ↇↂ" + (n === 6e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 5e4) return "ↇ" + (n === 5e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 4e4) return "ↂↇ" + (n === 4e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 3e4) return "ↂↂↂ" + (n === 3e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 2e4) return "ↂↂ" + (n === 2e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 1e4) return "ↂ" + (n === 1e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 9e3) return "Mↂ" + (n === 9e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 8e3) return "ↁMMM" + (n === 8e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 7e3) return "ↁMM" + (n === 7e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 6e3) return "ↁM" + (n === 6e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 5e3) return "ↁ" + (n === 5e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 4e3) return "Mↁ" + (n === 4e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 3e3) return "MMM" + (n === 3e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 2e3) return "MM" + (n === 2e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 1e3) return "M" + (n === 1e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 900) return "CM" + (n === 900 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 800) return "DCCC" + (n === 800 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 700) return "DCC" + (n === 700 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 600) return "DC" + (n === 600 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 500) return "D" + (n === 500 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 400) return "CD" + (n === 400 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 300) return "CCC" + (n === 300 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 200) return "CC" + (n === 200 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 100) return "C" + (n === 100 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 90) return "XC" + (n === 90 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 80) return "LXXX" + (n === 80 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 70) return "LXX" + (n === 70 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 60) return "LX" + (n === 60 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 50) return "L" + (n === 50 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 40) return "XL" + (n === 40 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 30) return "XXX" + (n === 30 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 20) return "XX" + (n === 20 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 10) return "X" + (n === 10 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 9) return "IX";
    if (n >= 8) return "VIII";
    if (n >= 7) return "VII";
    if (n >= 6) return "VI";
    if (n >= 5) return "V";
    if (n >= 4) return "IV";
    if (n >= 3) return "III";
    if (n >= 2) return "II";
    if (n >= 1) return "I";
    if (n >= 0) return "N";
};

inter.renderRomanLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderRomanLower(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 5e3) return this.renderNumber(n, "#,##0");
    if (n >= 4e3) return "mmmm" + (n === 4e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 3e3) return "mmm" + (n === 3e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 2e3) return "mm" + (n === 2e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 1e3) return "m" + (n === 1e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 900) return "cm" + (n === 900 ? "" : this.renderRomanLower(n % 100));
    if (n >= 800) return "dccc" + (n === 800 ? "" : this.renderRomanLower(n % 100));
    if (n >= 700) return "dcc" + (n === 700 ? "" : this.renderRomanLower(n % 100));
    if (n >= 600) return "dc" + (n === 600 ? "" : this.renderRomanLower(n % 100));
    if (n >= 500) return "d" + (n === 500 ? "" : this.renderRomanLower(n % 100));
    if (n >= 400) return "cd" + (n === 400 ? "" : this.renderRomanLower(n % 100));
    if (n >= 300) return "ccc" + (n === 300 ? "" : this.renderRomanLower(n % 100));
    if (n >= 200) return "cc" + (n === 200 ? "" : this.renderRomanLower(n % 100));
    if (n >= 100) return "c" + (n === 100 ? "" : this.renderRomanLower(n % 100));
    if (n >= 90) return "xc" + (n === 90 ? "" : this.renderRomanLower(n % 10));
    if (n >= 80) return "lxxx" + (n === 80 ? "" : this.renderRomanLower(n % 10));
    if (n >= 70) return "lxx" + (n === 70 ? "" : this.renderRomanLower(n % 10));
    if (n >= 60) return "lx" + (n === 60 ? "" : this.renderRomanLower(n % 10));
    if (n >= 50) return "l" + (n === 50 ? "" : this.renderRomanLower(n % 10));
    if (n >= 40) return "xl" + (n === 40 ? "" : this.renderRomanLower(n % 10));
    if (n >= 30) return "xxx" + (n === 30 ? "" : this.renderRomanLower(n % 10));
    if (n >= 20) return "xx" + (n === 20 ? "" : this.renderRomanLower(n % 10));
    if (n >= 10) return "x" + (n === 10 ? "" : this.renderRomanLower(n % 10));
    if (n >= 9) return "ix";
    if (n >= 8) return "viii";
    if (n >= 7) return "vii";
    if (n >= 6) return "vi";
    if (n >= 5) return "v";
    if (n >= 4) return "iv";
    if (n >= 3) return "iii";
    if (n >= 2) return "ii";
    if (n >= 1) return "i";
    if (n >= 0) return "n";
};

inter.renderTamil = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderTamil(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e8) return this.renderNumber(n, "#,##,##0");
    if (n >= 1e6) return this.renderTamil(Math.floor(n / 1e6)) + "௱௲" + (n === 1e6 ? "" : this.renderTamilThousands(n % 1e6));
    if (n >= 2e3) return this.renderTamil(Math.floor(n / 1e3)) + "௲" + (n === 2e3 ? "" : this.renderTamil(n % 1e3));
    if (n >= 1e3) return "௲" + (n === 1e3 ? "" : this.renderTamil(n % 1e3));
    if (n >= 200) return this.renderTamil(Math.floor(n / 100)) + "௱" + (n === 200 ? "" : this.renderTamil(n % 100));
    if (n >= 100) return "௱" + (n === 100 ? "" : this.renderTamil(n % 100));
    if (n >= 20) return this.renderTamil(Math.floor(n / 10)) + "௰" + (n === 20 ? "" : this.renderTamil(n % 10));
    if (n >= 10) return "௰" + (n === 10 ? "" : this.renderTamil(n % 10));
    if (n >= 9) return "௯";
    if (n >= 8) return "௮";
    if (n >= 7) return "௭";
    if (n >= 6) return "௬";
    if (n >= 5) return "௫";
    if (n >= 4) return "௪";
    if (n >= 3) return "௩";
    if (n >= 2) return "௨";
    if (n >= 1) return "௧";
    if (n >= 0) return "௦";
};

inter.renderCyrillicLowerThousands = function(n) {
    if (n >= 900) return "҂ц" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 800) return "҂ѿ" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 700) return "҂ѱ" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 600) return "҂х" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 500) return "҂ф" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 400) return "҂у" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 300) return "҂т" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 200) return "҂с" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 100) return "҂р" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 90) return "҂ч" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 80) return "҂п" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 70) return "҂ѻ" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 60) return "҂ѯ" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 50) return "҂н" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 40) return "҂м" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 30) return "҂л" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 20) return "҂к" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 19) return "҂ѳ҃҂і";
    if (n >= 18) return "҂и҃҂і";
    if (n >= 17) return "҂з҃҂і";
    if (n >= 16) return "҂ѕ҃҂і";
    if (n >= 15) return "҂є҃҂і";
    if (n >= 14) return "҂д҃҂і";
    if (n >= 13) return "҂г҃҂і";
    if (n >= 12) return "҂в҃҂і";
    if (n >= 11) return "҂а҃҂і";
    if (n >= 10) return "҃҂і";
    if (n >= 9) return "҃҂ѳ";
    if (n >= 8) return "҃҂и";
    if (n >= 7) return "҃҂з";
    if (n >= 6) return "҃҂ѕ";
    if (n >= 5) return "҃҂є";
    if (n >= 4) return "҃҂д";
    if (n >= 3) return "҃҂г";
    if (n >= 2) return "҃҂в";
    if (n >= 1) return "҃҂а";
    if (n >= 0) return "҃";
};

inter.renderCyrillicLower110 = function(n) {
    if (n >= 10) return "і";
    if (n >= 9) return "ѳ";
    if (n >= 8) return "и";
    if (n >= 7) return "з";
    if (n >= 6) return "ѕ";
    if (n >= 5) return "є";
    if (n >= 4) return "д";
    if (n >= 3) return "г";
    if (n >= 2) return "в";
    if (n >= 1) return "а";
};

inter.renderCyrillicLowerPost = function(n) {
    if (n >= 1) return this.renderCyrillicLower(n);
    if (n >= 0) return "҃";
};

inter.renderCyrillicLowerFinal = function(n) {
    if (n >= 91) return "ч" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 90) return "҃ч";
    if (n >= 81) return "п" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 80) return "҃п";
    if (n >= 71) return "ѻ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 70) return "҃ѻ";
    if (n >= 61) return "ѯ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 60) return "҃ѯ";
    if (n >= 51) return "н" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 50) return "҃н";
    if (n >= 41) return "м" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 40) return "҃м";
    if (n >= 31) return "л" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 30) return "҃л";
    if (n >= 21) return "к" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 20) return "҃к";
    if (n >= 19) return "ѳ҃і";
    if (n >= 18) return "и҃і";
    if (n >= 17) return "з҃і";
    if (n >= 16) return "ѕ҃і";
    if (n >= 15) return "є҃і";
    if (n >= 14) return "д҃і";
    if (n >= 13) return "г҃і";
    if (n >= 12) return "в҃і";
    if (n >= 11) return "а҃і";
    if (n >= 1) return "҃" + this.renderCyrillicLower110(n);
    if (n >= 0) return "҃";
};

inter.renderEthiopicP3 = function(n) {
    if (n >= 1e12) return this.renderEthiopic(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1) return "፼፼፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼፼፼";
};

inter.renderEthiopicP2 = function(n) {
    if (n >= 1e8) return this.renderEthiopic(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1) return "፼፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼፼";
};

inter.renderEthiopicP1 = function(n) {
    if (n >= 1e4) return this.renderEthiopic(Math.floor(n / 1e4)) + "፼" + (n === 1e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 1) return "፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼";
};

inter.renderGreekNumeralMajuscules = function(n) {
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e16) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e16)) + "ΜΜΜΜ" + (n === 1e16 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e16));
    if (n >= 1e12) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e12)) + "ΜΜΜ" + (n === 1e12 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e12));
    if (n >= 1e8) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e8)) + "ΜΜ" + (n === 1e8 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e8));
    if (n >= 1e4) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e4)) + "Μ" + (n === 1e4 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e4));
    if (n >= 1e3) return "͵" + this.renderGreekNumeralMajuscules(Math.floor(n / 1e3)) + (n === 1e3 ? "" : this.renderGreekNumeralMajuscules(n % 1e3));
    if (n >= 900) return "Ϡ" + (n === 900 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 800) return "Ω" + (n === 800 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 700) return "Ψ" + (n === 700 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 600) return "Χ" + (n === 600 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 500) return "Φ" + (n === 500 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 400) return "Υ" + (n === 400 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 300) return "Τ" + (n === 300 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 200) return "Σ" + (n === 200 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 100) return "Ρ" + (n === 100 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 90) return "Ϟ" + (n === 90 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 80) return "Π" + (n === 80 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 70) return "Ο" + (n === 70 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 60) return "Ξ" + (n === 60 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 50) return "Ν" + (n === 50 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 40) return "Μ" + (n === 40 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 30) return "Λ" + (n === 30 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 20) return "Κ" + (n === 20 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 10) return "Ι" + (n === 10 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 9) return "Θ";
    if (n >= 8) return "Η";
    if (n >= 7) return "Ζ";
    if (n >= 6) return "Ϝ";
    if (n >= 5) return "Ε";
    if (n >= 4) return "Δ";
    if (n >= 3) return "Γ";
    if (n >= 2) return "Β";
    if (n >= 1) return "Α";
    if (n >= 0) return "𐆊";
};

inter.renderGreekNumeralMinuscules = function(n) {
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e16) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e16)) + "μμμμ" + (n === 1e16 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e16));
    if (n >= 1e12) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e12)) + "μμμ" + (n === 1e12 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e12));
    if (n >= 1e8) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e8)) + "μμ" + (n === 1e8 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e8));
    if (n >= 1e4) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e4)) + "μ" + (n === 1e4 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e4));
    if (n >= 1e3) return "͵" + this.renderGreekNumeralMinuscules(Math.floor(n / 1e3)) + (n === 1e3 ? "" : this.renderGreekNumeralMinuscules(n % 1e3));
    if (n >= 900) return "ϡ" + (n === 900 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 800) return "ω" + (n === 800 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 700) return "ψ" + (n === 700 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 600) return "χ" + (n === 600 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 500) return "φ" + (n === 500 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 400) return "υ" + (n === 400 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 300) return "τ" + (n === 300 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 200) return "σ" + (n === 200 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 100) return "ρ" + (n === 100 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 90) return "ϟ" + (n === 90 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 80) return "π" + (n === 80 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 70) return "ο" + (n === 70 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 60) return "ξ" + (n === 60 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 50) return "ν" + (n === 50 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 40) return "μ" + (n === 40 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 30) return "λ" + (n === 30 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 20) return "κ" + (n === 20 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 10) return "ι" + (n === 10 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 9) return "θ";
    if (n >= 8) return "η";
    if (n >= 7) return "ζ";
    if (n >= 6) return "ϝ";
    if (n >= 5) return "ε";
    if (n >= 4) return "δ";
    if (n >= 3) return "γ";
    if (n >= 2) return "β";
    if (n >= 1) return "α";
    if (n >= 0) return "𐆊";
};

inter.renderHebrewThousands = function(n) {
    if (n >= 401) return this.renderHebrew(n) + "׳";
    if (n >= 100) return this.renderHebrew(n) + (n === 100 ? "" : "׳");
    if (n >= 10) return this.renderHebrew(n) + (n === 10 ? "" : "׳");
    if (n >= 0) return this.renderHebrew(n);
};

inter.renderHebrew099 = function(n) {
    if (n >= 91) return "צ״" + this.renderHebrewItem(n % 10);
    if (n >= 90) return "״צ";
    if (n >= 81) return "פ״" + this.renderHebrewItem(n % 10);
    if (n >= 80) return "״פ";
    if (n >= 71) return "ע״" + this.renderHebrewItem(n % 10);
    if (n >= 70) return "״ע";
    if (n >= 61) return "ס״" + this.renderHebrewItem(n % 10);
    if (n >= 60) return "״ס";
    if (n >= 51) return "נ״" + this.renderHebrewItem(n % 10);
    if (n >= 50) return "״נ";
    if (n >= 41) return "מ״" + this.renderHebrewItem(n % 10);
    if (n >= 40) return "״מ";
    if (n >= 31) return "ל״" + this.renderHebrewItem(n % 10);
    if (n >= 30) return "״ל";
    if (n >= 21) return "כ״" + this.renderHebrewItem(n % 10);
    if (n >= 20) return "״כ";
    if (n >= 17) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "ט״ז";
    if (n >= 15) return "ט״ו";
    if (n >= 11) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 1) return "״" + this.renderHebrewItem(n);
    if (n >= 0) return "׳";
};

inter.renderHebrewItem = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderHebrewItem(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 2100) return this.renderNumber(n, "#,##0");
    if (n >= 2e3) return "תתתתת" + (n === 2e3 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1900) return "תתתתש" + (n === 1900 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1800) return "תתתתר" + (n === 1800 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1700) return "תתתתק" + (n === 1700 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1600) return "תתתת" + (n === 1600 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1500) return "תתתש" + (n === 1500 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1400) return "תתתר" + (n === 1400 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1300) return "תתתק" + (n === 1300 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1200) return "תתת" + (n === 1200 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1100) return "תתש" + (n === 1100 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1e3) return "תתר" + (n === 1e3 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 900) return "תתק" + (n === 900 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 800) return "תת" + (n === 800 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 745) return "תש" + this.renderHebrewItem(n % 100);
    if (n >= 744) return "תשדמ";
    if (n >= 700) return "תש" + (n === 700 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 699) return "תר" + this.renderHebrewItem(n % 100);
    if (n >= 698) return "תרחצ";
    if (n >= 600) return "תר" + (n === 600 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 500) return "תק" + (n === 500 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 400) return "ת" + (n === 400 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 345) return "ש" + this.renderHebrewItem(n % 100);
    if (n >= 344) return "שדמ";
    if (n >= 305) return "ש" + this.renderHebrewItem(n % 100);
    if (n >= 304) return "דש";
    if (n >= 300) return "ש" + (n === 300 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 299) return "ר" + this.renderHebrewItem(n % 100);
    if (n >= 298) return "רחצ";
    if (n >= 200) return "ר" + (n === 200 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 100) return "ק" + (n === 100 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 90) return "צ" + (n === 90 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 80) return "פ" + (n === 80 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 70) return "ע" + (n === 70 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 60) return "ס" + (n === 60 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 50) return "נ" + (n === 50 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 40) return "מ" + (n === 40 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 30) return "ל" + (n === 30 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 20) return "כ" + (n === 20 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 17) return "י" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "טז";
    if (n >= 15) return "טו";
    if (n >= 10) return "י" + (n === 10 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 9) return "ט";
    if (n >= 8) return "ח";
    if (n >= 7) return "ז";
    if (n >= 6) return "ו";
    if (n >= 5) return "ה";
    if (n >= 4) return "ד";
    if (n >= 3) return "ג";
    if (n >= 2) return "ב";
    if (n >= 1) return "א";
    if (n >= 0) return "״";
};

inter.renderTamilThousands = function(n) {
    if (n >= 1e3) return this.renderTamilThousands(Math.floor(n / 1e3)) + "௲" + (n === 1e3 ? "" : this.renderTamilThousands(n % 1e3));
    if (n >= 0) return this.renderTamil(n);
};

inter.renderEthiopicP = function(n) {
    if (n >= 1e16) return this.renderEthiopicP(Math.floor(n / 1e16)) + "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 1e12) return this.renderEthiopicP(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1e8) return this.renderEthiopicP(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1e4) return this.renderEthiopicP(Math.floor(n / 1e4)) + "፼" + (n === 1e4 ? "" : this.renderEthiopicP(n % 1e4));
    if (n >= 1) return this.renderEthiopic(n);
};

inter.territories = _i1a7347;

inter.countries = _i1a7347;

inter.languages = [ {
    id: "ak",
    displayName: "Takant",
    nativeDisplayName: "Akan"
}, {
    id: "ms",
    displayName: "Tamalawit",
    nativeDisplayName: "Bahasa Melayu"
}, {
    id: "de",
    displayName: "Talimant",
    nativeDisplayName: "Deutsch"
}, {
    id: "en",
    displayName: "Tanglizt",
    nativeDisplayName: "English"
}, {
    id: "es",
    displayName: "Tasbnyulit",
    nativeDisplayName: "español"
}, {
    id: "fr",
    displayName: "Tafransist",
    nativeDisplayName: "français"
}, {
    id: "ha",
    displayName: "Tahawsat",
    nativeDisplayName: "Hausa"
}, {
    id: "ig",
    displayName: "Tigbut",
    nativeDisplayName: "Igbo"
}, {
    id: "id",
    displayName: "Tandunisit",
    nativeDisplayName: "Indonesia"
}, {
    id: "zu",
    displayName: "Tazulut",
    nativeDisplayName: "isiZulu"
}, {
    id: "it",
    displayName: "Taṭalyant",
    nativeDisplayName: "italiano"
}, {
    id: "rw",
    displayName: "Taruwandit",
    nativeDisplayName: "Kinyarwanda"
}, {
    id: "hu",
    displayName: "Tahnɣarit",
    nativeDisplayName: "magyar"
}, {
    id: "nl",
    displayName: "Tahulandit",
    nativeDisplayName: "Nederlands"
}, {
    id: "pl",
    displayName: "Tabulunit",
    nativeDisplayName: "polski"
}, {
    id: "pt",
    displayName: "Tabṛṭqizt",
    nativeDisplayName: "português"
}, {
    id: "ro",
    displayName: "Tarumanit",
    nativeDisplayName: "română"
}, {
    id: "so",
    displayName: "Tasumalit",
    nativeDisplayName: "Soomaali"
}, {
    id: "sv",
    displayName: "Taswidit",
    nativeDisplayName: "svenska"
}, {
    id: "jv",
    displayName: "Tajavanit"
}, {
    id: "vi",
    displayName: "Tafitnamit",
    nativeDisplayName: "Tiếng Việt"
}, {
    id: "tr",
    displayName: "Taturkit",
    nativeDisplayName: "Türkçe"
}, {
    id: "yo",
    displayName: "Tayrubat",
    nativeDisplayName: "Èdè Yorùbá"
}, {
    id: "cs",
    displayName: "Tatcikit",
    nativeDisplayName: "čeština"
}, {
    id: "el",
    displayName: "Tagrigit",
    nativeDisplayName: "Ελληνικά"
}, {
    id: "be",
    displayName: "Tabilarust",
    nativeDisplayName: "беларуская"
}, {
    id: "bg",
    displayName: "Tablɣarit",
    nativeDisplayName: "български"
}, {
    id: "ru",
    displayName: "Tarusit",
    nativeDisplayName: "русский"
}, {
    id: "uk",
    displayName: "Tukranit",
    nativeDisplayName: "українська"
}, {
    id: "ur",
    displayName: "Turdut",
    nativeDisplayName: "اردو"
}, {
    id: "ar",
    displayName: "Taɛrabt",
    nativeDisplayName: "العربية"
}, {
    id: "fa",
    displayName: "Tafursit",
    nativeDisplayName: "فارسی"
}, {
    id: "ne",
    displayName: "Tanibalit",
    nativeDisplayName: "नेपाली"
}, {
    id: "hi",
    displayName: "Tahindit",
    nativeDisplayName: "हिन्दी"
}, {
    id: "bn",
    displayName: "Tabnɣalit",
    nativeDisplayName: "বাংলা"
}, {
    id: "pa",
    displayName: "Tabnjabit",
    nativeDisplayName: "ਪੰਜਾਬੀ"
}, {
    id: "ta",
    displayName: "Tatamilt",
    nativeDisplayName: "தமிழ்"
}, {
    id: "th",
    displayName: "Tataylandit",
    nativeDisplayName: "ไทย"
}, {
    id: "my",
    displayName: "Tabirmanit",
    nativeDisplayName: "ဗမာ"
}, {
    id: "am",
    displayName: "Tamharit",
    nativeDisplayName: "አማርኛ"
}, {
    id: "km",
    displayName: "Taxmirt",
    nativeDisplayName: "ខ្មែរ"
}, {
    id: "shi",
    displayName: "Tashelḥiyt",
    nativeDisplayName: "ⵜⴰⵛⵍⵃⵉⵜ"
}, {
    id: "zh",
    displayName: "Tacinwit",
    nativeDisplayName: "中文"
}, {
    id: "ja",
    displayName: "Tajabbunit",
    nativeDisplayName: "日本語"
}, {
    id: "ko",
    displayName: "Takurit",
    nativeDisplayName: "한국어"
} ];

inter.currencies = [ {
    id: "BWP",
    displayName: "abula n butswana",
    symbol: "P"
}, {
    id: "KES",
    displayName: "acilin n kinya"
}, {
    id: "UGX",
    displayName: "acilin n uɣanda"
}, {
    id: "SOS",
    displayName: "acilin n ṣṣumal"
}, {
    id: "TZS",
    displayName: "acilin n ṭanẓanya"
}, {
    id: "BHD",
    displayName: "adinar n bḥrayn"
}, {
    id: "DZD",
    displayName: "adinar n dzayr"
}, {
    id: "LYD",
    displayName: "adinar n libya"
}, {
    id: "SDG",
    displayName: "adinar n ssudan"
}, {
    id: "TND",
    displayName: "adinar n tuns"
}, {
    id: "AED",
    displayName: "adrim n limarat"
}, {
    id: "MAD",
    displayName: "adrim n lmɣrib"
}, {
    id: "STD",
    displayName: "adubra n sanṭumi",
    symbol: "Db"
}, {
    id: "USD",
    displayName: "adular n iwunak imunn",
    symbol: "US$"
}, {
    id: "CAD",
    displayName: "adular n kanada",
    symbol: "CA$"
}, {
    id: "LRD",
    displayName: "adular n libirya",
    symbol: "$"
}, {
    id: "NAD",
    displayName: "adular n namibya",
    symbol: "$"
}, {
    id: "AUD",
    displayName: "adular n ustralya",
    symbol: "A$"
}, {
    id: "ZWD",
    displayName: "adular n zimbabwi"
}, {
    id: "RWF",
    displayName: "afrank n rwanda",
    symbol: "RF"
}, {
    id: "CHF",
    displayName: "afrank n swisra"
}, {
    id: "GBP",
    displayName: "ajnih astrlini n nngliz",
    symbol: "£"
}, {
    id: "EGP",
    displayName: "ajnih n miṣṛ",
    symbol: "E£"
}, {
    id: "SHP",
    displayName: "ajnih n santilin",
    symbol: "£"
}, {
    id: "SDP",
    displayName: "ajnih n ssudan"
}, {
    id: "ZMW",
    displayName: "akwaca n zambya",
    symbol: "ZK"
}, {
    id: "ZMK",
    displayName: "akwaca n zambya (1968–2012)"
}, {
    id: "MZM",
    displayName: "amitikl n muznbiq"
}, {
    id: "ZAR",
    displayName: "arand n afriqya n iffus",
    symbol: "R"
}, {
    id: "INR",
    displayName: "arubi n lhind",
    symbol: "₹"
}, {
    id: "MUR",
    displayName: "arubi n muris",
    symbol: "Rs"
}, {
    id: "SCR",
    displayName: "arubi n ssicil"
}, {
    id: "SAR",
    displayName: "aryal n ssaɛudiya"
}, {
    id: "CNY",
    displayName: "ayan n ccinwa",
    symbol: "CN¥"
}, {
    id: "JPY",
    displayName: "ayan n lyaban",
    symbol: "JP¥"
}, {
    id: "ETB",
    displayName: "bir n ityubya"
}, {
    id: "GMD",
    displayName: "dalasi n gambya"
}, {
    id: "BIF",
    displayName: "frank n burundi"
}, {
    id: "DJF",
    displayName: "frank n djibuti"
}, {
    id: "CDF",
    displayName: "frank n kungu"
}, {
    id: "MGA",
    displayName: "frank n madaɣacqar",
    symbol: "Ar"
}, {
    id: "KMF",
    displayName: "frank n qumuṛ",
    symbol: "CF"
}, {
    id: "GNS",
    displayName: "frank n ɣinya"
}, {
    id: "XAF",
    displayName: "frank ṣifa",
    symbol: "FCFA"
}, {
    id: "XOF",
    displayName: "frank ṣifa bisaw",
    symbol: "CFA"
}, {
    id: "CVE",
    displayName: "iskudu n kabbirdi"
}, {
    id: "MWK",
    displayName: "kwaca n malawi"
}, {
    id: "AOA",
    displayName: "kwanza n angula",
    symbol: "Kz"
}, {
    id: "SZL",
    displayName: "lilanjini"
}, {
    id: "SLL",
    displayName: "liyun"
}, {
    id: "LSL",
    displayName: "luti n liṣuṭu"
}, {
    id: "ERN",
    displayName: "nafka n iritirya"
}, {
    id: "NGN",
    displayName: "nayra n nijirya",
    symbol: "₦"
}, {
    id: "GHC",
    displayName: "sidi n ɣana"
}, {
    id: "MRO",
    displayName: "uqiyya n muṛiṭanya"
}, {
    id: "EUR",
    displayName: "uru",
    symbol: "€"
} ];

module.exports = inter;
