// inter/build/vai_latn.js
function defaults(targetObj) {
    for (var i = 1; i < arguments.length; i += 1) {
        if (arguments[i] && typeof arguments[i] === "object") {
            for (var propertyName in arguments[i]) {
                if (typeof targetObj[propertyName] === "undefined") {
                    targetObj[propertyName] = arguments[i][propertyName];
                }
            }
        }
    }
    return targetObj;
}

var P = require("./vai.js"), inter = defaults({
    renderers: {}
}, P);

var _bbn61 = [ "Q1", "Q2", "Q3", "Q4" ], _bbn62 = {
    all: {
        leap: "{0}bis"
    }
}, _bbn63 = {
    a: "h a – h a",
    h: "h–h a"
}, _bbn64 = {
    displayName: "Dayperiod"
}, _bbn65 = [ "Before R.O.C.", "R.O.C." ], _bbn66 = {
    a: "h a – h a v",
    h: "h–h a v"
}, _bbn67 = [ "S", "M", "T", "W", "T", "F", "S" ], _bbn68 = {
    displayName: "wikiyɛma tele"
}, _bbn69 = {
    d: "MMM d–d",
    M: "MMM d – MMM d"
}, _bbn610 = {
    "0": "wɛlɛ",
    "1": "sina",
    "-1": "kunu"
}, _bbn611 = {
    wide: [ "AH" ],
    abbreviated: [ "AH" ]
}, _bbn612 = {
    H: "HH:mm–HH:mm",
    m: "HH:mm–HH:mm"
}, _bbn613 = {
    M: "y-MM – y-MM",
    y: "y-MM – y-MM"
}, _bbn614 = {
    M: "U MMM–MMM",
    y: "U MMM – U MMM"
}, _bbn615 = {
    H: "HH:mm–HH:mm v",
    m: "HH:mm–HH:mm v"
}, _bbn616 = {
    d: "MM-dd – MM-dd",
    M: "MM-dd – MM-dd"
}, _bbn617 = {
    M: "G y MMM–MMM",
    y: "G y MMM – y MMM"
}, _bbn618 = {
    M: "U MMMM–MMMM",
    y: "U MMMM – U MMMM"
}, _bbn619 = {
    M: "G y MMMM–MMMM",
    y: "G y MMMM – y MMMM"
}, _bbn620 = {
    M: "GGGGG y-MM – y-MM",
    y: "GGGGG y-MM – y-MM"
}, _bbn621 = {
    "0": "this year",
    "1": "next year",
    "-1": "last year"
}, _bbn622 = {
    "0": "this week",
    "1": "next week",
    "-1": "last week"
}, _bbn623 = {
    future: {
        other: "+{0} y"
    },
    past: {
        other: "-{0} y"
    }
}, _bbn624 = {
    future: {
        other: "+{0} m"
    },
    past: {
        other: "-{0} m"
    }
}, _bbn625 = {
    future: {
        other: "+{0} w"
    },
    past: {
        other: "-{0} w"
    }
}, _bbn626 = {
    future: {
        other: "+{0} d"
    },
    past: {
        other: "-{0} d"
    }
}, _bbn627 = {
    future: {
        other: "+{0} h"
    },
    past: {
        other: "-{0} h"
    }
}, _bbn628 = {
    future: {
        other: "+{0} s"
    },
    past: {
        other: "-{0} s"
    }
}, _bbn629 = {
    future: {
        other: "+{0} Q"
    },
    past: {
        other: "-{0} Q"
    }
}, _bbn630 = {
    d: "MM-dd, E – MM-dd, E",
    M: "MM-dd, E – MM-dd, E"
}, _bbn631 = {
    d: "MMM d, E – MMM d, E",
    M: "MMM d, E – MMM d, E"
}, _bbn632 = {
    "0": "this month",
    "1": "next month",
    "-1": "last month"
}, _bbn633 = {
    wide: [ "ERA0", "ERA1" ],
    abbreviated: [ "ERA0", "ERA1" ]
}, _bbn634 = {
    future: {
        other: "+{0} min"
    },
    past: {
        other: "-{0} min"
    }
}, _bbn635 = [ "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12" ], _bbn636 = {
    "0": "this Sunday",
    "1": "next Sunday",
    "-1": "last Sunday"
}, _bbn637 = {
    "0": "this Monday",
    "1": "next Monday",
    "-1": "last Monday"
}, _bbn638 = {
    "0": "this Friday",
    "1": "next Friday",
    "-1": "last Friday"
}, _bbn639 = {
    a: "h:mm a – h:mm a",
    h: "h:mm–h:mm a",
    m: "h:mm–h:mm a"
}, _bbn640 = {
    "0": "this quarter",
    "1": "next quarter",
    "-1": "last quarter"
}, _bbn641 = {
    "0": "this Tuesday",
    "1": "next Tuesday",
    "-1": "last Tuesday"
}, _bbn642 = {
    d: "U MMM d–d",
    M: "U MMM d – MMM d",
    y: "U MMM d – U MMM d"
}, _bbn643 = [ "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13" ], _bbn644 = {
    "0": "this Thursday",
    "1": "next Thursday",
    "-1": "last Thursday"
}, _bbn645 = {
    "0": "this Saturday",
    "1": "next Saturday",
    "-1": "last Saturday"
}, _bbn646 = {
    a: "h:mm a – h:mm a v",
    h: "h:mm–h:mm a v",
    m: "h:mm–h:mm a v"
}, _bbn647 = {
    id: "US",
    displayName: "Poo",
    regionId: "021",
    hasTimeZones: true
}, _bbn648 = {
    "0": "this Wednesday",
    "1": "next Wednesday",
    "-1": "last Wednesday"
}, _bbn649 = {
    id: "FJ",
    displayName: "Fíji",
    regionId: "054",
    hasTimeZones: true
}, _bbn650 = {
    id: "GH",
    displayName: "Gana",
    regionId: "011",
    hasTimeZones: true
}, _bbn651 = {
    id: "GN",
    displayName: "Gini",
    regionId: "011",
    hasTimeZones: true
}, _bbn652 = {
    id: "IR",
    displayName: "Iraŋ",
    regionId: "034",
    hasTimeZones: true
}, _bbn653 = {
    id: "ML",
    displayName: "Mali",
    regionId: "011",
    hasTimeZones: true
}, _bbn654 = {
    id: "NP",
    displayName: "Nepa",
    regionId: "034",
    hasTimeZones: true
}, _bbn655 = {
    id: "NU",
    displayName: "Niwe",
    regionId: "061",
    hasTimeZones: true
}, _bbn656 = {
    id: "NR",
    displayName: "Noru",
    regionId: "057",
    hasTimeZones: true
}, _bbn657 = {
    id: "OM",
    displayName: "Omaŋ",
    regionId: "145",
    hasTimeZones: true
}, _bbn658 = {
    id: "PW",
    displayName: "Palo",
    regionId: "057",
    hasTimeZones: true
}, _bbn659 = {
    id: "PE",
    displayName: "Pɛru",
    regionId: "005",
    hasTimeZones: true
}, _bbn660 = {
    id: "TG",
    displayName: "Togo",
    regionId: "011",
    hasTimeZones: true
}, _bbn661 = {
    id: "BJ",
    displayName: "Bhɛni",
    regionId: "011",
    hasTimeZones: true
}, _bbn662 = {
    id: "TD",
    displayName: "Chádi",
    regionId: "017",
    hasTimeZones: true
}, _bbn663 = {
    id: "CL",
    displayName: "Chéli",
    regionId: "005",
    hasTimeZones: true
}, _bbn664 = {
    id: "FR",
    displayName: "Fɛŋsi",
    regionId: "155",
    hasTimeZones: true
}, _bbn665 = {
    id: "HT",
    displayName: "Háiti",
    regionId: "029",
    hasTimeZones: true
}, _bbn666 = {
    id: "GR",
    displayName: "Hɛlɛŋ",
    regionId: "039",
    hasTimeZones: true
}, _bbn667 = {
    id: "IQ",
    displayName: "Iraki",
    regionId: "145",
    hasTimeZones: true
}, _bbn668 = {
    id: "DE",
    displayName: "Jamáĩ",
    regionId: "155",
    hasTimeZones: true
}, _bbn669 = {
    id: "JP",
    displayName: "Japaŋ",
    regionId: "030",
    hasTimeZones: true
}, _bbn670 = {
    id: "KE",
    displayName: "Kénya",
    regionId: "014",
    hasTimeZones: true
}, _bbn671 = {
    id: "CG",
    displayName: "Kóngo",
    regionId: "017",
    hasTimeZones: true
}, _bbn672 = {
    id: "NE",
    displayName: "Naĩja",
    regionId: "011",
    hasTimeZones: true
}, _bbn673 = {
    id: "NO",
    displayName: "Nɔɔwe",
    regionId: "154",
    hasTimeZones: true
}, _bbn674 = {
    id: "PL",
    displayName: "Pólɛŋ",
    regionId: "151",
    hasTimeZones: true
}, _bbn675 = {
    id: "SD",
    displayName: "Sudɛŋ",
    regionId: "015",
    hasTimeZones: true
}, _bbn676 = {
    id: "SY",
    displayName: "Síyaŋ",
    regionId: "145",
    hasTimeZones: true
}, _bbn677 = {
    id: "TO",
    displayName: "Tɔnga",
    regionId: "061",
    hasTimeZones: true
}, _bbn678 = {
    id: "IT",
    displayName: "Ítali",
    regionId: "039",
    hasTimeZones: true
}, _bbn679 = {
    d: "G y MMM d–d",
    M: "G y MMM d – MMM d",
    y: "G y MMM d – y MMM d"
}, _bbn680 = [ "lahadi", "tɛɛnɛɛ", "talata", "alaba", "aimisa", "aijima", "siɓiti" ], _bbn681 = {
    relative: _bbn636
}, _bbn682 = {
    relative: _bbn637
}, _bbn683 = {
    relative: _bbn638
}, _bbn684 = {
    id: "AW",
    displayName: "Arubha",
    regionId: "029",
    hasTimeZones: true
}, _bbn685 = {
    id: "AD",
    displayName: "Aŋdóra",
    regionId: "039",
    hasTimeZones: true
}, _bbn686 = {
    id: "AI",
    displayName: "Aŋgíla",
    regionId: "029",
    hasTimeZones: true
}, _bbn687 = {
    id: "AO",
    displayName: "Aŋgóla",
    regionId: "017",
    hasTimeZones: true
}, _bbn688 = {
    id: "BH",
    displayName: "Bharɛŋ",
    regionId: "145",
    hasTimeZones: true
}, _bbn689 = {
    id: "BZ",
    displayName: "Bheliz",
    regionId: "013",
    hasTimeZones: true
}, _bbn690 = {
    id: "BT",
    displayName: "Bhutaŋ",
    regionId: "034",
    hasTimeZones: true
}, _bbn691 = {
    id: "CN",
    displayName: "Cháína",
    regionId: "030",
    hasTimeZones: true
}, _bbn692 = {
    id: "FI",
    displayName: "Fiŋlɛŋ",
    regionId: "154",
    hasTimeZones: true
}, _bbn693 = {
    id: "GA",
    displayName: "Gabhɔŋ",
    regionId: "017",
    hasTimeZones: true
}, _bbn694 = {
    id: "GU",
    displayName: "Guwami",
    regionId: "057",
    hasTimeZones: true
}, _bbn695 = {
    id: "GY",
    displayName: "Guyana",
    regionId: "005",
    hasTimeZones: true
}, _bbn696 = {
    id: "JO",
    displayName: "Jɔɔdaŋ",
    regionId: "145",
    hasTimeZones: true
}, _bbn697 = {
    id: "QA",
    displayName: "Kataha",
    regionId: "145",
    hasTimeZones: true
}, _bbn698 = {
    id: "KW",
    displayName: "Kuweti",
    regionId: "145",
    hasTimeZones: true
}, _bbn699 = {
    id: "CA",
    displayName: "Kánáda",
    regionId: "021",
    hasTimeZones: true
}, _bbn6100 = {
    id: "SA",
    displayName: "Lahabu",
    regionId: "145",
    hasTimeZones: true
}, _bbn6101 = {
    id: "LA",
    displayName: "Lawosi",
    regionId: "035",
    hasTimeZones: true
}, _bbn6102 = {
    id: "LS",
    displayName: "Lisóto",
    regionId: "018",
    hasTimeZones: true
}, _bbn6103 = {
    id: "MW",
    displayName: "Malawi",
    regionId: "014",
    hasTimeZones: true
}, _bbn6104 = {
    id: "MT",
    displayName: "Malita",
    regionId: "039",
    hasTimeZones: true
}, _bbn6105 = {
    id: "YT",
    displayName: "Mavote",
    regionId: "014",
    hasTimeZones: true
}, _bbn6106 = {
    id: "EG",
    displayName: "Míséla",
    regionId: "015",
    hasTimeZones: true
}, _bbn6107 = {
    id: "MC",
    displayName: "Mɔnako",
    regionId: "155",
    hasTimeZones: true
}, _bbn6108 = {
    id: "MA",
    displayName: "Mɔroko",
    regionId: "015",
    hasTimeZones: true
}, _bbn6109 = {
    id: "MX",
    displayName: "Mɛsíko",
    regionId: "013",
    hasTimeZones: true
}, _bbn6110 = {
    id: "PA",
    displayName: "Panama",
    regionId: "013",
    hasTimeZones: true
}, _bbn6111 = {
    id: "ES",
    displayName: "Panyɛĩ",
    regionId: "039",
    hasTimeZones: true
}, _bbn6112 = {
    id: "WS",
    displayName: "Samowa",
    regionId: "061",
    hasTimeZones: true
}, _bbn6113 = {
    id: "SC",
    displayName: "Seshɛɛ",
    regionId: "014",
    hasTimeZones: true
}, _bbn6114 = {
    id: "TW",
    displayName: "Taiwaŋ",
    regionId: "030",
    hasTimeZones: true
}, _bbn6115 = {
    id: "TK",
    displayName: "Tokelo",
    regionId: "061",
    hasTimeZones: true
}, _bbn6116 = {
    id: "TV",
    displayName: "Tuválu",
    regionId: "061",
    hasTimeZones: true
}, _bbn6117 = {
    id: "WF",
    displayName: "Walísi",
    regionId: "061",
    hasTimeZones: true
}, _bbn6118 = {
    id: "YE",
    displayName: "Yemɛni",
    regionId: "145",
    hasTimeZones: true
}, _bbn6119 = {
    id: "IN",
    displayName: "Índiya",
    regionId: "034",
    hasTimeZones: true
}, _bbn6120 = {
    id: "DZ",
    displayName: "Agiriya",
    regionId: "015",
    hasTimeZones: true
}, _bbn6121 = {
    id: "AM",
    displayName: "Améniya",
    regionId: "145",
    hasTimeZones: true
}, _bbn6122 = {
    id: "BM",
    displayName: "Bhɛmuda",
    regionId: "021",
    hasTimeZones: true
}, _bbn6123 = {
    id: "PH",
    displayName: "Félepiŋ",
    regionId: "035",
    hasTimeZones: true
}, _bbn6124 = {
    id: "GM",
    displayName: "Gambiya",
    regionId: "011",
    hasTimeZones: true
}, _bbn6125 = {
    id: "HN",
    displayName: "Hɔndura",
    regionId: "013",
    hasTimeZones: true
}, _bbn6126 = {
    id: "JM",
    displayName: "Jamaika",
    regionId: "029",
    hasTimeZones: true
}, _bbn6127 = {
    id: "DJ",
    displayName: "Jibhuti",
    regionId: "014",
    hasTimeZones: true
}, _bbn6128 = {
    id: "GE",
    displayName: "Jɔɔjiya",
    regionId: "145",
    hasTimeZones: true
}, _bbn6129 = {
    id: "CM",
    displayName: "Kameruŋ",
    regionId: "017",
    hasTimeZones: true
}, _bbn6130 = {
    id: "CU",
    displayName: "Kiyubha",
    regionId: "029",
    hasTimeZones: true
}, _bbn6131 = {
    id: "LB",
    displayName: "Lebhanɔ",
    regionId: "145",
    hasTimeZones: true
}, _bbn6132 = {
    id: "LY",
    displayName: "Lebhiya",
    regionId: "015",
    hasTimeZones: true
}, _bbn6133 = {
    id: "LU",
    displayName: "Lusimbɔ",
    regionId: "155",
    hasTimeZones: true
}, _bbn6134 = {
    id: "MU",
    displayName: "Mɔreshɔ",
    regionId: "014",
    hasTimeZones: true
}, _bbn6135 = {
    id: "NL",
    displayName: "Nidɔlɛŋ",
    regionId: "155",
    hasTimeZones: true
}, _bbn6136 = {
    id: "PY",
    displayName: "Paragɔe",
    regionId: "005",
    hasTimeZones: true
}, _bbn6137 = {
    id: "PN",
    displayName: "Pitikɛŋ",
    regionId: "061",
    hasTimeZones: true
}, _bbn6138 = {
    id: "PT",
    displayName: "Potokíi",
    regionId: "039",
    hasTimeZones: true
}, _bbn6139 = {
    id: "RW",
    displayName: "Rawunda",
    regionId: "014",
    hasTimeZones: true
}, _bbn6140 = {
    id: "RE",
    displayName: "Renyɔɔ̃",
    regionId: "014",
    hasTimeZones: true
}, _bbn6141 = {
    id: "RU",
    displayName: "Rɔshiya",
    regionId: "151",
    hasTimeZones: true
}, _bbn6142 = {
    id: "CY",
    displayName: "Saɛpurɔ",
    regionId: "145",
    hasTimeZones: true
}, _bbn6143 = {
    id: "SE",
    displayName: "Suwidɛŋ",
    regionId: "154",
    hasTimeZones: true
}, _bbn6144 = {
    id: "TR",
    displayName: "Tɔ́ɔ́ki",
    regionId: "145",
    hasTimeZones: true
}, _bbn6145 = {
    id: "UG",
    displayName: "Yuganda",
    regionId: "014",
    hasTimeZones: true
}, _bbn6146 = {
    id: "UA",
    displayName: "Yukuréŋ",
    regionId: "151",
    hasTimeZones: true
}, _bbn6147 = {
    id: "ZM",
    displayName: "Zambiya",
    regionId: "014",
    hasTimeZones: true
}, _bbn6148 = {
    id: "ER",
    displayName: "Ɛritera",
    regionId: "014",
    hasTimeZones: true
}, _bbn6149 = {
    id: "AL",
    displayName: "Abhaniya",
    regionId: "039",
    hasTimeZones: true
}, _bbn6150 = {
    id: "AR",
    displayName: "Ajɛŋtína",
    regionId: "005",
    hasTimeZones: true
}, _bbn6151 = {
    id: "CD",
    displayName: "Avorekoo",
    regionId: "017",
    hasTimeZones: true
}, _bbn6152 = {
    id: "BS",
    displayName: "Bahámasi",
    regionId: "029",
    hasTimeZones: true
}, _bbn6153 = {
    id: "BB",
    displayName: "Bhabhedo",
    regionId: "029",
    hasTimeZones: true
}, _bbn6154 = {
    id: "BI",
    displayName: "Bhurundi",
    regionId: "014",
    hasTimeZones: true
}, _bbn6155 = {
    id: "BN",
    displayName: "Bhurunɛĩ",
    regionId: "035",
    hasTimeZones: true
}, _bbn6156 = {
    id: "BE",
    displayName: "Bhɛgiyɔŋ",
    regionId: "155",
    hasTimeZones: true
}, _bbn6157 = {
    id: "DK",
    displayName: "Danimaha",
    regionId: "154",
    hasTimeZones: true
}, _bbn6158 = {
    id: "GD",
    displayName: "Gurinéda",
    regionId: "029",
    hasTimeZones: true
}, _bbn6159 = {
    id: "HU",
    displayName: "Hɔ́ngare",
    regionId: "151",
    hasTimeZones: true
}, _bbn6160 = {
    id: "KM",
    displayName: "Komorosi",
    regionId: "014",
    hasTimeZones: true
}, _bbn6161 = {
    id: "HR",
    displayName: "Koresiya",
    regionId: "039",
    hasTimeZones: true
}, _bbn6162 = {
    id: "CI",
    displayName: "Kódivówa",
    regionId: "011",
    hasTimeZones: true
}, _bbn6163 = {
    id: "LR",
    displayName: "Laibhiya",
    regionId: "011",
    hasTimeZones: true
}, _bbn6164 = {
    id: "LV",
    displayName: "Lativiya",
    regionId: "154",
    hasTimeZones: true
}, _bbn6165 = {
    id: "MY",
    displayName: "Malesiya",
    regionId: "035",
    hasTimeZones: true
}, _bbn6166 = {
    id: "MV",
    displayName: "Malidavi",
    regionId: "034",
    hasTimeZones: true
}, _bbn6167 = {
    id: "MQ",
    displayName: "Matiniki",
    regionId: "029",
    hasTimeZones: true
}, _bbn6168 = {
    id: "MM",
    displayName: "Miyamaha",
    regionId: "035",
    hasTimeZones: true
}, _bbn6169 = {
    id: "MD",
    displayName: "Mɔlidova",
    regionId: "151",
    hasTimeZones: true
}, _bbn6170 = {
    id: "MS",
    displayName: "Mɔserati",
    regionId: "029",
    hasTimeZones: true
}, _bbn6171 = {
    id: "NA",
    displayName: "Namibiya",
    regionId: "018",
    hasTimeZones: true
}, _bbn6172 = {
    id: "SN",
    displayName: "Sinigaha",
    regionId: "011",
    hasTimeZones: true
}, _bbn6173 = {
    id: "SO",
    displayName: "Somaliya",
    regionId: "014",
    hasTimeZones: true
}, _bbn6174 = {
    id: "TN",
    displayName: "Tunisiya",
    regionId: "015",
    hasTimeZones: true
}, _bbn6175 = {
    id: "VU",
    displayName: "Vanuwátu",
    regionId: "054",
    hasTimeZones: true
}, _bbn6176 = {
    id: "VN",
    displayName: "Viyanami",
    regionId: "035",
    hasTimeZones: true
}, _bbn6177 = {
    id: "UY",
    displayName: "Yuwegɔwe",
    regionId: "005",
    hasTimeZones: true
}, _bbn6178 = {
    id: "AT",
    displayName: "Ɔ́situwa",
    regionId: "155",
    hasTimeZones: true
}, _bbn6179 = {
    id: "EC",
    displayName: "Ɛ́kúwédɔ",
    regionId: "005",
    hasTimeZones: true
}, _bbn6180 = {
    relative: _bbn641
}, _bbn6181 = {
    id: "BO",
    displayName: "Bholiviya",
    regionId: "005",
    hasTimeZones: true
}, _bbn6182 = {
    id: "BW",
    displayName: "Bhosuwana",
    regionId: "018",
    hasTimeZones: true
}, _bbn6183 = {
    id: "BR",
    displayName: "Bhurazeli",
    regionId: "005",
    hasTimeZones: true
}, _bbn6184 = {
    id: "BG",
    displayName: "Bhɔgeriya",
    regionId: "151",
    hasTimeZones: true
}, _bbn6185 = {
    id: "BA",
    displayName: "Bhɔsiniya",
    regionId: "039",
    hasTimeZones: true
}, _bbn6186 = {
    id: "BY",
    displayName: "Bhɛlarusi",
    regionId: "151",
    hasTimeZones: true
}, _bbn6187 = {
    id: "DM",
    displayName: "Domíiníka",
    regionId: "029",
    hasTimeZones: true
}, _bbn6188 = {
    id: "GI",
    displayName: "Jibhurata",
    regionId: "039",
    hasTimeZones: true
}, _bbn6189 = {
    id: "KZ",
    displayName: "Kazasitaŋ",
    regionId: "143",
    hasTimeZones: true
}, _bbn6190 = {
    id: "KG",
    displayName: "Kigisitaŋ",
    regionId: "143",
    hasTimeZones: true
}, _bbn6191 = {
    id: "KI",
    displayName: "Kiribhati",
    regionId: "057",
    hasTimeZones: true
}, _bbn6192 = {
    id: "CO",
    displayName: "Kɔlɔmbiya",
    regionId: "005",
    hasTimeZones: true
}, _bbn6193 = {
    id: "MZ",
    displayName: "Mozambiki",
    regionId: "014",
    hasTimeZones: true
}, _bbn6194 = {
    id: "MN",
    displayName: "Mɔngoliya",
    regionId: "030",
    hasTimeZones: true
}, _bbn6195 = {
    id: "NG",
    displayName: "Naĩjiriya",
    regionId: "011",
    hasTimeZones: true
}, _bbn6196 = {
    id: "PK",
    displayName: "Pakisitaŋ",
    regionId: "034",
    hasTimeZones: true
}, _bbn6197 = {
    id: "RO",
    displayName: "Romininya",
    regionId: "151",
    hasTimeZones: true
}, _bbn6198 = {
    id: "VC",
    displayName: "Siŋ Viŋsi",
    regionId: "029",
    hasTimeZones: true
}, _bbn6199 = {
    id: "SR",
    displayName: "Surinambe",
    regionId: "005",
    hasTimeZones: true
}, _bbn6200 = {
    id: "SG",
    displayName: "Síingapoo",
    regionId: "035",
    hasTimeZones: true
}, _bbn6201 = {
    id: "TH",
    displayName: "Tai Lumaã",
    regionId: "035",
    hasTimeZones: true
}, _bbn6202 = {
    id: "IE",
    displayName: "Áre Lumaã",
    regionId: "154",
    hasTimeZones: true
}, _bbn6203 = {
    id: "ET",
    displayName: "Ítiyópiya",
    regionId: "014",
    hasTimeZones: true
}, _bbn6204 = {
    d: "y-MM-dd – y-MM-dd",
    M: "y-MM-dd – y-MM-dd",
    y: "y-MM-dd – y-MM-dd"
}, _bbn6205 = {
    id: "AZ",
    displayName: "Azabhaijaŋ",
    regionId: "145",
    hasTimeZones: true
}, _bbn6206 = {
    id: "GP",
    displayName: "Guwadelupe",
    regionId: "029",
    hasTimeZones: true
}, _bbn6207 = {
    id: "GT",
    displayName: "Guwatɛmala",
    regionId: "013",
    hasTimeZones: true
}, _bbn6208 = {
    id: "KH",
    displayName: "Kaŋbhodiya",
    regionId: "035",
    hasTimeZones: true
}, _bbn6209 = {
    id: "CK",
    displayName: "Kóki Tiŋŋɛ",
    regionId: "061",
    hasTimeZones: true
}, _bbn6210 = {
    id: "MG",
    displayName: "Madagasita",
    regionId: "014",
    hasTimeZones: true
}, _bbn6211 = {
    id: "FM",
    displayName: "Mikonisiya",
    regionId: "057",
    hasTimeZones: true
}, _bbn6212 = {
    id: "NI",
    displayName: "Nikaraguwa",
    regionId: "013",
    hasTimeZones: true
}, _bbn6213 = {
    id: "NF",
    displayName: "Nɔfɔ Tiŋŋɛ",
    regionId: "053",
    hasTimeZones: true
}, _bbn6214 = {
    id: "SH",
    displayName: "Siŋ Hɛlina",
    regionId: "011",
    hasTimeZones: true
}, _bbn6215 = {
    id: "LC",
    displayName: "Siŋ Lusiya",
    regionId: "029",
    hasTimeZones: true
}, _bbn6216 = {
    id: "SK",
    displayName: "Sulovakiya",
    regionId: "151",
    hasTimeZones: true
}, _bbn6217 = {
    id: "LK",
    displayName: "Suri Laŋka",
    regionId: "034",
    hasTimeZones: true
}, _bbn6218 = {
    id: "TZ",
    displayName: "Taŋzaninya",
    regionId: "014",
    hasTimeZones: true
}, _bbn6219 = {
    id: "VE",
    displayName: "Vɛnɛzuwela",
    regionId: "005",
    hasTimeZones: true
}, _bbn6220 = {
    id: "ZW",
    displayName: "Zimbabhuwe",
    regionId: "014",
    hasTimeZones: true
}, _bbn6221 = {
    id: "IS",
    displayName: "Áisi Lumaã",
    regionId: "154",
    hasTimeZones: true
}, _bbn6222 = {
    id: "ID",
    displayName: "Índonisiya",
    regionId: "035",
    hasTimeZones: true
}, _bbn6223 = {
    id: "SV",
    displayName: "Ɛlɛ Sávádɔ",
    regionId: "013",
    hasTimeZones: true
}, _bbn6224 = {
    id: "EE",
    displayName: "Ɛsitóninya",
    regionId: "154",
    hasTimeZones: true
}, _bbn6225 = {
    id: "CZ",
    displayName: "Chɛki Boloe",
    regionId: "151",
    hasTimeZones: true
}, _bbn6226 = {
    id: "CR",
    displayName: "Kósíta Ríko",
    regionId: "013",
    hasTimeZones: true
}, _bbn6227 = {
    id: "LT",
    displayName: "Lituweninya",
    regionId: "154",
    hasTimeZones: true
}, _bbn6228 = {
    id: "MK",
    displayName: "Masedoninya",
    regionId: "039",
    hasTimeZones: true
}, _bbn6229 = {
    id: "MH",
    displayName: "Masha Tiŋŋɛ",
    regionId: "057",
    hasTimeZones: true
}, _bbn6230 = {
    id: "MR",
    displayName: "Mɔretaninya",
    regionId: "011",
    hasTimeZones: true
}, _bbn6231 = {
    id: "PR",
    displayName: "Piyuto Riko",
    regionId: "029",
    hasTimeZones: true
}, _bbn6232 = {
    id: "AS",
    displayName: "Poo Sambowa",
    regionId: "061",
    hasTimeZones: true
}, _bbn6233 = {
    id: "SM",
    displayName: "Saŋ Marindo",
    regionId: "039",
    hasTimeZones: true
}, _bbn6234 = {
    id: "SI",
    displayName: "Suloveninya",
    regionId: "039",
    hasTimeZones: true
}, _bbn6235 = {
    id: "TJ",
    displayName: "Tajikisitaŋ",
    regionId: "143",
    hasTimeZones: true
}, _bbn6236 = {
    id: "NZ",
    displayName: "Zilɛŋ Námaá",
    regionId: "053",
    hasTimeZones: true
}, _bbn6237 = {
    relative: _bbn644
}, _bbn6238 = {
    relative: _bbn645
}, _bbn6239 = {
    id: "IL",
    displayName: "Bhanísiláila",
    regionId: "145",
    hasTimeZones: true
}, _bbn6240 = {
    id: "BF",
    displayName: "Bhokina Fáso",
    regionId: "011",
    hasTimeZones: true
}, _bbn6241 = {
    id: "GF",
    displayName: "Fɛŋsi Giwana",
    regionId: "005",
    hasTimeZones: true
}, _bbn6242 = {
    id: "SL",
    displayName: "Gbeya Bahawɔ",
    regionId: "011",
    hasTimeZones: true
}, _bbn6243 = {
    id: "GW",
    displayName: "Gini Bhisawo",
    regionId: "011",
    hasTimeZones: true
}, _bbn6244 = {
    id: "KY",
    displayName: "Keemaŋ Tiŋŋɛ",
    regionId: "029",
    hasTimeZones: true
}, _bbn6245 = {
    id: "SZ",
    displayName: "Suwazi Lumaã",
    regionId: "018",
    hasTimeZones: true
}, _bbn6246 = {
    id: "CH",
    displayName: "Suweza Lumaã",
    regionId: "155",
    hasTimeZones: true
}, _bbn6247 = {
    id: "UZ",
    displayName: "Yubhɛkisitaŋ",
    regionId: "143",
    hasTimeZones: true
}, _bbn6248 = {
    id: "AU",
    displayName: "Ɔsituwéeliya",
    regionId: "053",
    hasTimeZones: true
}, _bbn6249 = {
    id: "AF",
    displayName: "Afigándesitaŋ",
    regionId: "034",
    hasTimeZones: true
}, _bbn6250 = {
    id: "TM",
    displayName: "Tukimɛnisitaŋ",
    regionId: "143",
    hasTimeZones: true
}, _bbn6251 = {
    id: "VA",
    displayName: "Vatikaŋ Ɓoloe",
    regionId: "039",
    hasTimeZones: true
}, _bbn6252 = {
    abbreviated: {
        leap: "{0}bis"
    },
    narrow: {
        leap: "{0}b"
    },
    wide: {
        leap: "{0}bis"
    }
}, _bbn6253 = {
    id: "BD",
    displayName: "Bhangiladɛ̀shi",
    regionId: "034",
    hasTimeZones: true
}, _bbn6254 = {
    id: "GB",
    displayName: "Yunaitɛ Kíŋdɔŋ",
    regionId: "154",
    hasTimeZones: true
}, _bbn6255 = {
    relative: _bbn648
}, _bbn6256 = {
    id: "DO",
    displayName: "Domíiníka Ɓoloe",
    regionId: "029",
    hasTimeZones: true
}, _bbn6257 = {
    id: "GQ",
    displayName: "Dúúnyá Tɛ Giini",
    regionId: "017",
    hasTimeZones: true
}, _bbn6258 = {
    id: "CV",
    displayName: "Kepi Vɛdi Tiŋŋɛ",
    regionId: "011",
    hasTimeZones: true
}, _bbn6259 = {
    id: "TL",
    displayName: "Tele Ɓɔ́ Timɔɔ̃",
    regionId: "035",
    hasTimeZones: true
}, _bbn6260 = [ "M01", "M02", "M03", "M04", "M05", "M06", "M07", "M08", "M09", "M10", "M11", "M12" ], _bbn6261 = {
    id: "PF",
    displayName: "Fɛŋsi Polinísiya",
    regionId: "061",
    hasTimeZones: true
}, _bbn6262 = {
    id: "GL",
    displayName: "Jamba Kuwa Lumaã",
    regionId: "021",
    hasTimeZones: true
}, _bbn6263 = {
    id: "IO",
    displayName: "Jengéesi Gbawoe Índiya Kɔiyɛ Lɔ",
    hasTimeZones: true
}, _bbn6264 = {
    id: "SB",
    displayName: "Sulaimaãna Tiŋŋɛ",
    regionId: "054",
    hasTimeZones: true
}, _bbn6265 = {
    id: "AG",
    displayName: "Aŋtígwa Ɓahabhuda",
    regionId: "029",
    hasTimeZones: true
}, _bbn6266 = {
    id: "FK",
    displayName: "Fáháki Luma Tiŋŋɛ",
    regionId: "005",
    hasTimeZones: true
}, _bbn6267 = {
    id: "NC",
    displayName: "Kalidoninya Námaá",
    regionId: "054",
    hasTimeZones: true
}, _bbn6268 = {
    id: "KP",
    displayName: "Koriya Kɔi Kaŋndɔ",
    regionId: "030",
    hasTimeZones: true
}, _bbn6269 = {
    id: "PG",
    displayName: "Papuwa Gini Námaá",
    regionId: "054",
    hasTimeZones: true
}, _bbn6270 = [ undefined, "zi", "chou", "yin", "mao", "chen", "si", "wu", "wei", "shen", "you", "xu" ], _bbn6271 = {
    id: "KR",
    displayName: "Koriya Kɔi Leŋŋɛ Lɔ",
    regionId: "030",
    hasTimeZones: true
}, _bbn6272 = {
    id: "KN",
    displayName: "Siŋ Kisi ɓɛ́ Nevisi",
    regionId: "029",
    hasTimeZones: true
}, _bbn6273 = {
    id: "ZA",
    displayName: "Afirika Kɔi Leŋŋɛ Lɔ",
    regionId: "018",
    hasTimeZones: true
}, _bbn6274 = {
    id: "AE",
    displayName: "Yunaitɛ Arabhi Ɛmire",
    regionId: "145",
    hasTimeZones: true
}, _bbn6275 = {
    id: "PM",
    displayName: "Siŋ Piiyɛ ɓɛ́ Mikelɔŋ",
    regionId: "021",
    hasTimeZones: true
}, _bbn6276 = {
    id: "TT",
    displayName: "Turindeda ɓɛ́ Tobhego",
    regionId: "029",
    hasTimeZones: true
}, _bbn6277 = {
    abbreviated: {
        am: "AM",
        pm: "PM"
    },
    narrow: {
        am: "AM",
        pm: "PM"
    },
    wide: {
        am: "AM",
        pm: "PM"
    }
}, _bbn6278 = {
    id: "VI",
    displayName: "Poo Bhɛɛ lɔ Musu Tiŋŋɛ",
    regionId: "029",
    hasTimeZones: true
}, _bbn6279 = {
    id: "ST",
    displayName: "Sawo Tombe ɓɛ a Gbawoe",
    regionId: "017",
    hasTimeZones: true
}, _bbn6280 = {
    id: "CF",
    displayName: "Áfíríka Lumaã Tɛ Boloe",
    regionId: "017",
    hasTimeZones: true
}, _bbn6281 = {
    displayName: "hawa",
    relativeTime: _bbn627
}, _bbn6282 = {
    d: "U MMM d, E – MMM d, E",
    M: "U MMM d, E – MMM d, E",
    y: "U MMM d, E – U MMM d, E"
}, _bbn6283 = {
    id: "MP",
    displayName: "Kɔi Kaŋndɔ Mariyana Tiŋŋɛ",
    regionId: "057",
    hasTimeZones: true
}, _bbn6284 = {
    id: "TC",
    displayName: "Tukisi ɓɛ̀ Kaikóosi Tiŋŋɛ",
    regionId: "029",
    hasTimeZones: true
}, _bbn6285 = {
    id: "VG",
    displayName: "Jengéesi Bhɛɛ Lɔ Musu Tiŋŋɛ",
    regionId: "029",
    hasTimeZones: true
}, _bbn6286 = {
    d: "GGGGG y-MM-dd – y-MM-dd",
    M: "GGGGG y-MM-dd – y-MM-dd",
    y: "GGGGG y-MM-dd – y-MM-dd"
}, _bbn6287 = {
    displayName: "mini",
    relativeTime: _bbn634
}, _bbn6288 = {
    d: "y-MM-dd, E – y-MM-dd, E",
    M: "y-MM-dd, E – y-MM-dd, E",
    y: "y-MM-dd, E – y-MM-dd, E"
}, _bbn6289 = {
    d: "G y MMM d, E – MMM d, E",
    M: "G y MMM d, E – MMM d, E",
    y: "G y MMM d, E – y MMM d, E"
}, _bbn6290 = {
    abbreviated: _bbn61,
    narrow: [ "1", "2", "3", "4" ],
    wide: _bbn61
}, _bbn6291 = {
    id: "PS",
    displayName: "Palesitininya Tele Jii Kɔiyɛ lá hĩ Gaza",
    regionId: "145",
    hasTimeZones: true
}, _bbn6292 = {
    d: "GGGGG y-MM-dd, E – y-MM-dd, E",
    M: "GGGGG y-MM-dd, E – y-MM-dd, E",
    y: "GGGGG y-MM-dd, E – y-MM-dd, E"
}, _bbn6293 = {
    displayName: "jaki-jaka",
    relative: [ "now" ],
    relativeTime: _bbn628
}, _bbn6294 = [ "Muh.", "Saf.", "Rab. I", "Rab. II", "Jum. I", "Jum. II", "Raj.", "Sha.", "Ram.", "Shaw.", "Dhuʻl-Q.", "Dhuʻl-H." ], _bbn6295 = [ "Tishri", "Heshvan", "Kislev", "Tevet", "Shevat", "Adar I", "Adar", "Nisan", "Iyar", "Sivan", "Tamuz", "Av", "Elul" ], _bbn6296 = [ "luukao kemã", "ɓandaɓu", "vɔɔ", "fulu", "goo", "6", "7", "kɔnde", "saah", "galo", "kenpkato ɓololɔ", "luukao lɔma" ], _bbn6297 = [ "Farvardin", "Ordibehesht", "Khordad", "Tir", "Mordad", "Shahrivar", "Mehr", "Aban", "Azar", "Dey", "Bahman", "Esfand" ], _bbn6298 = [ "Tout", "Baba", "Hator", "Kiahk", "Toba", "Amshir", "Baramhat", "Baramouda", "Bashans", "Paona", "Epep", "Mesra", "Nasie" ], _bbn6299 = [ "Meskerem", "Tekemt", "Hedar", "Tahsas", "Ter", "Yekatit", "Megabit", "Miazia", "Genbot", "Sene", "Hamle", "Nehasse", "Pagumen" ], _bbn6300 = [ "Chaitra", "Vaisakha", "Jyaistha", "Asadha", "Sravana", "Bhadra", "Asvina", "Kartika", "Agrahayana", "Pausa", "Magha", "Phalguna" ], _bbn6301 = {
    displayName: "tele",
    relative: _bbn610,
    relativeTime: _bbn626
}, _bbn6302 = [ "Muharram", "Safar", "Rabiʻ I", "Rabiʻ II", "Jumada I", "Jumada II", "Rajab", "Shaʻban", "Ramadan", "Shawwal", "Dhuʻl-Qiʻdah", "Dhuʻl-Hijjah" ], _bbn6303 = {
    displayName: "saŋ",
    relative: _bbn621,
    relativeTime: _bbn623
}, _bbn6304 = {
    displayName: "wiki",
    relative: _bbn622,
    relativeTime: _bbn625
}, _bbn6305 = {
    displayName: "kalo",
    relative: _bbn632,
    relativeTime: _bbn624
}, _bbn6306 = {
    displayName: "Quarter",
    relative: _bbn640,
    relativeTime: _bbn629
}, _bbn6307 = {
    format: _bbn6277,
    standAlone: _bbn6277
}, _bbn6308 = {
    format: _bbn6252,
    numeric: _bbn62,
    standAlone: _bbn6252
}, _bbn6309 = {
    format: _bbn6290,
    standAlone: _bbn6290
}, _bbn6310 = {
    abbreviated: _bbn6260,
    narrow: _bbn635,
    wide: _bbn6260
}, _bbn6311 = {
    abbreviated: _bbn680,
    narrow: _bbn67,
    wide: _bbn680,
    "short": _bbn680
}, _bbn6312 = {
    abbreviated: _bbn6270,
    narrow: _bbn6270,
    wide: _bbn6270
}, _bbn6313 = {
    format: _bbn6312
}, _bbn6314 = {
    abbreviated: _bbn6296,
    narrow: _bbn635,
    wide: _bbn6296
}, _bbn6315 = {
    abbreviated: _bbn6295,
    narrow: _bbn643,
    wide: _bbn6295
}, _bbn6316 = {
    abbreviated: _bbn6297,
    narrow: _bbn635,
    wide: _bbn6297
}, _bbn6317 = {
    abbreviated: _bbn6298,
    narrow: _bbn643,
    wide: _bbn6298
}, _bbn6318 = {
    abbreviated: _bbn6294,
    narrow: _bbn635,
    wide: _bbn6302
}, _bbn6319 = {
    abbreviated: _bbn6299,
    narrow: _bbn643,
    wide: _bbn6299
}, _bbn6320 = {
    abbreviated: _bbn6300,
    narrow: _bbn635,
    wide: _bbn6300
}, _bbn6321 = {
    fullDate: "r(U) MMMM d, EEEE",
    longDate: "r(U) MMMM d",
    mediumDate: "r MMM d",
    shortDate: "r-MM-dd",
    fullTime: "h:mm:ss a zzzz",
    longTime: "h:mm:ss a z",
    mediumTime: "h:mm:ss a",
    shortTime: "h:mm a",
    shortDateTime: "r-MM-dd h:mm a",
    mediumDateTime: "r MMM d h:mm:ss a",
    longDateTime: "r(U) MMMM d h:mm:ss a z",
    fullDateTime: "r(U) MMMM d, EEEE h:mm:ss a zzzz"
}, _bbn6322 = {
    fullDate: "EEEE, d MMMM y G",
    longDate: "d MMMM y G",
    mediumDate: "d MMM y G",
    shortDate: "dd/MM/y G",
    fullTime: "h:mm:ss a zzzz",
    longTime: "h:mm:ss a z",
    mediumTime: "h:mm:ss a",
    shortTime: "h:mm a",
    shortDateTime: "dd/MM/y G h:mm a",
    mediumDateTime: "d MMM y G h:mm:ss a",
    longDateTime: "d MMMM y G h:mm:ss a z",
    fullDateTime: "EEEE, d MMMM y G h:mm:ss a zzzz"
}, _bbn6323 = {
    format: _bbn6310,
    standAlone: _bbn6310
}, _bbn6324 = {
    d: "d",
    E: "ccc",
    Ed: "d, E",
    Gy: "r U",
    GyMMM: "r(U) MMM",
    GyMMMd: "r MMM d",
    GyMMMEd: "r(U) MMM d, E",
    h: "h a",
    H: "HH",
    hm: "h:mm a",
    Hm: "HH:mm",
    hms: "h:mm:ss a",
    Hms: "HH:mm:ss",
    M: "L",
    Md: "MM-dd",
    MEd: "MM-dd, E",
    MMM: "LLL",
    MMMd: "MMM d",
    MMMEd: "MMM d, E",
    MMMMd: "MMMM d",
    ms: "mm:ss",
    UM: "U MM",
    UMd: "U MM-d",
    UMMM: "U MMM",
    UMMMd: "U MMM d",
    y: "r(U)",
    yMd: "r-MM-dd",
    yyyy: "r(U)",
    yyyyM: "r-MM",
    yyyyMd: "r-MM-dd",
    yyyyMEd: "r-MM-dd, E",
    yyyyMMM: "r(U) MMM",
    yyyyMMMd: "r MMM d",
    yyyyMMMEd: "r(U) MMM d, E",
    yyyyMMMM: "r(U) MMMM",
    yyyyQQQ: "r(U) QQQ",
    yyyyQQQQ: "r(U) QQQQ"
}, _bbn6325 = {
    format: _bbn6311,
    standAlone: _bbn6311
}, _bbn6326 = [ undefined, "jia-zi", "yi-chou", "bing-yin", "ding-mao", "wu-chen", "ji-si", "geng-wu", "xin-wei", "ren-shen", "gui-you", "jia-xu", "yi-hai", "bing-zi", "ding-chou", "wu-yin", "ji-mao", "geng-chen", "xin-si", "ren-wu", "gui-wei", "jia-shen", "yi-you", "bing-xu", "ding-hai", "wu-zi", "ji-chou", "geng-yin", "xin-mao", "ren-chen", "gui-si", "jia-wu", "yi-wei", "bing-shen", "ding-you", "wu-xu", "ji-hai", "geng-zi", "xin-chou", "ren-yin", "gui-mao", "jia-chen", "yi-si", "bing-wu", "ding-wei", "wu-shen", "ji-you", "geng-xu", "xin-hai", "ren-zi", "gui-chou", "jia-yin", "yi-mao", "bing-chen", "ding-si", "wu-wu", "ji-wei", "geng-shen", "xin-you", "ren-xu" ], _bbn6327 = {
    format: _bbn6314,
    standAlone: _bbn6314
}, _bbn6328 = {
    d: "d",
    Ed: "E d",
    hm: "h:mm a",
    Hm: "HH:mm",
    hms: "h:mm:ss a",
    Hms: "HH:mm:ss",
    M: "L",
    Md: "M/d",
    MEd: "E, M/d",
    MMM: "LLL",
    MMMd: "MMM d",
    MMMEd: "E, MMM d",
    ms: "m:ss",
    y: "y",
    yM: "M/y",
    yMd: "M/d/y",
    yMEd: "E, M/d/y",
    yMMM: "MMM y",
    yMMMd: "MMM d y",
    yMMMEd: "E, MMM d, y",
    yQQQ: "QQQ y",
    yQQQQ: "QQQQ y",
    E: "ccc",
    Ehm: "E h:mm a",
    EHm: "E HH:mm",
    Ehms: "E h:mm:ss a",
    EHms: "E HH:mm:ss",
    Gy: "G y",
    GyMMM: "G y MMM",
    GyMMMd: "G y MMM d",
    GyMMMEd: "G y MMM d, E",
    h: "h a",
    H: "HH",
    MMMMd: "MMMM d",
    yyyy: "G y",
    yyyyM: "GGGGG y-MM",
    yyyyMd: "GGGGG y-MM-dd",
    yyyyMEd: "GGGGG y-MM-dd, E",
    yyyyMMM: "G y MMM",
    yyyyMMMd: "G y MMM d",
    yyyyMMMEd: "G y MMM d, E",
    yyyyMMMM: "G y MMMM",
    yyyyQQQ: "G y QQQ",
    yyyyQQQQ: "G y QQQQ"
}, _bbn6329 = {
    format: _bbn6318,
    standAlone: _bbn6318
}, _bbn6330 = {
    basic: _bbn6321,
    cldr: _bbn6324
}, _bbn6331 = {
    d: {
        d: "d–d"
    },
    h: _bbn63,
    H: {
        H: "HH–HH"
    },
    hm: _bbn639,
    Hm: _bbn612,
    hmv: _bbn646,
    Hmv: _bbn615,
    hv: _bbn66,
    Hv: {
        H: "HH–HH v"
    },
    M: {
        M: "MM–MM"
    },
    Md: _bbn616,
    MEd: _bbn630,
    MMM: {
        M: "LLL–LLL"
    },
    MMMd: _bbn69,
    MMMEd: _bbn631,
    y: {
        y: "U–U"
    },
    yM: _bbn613,
    yMd: _bbn6204,
    yMEd: _bbn6288,
    yMMM: _bbn614,
    yMMMd: _bbn642,
    yMMMEd: _bbn6282,
    yMMMM: _bbn618
}, _bbn6332 = {
    basic: _bbn6322,
    cldr: _bbn6328
}, _bbn6333 = {
    d: {
        d: "d–d"
    },
    h: _bbn63,
    H: {
        H: "HH–HH"
    },
    hm: _bbn639,
    Hm: _bbn612,
    hmv: _bbn646,
    Hmv: _bbn615,
    hv: _bbn66,
    Hv: {
        H: "HH–HH v"
    },
    M: {
        M: "MM–MM"
    },
    Md: _bbn616,
    MEd: _bbn630,
    MMM: {
        M: "LLL–LLL"
    },
    MMMd: _bbn69,
    MMMEd: _bbn631,
    y: {
        y: "G y–y"
    },
    yM: _bbn620,
    yMd: _bbn6286,
    yMEd: _bbn6292,
    yMMM: _bbn617,
    yMMMd: _bbn679,
    yMMMEd: _bbn6289,
    yMMMM: _bbn619
}, _bbn6334 = {
    year: _bbn6303,
    month: _bbn6305,
    week: _bbn6304,
    day: _bbn6301,
    weekday: _bbn68,
    hour: _bbn6281,
    minute: _bbn6287,
    second: _bbn6293,
    era: {
        displayName: "Era"
    },
    quarter: _bbn6306,
    dayperiod: _bbn64,
    zone: {
        displayName: "Zone"
    },
    sun: _bbn681,
    mon: _bbn682,
    tue: _bbn6180,
    wed: _bbn6255,
    thu: _bbn6237,
    fri: _bbn683,
    sat: _bbn6238
}, _bbn6335 = {
    dateFormats: {
        basic: {},
        cldr: {}
    },
    fields: _bbn6334
}, _bbn6336 = {
    abbreviated: _bbn6326,
    narrow: _bbn6326,
    wide: _bbn6326
}, _bbn6337 = {
    format: _bbn6336
}, _bbn6338 = [ "Taika (645–650)", "Hakuchi (650–671)", "Hakuhō (672–686)", "Shuchō (686–701)", "Taihō (701–704)", "Keiun (704–708)", "Wadō (708–715)", "Reiki (715–717)", "Yōrō (717–724)", "Jinki (724–729)", "Tenpyō (729–749)", "Tenpyō-kampō (749-749)", "Tenpyō-shōhō (749-757)", "Tenpyō-hōji (757-765)", "Tenpyō-jingo (765-767)", "Jingo-keiun (767-770)", "Hōki (770–780)", "Ten-ō (781-782)", "Enryaku (782–806)", "Daidō (806–810)", "Kōnin (810–824)", "Tenchō (824–834)", "Jōwa (834–848)", "Kajō (848–851)", "Ninju (851–854)", "Saikō (854–857)", "Ten-an (857-859)", "Jōgan (859–877)", "Gangyō (877–885)", "Ninna (885–889)", "Kanpyō (889–898)", "Shōtai (898–901)", "Engi (901–923)", "Enchō (923–931)", "Jōhei (931–938)", "Tengyō (938–947)", "Tenryaku (947–957)", "Tentoku (957–961)", "Ōwa (961–964)", "Kōhō (964–968)", "Anna (968–970)", "Tenroku (970–973)", "Ten’en (973–976)", "Jōgen (976–978)", "Tengen (978–983)", "Eikan (983–985)", "Kanna (985–987)", "Eien (987–989)", "Eiso (989–990)", "Shōryaku (990–995)", "Chōtoku (995–999)", "Chōhō (999–1004)", "Kankō (1004–1012)", "Chōwa (1012–1017)", "Kannin (1017–1021)", "Jian (1021–1024)", "Manju (1024–1028)", "Chōgen (1028–1037)", "Chōryaku (1037–1040)", "Chōkyū (1040–1044)", "Kantoku (1044–1046)", "Eishō (1046–1053)", "Tengi (1053–1058)", "Kōhei (1058–1065)", "Jiryaku (1065–1069)", "Enkyū (1069–1074)", "Shōho (1074–1077)", "Shōryaku (1077–1081)", "Eihō (1081–1084)", "Ōtoku (1084–1087)", "Kanji (1087–1094)", "Kahō (1094–1096)", "Eichō (1096–1097)", "Jōtoku (1097–1099)", "Kōwa (1099–1104)", "Chōji (1104–1106)", "Kashō (1106–1108)", "Tennin (1108–1110)", "Ten-ei (1110-1113)", "Eikyū (1113–1118)", "Gen’ei (1118–1120)", "Hōan (1120–1124)", "Tenji (1124–1126)", "Daiji (1126–1131)", "Tenshō (1131–1132)", "Chōshō (1132–1135)", "Hōen (1135–1141)", "Eiji (1141–1142)", "Kōji (1142–1144)", "Ten’yō (1144–1145)", "Kyūan (1145–1151)", "Ninpei (1151–1154)", "Kyūju (1154–1156)", "Hōgen (1156–1159)", "Heiji (1159–1160)", "Eiryaku (1160–1161)", "Ōho (1161–1163)", "Chōkan (1163–1165)", "Eiman (1165–1166)", "Nin’an (1166–1169)", "Kaō (1169–1171)", "Shōan (1171–1175)", "Angen (1175–1177)", "Jishō (1177–1181)", "Yōwa (1181–1182)", "Juei (1182–1184)", "Genryaku (1184–1185)", "Bunji (1185–1190)", "Kenkyū (1190–1199)", "Shōji (1199–1201)", "Kennin (1201–1204)", "Genkyū (1204–1206)", "Ken’ei (1206–1207)", "Jōgen (1207–1211)", "Kenryaku (1211–1213)", "Kenpō (1213–1219)", "Jōkyū (1219–1222)", "Jōō (1222–1224)", "Gennin (1224–1225)", "Karoku (1225–1227)", "Antei (1227–1229)", "Kanki (1229–1232)", "Jōei (1232–1233)", "Tenpuku (1233–1234)", "Bunryaku (1234–1235)", "Katei (1235–1238)", "Ryakunin (1238–1239)", "En’ō (1239–1240)", "Ninji (1240–1243)", "Kangen (1243–1247)", "Hōji (1247–1249)", "Kenchō (1249–1256)", "Kōgen (1256–1257)", "Shōka (1257–1259)", "Shōgen (1259–1260)", "Bun’ō (1260–1261)", "Kōchō (1261–1264)", "Bun’ei (1264–1275)", "Kenji (1275–1278)", "Kōan (1278–1288)", "Shōō (1288–1293)", "Einin (1293–1299)", "Shōan (1299–1302)", "Kengen (1302–1303)", "Kagen (1303–1306)", "Tokuji (1306–1308)", "Enkyō (1308–1311)", "Ōchō (1311–1312)", "Shōwa (1312–1317)", "Bunpō (1317–1319)", "Genō (1319–1321)", "Genkō (1321–1324)", "Shōchū (1324–1326)", "Karyaku (1326–1329)", "Gentoku (1329–1331)", "Genkō (1331–1334)", "Kenmu (1334–1336)", "Engen (1336–1340)", "Kōkoku (1340–1346)", "Shōhei (1346–1370)", "Kentoku (1370–1372)", "Bunchū (1372–1375)", "Tenju (1375–1379)", "Kōryaku (1379–1381)", "Kōwa (1381–1384)", "Genchū (1384–1392)", "Meitoku (1384–1387)", "Kakei (1387–1389)", "Kōō (1389–1390)", "Meitoku (1390–1394)", "Ōei (1394–1428)", "Shōchō (1428–1429)", "Eikyō (1429–1441)", "Kakitsu (1441–1444)", "Bun’an (1444–1449)", "Hōtoku (1449–1452)", "Kyōtoku (1452–1455)", "Kōshō (1455–1457)", "Chōroku (1457–1460)", "Kanshō (1460–1466)", "Bunshō (1466–1467)", "Ōnin (1467–1469)", "Bunmei (1469–1487)", "Chōkyō (1487–1489)", "Entoku (1489–1492)", "Meiō (1492–1501)", "Bunki (1501–1504)", "Eishō (1504–1521)", "Taiei (1521–1528)", "Kyōroku (1528–1532)", "Tenbun (1532–1555)", "Kōji (1555–1558)", "Eiroku (1558–1570)", "Genki (1570–1573)", "Tenshō (1573–1592)", "Bunroku (1592–1596)", "Keichō (1596–1615)", "Genna (1615–1624)", "Kan’ei (1624–1644)", "Shōho (1644–1648)", "Keian (1648–1652)", "Jōō (1652–1655)", "Meireki (1655–1658)", "Manji (1658–1661)", "Kanbun (1661–1673)", "Enpō (1673–1681)", "Tenna (1681–1684)", "Jōkyō (1684–1688)", "Genroku (1688–1704)", "Hōei (1704–1711)", "Shōtoku (1711–1716)", "Kyōhō (1716–1736)", "Genbun (1736–1741)", "Kanpō (1741–1744)", "Enkyō (1744–1748)", "Kan’en (1748–1751)", "Hōreki (1751–1764)", "Meiwa (1764–1772)", "An’ei (1772–1781)", "Tenmei (1781–1789)", "Kansei (1789–1801)", "Kyōwa (1801–1804)", "Bunka (1804–1818)", "Bunsei (1818–1830)", "Tenpō (1830–1844)", "Kōka (1844–1848)", "Kaei (1848–1854)", "Ansei (1854–1860)", "Man’en (1860–1861)", "Bunkyū (1861–1864)", "Genji (1864–1865)", "Keiō (1865–1868)", "Meiji", "Taishō", "Shōwa", "Heisei" ], _bbn6339 = {
    dateFormats: _bbn6332,
    datePartsOrder: "dmy",
    dateIntervalFormats: _bbn6333,
    dateIntervalFallbackFormat: "{0} – {1}",
    dayNames: _bbn6325,
    dayPeriods: _bbn6307,
    monthNames: _bbn6329,
    eraNames: _bbn611,
    quarterNames: _bbn6309,
    fields: _bbn6334
}, _bbn6340 = {
    dayParts: _bbn6313,
    days: _bbn6337,
    months: _bbn6337,
    years: _bbn6337,
    zodiacs: _bbn6313
}, _bbn6341 = {
    dateFormats: _bbn6330,
    datePartsOrder: "md",
    dateIntervalFormats: _bbn6331,
    dateIntervalFallbackFormat: "{0} – {1}",
    dayNames: _bbn6325,
    dayPeriods: _bbn6307,
    cyclicNames: _bbn6340,
    monthNames: _bbn6323,
    monthPatterns: _bbn6308,
    quarterNames: _bbn6309,
    fields: _bbn6334
}, _bbn6342 = [ _bbn6149, _bbn6249, _bbn6273, _bbn6120, _bbn6150, _bbn6121, _bbn684, _bbn6151, _bbn6205, _bbn685, _bbn686, _bbn687, _bbn6265, _bbn6152, _bbn6153, _bbn6253, _bbn6239, _bbn688, _bbn689, _bbn6240, _bbn6181, _bbn6182, _bbn6183, _bbn6154, _bbn6155, _bbn690, _bbn6184, _bbn6185, _bbn6156, _bbn6186, _bbn6122, _bbn661, _bbn662, _bbn691, _bbn663, _bbn6225, _bbn6157, _bbn6187, _bbn6256, _bbn6257, _bbn692, _bbn6266, _bbn6123, _bbn649, _bbn664, _bbn6241, _bbn6261, _bbn693, _bbn6124, _bbn650, _bbn6242, _bbn651, _bbn6243, _bbn6158, _bbn6206, _bbn694, _bbn6207, _bbn695, _bbn665, _bbn6125, _bbn6159, _bbn666, _bbn667, _bbn652, _bbn6126, _bbn6262, _bbn668, _bbn669, _bbn6285, _bbn6263, _bbn6188, _bbn6127, _bbn696, _bbn6128, _bbn6267, _bbn6129, _bbn697, _bbn6189, _bbn6208, _bbn6244, _bbn6258, _bbn6190, _bbn6191, _bbn6130, _bbn6160, _bbn6161, _bbn6268, _bbn6271, _bbn698, _bbn699, _bbn670, _bbn6162, _bbn6209, _bbn671, _bbn6226, _bbn6283, _bbn6192, _bbn6100, _bbn6163, _bbn6164, _bbn6101, _bbn6131, _bbn6132, _bbn6102, _bbn6227, _bbn6133, _bbn6210, _bbn6103, _bbn6165, _bbn653, _bbn6166, _bbn6104, _bbn6228, _bbn6229, _bbn6167, _bbn6105, _bbn6211, _bbn6168, _bbn6193, _bbn6106, _bbn6169, _bbn6107, _bbn6194, _bbn6134, _bbn6230, _bbn6108, _bbn6170, _bbn6109, _bbn6171, _bbn672, _bbn6195, _bbn654, _bbn6135, _bbn6212, _bbn655, _bbn656, _bbn6213, _bbn673, _bbn657, _bbn6196, _bbn6291, _bbn658, _bbn6110, _bbn6111, _bbn6269, _bbn6136, _bbn6137, _bbn6231, _bbn647, _bbn6278, _bbn6232, _bbn6138, _bbn674, _bbn659, _bbn6139, _bbn6140, _bbn6197, _bbn6141, _bbn6112, _bbn6279, _bbn6233, _bbn6142, _bbn6113, _bbn6172, _bbn6214, _bbn6272, _bbn6215, _bbn6275, _bbn6198, _bbn6173, _bbn675, _bbn6264, _bbn6216, _bbn6234, _bbn6217, _bbn6199, _bbn6245, _bbn6246, _bbn6143, _bbn6200, _bbn676, _bbn6201, _bbn6114, _bbn6235, _bbn6218, _bbn6259, _bbn660, _bbn6115, _bbn6250, _bbn6284, _bbn6174, _bbn6276, _bbn6116, _bbn677, _bbn6144, _bbn6175, _bbn6251, _bbn6176, _bbn6219, _bbn6117, _bbn6118, _bbn6247, _bbn6145, _bbn6146, _bbn6274, _bbn6254, _bbn6177, _bbn6147, _bbn6236, _bbn6220, _bbn6280, _bbn6221, _bbn6202, _bbn6119, _bbn6222, _bbn678, _bbn6203, _bbn6248, _bbn6178, _bbn6223, _bbn6148, _bbn6224, _bbn6179 ];

inter.id = "vai_latn";

inter.characters = {
    exemplar: {
        "default": [ "a", "á", "ã", "b", "ɓ", "c", "d", "ɗ", "e", "é", "ẽ", "ɛ", "{ɛ\\u0301}", "{ɛ\\u0303}", "f", "g", "h", "i", "í", "ĩ", "j", "k", "l", "m", "n", "ŋ", "o", "ó", "õ", "ɔ", "{ɔ\\u0301}", "{ɔ\\u0303}", "p", "q", "r", "s", "t", "u", "ú", "ũ", "v", "w", "x", "y", "z" ],
        index: [ "A", "B", "Ɓ", "C", "D", "Ɗ", "E", "Ɛ", "F", "G", "H", "I", "J", "K", "L", "M", "N", "Ŋ", "O", "Ɔ", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z" ],
        auxiliary: [ "" ],
        punctuation: [ "\\-", ",", ";", "\\:", "!", "?", ".", "(", ")", "\\[", "\\]", "\\{", "\\}" ]
    },
    ellipsis: {
        "final": "{0}…",
        initial: "…{0}",
        medial: "{0}…{1}",
        "word-final": "{0} …",
        "word-initial": "… {0}",
        "word-medial": "{0} … {1}"
    },
    moreInformation: "?"
};

inter.calendars = {
    buddhist: {
        dateFormats: _bbn6332,
        datePartsOrder: "dmy",
        dateIntervalFormats: _bbn6333,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _bbn6325,
        dayPeriods: _bbn6307,
        monthNames: _bbn6327,
        eraNames: {
            wide: [ "BE" ],
            abbreviated: [ "BE" ]
        },
        quarterNames: _bbn6309,
        fields: _bbn6334
    },
    chinese: _bbn6341,
    coptic: {
        dateFormats: _bbn6332,
        datePartsOrder: "dmy",
        dateIntervalFormats: _bbn6333,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _bbn6325,
        dayPeriods: _bbn6307,
        monthNames: {
            format: _bbn6317,
            standAlone: _bbn6317
        },
        eraNames: _bbn633,
        quarterNames: _bbn6309,
        fields: _bbn6334
    },
    dangi: _bbn6341,
    ethioaa: _bbn6335,
    ethiopic: {
        dateFormats: _bbn6332,
        datePartsOrder: "dmy",
        dateIntervalFormats: _bbn6333,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _bbn6325,
        dayPeriods: _bbn6307,
        monthNames: {
            format: _bbn6319,
            standAlone: _bbn6319
        },
        eraNames: _bbn633,
        quarterNames: _bbn6309,
        fields: _bbn6334
    },
    gregorian: {
        dateFormats: {
            basic: {
                fullDate: "EEEE, d MMMM y",
                longDate: "d MMMM y",
                mediumDate: "d MMM y",
                shortDate: "dd/MM/y",
                fullTime: "h:mm:ss a zzzz",
                longTime: "h:mm:ss a z",
                mediumTime: "h:mm:ss a",
                shortTime: "h:mm a",
                shortDateTime: "dd/MM/y h:mm a",
                mediumDateTime: "d MMM y h:mm:ss a",
                longDateTime: "d MMMM y h:mm:ss a z",
                fullDateTime: "EEEE, d MMMM y h:mm:ss a zzzz"
            },
            cldr: {
                d: "d",
                Ed: "E d",
                hm: "h:mm a",
                Hm: "HH:mm",
                hms: "h:mm:ss a",
                Hms: "HH:mm:ss",
                M: "L",
                Md: "M/d",
                MEd: "E, M/d",
                MMM: "LLL",
                MMMd: "MMM d",
                MMMEd: "E, MMM d",
                ms: "m:ss",
                y: "y",
                yM: "M/y",
                yMd: "M/d/y",
                yMEd: "E, M/d/y",
                yMMM: "MMM y",
                yMMMd: "MMM d y",
                yMMMEd: "E, MMM d, y",
                yQQQ: "QQQ y",
                yQQQQ: "QQQQ y",
                E: "ccc",
                Ehm: "E h:mm a",
                EHm: "E HH:mm",
                Ehms: "E h:mm:ss a",
                EHms: "E HH:mm:ss",
                Gy: "G y",
                GyMMM: "G y MMM",
                GyMMMd: "G y MMM d",
                GyMMMEd: "G y MMM d, E",
                h: "h a",
                H: "HH",
                hmsv: "h:mm:ss a v",
                Hmsv: "HH:mm:ss v",
                hmv: "h:mm a v",
                Hmv: "HH:mm v",
                MMMMd: "MMMM d",
                yMMMM: "y MMMM"
            }
        },
        datePartsOrder: "dmy",
        dateIntervalFormats: {
            d: {
                d: "d–d"
            },
            h: _bbn63,
            H: {
                H: "HH–HH"
            },
            hm: _bbn639,
            Hm: _bbn612,
            hmv: _bbn646,
            Hmv: _bbn615,
            hv: _bbn66,
            Hv: {
                H: "HH–HH v"
            },
            M: {
                M: "MM–MM"
            },
            Md: _bbn616,
            MEd: _bbn630,
            MMM: {
                M: "LLL–LLL"
            },
            MMMd: _bbn69,
            MMMEd: _bbn631,
            y: {
                y: "y–y"
            },
            yM: _bbn613,
            yMd: _bbn6204,
            yMEd: _bbn6288,
            yMMM: {
                M: "y MMM–MMM",
                y: "y MMM – y MMM"
            },
            yMMMd: {
                d: "y MMM d–d",
                M: "y MMM d – MMM d",
                y: "y MMM d – y MMM d"
            },
            yMMMEd: {
                d: "y MMM d, E – MMM d, E",
                M: "y MMM d, E – MMM d, E",
                y: "y MMM d, E – y MMM d, E"
            },
            yMMMM: {
                M: "y MMMM–MMMM",
                y: "y MMMM – y MMMM"
            }
        },
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _bbn6325,
        dayPeriods: _bbn6307,
        monthNames: _bbn6327,
        eraNames: {
            wide: [ "BCE", "CE" ],
            abbreviated: [ "BCE", "CE" ]
        },
        quarterNames: _bbn6309,
        fields: _bbn6334
    },
    hebrew: {
        dateFormats: _bbn6332,
        datePartsOrder: "dmy",
        dateIntervalFormats: _bbn6333,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _bbn6325,
        dayPeriods: _bbn6307,
        monthNames: {
            format: _bbn6315,
            standAlone: _bbn6315
        },
        eraNames: {
            wide: [ "AM" ],
            abbreviated: [ "AM" ]
        },
        quarterNames: _bbn6309,
        fields: _bbn6334
    },
    indian: {
        dateFormats: _bbn6332,
        datePartsOrder: "dmy",
        dateIntervalFormats: _bbn6333,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _bbn6325,
        dayPeriods: _bbn6307,
        monthNames: {
            format: _bbn6320,
            standAlone: _bbn6320
        },
        eraNames: {
            wide: [ "Saka" ],
            abbreviated: [ "Saka" ]
        },
        quarterNames: _bbn6309,
        fields: _bbn6334
    },
    islamic: _bbn6339,
    "islamic-umalqura": _bbn6339,
    "islamic-tbla": _bbn6339,
    "islamic-civil": _bbn6339,
    "islamic-rgsa": _bbn6339,
    iso8601: _bbn6335,
    japanese: {
        dateFormats: _bbn6332,
        datePartsOrder: "dmy",
        dateIntervalFormats: _bbn6333,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _bbn6325,
        dayPeriods: _bbn6307,
        monthNames: _bbn6327,
        eraNames: {
            wide: _bbn6338,
            abbreviated: _bbn6338
        },
        quarterNames: _bbn6309,
        fields: _bbn6334
    },
    persian: {
        dateFormats: _bbn6332,
        datePartsOrder: "dmy",
        dateIntervalFormats: _bbn6333,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _bbn6325,
        dayPeriods: _bbn6307,
        monthNames: {
            format: _bbn6316,
            standAlone: _bbn6316
        },
        eraNames: {
            wide: [ "AP" ],
            abbreviated: [ "AP" ]
        },
        quarterNames: _bbn6309,
        fields: _bbn6334
    },
    roc: {
        dateFormats: _bbn6332,
        datePartsOrder: "dmy",
        dateIntervalFormats: _bbn6333,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _bbn6325,
        dayPeriods: _bbn6307,
        monthNames: _bbn6327,
        eraNames: {
            wide: _bbn65,
            abbreviated: _bbn65
        },
        quarterNames: _bbn6309,
        fields: _bbn6334
    },
    islamicc: _bbn6335
};

inter.pluralRule = function(n) {
    return "other";
};

inter.renderDigitsOrdinal = function(n) {
    if (n < 0) return "−" + this.renderDigitsOrdinal(-n);
    if (n >= 0) return this.renderNumber(n, "#,##0") + ".";
};

inter.renderSpelloutNumbering = function(n) {
    if (n < 0) return "−" + this.renderSpelloutNumbering(-n);
    if (n >= 0) return this.renderNumber(n, "#,##0.#");
};

inter.renderArmenianUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderArmenianUpper(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e4) return this.renderNumber(n, "#,##0");
    if (n >= 9e3) return "Ք" + (n === 9e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 8e3) return "Փ" + (n === 8e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 7e3) return "Ւ" + (n === 7e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 6e3) return "Ց" + (n === 6e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 5e3) return "Ր" + (n === 5e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 4e3) return "Տ" + (n === 4e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 3e3) return "Վ" + (n === 3e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 2e3) return "Ս" + (n === 2e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 1e3) return "Ռ" + (n === 1e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 900) return "Ջ" + (n === 900 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 800) return "Պ" + (n === 800 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 700) return "Չ" + (n === 700 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 600) return "Ո" + (n === 600 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 500) return "Շ" + (n === 500 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 400) return "Ն" + (n === 400 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 300) return "Յ" + (n === 300 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 200) return "Մ" + (n === 200 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 100) return "Ճ" + (n === 100 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 90) return "Ղ" + (n === 90 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 80) return "Ձ" + (n === 80 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 70) return "Հ" + (n === 70 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 60) return "Կ" + (n === 60 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 50) return "Ծ" + (n === 50 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 40) return "Խ" + (n === 40 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 30) return "Լ" + (n === 30 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 20) return "Ի" + (n === 20 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 10) return "Ժ" + (n === 10 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 9) return "Թ";
    if (n >= 8) return "Ը";
    if (n >= 7) return "Է";
    if (n >= 6) return "Զ";
    if (n >= 5) return "Ե";
    if (n >= 4) return "Դ";
    if (n >= 3) return "Գ";
    if (n >= 2) return "Բ";
    if (n >= 1) return "Ա";
    if (n >= 0) return "0";
};

inter.renderArmenianLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderArmenianLower(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e4) return this.renderNumber(n, "#,##0");
    if (n >= 9e3) return "ք" + (n === 9e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 8e3) return "փ" + (n === 8e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 7e3) return "ւ" + (n === 7e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 6e3) return "ց" + (n === 6e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 5e3) return "ր" + (n === 5e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 4e3) return "տ" + (n === 4e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 3e3) return "վ" + (n === 3e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 2e3) return "ս" + (n === 2e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 1e3) return "ռ" + (n === 1e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 900) return "ջ" + (n === 900 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 800) return "պ" + (n === 800 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 700) return "չ" + (n === 700 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 600) return "ո" + (n === 600 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 500) return "շ" + (n === 500 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 400) return "ն" + (n === 400 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 300) return "յ" + (n === 300 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 200) return "մ" + (n === 200 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 100) return "ճ" + (n === 100 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 90) return "ղ" + (n === 90 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 80) return "ձ" + (n === 80 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 70) return "հ" + (n === 70 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 60) return "կ" + (n === 60 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 50) return "ծ" + (n === 50 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 40) return "խ" + (n === 40 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 30) return "լ" + (n === 30 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 20) return "ի" + (n === 20 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 10) return "ժ" + (n === 10 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 9) return "թ";
    if (n >= 8) return "ը";
    if (n >= 7) return "է";
    if (n >= 6) return "զ";
    if (n >= 5) return "ե";
    if (n >= 4) return "դ";
    if (n >= 3) return "գ";
    if (n >= 2) return "բ";
    if (n >= 1) return "ա";
    if (n >= 0) return "0";
};

inter.renderCyrillicLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderCyrillicLower(-n);
    if (isFractional && n > 1) return this.renderCyrillicLower(Math.floor(n)) + "." + this.renderCyrillicLower(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e15) return "҂҂҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e15)) + (n === 1e15 ? "" : " " + this.renderCyrillicLower(n % 1e15));
    if (n >= 1e12) return "҂҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e12)) + (n === 1e12 ? "" : " " + this.renderCyrillicLower(n % 1e12));
    if (n >= 1e9) return "҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e9)) + (n === 1e9 ? "" : " " + this.renderCyrillicLower(n % 1e9));
    if (n >= 1e6) return "҂҂" + this.renderCyrillicLower(Math.floor(n / 1e6)) + (n === 1e6 ? "" : " " + this.renderCyrillicLower(n % 1e6));
    if (n >= 11e3) return this.renderCyrillicLowerThousands(Math.floor(n / 1e4)) + (n === 11e3 ? "" : " " + this.renderCyrillicLower(n % 1e4));
    if (n >= 1e4) return "҂" + this.renderCyrillicLower(Math.floor(n / 1e4)) + (n === 1e4 ? "" : " " + this.renderCyrillicLower(n % 1e4));
    if (n >= 1e3) return "҂" + this.renderCyrillicLower110(Math.floor(n / 1e3)) + this.renderCyrillicLowerPost(n % 1e3);
    if (n >= 900) return "ц" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 801) return "ѿ" + this.renderCyrillicLower(n % 100);
    if (n >= 800) return "ѿ҃";
    if (n >= 700) return "ѱ" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 600) return "х" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 500) return "ф" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 400) return "у" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 300) return "т" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 200) return "с" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 100) return "р" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 90) return "ч" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 80) return "п" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 70) return "ѻ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 60) return "ѯ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 50) return "н" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 40) return "м" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 30) return "л" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 20) return "к" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 19) return "ѳ҃і";
    if (n >= 18) return "и҃і";
    if (n >= 17) return "з҃і";
    if (n >= 16) return "ѕ҃і";
    if (n >= 15) return "є҃і";
    if (n >= 14) return "д҃і";
    if (n >= 13) return "г҃і";
    if (n >= 12) return "в҃і";
    if (n >= 11) return "а҃і";
    if (n >= 1) return this.renderCyrillicLower110(n) + "҃";
    if (n >= 0) return "0҃";
};

inter.renderEthiopic = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderEthiopic(-n);
    if (isFractional && n > 1) return this.renderEthiopic(Math.floor(n)) + "፡" + this.renderEthiopic(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 1e18) return this.renderNumber(n, "#,###0");
    if (n >= 2e16) return this.renderEthiopic(Math.floor(n / 1e16)) + "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 1e16) return "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 2e12) return this.renderEthiopic(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1e12) return "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 2e8) return this.renderEthiopic(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1e8) return "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 2e4) return this.renderEthiopic(Math.floor(n / 1e4)) + "፼" + (n === 2e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 1e4) return "፼" + (n === 1e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 200) return this.renderEthiopic(Math.floor(n / 100)) + "፻" + (n === 200 ? "" : this.renderEthiopic(n % 100));
    if (n >= 100) return "፻" + (n === 100 ? "" : this.renderEthiopic(n % 100));
    if (n >= 90) return "፺" + (n === 90 ? "" : this.renderEthiopic(n % 10));
    if (n >= 80) return "፹" + (n === 80 ? "" : this.renderEthiopic(n % 10));
    if (n >= 70) return "፸" + (n === 70 ? "" : this.renderEthiopic(n % 10));
    if (n >= 60) return "፷" + (n === 60 ? "" : this.renderEthiopic(n % 10));
    if (n >= 50) return "፶" + (n === 50 ? "" : this.renderEthiopic(n % 10));
    if (n >= 40) return "፵" + (n === 40 ? "" : this.renderEthiopic(n % 10));
    if (n >= 30) return "፴" + (n === 30 ? "" : this.renderEthiopic(n % 10));
    if (n >= 20) return "፳" + (n === 20 ? "" : this.renderEthiopic(n % 10));
    if (n >= 10) return "፲" + (n === 10 ? "" : this.renderEthiopic(n % 10));
    if (n >= 9) return "፱";
    if (n >= 8) return "፰";
    if (n >= 7) return "፯";
    if (n >= 6) return "፮";
    if (n >= 5) return "፭";
    if (n >= 4) return "፬";
    if (n >= 3) return "፫";
    if (n >= 2) return "፪";
    if (n >= 1) return "፩";
    if (n >= 0) return "ባዶ";
};

inter.renderGeorgian = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGeorgian(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 2e4) return this.renderNumber(n, "#,##0");
    if (n >= 1e4) return "ჯ" + (n === 1e4 ? "" : this.renderGeorgian(n % 1e4));
    if (n >= 9e3) return "ჵ" + (n === 9e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 8e3) return "ჴ" + (n === 8e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 7e3) return "ხ" + (n === 7e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 6e3) return "ჭ" + (n === 6e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 5e3) return "წ" + (n === 5e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 4e3) return "ძ" + (n === 4e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 3e3) return "ც" + (n === 3e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 2e3) return "ჩ" + (n === 2e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 1e3) return "შ" + (n === 1e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 900) return "ყ" + (n === 900 ? "" : this.renderGeorgian(n % 100));
    if (n >= 800) return "ღ" + (n === 800 ? "" : this.renderGeorgian(n % 100));
    if (n >= 700) return "ქ" + (n === 700 ? "" : this.renderGeorgian(n % 100));
    if (n >= 600) return "ფ" + (n === 600 ? "" : this.renderGeorgian(n % 100));
    if (n >= 500) return "ჳ" + (n === 500 ? "" : this.renderGeorgian(n % 100));
    if (n >= 400) return "უ" + (n === 400 ? "" : this.renderGeorgian(n % 100));
    if (n >= 300) return "ტ" + (n === 300 ? "" : this.renderGeorgian(n % 100));
    if (n >= 200) return "ს" + (n === 200 ? "" : this.renderGeorgian(n % 100));
    if (n >= 100) return "რ" + (n === 100 ? "" : this.renderGeorgian(n % 100));
    if (n >= 90) return "ჟ" + (n === 90 ? "" : this.renderGeorgian(n % 10));
    if (n >= 80) return "პ" + (n === 80 ? "" : this.renderGeorgian(n % 10));
    if (n >= 70) return "ო" + (n === 70 ? "" : this.renderGeorgian(n % 10));
    if (n >= 60) return "ჲ" + (n === 60 ? "" : this.renderGeorgian(n % 10));
    if (n >= 50) return "ნ" + (n === 50 ? "" : this.renderGeorgian(n % 10));
    if (n >= 40) return "მ" + (n === 40 ? "" : this.renderGeorgian(n % 10));
    if (n >= 30) return "ლ" + (n === 30 ? "" : this.renderGeorgian(n % 10));
    if (n >= 20) return "კ" + (n === 20 ? "" : this.renderGeorgian(n % 10));
    if (n >= 10) return "ი" + (n === 10 ? "" : this.renderGeorgian(n % 10));
    if (n >= 9) return "თ";
    if (n >= 8) return "ჱ";
    if (n >= 7) return "ზ";
    if (n >= 6) return "ვ";
    if (n >= 5) return "ე";
    if (n >= 4) return "დ";
    if (n >= 3) return "გ";
    if (n >= 2) return "ბ";
    if (n >= 1) return "ა";
    if (n >= 0) return this.renderNumber(n, "#,##0");
};

inter.renderGreekUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGreekUpper(-n);
    if (isFractional && n > 1) return this.renderGreekUpper(Math.floor(n)) + "." + this.renderGreekUpper(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 0) return this.renderGreekNumeralMajuscules(n) + "´";
};

inter.renderGreekLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGreekLower(-n);
    if (isFractional && n > 1) return this.renderGreekLower(Math.floor(n)) + "." + this.renderGreekLower(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 0) return this.renderGreekNumeralMinuscules(n) + "´";
};

inter.renderSpelloutCardinal = function(n) {
    if (n < 0) return "−" + this.renderSpelloutCardinal(-n);
    if (n >= 0) return this.renderNumber(n, "#,##0.#");
};

inter.renderHebrew = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderHebrew(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1000001) return this.renderNumber(n, "#,##0");
    if (n >= 1e6) return "אלף אלפים";
    if (n >= 3001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 3001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 3e3) return this.renderHebrew(Math.floor(n / 1e3)) + " אלפים";
    if (n >= 2001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 2001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 2e3) return "אלפיים";
    if (n >= 1001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 1001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 1e3) return "אלף";
    if (n >= 901) return "תתק" + this.renderHebrew099(n % 100);
    if (n >= 900) return "תת״ק";
    if (n >= 801) return "תת" + this.renderHebrew099(n % 100);
    if (n >= 800) return "ת״ת";
    if (n >= 745) return "תש" + this.renderHebrew099(n % 100);
    if (n >= 744) return "תשד״מ";
    if (n >= 701) return "תש" + this.renderHebrew099(n % 100);
    if (n >= 700) return "ת״ש";
    if (n >= 699) return "תר" + this.renderHebrew099(n % 100);
    if (n >= 698) return "תרח״צ";
    if (n >= 601) return "תר" + this.renderHebrew099(n % 100);
    if (n >= 600) return "ת״ר";
    if (n >= 501) return "תק" + this.renderHebrew099(n % 100);
    if (n >= 500) return "ת״ק";
    if (n >= 400) return "ת" + this.renderHebrew099(n % 100);
    if (n >= 345) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 344) return "שד״מ";
    if (n >= 305) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 304) return "ד״ש";
    if (n >= 300) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 299) return "ר" + this.renderHebrew099(n % 100);
    if (n >= 298) return "רח״צ";
    if (n >= 200) return "ר" + this.renderHebrew099(n % 100);
    if (n >= 100) return "ק" + this.renderHebrew099(n % 100);
    if (n >= 91) return "צ״" + this.renderHebrewItem(n % 10);
    if (n >= 90) return "צ׳";
    if (n >= 81) return "פ״" + this.renderHebrewItem(n % 10);
    if (n >= 80) return "פ׳";
    if (n >= 71) return "ע״" + this.renderHebrewItem(n % 10);
    if (n >= 70) return "ע׳";
    if (n >= 61) return "ס״" + this.renderHebrewItem(n % 10);
    if (n >= 60) return "ס׳";
    if (n >= 51) return "נ״" + this.renderHebrewItem(n % 10);
    if (n >= 50) return "נ׳";
    if (n >= 41) return "מ״" + this.renderHebrewItem(n % 10);
    if (n >= 40) return "מ׳";
    if (n >= 31) return "ל״" + this.renderHebrewItem(n % 10);
    if (n >= 30) return "ל׳";
    if (n >= 21) return "כ״" + this.renderHebrewItem(n % 10);
    if (n >= 20) return "כ׳";
    if (n >= 17) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "ט״ז";
    if (n >= 15) return "ט״ו";
    if (n >= 11) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 0) return this.renderHebrewItem(n) + "׳";
};

inter.renderRomanUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderRomanUpper(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 4e5) return this.renderNumber(n, "#,##0");
    if (n >= 3e5) return "ↈↈↈ" + (n === 3e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 2e5) return "ↈↈ" + (n === 2e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 1e5) return "ↈ" + (n === 1e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 9e4) return "ↂↈ" + (n === 9e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 8e4) return "ↇↂↂↂ" + (n === 8e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 7e4) return "ↇↂↂ" + (n === 7e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 6e4) return "ↇↂ" + (n === 6e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 5e4) return "ↇ" + (n === 5e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 4e4) return "ↂↇ" + (n === 4e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 3e4) return "ↂↂↂ" + (n === 3e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 2e4) return "ↂↂ" + (n === 2e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 1e4) return "ↂ" + (n === 1e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 9e3) return "Mↂ" + (n === 9e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 8e3) return "ↁMMM" + (n === 8e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 7e3) return "ↁMM" + (n === 7e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 6e3) return "ↁM" + (n === 6e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 5e3) return "ↁ" + (n === 5e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 4e3) return "Mↁ" + (n === 4e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 3e3) return "MMM" + (n === 3e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 2e3) return "MM" + (n === 2e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 1e3) return "M" + (n === 1e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 900) return "CM" + (n === 900 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 800) return "DCCC" + (n === 800 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 700) return "DCC" + (n === 700 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 600) return "DC" + (n === 600 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 500) return "D" + (n === 500 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 400) return "CD" + (n === 400 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 300) return "CCC" + (n === 300 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 200) return "CC" + (n === 200 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 100) return "C" + (n === 100 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 90) return "XC" + (n === 90 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 80) return "LXXX" + (n === 80 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 70) return "LXX" + (n === 70 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 60) return "LX" + (n === 60 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 50) return "L" + (n === 50 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 40) return "XL" + (n === 40 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 30) return "XXX" + (n === 30 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 20) return "XX" + (n === 20 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 10) return "X" + (n === 10 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 9) return "IX";
    if (n >= 8) return "VIII";
    if (n >= 7) return "VII";
    if (n >= 6) return "VI";
    if (n >= 5) return "V";
    if (n >= 4) return "IV";
    if (n >= 3) return "III";
    if (n >= 2) return "II";
    if (n >= 1) return "I";
    if (n >= 0) return "N";
};

inter.renderRomanLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderRomanLower(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 5e3) return this.renderNumber(n, "#,##0");
    if (n >= 4e3) return "mmmm" + (n === 4e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 3e3) return "mmm" + (n === 3e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 2e3) return "mm" + (n === 2e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 1e3) return "m" + (n === 1e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 900) return "cm" + (n === 900 ? "" : this.renderRomanLower(n % 100));
    if (n >= 800) return "dccc" + (n === 800 ? "" : this.renderRomanLower(n % 100));
    if (n >= 700) return "dcc" + (n === 700 ? "" : this.renderRomanLower(n % 100));
    if (n >= 600) return "dc" + (n === 600 ? "" : this.renderRomanLower(n % 100));
    if (n >= 500) return "d" + (n === 500 ? "" : this.renderRomanLower(n % 100));
    if (n >= 400) return "cd" + (n === 400 ? "" : this.renderRomanLower(n % 100));
    if (n >= 300) return "ccc" + (n === 300 ? "" : this.renderRomanLower(n % 100));
    if (n >= 200) return "cc" + (n === 200 ? "" : this.renderRomanLower(n % 100));
    if (n >= 100) return "c" + (n === 100 ? "" : this.renderRomanLower(n % 100));
    if (n >= 90) return "xc" + (n === 90 ? "" : this.renderRomanLower(n % 10));
    if (n >= 80) return "lxxx" + (n === 80 ? "" : this.renderRomanLower(n % 10));
    if (n >= 70) return "lxx" + (n === 70 ? "" : this.renderRomanLower(n % 10));
    if (n >= 60) return "lx" + (n === 60 ? "" : this.renderRomanLower(n % 10));
    if (n >= 50) return "l" + (n === 50 ? "" : this.renderRomanLower(n % 10));
    if (n >= 40) return "xl" + (n === 40 ? "" : this.renderRomanLower(n % 10));
    if (n >= 30) return "xxx" + (n === 30 ? "" : this.renderRomanLower(n % 10));
    if (n >= 20) return "xx" + (n === 20 ? "" : this.renderRomanLower(n % 10));
    if (n >= 10) return "x" + (n === 10 ? "" : this.renderRomanLower(n % 10));
    if (n >= 9) return "ix";
    if (n >= 8) return "viii";
    if (n >= 7) return "vii";
    if (n >= 6) return "vi";
    if (n >= 5) return "v";
    if (n >= 4) return "iv";
    if (n >= 3) return "iii";
    if (n >= 2) return "ii";
    if (n >= 1) return "i";
    if (n >= 0) return "n";
};

inter.renderTamil = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderTamil(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e8) return this.renderNumber(n, "#,##,##0");
    if (n >= 1e6) return this.renderTamil(Math.floor(n / 1e6)) + "௱௲" + (n === 1e6 ? "" : this.renderTamilThousands(n % 1e6));
    if (n >= 2e3) return this.renderTamil(Math.floor(n / 1e3)) + "௲" + (n === 2e3 ? "" : this.renderTamil(n % 1e3));
    if (n >= 1e3) return "௲" + (n === 1e3 ? "" : this.renderTamil(n % 1e3));
    if (n >= 200) return this.renderTamil(Math.floor(n / 100)) + "௱" + (n === 200 ? "" : this.renderTamil(n % 100));
    if (n >= 100) return "௱" + (n === 100 ? "" : this.renderTamil(n % 100));
    if (n >= 20) return this.renderTamil(Math.floor(n / 10)) + "௰" + (n === 20 ? "" : this.renderTamil(n % 10));
    if (n >= 10) return "௰" + (n === 10 ? "" : this.renderTamil(n % 10));
    if (n >= 9) return "௯";
    if (n >= 8) return "௮";
    if (n >= 7) return "௭";
    if (n >= 6) return "௬";
    if (n >= 5) return "௫";
    if (n >= 4) return "௪";
    if (n >= 3) return "௩";
    if (n >= 2) return "௨";
    if (n >= 1) return "௧";
    if (n >= 0) return "௦";
};

inter.renderCyrillicLowerThousands = function(n) {
    if (n >= 900) return "҂ц" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 800) return "҂ѿ" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 700) return "҂ѱ" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 600) return "҂х" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 500) return "҂ф" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 400) return "҂у" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 300) return "҂т" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 200) return "҂с" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 100) return "҂р" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 90) return "҂ч" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 80) return "҂п" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 70) return "҂ѻ" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 60) return "҂ѯ" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 50) return "҂н" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 40) return "҂м" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 30) return "҂л" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 20) return "҂к" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 19) return "҂ѳ҃҂і";
    if (n >= 18) return "҂и҃҂і";
    if (n >= 17) return "҂з҃҂і";
    if (n >= 16) return "҂ѕ҃҂і";
    if (n >= 15) return "҂є҃҂і";
    if (n >= 14) return "҂д҃҂і";
    if (n >= 13) return "҂г҃҂і";
    if (n >= 12) return "҂в҃҂і";
    if (n >= 11) return "҂а҃҂і";
    if (n >= 10) return "҃҂і";
    if (n >= 9) return "҃҂ѳ";
    if (n >= 8) return "҃҂и";
    if (n >= 7) return "҃҂з";
    if (n >= 6) return "҃҂ѕ";
    if (n >= 5) return "҃҂є";
    if (n >= 4) return "҃҂д";
    if (n >= 3) return "҃҂г";
    if (n >= 2) return "҃҂в";
    if (n >= 1) return "҃҂а";
    if (n >= 0) return "҃";
};

inter.renderCyrillicLower110 = function(n) {
    if (n >= 10) return "і";
    if (n >= 9) return "ѳ";
    if (n >= 8) return "и";
    if (n >= 7) return "з";
    if (n >= 6) return "ѕ";
    if (n >= 5) return "є";
    if (n >= 4) return "д";
    if (n >= 3) return "г";
    if (n >= 2) return "в";
    if (n >= 1) return "а";
};

inter.renderCyrillicLowerPost = function(n) {
    if (n >= 1) return this.renderCyrillicLower(n);
    if (n >= 0) return "҃";
};

inter.renderCyrillicLowerFinal = function(n) {
    if (n >= 91) return "ч" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 90) return "҃ч";
    if (n >= 81) return "п" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 80) return "҃п";
    if (n >= 71) return "ѻ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 70) return "҃ѻ";
    if (n >= 61) return "ѯ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 60) return "҃ѯ";
    if (n >= 51) return "н" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 50) return "҃н";
    if (n >= 41) return "м" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 40) return "҃м";
    if (n >= 31) return "л" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 30) return "҃л";
    if (n >= 21) return "к" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 20) return "҃к";
    if (n >= 19) return "ѳ҃і";
    if (n >= 18) return "и҃і";
    if (n >= 17) return "з҃і";
    if (n >= 16) return "ѕ҃і";
    if (n >= 15) return "є҃і";
    if (n >= 14) return "д҃і";
    if (n >= 13) return "г҃і";
    if (n >= 12) return "в҃і";
    if (n >= 11) return "а҃і";
    if (n >= 1) return "҃" + this.renderCyrillicLower110(n);
    if (n >= 0) return "҃";
};

inter.renderEthiopicP3 = function(n) {
    if (n >= 1e12) return this.renderEthiopic(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1) return "፼፼፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼፼፼";
};

inter.renderEthiopicP2 = function(n) {
    if (n >= 1e8) return this.renderEthiopic(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1) return "፼፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼፼";
};

inter.renderEthiopicP1 = function(n) {
    if (n >= 1e4) return this.renderEthiopic(Math.floor(n / 1e4)) + "፼" + (n === 1e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 1) return "፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼";
};

inter.renderGreekNumeralMajuscules = function(n) {
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e16) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e16)) + "ΜΜΜΜ" + (n === 1e16 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e16));
    if (n >= 1e12) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e12)) + "ΜΜΜ" + (n === 1e12 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e12));
    if (n >= 1e8) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e8)) + "ΜΜ" + (n === 1e8 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e8));
    if (n >= 1e4) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e4)) + "Μ" + (n === 1e4 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e4));
    if (n >= 1e3) return "͵" + this.renderGreekNumeralMajuscules(Math.floor(n / 1e3)) + (n === 1e3 ? "" : this.renderGreekNumeralMajuscules(n % 1e3));
    if (n >= 900) return "Ϡ" + (n === 900 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 800) return "Ω" + (n === 800 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 700) return "Ψ" + (n === 700 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 600) return "Χ" + (n === 600 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 500) return "Φ" + (n === 500 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 400) return "Υ" + (n === 400 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 300) return "Τ" + (n === 300 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 200) return "Σ" + (n === 200 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 100) return "Ρ" + (n === 100 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 90) return "Ϟ" + (n === 90 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 80) return "Π" + (n === 80 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 70) return "Ο" + (n === 70 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 60) return "Ξ" + (n === 60 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 50) return "Ν" + (n === 50 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 40) return "Μ" + (n === 40 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 30) return "Λ" + (n === 30 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 20) return "Κ" + (n === 20 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 10) return "Ι" + (n === 10 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 9) return "Θ";
    if (n >= 8) return "Η";
    if (n >= 7) return "Ζ";
    if (n >= 6) return "Ϝ";
    if (n >= 5) return "Ε";
    if (n >= 4) return "Δ";
    if (n >= 3) return "Γ";
    if (n >= 2) return "Β";
    if (n >= 1) return "Α";
    if (n >= 0) return "𐆊";
};

inter.renderGreekNumeralMinuscules = function(n) {
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e16) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e16)) + "μμμμ" + (n === 1e16 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e16));
    if (n >= 1e12) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e12)) + "μμμ" + (n === 1e12 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e12));
    if (n >= 1e8) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e8)) + "μμ" + (n === 1e8 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e8));
    if (n >= 1e4) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e4)) + "μ" + (n === 1e4 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e4));
    if (n >= 1e3) return "͵" + this.renderGreekNumeralMinuscules(Math.floor(n / 1e3)) + (n === 1e3 ? "" : this.renderGreekNumeralMinuscules(n % 1e3));
    if (n >= 900) return "ϡ" + (n === 900 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 800) return "ω" + (n === 800 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 700) return "ψ" + (n === 700 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 600) return "χ" + (n === 600 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 500) return "φ" + (n === 500 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 400) return "υ" + (n === 400 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 300) return "τ" + (n === 300 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 200) return "σ" + (n === 200 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 100) return "ρ" + (n === 100 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 90) return "ϟ" + (n === 90 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 80) return "π" + (n === 80 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 70) return "ο" + (n === 70 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 60) return "ξ" + (n === 60 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 50) return "ν" + (n === 50 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 40) return "μ" + (n === 40 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 30) return "λ" + (n === 30 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 20) return "κ" + (n === 20 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 10) return "ι" + (n === 10 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 9) return "θ";
    if (n >= 8) return "η";
    if (n >= 7) return "ζ";
    if (n >= 6) return "ϝ";
    if (n >= 5) return "ε";
    if (n >= 4) return "δ";
    if (n >= 3) return "γ";
    if (n >= 2) return "β";
    if (n >= 1) return "α";
    if (n >= 0) return "𐆊";
};

inter.renderHebrewThousands = function(n) {
    if (n >= 401) return this.renderHebrew(n) + "׳";
    if (n >= 100) return this.renderHebrew(n) + (n === 100 ? "" : "׳");
    if (n >= 10) return this.renderHebrew(n) + (n === 10 ? "" : "׳");
    if (n >= 0) return this.renderHebrew(n);
};

inter.renderHebrew099 = function(n) {
    if (n >= 91) return "צ״" + this.renderHebrewItem(n % 10);
    if (n >= 90) return "״צ";
    if (n >= 81) return "פ״" + this.renderHebrewItem(n % 10);
    if (n >= 80) return "״פ";
    if (n >= 71) return "ע״" + this.renderHebrewItem(n % 10);
    if (n >= 70) return "״ע";
    if (n >= 61) return "ס״" + this.renderHebrewItem(n % 10);
    if (n >= 60) return "״ס";
    if (n >= 51) return "נ״" + this.renderHebrewItem(n % 10);
    if (n >= 50) return "״נ";
    if (n >= 41) return "מ״" + this.renderHebrewItem(n % 10);
    if (n >= 40) return "״מ";
    if (n >= 31) return "ל״" + this.renderHebrewItem(n % 10);
    if (n >= 30) return "״ל";
    if (n >= 21) return "כ״" + this.renderHebrewItem(n % 10);
    if (n >= 20) return "״כ";
    if (n >= 17) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "ט״ז";
    if (n >= 15) return "ט״ו";
    if (n >= 11) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 1) return "״" + this.renderHebrewItem(n);
    if (n >= 0) return "׳";
};

inter.renderHebrewItem = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderHebrewItem(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 2100) return this.renderNumber(n, "#,##0");
    if (n >= 2e3) return "תתתתת" + (n === 2e3 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1900) return "תתתתש" + (n === 1900 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1800) return "תתתתר" + (n === 1800 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1700) return "תתתתק" + (n === 1700 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1600) return "תתתת" + (n === 1600 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1500) return "תתתש" + (n === 1500 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1400) return "תתתר" + (n === 1400 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1300) return "תתתק" + (n === 1300 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1200) return "תתת" + (n === 1200 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1100) return "תתש" + (n === 1100 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1e3) return "תתר" + (n === 1e3 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 900) return "תתק" + (n === 900 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 800) return "תת" + (n === 800 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 745) return "תש" + this.renderHebrewItem(n % 100);
    if (n >= 744) return "תשדמ";
    if (n >= 700) return "תש" + (n === 700 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 699) return "תר" + this.renderHebrewItem(n % 100);
    if (n >= 698) return "תרחצ";
    if (n >= 600) return "תר" + (n === 600 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 500) return "תק" + (n === 500 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 400) return "ת" + (n === 400 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 345) return "ש" + this.renderHebrewItem(n % 100);
    if (n >= 344) return "שדמ";
    if (n >= 305) return "ש" + this.renderHebrewItem(n % 100);
    if (n >= 304) return "דש";
    if (n >= 300) return "ש" + (n === 300 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 299) return "ר" + this.renderHebrewItem(n % 100);
    if (n >= 298) return "רחצ";
    if (n >= 200) return "ר" + (n === 200 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 100) return "ק" + (n === 100 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 90) return "צ" + (n === 90 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 80) return "פ" + (n === 80 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 70) return "ע" + (n === 70 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 60) return "ס" + (n === 60 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 50) return "נ" + (n === 50 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 40) return "מ" + (n === 40 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 30) return "ל" + (n === 30 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 20) return "כ" + (n === 20 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 17) return "י" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "טז";
    if (n >= 15) return "טו";
    if (n >= 10) return "י" + (n === 10 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 9) return "ט";
    if (n >= 8) return "ח";
    if (n >= 7) return "ז";
    if (n >= 6) return "ו";
    if (n >= 5) return "ה";
    if (n >= 4) return "ד";
    if (n >= 3) return "ג";
    if (n >= 2) return "ב";
    if (n >= 1) return "א";
    if (n >= 0) return "״";
};

inter.renderTamilThousands = function(n) {
    if (n >= 1e3) return this.renderTamilThousands(Math.floor(n / 1e3)) + "௲" + (n === 1e3 ? "" : this.renderTamilThousands(n % 1e3));
    if (n >= 0) return this.renderTamil(n);
};

inter.renderEthiopicP = function(n) {
    if (n >= 1e16) return this.renderEthiopicP(Math.floor(n / 1e16)) + "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 1e12) return this.renderEthiopicP(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1e8) return this.renderEthiopicP(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1e4) return this.renderEthiopicP(Math.floor(n / 1e4)) + "፼" + (n === 1e4 ? "" : this.renderEthiopicP(n % 1e4));
    if (n >= 1) return this.renderEthiopic(n);
};

inter.territories = _bbn6342;

inter.countries = _bbn6342;

inter.languages = [ {
    id: "ak",
    displayName: "Akaŋ",
    nativeDisplayName: "Akan"
}, {
    id: "ms",
    displayName: "Maléee",
    nativeDisplayName: "Bahasa Melayu"
}, {
    id: "de",
    displayName: "Jamáĩ",
    nativeDisplayName: "Deutsch"
}, {
    id: "en",
    displayName: "Poo",
    nativeDisplayName: "English"
}, {
    id: "es",
    displayName: "Panyɛĩ",
    nativeDisplayName: "español"
}, {
    id: "fr",
    displayName: "Fɛŋsi",
    nativeDisplayName: "français"
}, {
    id: "ha",
    displayName: "Hawusa",
    nativeDisplayName: "Hausa"
}, {
    id: "ig",
    displayName: "Ígbo",
    nativeDisplayName: "Igbo"
}, {
    id: "id",
    displayName: "Índonisiyɛŋ",
    nativeDisplayName: "Indonesia"
}, {
    id: "zu",
    displayName: "Zúlu",
    nativeDisplayName: "isiZulu"
}, {
    id: "it",
    displayName: "Itáliyɛŋ",
    nativeDisplayName: "italiano"
}, {
    id: "jv",
    displayName: "Javaníĩ"
}, {
    id: "rw",
    displayName: "Rawunda",
    nativeDisplayName: "Kinyarwanda"
}, {
    id: "hu",
    displayName: "Hɔŋgérɛŋ",
    nativeDisplayName: "magyar"
}, {
    id: "nl",
    displayName: "Dɔchi",
    nativeDisplayName: "Nederlands"
}, {
    id: "pl",
    displayName: "Pɔ́lési",
    nativeDisplayName: "polski"
}, {
    id: "pt",
    displayName: "Potokíi",
    nativeDisplayName: "português"
}, {
    id: "ro",
    displayName: "Romíniyɛŋ",
    nativeDisplayName: "română"
}, {
    id: "so",
    displayName: "Somáli",
    nativeDisplayName: "Soomaali"
}, {
    id: "sv",
    displayName: "Súwídɛŋ",
    nativeDisplayName: "svenska"
}, {
    id: "vi",
    displayName: "Viyamíĩ",
    nativeDisplayName: "Tiếng Việt"
}, {
    id: "tr",
    displayName: "Tɔ́ki",
    nativeDisplayName: "Türkçe"
}, {
    id: "yo",
    displayName: "Yóróbha",
    nativeDisplayName: "Èdè Yorùbá"
}, {
    id: "cs",
    displayName: "Chɛ",
    nativeDisplayName: "čeština"
}, {
    id: "el",
    displayName: "Hɛlɛŋ",
    nativeDisplayName: "Ελληνικά"
}, {
    id: "be",
    displayName: "Bhelarusaŋ",
    nativeDisplayName: "беларуская"
}, {
    id: "bg",
    displayName: "Bhɔgerɛŋ",
    nativeDisplayName: "български"
}, {
    id: "ru",
    displayName: "Rɔshiyɛŋ",
    nativeDisplayName: "русский"
}, {
    id: "uk",
    displayName: "Yukureniyɛŋ",
    nativeDisplayName: "українська"
}, {
    id: "ur",
    displayName: "Ɔdu",
    nativeDisplayName: "اردو"
}, {
    id: "ar",
    displayName: "Lahabu",
    nativeDisplayName: "العربية"
}, {
    id: "fa",
    displayName: "Pɛɛsiyɛŋ",
    nativeDisplayName: "فارسی"
}, {
    id: "ne",
    displayName: "Nipali",
    nativeDisplayName: "नेपाली"
}, {
    id: "hi",
    displayName: "Híiŋdi",
    nativeDisplayName: "हिन्दी"
}, {
    id: "bn",
    displayName: "Bhɛŋgáli",
    nativeDisplayName: "বাংলা"
}, {
    id: "pa",
    displayName: "Puŋjabhi",
    nativeDisplayName: "ਪੰਜਾਬੀ"
}, {
    id: "ta",
    displayName: "Tamíli",
    nativeDisplayName: "தமிழ்"
}, {
    id: "th",
    displayName: "Tái",
    nativeDisplayName: "ไทย"
}, {
    id: "my",
    displayName: "Bhɛmísi",
    nativeDisplayName: "ဗမာ"
}, {
    id: "am",
    displayName: "Amihári",
    nativeDisplayName: "አማርኛ"
}, {
    id: "km",
    displayName: "Kimɛɛ̃ tɛ",
    nativeDisplayName: "ខ្មែរ"
}, {
    id: "zh",
    displayName: "Chaniĩ",
    nativeDisplayName: "中文"
}, {
    id: "ja",
    displayName: "Japaníĩ",
    nativeDisplayName: "日本語"
}, {
    id: "vai",
    displayName: "Vai",
    nativeDisplayName: "ꕙꔤ"
}, {
    id: "ko",
    displayName: "Koríyɛŋ",
    nativeDisplayName: "한국어"
} ];

inter.currencies = [ {
    id: "ZAR",
    displayName: "Afirika Kɔi Leŋŋɛ lɔ Randi",
    symbol: "R"
}, {
    id: "DZD",
    displayName: "Agiriya Dina"
}, {
    id: "AOA",
    displayName: "Angola Kuwaŋza",
    symbol: "Kz"
}, {
    id: "BHD",
    displayName: "Bharɛŋ Dina"
}, {
    id: "BWP",
    displayName: "Bhosuwana Pula",
    symbol: "P"
}, {
    id: "BIF",
    displayName: "Bhurundi Furaŋki"
}, {
    id: "CNY",
    displayName: "Chaníĩ Yuwaŋ Rɛŋmimbi",
    symbol: "CN¥"
}, {
    id: "GMD",
    displayName: "Gambiya Dalasi"
}, {
    id: "GHC",
    displayName: "Gana Sidi"
}, {
    id: "GNS",
    displayName: "Gini Furaŋki"
}, {
    id: "JPY",
    displayName: "Japaniĩ Yɛŋ",
    symbol: "JP¥"
}, {
    id: "GBP",
    displayName: "Jengési Pɔɔ̃",
    symbol: "£"
}, {
    id: "DJF",
    displayName: "Jibhuti Furaŋki"
}, {
    id: "CAD",
    displayName: "Kanada Dala",
    symbol: "CA$"
}, {
    id: "KMF",
    displayName: "Komoro Furaŋki",
    symbol: "CF"
}, {
    id: "KES",
    displayName: "Kénya Siyeŋ"
}, {
    id: "CDF",
    displayName: "Kóngo Furaŋki"
}, {
    id: "SAR",
    displayName: "Lahabu Sawodi Riya"
}, {
    id: "LRD",
    displayName: "Laibhiya Dala",
    symbol: "$"
}, {
    id: "LYD",
    displayName: "Libhiya Dina"
}, {
    id: "SZL",
    displayName: "Lilangeni"
}, {
    id: "LSL",
    displayName: "Lisóto Loti"
}, {
    id: "SLL",
    displayName: "Liyɔɔ̀"
}, {
    id: "MGA",
    displayName: "Malagasi Ariyari",
    symbol: "Ar"
}, {
    id: "MWK",
    displayName: "Malawi Kuwacha"
}, {
    id: "MZM",
    displayName: "Mozambiki Mɛtikali"
}, {
    id: "EGP",
    displayName: "Míséla Pɔɔ̃",
    symbol: "E£"
}, {
    id: "MUR",
    displayName: "Mɔreshɔ Rupi",
    symbol: "Rs"
}, {
    id: "MRO",
    displayName: "Mɔretani Yugiya"
}, {
    id: "MAD",
    displayName: "Mɔroko Dihami"
}, {
    id: "NAD",
    displayName: "Namibiya Dala",
    symbol: "$"
}, {
    id: "NGN",
    displayName: "Naĩjiriya Naĩra",
    symbol: "₦"
}, {
    id: "USD",
    displayName: "Poo Dala",
    symbol: "US$"
}, {
    id: "RWF",
    displayName: "Rawunda Furaŋki",
    symbol: "RF"
}, {
    id: "STD",
    displayName: "Sawo Tombe ɓɛ a Gbawo Dobura",
    symbol: "Db"
}, {
    id: "SHP",
    displayName: "Siŋ Hɛlina Pɔɔ̃",
    symbol: "£"
}, {
    id: "SOS",
    displayName: "Somaliya Siyeŋ"
}, {
    id: "SDG",
    displayName: "Sudaniĩ Pɔɔ̃"
}, {
    id: "CHF",
    displayName: "Suwesi Furaŋki"
}, {
    id: "TZS",
    displayName: "Taŋzaniya Siyeŋ"
}, {
    id: "TND",
    displayName: "Tunisiya Dina"
}, {
    id: "UGX",
    displayName: "Yuganda Siyeŋ"
}, {
    id: "AED",
    displayName: "Yunaitɛ Arabhi Ɛmire Dihami"
}, {
    id: "EUR",
    displayName: "Yuro",
    symbol: "€"
}, {
    id: "ZMW",
    displayName: "Zambiya Kuwacha",
    symbol: "ZK"
}, {
    id: "ZMK",
    displayName: "Zambiya Kuwacha (1968–2012)"
}, {
    id: "ZWD",
    displayName: "Zimbhabhuwe Dala"
}, {
    id: "XOF",
    displayName: "Áfíríka Tele Jíí Sifa",
    symbol: "CFA"
}, {
    id: "XAF",
    displayName: "Áfíríka Tɛ Sifa",
    symbol: "FCFA"
}, {
    id: "INR",
    displayName: "Índiya Rupi",
    symbol: "₹"
}, {
    id: "ETB",
    displayName: "Ítiyopiya Bhii"
}, {
    id: "AUD",
    displayName: "Ɔ́situwa Dala",
    symbol: "A$"
}, {
    id: "ERN",
    displayName: "Ɛritera Nakifa"
}, {
    id: "CVE",
    displayName: "Ɛsikudo Cabovɛdiyano"
} ];

module.exports = inter;
