// inter/build/yo_bj.js
function defaults(targetObj) {
    for (var i = 1; i < arguments.length; i += 1) {
        if (arguments[i] && typeof arguments[i] === "object") {
            for (var propertyName in arguments[i]) {
                if (typeof targetObj[propertyName] === "undefined") {
                    targetObj[propertyName] = arguments[i][propertyName];
                }
            }
        }
    }
    return targetObj;
}

var P = require("./yo.js"), inter = defaults({
    renderers: {}
}, P);

var _h7js1 = [ "K1", "K2", "K3", "K4" ], _h7js2 = {
    all: {
        leap: "{0}bis"
    }
}, _h7js3 = {
    am: "Àárɔ̀",
    pm: "Ɔ̀sán"
}, _h7js4 = {
    a: "h a – h a",
    h: "h–h a"
}, _h7js5 = {
    displayName: "Ɔjɔ́ Ɔ̀sɛ̀"
}, _h7js6 = [ "Before R.O.C.", "R.O.C." ], _h7js7 = {
    displayName: "Àárɔ̀/ɔ̀sán"
}, _h7js8 = {
    a: "h a – h a v",
    h: "h–h a v"
}, _h7js9 = [ "S", "M", "T", "W", "T", "F", "S" ], _h7js10 = {
    d: "MMM d–d",
    M: "MMM d – MMM d"
}, _h7js11 = {
    displayName: "Ibi Àkókò Àgbáyé"
}, _h7js12 = {
    wide: [ "AH" ],
    abbreviated: [ "AH" ]
}, _h7js13 = {
    H: "HH:mm–HH:mm",
    m: "HH:mm–HH:mm"
}, _h7js14 = {
    M: "y-MM – y-MM",
    y: "y-MM – y-MM"
}, _h7js15 = {
    M: "U MMM–MMM",
    y: "U MMM – U MMM"
}, _h7js16 = {
    H: "HH:mm–HH:mm v",
    m: "HH:mm–HH:mm v"
}, _h7js17 = {
    d: "MM-dd – MM-dd",
    M: "MM-dd – MM-dd"
}, _h7js18 = {
    M: "G y MMM–MMM",
    y: "G y MMM – y MMM"
}, _h7js19 = {
    M: "U MMMM–MMMM",
    y: "U MMMM – U MMMM"
}, _h7js20 = {
    M: "G y MMMM–MMMM",
    y: "G y MMMM – y MMMM"
}, _h7js21 = {
    M: "GGGGG y-MM – y-MM",
    y: "GGGGG y-MM – y-MM"
}, _h7js22 = {
    "0": "this year",
    "1": "next year",
    "-1": "last year"
}, _h7js23 = {
    "0": "this week",
    "1": "next week",
    "-1": "last week"
}, _h7js24 = {
    future: {
        other: "+{0} y"
    },
    past: {
        other: "-{0} y"
    }
}, _h7js25 = {
    future: {
        other: "+{0} w"
    },
    past: {
        other: "-{0} w"
    }
}, _h7js26 = {
    future: {
        other: "+{0} d"
    },
    past: {
        other: "-{0} d"
    }
}, _h7js27 = {
    future: {
        other: "+{0} s"
    },
    past: {
        other: "-{0} s"
    }
}, _h7js28 = {
    future: {
        other: "+{0} m"
    },
    past: {
        other: "-{0} m"
    }
}, _h7js29 = {
    future: {
        other: "+{0} h"
    },
    past: {
        other: "-{0} h"
    }
}, _h7js30 = {
    future: {
        other: "+{0} Q"
    },
    past: {
        other: "-{0} Q"
    }
}, _h7js31 = {
    d: "MM-dd, E – MM-dd, E",
    M: "MM-dd, E – MM-dd, E"
}, _h7js32 = {
    d: "MMM d, E – MMM d, E",
    M: "MMM d, E – MMM d, E"
}, _h7js33 = {
    "0": "this month",
    "1": "next month",
    "-1": "last month"
}, _h7js34 = {
    wide: [ "ERA0", "ERA1" ],
    abbreviated: [ "ERA0", "ERA1" ]
}, _h7js35 = {
    future: {
        other: "+{0} min"
    },
    past: {
        other: "-{0} min"
    }
}, _h7js36 = [ "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12" ], _h7js37 = {
    "0": "this Sunday",
    "1": "next Sunday",
    "-1": "last Sunday"
}, _h7js38 = {
    "0": "this Monday",
    "1": "next Monday",
    "-1": "last Monday"
}, _h7js39 = {
    "0": "this Friday",
    "1": "next Friday",
    "-1": "last Friday"
}, _h7js40 = {
    a: "h:mm a – h:mm a",
    h: "h:mm–h:mm a",
    m: "h:mm–h:mm a"
}, _h7js41 = [ "Kɔ́tà Kínní", "Kɔ́tà Kejì", "Kɔ́à Keta", "Kɔ́tà Kɛrin" ], _h7js42 = {
    "0": "Òní",
    "1": "Ɔ̀la",
    "2": "òtúùnla",
    "-2": "íjɛta",
    "-1": "Àná"
}, _h7js43 = {
    "0": "this quarter",
    "1": "next quarter",
    "-1": "last quarter"
}, _h7js44 = {
    "0": "this Tuesday",
    "1": "next Tuesday",
    "-1": "last Tuesday"
}, _h7js45 = {
    d: "U MMM d–d",
    M: "U MMM d – MMM d",
    y: "U MMM d – U MMM d"
}, _h7js46 = [ "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13" ], _h7js47 = {
    "0": "this Thursday",
    "1": "next Thursday",
    "-1": "last Thursday"
}, _h7js48 = {
    "0": "this Saturday",
    "1": "next Saturday",
    "-1": "last Saturday"
}, _h7js49 = {
    a: "h:mm a – h:mm a v",
    h: "h:mm–h:mm a v",
    m: "h:mm–h:mm a v"
}, _h7js50 = [ "Àìkú", "Ajé", "Ìsɛ́gun", "Ɔjɔ́rú", "Ɔjɔ́bɔ", "Ɛtì", "Àbámɛ́ta" ], _h7js51 = {
    "0": "this Wednesday",
    "1": "next Wednesday",
    "-1": "last Wednesday"
}, _h7js52 = {
    d: "G y MMM d–d",
    M: "G y MMM d – MMM d",
    y: "G y MMM d – y MMM d"
}, _h7js53 = {
    relative: _h7js37
}, _h7js54 = {
    relative: _h7js38
}, _h7js55 = {
    relative: _h7js39
}, _h7js56 = {
    relative: _h7js44
}, _h7js57 = {
    d: "y-MM-dd – y-MM-dd",
    M: "y-MM-dd – y-MM-dd",
    y: "y-MM-dd – y-MM-dd"
}, _h7js58 = {
    relative: _h7js47
}, _h7js59 = {
    relative: _h7js48
}, _h7js60 = {
    abbreviated: {
        leap: "{0}bis"
    },
    narrow: {
        leap: "{0}b"
    },
    wide: {
        leap: "{0}bis"
    }
}, _h7js61 = {
    id: "FJ",
    displayName: "Orílɛ́ède Fiji",
    regionId: "054",
    hasTimeZones: true
}, _h7js62 = {
    id: "GH",
    displayName: "Orílɛ́ède Gana",
    regionId: "011",
    hasTimeZones: true
}, _h7js63 = {
    id: "GN",
    displayName: "Orílɛ́ède Gene",
    regionId: "011",
    hasTimeZones: true
}, _h7js64 = {
    id: "QA",
    displayName: "Orílɛ́ède Kota",
    regionId: "145",
    hasTimeZones: true
}, _h7js65 = {
    id: "CU",
    displayName: "Orílɛ́ède Kúbà",
    regionId: "029",
    hasTimeZones: true
}, _h7js66 = {
    id: "ML",
    displayName: "Orílɛ́ède Mali",
    regionId: "011",
    hasTimeZones: true
}, _h7js67 = {
    id: "NP",
    displayName: "Orílɛ́ède Nepa",
    regionId: "034",
    hasTimeZones: true
}, _h7js68 = {
    id: "NU",
    displayName: "Orílɛ́ède Niue",
    regionId: "061",
    hasTimeZones: true
}, _h7js69 = {
    id: "PE",
    displayName: "Orílɛ́ède Peru",
    regionId: "005",
    hasTimeZones: true
}, _h7js70 = {
    id: "WS",
    displayName: "Orílɛ́ède Samɔ",
    regionId: "061",
    hasTimeZones: true
}, _h7js71 = {
    id: "TG",
    displayName: "Orílɛ́ède Togo",
    regionId: "011",
    hasTimeZones: true
}, _h7js72 = {
    id: "OM",
    displayName: "Orílɛ́ède Ɔɔma",
    regionId: "145",
    hasTimeZones: true
}, _h7js73 = {
    relative: _h7js51
}, _h7js74 = {
    id: "CD",
    displayName: "Orilɛ́ède Kóngò",
    regionId: "017",
    hasTimeZones: true
}, _h7js75 = {
    id: "GA",
    displayName: "Orílɛ́ède Gabon",
    regionId: "017",
    hasTimeZones: true
}, _h7js76 = {
    id: "GU",
    displayName: "Orílɛ́ède Guamu",
    regionId: "057",
    hasTimeZones: true
}, _h7js77 = {
    id: "GE",
    displayName: "Orílɛ́ède Gɔgia",
    regionId: "145",
    hasTimeZones: true
}, _h7js78 = {
    id: "HT",
    displayName: "Orílɛ́ède Haati",
    regionId: "029",
    hasTimeZones: true
}, _h7js79 = {
    id: "IN",
    displayName: "Orílɛ́ède India",
    regionId: "034",
    hasTimeZones: true
}, _h7js80 = {
    id: "IQ",
    displayName: "Orílɛ́ède Iraki",
    regionId: "145",
    hasTimeZones: true
}, _h7js81 = {
    id: "IR",
    displayName: "Orílɛ́ède Irani",
    regionId: "034",
    hasTimeZones: true
}, _h7js82 = {
    id: "KE",
    displayName: "Orílɛ́ède Kenya",
    regionId: "014",
    hasTimeZones: true
}, _h7js83 = {
    id: "CG",
    displayName: "Orílɛ́ède Kóngò",
    regionId: "017",
    hasTimeZones: true
}, _h7js84 = {
    id: "LA",
    displayName: "Orílɛ́ède Laosi",
    regionId: "035",
    hasTimeZones: true
}, _h7js85 = {
    id: "NR",
    displayName: "Orílɛ́ède Nauru",
    regionId: "057",
    hasTimeZones: true
}, _h7js86 = {
    id: "NE",
    displayName: "Orílɛ́ède Nàìjá",
    regionId: "011",
    hasTimeZones: true
}, _h7js87 = {
    id: "PW",
    displayName: "Orílɛ́ède Paalu",
    regionId: "057",
    hasTimeZones: true
}, _h7js88 = {
    id: "SY",
    displayName: "Orílɛ́ède Siria",
    regionId: "145",
    hasTimeZones: true
}, _h7js89 = {
    id: "TO",
    displayName: "Orílɛ́ède Tonga",
    regionId: "061",
    hasTimeZones: true
}, _h7js90 = {
    id: "TR",
    displayName: "Orílɛ́ède Tɔɔki",
    regionId: "145",
    hasTimeZones: true
}, _h7js91 = {
    id: "CL",
    displayName: "Orílɛ́ède shílè",
    regionId: "005",
    hasTimeZones: true
}, _h7js92 = {
    id: "AW",
    displayName: "Orílɛ́ède Árúbà",
    regionId: "029",
    hasTimeZones: true
}, _h7js93 = [ "M01", "M02", "M03", "M04", "M05", "M06", "M07", "M08", "M09", "M10", "M11", "M12" ], _h7js94 = {
    id: "AM",
    displayName: "Orílɛ́ède Améníà",
    regionId: "145",
    hasTimeZones: true
}, _h7js95 = {
    id: "BH",
    displayName: "Orílɛ́ède Báránì",
    regionId: "145",
    hasTimeZones: true
}, _h7js96 = {
    id: "BM",
    displayName: "Orílɛ́ède Bémúdà",
    regionId: "021",
    hasTimeZones: true
}, _h7js97 = {
    id: "BT",
    displayName: "Orílɛ́ède Bútánì",
    regionId: "034",
    hasTimeZones: true
}, _h7js98 = {
    id: "BJ",
    displayName: "Orílɛ́ède Bɛ̀nɛ̀",
    regionId: "011",
    hasTimeZones: true
}, _h7js99 = {
    id: "EC",
    displayName: "Orílɛ́ède Ekuádò",
    regionId: "005",
    hasTimeZones: true
}, _h7js100 = {
    id: "ET",
    displayName: "Orílɛ́ède Etopia",
    regionId: "014",
    hasTimeZones: true
}, _h7js101 = {
    id: "GM",
    displayName: "Orílɛ́ède Gambia",
    regionId: "011",
    hasTimeZones: true
}, _h7js102 = {
    id: "DE",
    displayName: "Orílɛ́ède Gemani",
    regionId: "155",
    hasTimeZones: true
}, _h7js103 = {
    id: "GD",
    displayName: "Orílɛ́ède Genada",
    regionId: "029",
    hasTimeZones: true
}, _h7js104 = {
    id: "GY",
    displayName: "Orílɛ́ède Guyana",
    regionId: "005",
    hasTimeZones: true
}, _h7js105 = {
    id: "SH",
    displayName: "Orílɛ́ède Hɛlena",
    regionId: "011",
    hasTimeZones: true
}, _h7js106 = {
    id: "JP",
    displayName: "Orílɛ́ède Japani",
    regionId: "030",
    hasTimeZones: true
}, _h7js107 = {
    id: "JO",
    displayName: "Orílɛ́ède Jɔdani",
    regionId: "145",
    hasTimeZones: true
}, _h7js108 = {
    id: "KW",
    displayName: "Orílɛ́ède Kuweti",
    regionId: "145",
    hasTimeZones: true
}, _h7js109 = {
    id: "CA",
    displayName: "Orílɛ́ède Kánádà",
    regionId: "021",
    hasTimeZones: true
}, _h7js110 = {
    id: "CY",
    displayName: "Orílɛ́ède Kúrúsì",
    regionId: "145",
    hasTimeZones: true
}, _h7js111 = {
    id: "LS",
    displayName: "Orílɛ́ède Lesoto",
    regionId: "018",
    hasTimeZones: true
}, _h7js112 = {
    id: "LY",
    displayName: "Orílɛ́ède Libiya",
    regionId: "015",
    hasTimeZones: true
}, _h7js113 = {
    id: "LC",
    displayName: "Orílɛ́ède Lushia",
    regionId: "029",
    hasTimeZones: true
}, _h7js114 = {
    id: "MT",
    displayName: "Orílɛ́ède Malata",
    regionId: "039",
    hasTimeZones: true
}, _h7js115 = {
    id: "MW",
    displayName: "Orílɛ́ède Malawi",
    regionId: "014",
    hasTimeZones: true
}, _h7js116 = {
    id: "YT",
    displayName: "Orílɛ́ède Mayote",
    regionId: "014",
    hasTimeZones: true
}, _h7js117 = {
    id: "MX",
    displayName: "Orílɛ́ède Mesiko",
    regionId: "013",
    hasTimeZones: true
}, _h7js118 = {
    id: "MC",
    displayName: "Orílɛ́ède Monako",
    regionId: "155",
    hasTimeZones: true
}, _h7js119 = {
    id: "MA",
    displayName: "Orílɛ́ède Moroko",
    regionId: "015",
    hasTimeZones: true
}, _h7js120 = {
    id: "NO",
    displayName: "Orílɛ́ède Nɔɔwii",
    regionId: "154",
    hasTimeZones: true
}, _h7js121 = {
    id: "PA",
    displayName: "Orílɛ́ède Panama",
    regionId: "013",
    hasTimeZones: true
}, _h7js122 = {
    id: "PT",
    displayName: "Orílɛ́ède Pɔtugi",
    regionId: "039",
    hasTimeZones: true
}, _h7js123 = {
    id: "RU",
    displayName: "Orílɛ́ède Rɔshia",
    regionId: "151",
    hasTimeZones: true
}, _h7js124 = {
    id: "ES",
    displayName: "Orílɛ́ède Sipani",
    regionId: "039",
    hasTimeZones: true
}, _h7js125 = {
    id: "SD",
    displayName: "Orílɛ́ède Sudani",
    regionId: "015",
    hasTimeZones: true
}, _h7js126 = {
    id: "SN",
    displayName: "Orílɛ́ède Sɛnɛga",
    regionId: "011",
    hasTimeZones: true
}, _h7js127 = {
    id: "TV",
    displayName: "Orílɛ́ède Tufalu",
    regionId: "061",
    hasTimeZones: true
}, _h7js128 = {
    id: "UG",
    displayName: "Orílɛ́ède Uganda",
    regionId: "014",
    hasTimeZones: true
}, _h7js129 = {
    id: "TD",
    displayName: "Orílɛ́ède shààdì",
    regionId: "017",
    hasTimeZones: true
}, _h7js130 = {
    id: "CN",
    displayName: "Orílɛ́ède sháínà",
    regionId: "030",
    hasTimeZones: true
}, _h7js131 = {
    id: "YE",
    displayName: "Orílɛ́ède yemeni",
    regionId: "145",
    hasTimeZones: true
}, _h7js132 = {
    id: "IE",
    displayName: "Orílɛ́ède Ailandi",
    regionId: "154",
    hasTimeZones: true
}, _h7js133 = {
    id: "BZ",
    displayName: "Orílɛ́ède Bèlísɛ̀",
    regionId: "013",
    hasTimeZones: true
}, _h7js134 = {
    id: "DK",
    displayName: "Orílɛ́ède Dɛ́mákì",
    regionId: "154",
    hasTimeZones: true
}, _h7js135 = {
    id: "ER",
    displayName: "Orílɛ́ède Eritira",
    regionId: "014",
    hasTimeZones: true
}, _h7js136 = {
    id: "VU",
    displayName: "Orílɛ́ède Faniatu",
    regionId: "054",
    hasTimeZones: true
}, _h7js137 = {
    id: "FR",
    displayName: "Orílɛ́ède Faranse",
    regionId: "155",
    hasTimeZones: true
}, _h7js138 = {
    id: "FI",
    displayName: "Orílɛ́ède Filandi",
    regionId: "154",
    hasTimeZones: true
}, _h7js139 = {
    id: "GR",
    displayName: "Orílɛ́ède Geriisi",
    regionId: "039",
    hasTimeZones: true
}, _h7js140 = {
    id: "HU",
    displayName: "Orílɛ́ède Hungari",
    regionId: "151",
    hasTimeZones: true
}, _h7js141 = {
    id: "IL",
    displayName: "Orílɛ́ède Iserɛli",
    regionId: "145",
    hasTimeZones: true
}, _h7js142 = {
    id: "IT",
    displayName: "Orílɛ́ède Italiyi",
    regionId: "039",
    hasTimeZones: true
}, _h7js143 = {
    id: "JM",
    displayName: "Orílɛ́ède Jamaika",
    regionId: "029",
    hasTimeZones: true
}, _h7js144 = {
    id: "LR",
    displayName: "Orílɛ́ède Laberia",
    regionId: "011",
    hasTimeZones: true
}, _h7js145 = {
    id: "LV",
    displayName: "Orílɛ́ède Latifia",
    regionId: "154",
    hasTimeZones: true
}, _h7js146 = {
    id: "MY",
    displayName: "Orílɛ́ède Malasia",
    regionId: "035",
    hasTimeZones: true
}, _h7js147 = {
    id: "MD",
    displayName: "Orílɛ́ède Modofia",
    regionId: "151",
    hasTimeZones: true
}, _h7js148 = {
    id: "MN",
    displayName: "Orílɛ́ède Mogolia",
    regionId: "030",
    hasTimeZones: true
}, _h7js149 = {
    id: "NA",
    displayName: "Orílɛ́ède Namibia",
    regionId: "018",
    hasTimeZones: true
}, _h7js150 = {
    id: "PL",
    displayName: "Orílɛ́ède Polandi",
    regionId: "151",
    hasTimeZones: true
}, _h7js151 = {
    id: "RW",
    displayName: "Orílɛ́ède Ruwanda",
    regionId: "014",
    hasTimeZones: true
}, _h7js152 = {
    id: "SG",
    displayName: "Orílɛ́ède Singapo",
    regionId: "035",
    hasTimeZones: true
}, _h7js153 = {
    id: "SO",
    displayName: "Orílɛ́ède Somalia",
    regionId: "014",
    hasTimeZones: true
}, _h7js154 = {
    id: "SE",
    displayName: "Orílɛ́ède Swidini",
    regionId: "154",
    hasTimeZones: true
}, _h7js155 = {
    id: "TW",
    displayName: "Orílɛ́ède Taiwani",
    regionId: "030",
    hasTimeZones: true
}, _h7js156 = {
    id: "TK",
    displayName: "Orílɛ́ède Tokelau",
    regionId: "061",
    hasTimeZones: true
}, _h7js157 = {
    id: "UA",
    displayName: "Orílɛ́ède Ukarini",
    regionId: "151",
    hasTimeZones: true
}, _h7js158 = {
    id: "AD",
    displayName: "Orílɛ́ède Ààndórà",
    regionId: "039",
    hasTimeZones: true
}, _h7js159 = {
    id: "AO",
    displayName: "Orílɛ́ède Ààngólà",
    regionId: "017",
    hasTimeZones: true
}, _h7js160 = {
    id: "EG",
    displayName: "Orílɛ́ède Égípítì",
    regionId: "015",
    hasTimeZones: true
}, _h7js161 = {
    id: "AR",
    displayName: "Orílɛ́ède Agentínà",
    regionId: "005",
    hasTimeZones: true
}, _h7js162 = {
    id: "AT",
    displayName: "Orílɛ́ède Asítíríà",
    regionId: "155",
    hasTimeZones: true
}, _h7js163 = {
    id: "BS",
    displayName: "Orílɛ́ède Bàhámásì",
    regionId: "029",
    hasTimeZones: true
}, _h7js164 = {
    id: "BR",
    displayName: "Orílɛ́ède Bàràsílì",
    regionId: "005",
    hasTimeZones: true
}, _h7js165 = {
    id: "BB",
    displayName: "Orílɛ́ède Bábádósì",
    regionId: "029",
    hasTimeZones: true
}, _h7js166 = {
    id: "BE",
    displayName: "Orílɛ́ède Bégíɔ́mù",
    regionId: "155",
    hasTimeZones: true
}, _h7js167 = {
    id: "BY",
    displayName: "Orílɛ́ède Bélárúsì",
    regionId: "151",
    hasTimeZones: true
}, _h7js168 = {
    id: "BG",
    displayName: "Orílɛ́ède Bùùgáríà",
    regionId: "151",
    hasTimeZones: true
}, _h7js169 = {
    id: "BI",
    displayName: "Orílɛ́ède Bùùrúndì",
    regionId: "014",
    hasTimeZones: true
}, _h7js170 = {
    id: "DJ",
    displayName: "Orílɛ́ède Díbɔ́ótì",
    regionId: "014",
    hasTimeZones: true
}, _h7js171 = {
    id: "DM",
    displayName: "Orílɛ́ède Dòmíníkà",
    regionId: "029",
    hasTimeZones: true
}, _h7js172 = {
    id: "EE",
    displayName: "Orílɛ́ède Esitonia",
    regionId: "154",
    hasTimeZones: true
}, _h7js173 = {
    id: "VA",
    displayName: "Orílɛ́ède Fatikani",
    regionId: "039",
    hasTimeZones: true
}, _h7js174 = {
    id: "VN",
    displayName: "Orílɛ́ède Fɛtinami",
    regionId: "035",
    hasTimeZones: true
}, _h7js175 = {
    id: "GP",
    displayName: "Orílɛ́ède Gadelope",
    regionId: "029",
    hasTimeZones: true
}, _h7js176 = {
    id: "KI",
    displayName: "Orílɛ́ède Kiribati",
    regionId: "057",
    hasTimeZones: true
}, _h7js177 = {
    id: "KM",
    displayName: "Orílɛ́ède Kòmòrósì",
    regionId: "014",
    hasTimeZones: true
}, _h7js178 = {
    id: "HR",
    displayName: "Orílɛ́ède Kòróátíà",
    regionId: "039",
    hasTimeZones: true
}, _h7js179 = {
    id: "LB",
    displayName: "Orílɛ́ède Lebanoni",
    regionId: "145",
    hasTimeZones: true
}, _h7js180 = {
    id: "LT",
    displayName: "Orílɛ́ède Lituania",
    regionId: "154",
    hasTimeZones: true
}, _h7js181 = {
    id: "LU",
    displayName: "Orílɛ́ède Lusemogi",
    regionId: "155",
    hasTimeZones: true
}, _h7js182 = {
    id: "MG",
    displayName: "Orílɛ́ède Madasika",
    regionId: "014",
    hasTimeZones: true
}, _h7js183 = {
    id: "MV",
    displayName: "Orílɛ́ède Maladifi",
    regionId: "034",
    hasTimeZones: true
}, _h7js184 = {
    id: "MM",
    displayName: "Orílɛ́ède Manamari",
    regionId: "035",
    hasTimeZones: true
}, _h7js185 = {
    id: "UY",
    displayName: "Orílɛ́ède Nruguayi",
    regionId: "005",
    hasTimeZones: true
}, _h7js186 = {
    id: "NG",
    displayName: "Orílɛ́ède Nàìjíríà",
    regionId: "011",
    hasTimeZones: true
}, _h7js187 = {
    id: "PY",
    displayName: "Orílɛ́ède Paraguye",
    regionId: "005",
    hasTimeZones: true
}, _h7js188 = {
    id: "PN",
    displayName: "Orílɛ́ède Pikarini",
    regionId: "061",
    hasTimeZones: true
}, _h7js189 = {
    id: "RE",
    displayName: "Orílɛ́ède Riuniyan",
    regionId: "014",
    hasTimeZones: true
}, _h7js190 = {
    id: "RO",
    displayName: "Orílɛ́ède Romaniya",
    regionId: "151",
    hasTimeZones: true
}, _h7js191 = {
    id: "SR",
    displayName: "Orílɛ́ède Surinami",
    regionId: "005",
    hasTimeZones: true
}, _h7js192 = {
    id: "TH",
    displayName: "Orílɛ́ède Tailandi",
    regionId: "035",
    hasTimeZones: true
}, _h7js193 = {
    id: "TN",
    displayName: "Orílɛ́ède Tunishia",
    regionId: "015",
    hasTimeZones: true
}, _h7js194 = {
    id: "PH",
    displayName: "Orílɛ́ède filipini",
    regionId: "035",
    hasTimeZones: true
}, _h7js195 = {
    id: "ZM",
    displayName: "Orílɛ́ède shamibia",
    regionId: "014",
    hasTimeZones: true
}, _h7js196 = {
    id: "CZ",
    displayName: "Orílɛ́ède shɛ́ɛ́kì",
    regionId: "151",
    hasTimeZones: true
}, _h7js197 = {
    id: "AU",
    displayName: "Orílɛ́ède Ástràlìá",
    regionId: "053",
    hasTimeZones: true
}, _h7js198 = {
    id: "SV",
    displayName: "Orílɛ́ède Ɛɛsáfádò",
    regionId: "013",
    hasTimeZones: true
}, _h7js199 = [ undefined, "zi", "chou", "yin", "mao", "chen", "si", "wu", "wei", "shen", "you", "xu" ], _h7js200 = {
    id: "IS",
    displayName: "Orílɛ́ède Ashilandi",
    regionId: "154",
    hasTimeZones: true
}, _h7js201 = {
    id: "BN",
    displayName: "Orílɛ́ède Búrúnɛ́lì",
    regionId: "035",
    hasTimeZones: true
}, _h7js202 = {
    id: "BO",
    displayName: "Orílɛ́ède Bɔ̀lífíyà",
    regionId: "005",
    hasTimeZones: true
}, _h7js203 = {
    id: "GL",
    displayName: "Orílɛ́ède Gerelandi",
    regionId: "021",
    hasTimeZones: true
}, _h7js204 = {
    id: "GT",
    displayName: "Orílɛ́ède Guatemala",
    regionId: "013",
    hasTimeZones: true
}, _h7js205 = {
    id: "HN",
    displayName: "Orílɛ́ède Hondurasi",
    regionId: "013",
    hasTimeZones: true
}, _h7js206 = {
    id: "ID",
    displayName: "Orílɛ́ède Indonesia",
    regionId: "035",
    hasTimeZones: true
}, _h7js207 = {
    id: "CM",
    displayName: "Orílɛ́ède Kamerúúnì",
    regionId: "017",
    hasTimeZones: true
}, _h7js208 = {
    id: "KH",
    displayName: "Orílɛ́ède Kàmùbódíà",
    regionId: "035",
    hasTimeZones: true
}, _h7js209 = {
    id: "CO",
    displayName: "Orílɛ́ède Kòlómíbìa",
    regionId: "005",
    hasTimeZones: true
}, _h7js210 = {
    id: "MR",
    displayName: "Orílɛ́ède Maritania",
    regionId: "011",
    hasTimeZones: true
}, _h7js211 = {
    id: "MU",
    displayName: "Orílɛ́ède Maritiusi",
    regionId: "014",
    hasTimeZones: true
}, _h7js212 = {
    id: "MK",
    displayName: "Orílɛ́ède Masidonia",
    regionId: "039",
    hasTimeZones: true
}, _h7js213 = {
    id: "MS",
    displayName: "Orílɛ́ède Motserati",
    regionId: "029",
    hasTimeZones: true
}, _h7js214 = {
    id: "NI",
    displayName: "Orílɛ́ède NIkaragua",
    regionId: "013",
    hasTimeZones: true
}, _h7js215 = {
    id: "NL",
    displayName: "Orílɛ́ède Nedalandi",
    regionId: "155",
    hasTimeZones: true
}, _h7js216 = {
    id: "PK",
    displayName: "Orílɛ́ède Pakisitan",
    regionId: "034",
    hasTimeZones: true
}, _h7js217 = {
    id: "PR",
    displayName: "Orílɛ́ède Pɔto Riko",
    regionId: "029",
    hasTimeZones: true
}, _h7js218 = {
    id: "SZ",
    displayName: "Orílɛ́ède Sashiland",
    regionId: "018",
    hasTimeZones: true
}, _h7js219 = {
    id: "SK",
    displayName: "Orílɛ́ède Silofakia",
    regionId: "151",
    hasTimeZones: true
}, _h7js220 = {
    id: "SI",
    displayName: "Orílɛ́ède Silofania",
    regionId: "039",
    hasTimeZones: true
}, _h7js221 = {
    id: "TZ",
    displayName: "Orílɛ́ède Tanshania",
    regionId: "014",
    hasTimeZones: true
}, _h7js222 = {
    id: "SC",
    displayName: "Orílɛ́ède seshɛlɛsi",
    regionId: "014",
    hasTimeZones: true
}, _h7js223 = {
    id: "ZW",
    displayName: "Orílɛ́ède shimibabe",
    regionId: "014",
    hasTimeZones: true
}, _h7js224 = {
    id: "AI",
    displayName: "Orílɛ́ède Ààngúlílà",
    regionId: "029",
    hasTimeZones: true
}, _h7js225 = {
    id: "DO",
    displayName: "Orilɛ́ède Dòmíníkánì",
    regionId: "029",
    hasTimeZones: true
}, _h7js226 = {
    id: "AZ",
    displayName: "Orílɛ́ède Asɛ́bájánì",
    regionId: "145",
    hasTimeZones: true
}, _h7js227 = {
    id: "VE",
    displayName: "Orílɛ́ède Fɛnɛshuɛla",
    regionId: "005",
    hasTimeZones: true
}, _h7js228 = {
    id: "GW",
    displayName: "Orílɛ́ède Gene-Busau",
    regionId: "011",
    hasTimeZones: true
}, _h7js229 = {
    id: "GI",
    displayName: "Orílɛ́ède Gibaratara",
    regionId: "039",
    hasTimeZones: true
}, _h7js230 = {
    id: "CI",
    displayName: "Orílɛ́ède Kóútè forà",
    regionId: "011",
    hasTimeZones: true
}, _h7js231 = {
    id: "MQ",
    displayName: "Orílɛ́ède Matinikuwi",
    regionId: "029",
    hasTimeZones: true
}, _h7js232 = {
    id: "GB",
    displayName: "Orílɛ́ède Omobabirin",
    regionId: "154",
    hasTimeZones: true
}, _h7js233 = {
    id: "LK",
    displayName: "Orílɛ́ède Siri Lanka",
    regionId: "034",
    hasTimeZones: true
}, _h7js234 = {
    id: "TJ",
    displayName: "Orílɛ́ède Takisitani",
    regionId: "143",
    hasTimeZones: true
}, _h7js235 = {
    id: "AL",
    displayName: "Orílɛ́ède Àlùbàníánì",
    regionId: "039",
    hasTimeZones: true
}, _h7js236 = {
    id: "DZ",
    displayName: "Orílɛ́ède Àlùgèríánì",
    regionId: "015",
    hasTimeZones: true
}, _h7js237 = {
    id: "KR",
    displayName: "Orílɛ́ède Ariwa Kɔria",
    regionId: "030",
    hasTimeZones: true
}, _h7js238 = {
    id: "BD",
    displayName: "Orílɛ́ède Bángáládésì",
    regionId: "034",
    hasTimeZones: true
}, _h7js239 = {
    id: "BW",
    displayName: "Orílɛ́ède Bɔ̀tìsúwánà",
    regionId: "018",
    hasTimeZones: true
}, _h7js240 = {
    id: "KP",
    displayName: "Orílɛ́ède Guusu Kɔria",
    regionId: "030",
    hasTimeZones: true
}, _h7js241 = {
    id: "FM",
    displayName: "Orílɛ́ède Makoronesia",
    regionId: "057",
    hasTimeZones: true
}, _h7js242 = {
    id: "MZ",
    displayName: "Orílɛ́ède Moshamibiku",
    regionId: "014",
    hasTimeZones: true
}, _h7js243 = {
    id: "SM",
    displayName: "Orílɛ́ède Sani Marino",
    regionId: "039",
    hasTimeZones: true
}, _h7js244 = {
    id: "SL",
    displayName: "Orílɛ́ède Siria looni",
    regionId: "011",
    hasTimeZones: true
}, _h7js245 = {
    id: "BF",
    displayName: "Orílɛ́ède Bùùkíná Fasò",
    regionId: "011",
    hasTimeZones: true
}, _h7js246 = {
    id: "KZ",
    displayName: "Orílɛ́ède Kashashatani",
    regionId: "143",
    hasTimeZones: true
}, _h7js247 = {
    id: "CR",
    displayName: "Orílɛ́ède Kuusita Ríkà",
    regionId: "013",
    hasTimeZones: true
}, _h7js248 = {
    id: "SA",
    displayName: "Orílɛ́ède Saudi Arabia",
    regionId: "145",
    hasTimeZones: true
}, _h7js249 = {
    id: "AF",
    displayName: "Orílɛ́ède Àfùgànístánì",
    regionId: "034",
    hasTimeZones: true
}, _h7js250 = {
    id: "ZA",
    displayName: "Orílɛ́ède Ariwa Afirika",
    regionId: "018",
    hasTimeZones: true
}, _h7js251 = {
    id: "GQ",
    displayName: "Orílɛ́ède Ekutoria Gini",
    regionId: "017",
    hasTimeZones: true
}, _h7js252 = {
    id: "CK",
    displayName: "Orílɛ́ède Etíokun Kùúkù",
    regionId: "061",
    hasTimeZones: true
}, _h7js253 = {
    id: "KG",
    displayName: "Orílɛ́ède Kurishisitani",
    regionId: "143",
    hasTimeZones: true
}, _h7js254 = {
    id: "TM",
    displayName: "Orílɛ́ède Tɔɔkimenisita",
    regionId: "143",
    hasTimeZones: true
}, _h7js255 = {
    id: "CH",
    displayName: "Orílɛ́ède switishilandi",
    regionId: "155",
    hasTimeZones: true
}, _h7js256 = {
    d: "U MMM d, E – MMM d, E",
    M: "U MMM d, E – MMM d, E",
    y: "U MMM d, E – U MMM d, E"
}, _h7js257 = {
    id: "MH",
    displayName: "Orílɛ́ède Etikun Máshali",
    regionId: "057",
    hasTimeZones: true
}, _h7js258 = {
    id: "KY",
    displayName: "Orílɛ́ède Etíokun Kámánì",
    regionId: "029",
    hasTimeZones: true
}, _h7js259 = {
    id: "GF",
    displayName: "Orílɛ́ède Firenshi Guana",
    regionId: "005",
    hasTimeZones: true
}, _h7js260 = {
    id: "UZ",
    displayName: "Orílɛ́ède Nshibɛkisitani",
    regionId: "143",
    hasTimeZones: true
}, _h7js261 = {
    id: "PG",
    displayName: "Orílɛ́ède Paapu ti Giini",
    regionId: "054",
    hasTimeZones: true
}, _h7js262 = {
    id: "NZ",
    displayName: "Orílɛ́ède shilandi Titun",
    regionId: "053",
    hasTimeZones: true
}, _h7js263 = {
    id: "TL",
    displayName: "Orílɛ́ède ÌlàOòrùn Tímɔ̀",
    regionId: "035",
    hasTimeZones: true
}, _h7js264 = {
    id: "NF",
    displayName: "Orílɛ́ède Etikun Nɔ́úfókì",
    regionId: "053",
    hasTimeZones: true
}, _h7js265 = {
    id: "SB",
    displayName: "Orílɛ́ède Etikun Solomoni",
    regionId: "054",
    hasTimeZones: true
}, _h7js266 = {
    id: "IO",
    displayName: "Orílɛ́ède Etíkun Índíánì ti Ìlú Bírítísì",
    hasTimeZones: true
}, _h7js267 = {
    id: "NC",
    displayName: "Orílɛ́ède Kaledonia Titun",
    regionId: "054",
    hasTimeZones: true
}, _h7js268 = {
    id: "KN",
    displayName: "Orílɛ́ède Kiiti ati Neefi",
    regionId: "029",
    hasTimeZones: true
}, _h7js269 = {
    id: "LI",
    displayName: "Orílɛ́ède Lɛshitɛnisiteni",
    regionId: "155",
    hasTimeZones: true
}, _h7js270 = {
    id: "WF",
    displayName: "Orílɛ́ède Wali ati futuna",
    regionId: "061",
    hasTimeZones: true
}, _h7js271 = {
    displayName: "wákàtí",
    relativeTime: _h7js29
}, _h7js272 = {
    id: "FK",
    displayName: "Orílɛ́ède Etikun Fakalandi",
    regionId: "005",
    hasTimeZones: true
}, _h7js273 = {
    id: "US",
    displayName: "Orílɛ́ède Orilɛede Amerika",
    regionId: "021",
    hasTimeZones: true
}, _h7js274 = [ "Ɔjɔ́ Àìkú", "Ɔjɔ́ Ajé", "Ɔjɔ́ Ìsɛ́gun", "Ɔjɔ́rú", "Ɔjɔ́bɔ", "Ɔjɔ́ Ɛtì", "Ɔjɔ́ Àbámɛ́ta" ], _h7js275 = {
    d: "GGGGG y-MM-dd – y-MM-dd",
    M: "GGGGG y-MM-dd – y-MM-dd",
    y: "GGGGG y-MM-dd – y-MM-dd"
}, _h7js276 = {
    d: "y-MM-dd, E – y-MM-dd, E",
    M: "y-MM-dd, E – y-MM-dd, E",
    y: "y-MM-dd, E – y-MM-dd, E"
}, _h7js277 = {
    id: "CV",
    displayName: "Orílɛ́ède Etíokun Kápé féndè",
    regionId: "011",
    hasTimeZones: true
}, _h7js278 = {
    id: "PF",
    displayName: "Orílɛ́ède Firenshi Polinesia",
    regionId: "061",
    hasTimeZones: true
}, _h7js279 = {
    id: "PM",
    displayName: "Orílɛ́ède Pɛɛri ati mikuloni",
    regionId: "021",
    hasTimeZones: true
}, _h7js280 = {
    id: "AS",
    displayName: "Sámóánì ti Orílɛ́ède Àméríkà",
    regionId: "061",
    hasTimeZones: true
}, _h7js281 = {
    id: "TT",
    displayName: "Orílɛ́ède Tirinida ati Tobaga",
    regionId: "029",
    hasTimeZones: true
}, _h7js282 = {
    id: "CF",
    displayName: "Orílɛ́ède Àrin gùngun Áfíríkà",
    regionId: "017",
    hasTimeZones: true
}, _h7js283 = {
    d: "G y MMM d, E – MMM d, E",
    M: "G y MMM d, E – MMM d, E",
    y: "G y MMM d, E – y MMM d, E"
}, _h7js284 = {
    displayName: "Ìsɛ́jú",
    relativeTime: _h7js35
}, _h7js285 = {
    id: "MP",
    displayName: "Orílɛ́ède Etikun Guusu Mariana",
    regionId: "057",
    hasTimeZones: true
}, _h7js286 = {
    id: "VC",
    displayName: "Orílɛ́ède Fisɛnnti ati Genadina",
    regionId: "029",
    hasTimeZones: true
}, _h7js287 = {
    id: "AG",
    displayName: "Orílɛ́ède Ààntígúà àti Báríbúdà",
    regionId: "029",
    hasTimeZones: true
}, _h7js288 = {
    id: "AE",
    displayName: "Orílɛ́ède Ɛmirate ti Awɔn Arabu",
    regionId: "145",
    hasTimeZones: true
}, _h7js289 = {
    id: "ST",
    displayName: "Orílɛ́ède Sao tomi ati piriishipi",
    regionId: "017",
    hasTimeZones: true
}, _h7js290 = {
    id: "TC",
    displayName: "Orílɛ́ède Tɔɔki ati Etikun Kakɔsi",
    regionId: "029",
    hasTimeZones: true
}, _h7js291 = {
    id: "BA",
    displayName: "Orílɛ́ède Bɔ̀síníà àti Ɛtisɛgófínà",
    regionId: "039",
    hasTimeZones: true
}, _h7js292 = {
    id: "VI",
    displayName: "Orílɛ́ède Etikun Fagini ti Amɛrika",
    regionId: "029",
    hasTimeZones: true
}, _h7js293 = {
    id: "PS",
    displayName: "Orílɛ́ède Iwɔorun Pakisitian ati Gasha",
    regionId: "145",
    hasTimeZones: true
}, _h7js294 = {
    id: "VG",
    displayName: "Orílɛ́ède Etíkun Fágínì ti ìlú Bírítísì",
    regionId: "029",
    hasTimeZones: true
}, _h7js295 = {
    abbreviated: _h7js3,
    narrow: _h7js3,
    wide: _h7js3
}, _h7js296 = [ "Shɛ́rɛ́", "Èrèlè", "Ɛrɛ̀nà", "Ìgbé", "Ɛ̀bibi", "Òkúdu", "Agɛmɔ", "Ògún", "Owewe", "Ɔ̀wàrà", "Bélú", "Ɔ̀pɛ̀" ], _h7js297 = {
    d: "GGGGG y-MM-dd, E – y-MM-dd, E",
    M: "GGGGG y-MM-dd, E – y-MM-dd, E",
    y: "GGGGG y-MM-dd, E – y-MM-dd, E"
}, _h7js298 = [ "Muh.", "Saf.", "Rab. I", "Rab. II", "Jum. I", "Jum. II", "Raj.", "Sha.", "Ram.", "Shaw.", "Dhuʻl-Q.", "Dhuʻl-H." ], _h7js299 = {
    displayName: "Ìsɛ́jú Ààyá",
    relative: [ "now" ],
    relativeTime: _h7js27
}, _h7js300 = [ "Tishri", "Heshvan", "Kislev", "Tevet", "Shevat", "Adar I", "Adar", "Nisan", "Iyar", "Sivan", "Tamuz", "Av", "Elul" ], _h7js301 = [ "Farvardin", "Ordibehesht", "Khordad", "Tir", "Mordad", "Shahrivar", "Mehr", "Aban", "Azar", "Dey", "Bahman", "Esfand" ], _h7js302 = [ "Tout", "Baba", "Hator", "Kiahk", "Toba", "Amshir", "Baramhat", "Baramouda", "Bashans", "Paona", "Epep", "Mesra", "Nasie" ], _h7js303 = [ "Meskerem", "Tekemt", "Hedar", "Tahsas", "Ter", "Yekatit", "Megabit", "Miazia", "Genbot", "Sene", "Hamle", "Nehasse", "Pagumen" ], _h7js304 = {
    abbreviated: _h7js1,
    narrow: [ "1", "2", "3", "4" ],
    wide: _h7js41
}, _h7js305 = [ "Chaitra", "Vaisakha", "Jyaistha", "Asadha", "Sravana", "Bhadra", "Asvina", "Kartika", "Agrahayana", "Pausa", "Magha", "Phalguna" ], _h7js306 = [ "Muharram", "Safar", "Rabiʻ I", "Rabiʻ II", "Jumada I", "Jumada II", "Rajab", "Shaʻban", "Ramadan", "Shawwal", "Dhuʻl-Qiʻdah", "Dhuʻl-Hijjah" ], _h7js307 = {
    displayName: "Ɔdún",
    relative: _h7js22,
    relativeTime: _h7js24
}, _h7js308 = {
    displayName: "Ɔ̀sè",
    relative: _h7js23,
    relativeTime: _h7js25
}, _h7js309 = {
    displayName: "Osù",
    relative: _h7js33,
    relativeTime: _h7js28
}, _h7js310 = {
    displayName: "Ɔjɔ́",
    relative: _h7js42,
    relativeTime: _h7js26
}, _h7js311 = {
    displayName: "Quarter",
    relative: _h7js43,
    relativeTime: _h7js30
}, _h7js312 = [ "Oshù Shɛ́rɛ́", "Oshù Èrèlè", "Oshù Ɛrɛ̀nà", "Oshù Ìgbé", "Oshù Ɛ̀bibi", "Oshù Òkúdu", "Oshù Agɛmɔ", "Oshù Ògún", "Oshù Owewe", "Oshù Ɔ̀wàrà", "Oshù Bélú", "Oshù Ɔ̀pɛ̀" ], _h7js313 = {
    format: _h7js60,
    numeric: _h7js2,
    standAlone: _h7js60
}, _h7js314 = {
    format: _h7js295,
    standAlone: _h7js295
}, _h7js315 = {
    abbreviated: _h7js93,
    narrow: _h7js36,
    wide: _h7js93
}, _h7js316 = {
    abbreviated: _h7js199,
    narrow: _h7js199,
    wide: _h7js199
}, _h7js317 = {
    format: _h7js304,
    standAlone: _h7js304
}, _h7js318 = {
    format: _h7js316
}, _h7js319 = {
    abbreviated: _h7js50,
    narrow: _h7js9,
    wide: _h7js274,
    "short": _h7js50
}, _h7js320 = {
    abbreviated: _h7js300,
    narrow: _h7js46,
    wide: _h7js300
}, _h7js321 = {
    abbreviated: _h7js301,
    narrow: _h7js36,
    wide: _h7js301
}, _h7js322 = {
    abbreviated: _h7js302,
    narrow: _h7js46,
    wide: _h7js302
}, _h7js323 = {
    abbreviated: _h7js298,
    narrow: _h7js36,
    wide: _h7js306
}, _h7js324 = {
    abbreviated: _h7js303,
    narrow: _h7js46,
    wide: _h7js303
}, _h7js325 = {
    abbreviated: _h7js305,
    narrow: _h7js36,
    wide: _h7js305
}, _h7js326 = {
    abbreviated: _h7js296,
    narrow: _h7js36,
    wide: _h7js312
}, _h7js327 = {
    fullDate: "r(U) MMMM d, EEEE",
    longDate: "r(U) MMMM d",
    mediumDate: "r MMM d",
    shortDate: "r-MM-dd",
    fullTime: "h:mm:ss a zzzz",
    longTime: "h:mm:ss a z",
    mediumTime: "h:mm:ss a",
    shortTime: "h:mm a",
    shortDateTime: "r-MM-dd h:mm a",
    mediumDateTime: "r MMM d h:mm:ss a",
    longDateTime: "r(U) MMMM d h:mm:ss a z",
    fullDateTime: "r(U) MMMM d, EEEE h:mm:ss a zzzz"
}, _h7js328 = {
    format: _h7js315,
    standAlone: _h7js315
}, _h7js329 = {
    d: "d",
    E: "ccc",
    Ed: "d, E",
    Gy: "r U",
    GyMMM: "r(U) MMM",
    GyMMMd: "r MMM d",
    GyMMMEd: "r(U) MMM d, E",
    h: "h a",
    H: "HH",
    hm: "h:mm a",
    Hm: "HH:mm",
    hms: "h:mm:ss a",
    Hms: "HH:mm:ss",
    M: "L",
    Md: "MM-dd",
    MEd: "MM-dd, E",
    MMM: "LLL",
    MMMd: "MMM d",
    MMMEd: "MMM d, E",
    MMMMd: "MMMM d",
    ms: "mm:ss",
    UM: "U MM",
    UMd: "U MM-d",
    UMMM: "U MMM",
    UMMMd: "U MMM d",
    y: "r(U)",
    yMd: "r-MM-dd",
    yyyy: "r(U)",
    yyyyM: "r-MM",
    yyyyMd: "r-MM-dd",
    yyyyMEd: "r-MM-dd, E",
    yyyyMMM: "r(U) MMM",
    yyyyMMMd: "r MMM d",
    yyyyMMMEd: "r(U) MMM d, E",
    yyyyMMMM: "r(U) MMMM",
    yyyyQQQ: "r(U) QQQ",
    yyyyQQQQ: "r(U) QQQQ"
}, _h7js330 = {
    format: _h7js319,
    standAlone: _h7js319
}, _h7js331 = [ undefined, "jia-zi", "yi-chou", "bing-yin", "ding-mao", "wu-chen", "ji-si", "geng-wu", "xin-wei", "ren-shen", "gui-you", "jia-xu", "yi-hai", "bing-zi", "ding-chou", "wu-yin", "ji-mao", "geng-chen", "xin-si", "ren-wu", "gui-wei", "jia-shen", "yi-you", "bing-xu", "ding-hai", "wu-zi", "ji-chou", "geng-yin", "xin-mao", "ren-chen", "gui-si", "jia-wu", "yi-wei", "bing-shen", "ding-you", "wu-xu", "ji-hai", "geng-zi", "xin-chou", "ren-yin", "gui-mao", "jia-chen", "yi-si", "bing-wu", "ding-wei", "wu-shen", "ji-you", "geng-xu", "xin-hai", "ren-zi", "gui-chou", "jia-yin", "yi-mao", "bing-chen", "ding-si", "wu-wu", "ji-wei", "geng-shen", "xin-you", "ren-xu" ], _h7js332 = {
    format: _h7js323,
    standAlone: _h7js323
}, _h7js333 = {
    format: _h7js326,
    standAlone: _h7js326
}, _h7js334 = {
    basic: _h7js327,
    cldr: _h7js329
}, _h7js335 = {
    d: {
        d: "d–d"
    },
    h: _h7js4,
    H: {
        H: "HH–HH"
    },
    hm: _h7js40,
    Hm: _h7js13,
    hmv: _h7js49,
    Hmv: _h7js16,
    hv: _h7js8,
    Hv: {
        H: "HH–HH v"
    },
    M: {
        M: "MM–MM"
    },
    Md: _h7js17,
    MEd: _h7js31,
    MMM: {
        M: "LLL–LLL"
    },
    MMMd: _h7js10,
    MMMEd: _h7js32,
    y: {
        y: "U–U"
    },
    yM: _h7js14,
    yMd: _h7js57,
    yMEd: _h7js276,
    yMMM: _h7js15,
    yMMMd: _h7js45,
    yMMMEd: _h7js256,
    yMMMM: _h7js19
}, _h7js336 = {
    d: {
        d: "d–d"
    },
    h: _h7js4,
    H: {
        H: "HH–HH"
    },
    hm: _h7js40,
    Hm: _h7js13,
    hmv: _h7js49,
    Hmv: _h7js16,
    hv: _h7js8,
    Hv: {
        H: "HH–HH v"
    },
    M: {
        M: "MM–MM"
    },
    Md: _h7js17,
    MEd: _h7js31,
    MMM: {
        M: "LLL–LLL"
    },
    MMMd: _h7js10,
    MMMEd: _h7js32,
    y: {
        y: "G y–y"
    },
    yM: _h7js21,
    yMd: _h7js275,
    yMEd: _h7js297,
    yMMM: _h7js18,
    yMMMd: _h7js52,
    yMMMEd: _h7js283,
    yMMMM: _h7js20
}, _h7js337 = {
    year: _h7js307,
    week: _h7js308,
    day: _h7js310,
    weekday: _h7js5,
    dayperiod: _h7js7,
    minute: _h7js284,
    second: _h7js299,
    era: {
        displayName: "Ìgbà"
    },
    month: _h7js309,
    hour: _h7js271,
    zone: _h7js11,
    quarter: _h7js311,
    sun: _h7js53,
    mon: _h7js54,
    tue: _h7js56,
    wed: _h7js73,
    thu: _h7js58,
    fri: _h7js55,
    sat: _h7js59
}, _h7js338 = {
    dateFormats: {
        basic: {},
        cldr: {}
    },
    fields: _h7js337
}, _h7js339 = {
    abbreviated: _h7js331,
    narrow: _h7js331,
    wide: _h7js331
}, _h7js340 = {
    format: _h7js339
}, _h7js341 = {
    dateFormats: P.calendars.buddhist.dateFormats,
    datePartsOrder: "dmy",
    dateIntervalFormats: _h7js336,
    dateIntervalFallbackFormat: "{0} – {1}",
    dayNames: _h7js330,
    dayPeriods: _h7js314,
    monthNames: _h7js332,
    eraNames: _h7js12,
    quarterNames: _h7js317,
    fields: _h7js337
}, _h7js342 = [ "Taika (645–650)", "Hakuchi (650–671)", "Hakuhō (672–686)", "Shuchō (686–701)", "Taihō (701–704)", "Keiun (704–708)", "Wadō (708–715)", "Reiki (715–717)", "Yōrō (717–724)", "Jinki (724–729)", "Tenpyō (729–749)", "Tenpyō-kampō (749-749)", "Tenpyō-shōhō (749-757)", "Tenpyō-hōji (757-765)", "Tenpyō-jingo (765-767)", "Jingo-keiun (767-770)", "Hōki (770–780)", "Ten-ō (781-782)", "Enryaku (782–806)", "Daidō (806–810)", "Kōnin (810–824)", "Tenchō (824–834)", "Jōwa (834–848)", "Kajō (848–851)", "Ninju (851–854)", "Saikō (854–857)", "Ten-an (857-859)", "Jōgan (859–877)", "Gangyō (877–885)", "Ninna (885–889)", "Kanpyō (889–898)", "Shōtai (898–901)", "Engi (901–923)", "Enchō (923–931)", "Jōhei (931–938)", "Tengyō (938–947)", "Tenryaku (947–957)", "Tentoku (957–961)", "Ōwa (961–964)", "Kōhō (964–968)", "Anna (968–970)", "Tenroku (970–973)", "Ten’en (973–976)", "Jōgen (976–978)", "Tengen (978–983)", "Eikan (983–985)", "Kanna (985–987)", "Eien (987–989)", "Eiso (989–990)", "Shōryaku (990–995)", "Chōtoku (995–999)", "Chōhō (999–1004)", "Kankō (1004–1012)", "Chōwa (1012–1017)", "Kannin (1017–1021)", "Jian (1021–1024)", "Manju (1024–1028)", "Chōgen (1028–1037)", "Chōryaku (1037–1040)", "Chōkyū (1040–1044)", "Kantoku (1044–1046)", "Eishō (1046–1053)", "Tengi (1053–1058)", "Kōhei (1058–1065)", "Jiryaku (1065–1069)", "Enkyū (1069–1074)", "Shōho (1074–1077)", "Shōryaku (1077–1081)", "Eihō (1081–1084)", "Ōtoku (1084–1087)", "Kanji (1087–1094)", "Kahō (1094–1096)", "Eichō (1096–1097)", "Jōtoku (1097–1099)", "Kōwa (1099–1104)", "Chōji (1104–1106)", "Kashō (1106–1108)", "Tennin (1108–1110)", "Ten-ei (1110-1113)", "Eikyū (1113–1118)", "Gen’ei (1118–1120)", "Hōan (1120–1124)", "Tenji (1124–1126)", "Daiji (1126–1131)", "Tenshō (1131–1132)", "Chōshō (1132–1135)", "Hōen (1135–1141)", "Eiji (1141–1142)", "Kōji (1142–1144)", "Ten’yō (1144–1145)", "Kyūan (1145–1151)", "Ninpei (1151–1154)", "Kyūju (1154–1156)", "Hōgen (1156–1159)", "Heiji (1159–1160)", "Eiryaku (1160–1161)", "Ōho (1161–1163)", "Chōkan (1163–1165)", "Eiman (1165–1166)", "Nin’an (1166–1169)", "Kaō (1169–1171)", "Shōan (1171–1175)", "Angen (1175–1177)", "Jishō (1177–1181)", "Yōwa (1181–1182)", "Juei (1182–1184)", "Genryaku (1184–1185)", "Bunji (1185–1190)", "Kenkyū (1190–1199)", "Shōji (1199–1201)", "Kennin (1201–1204)", "Genkyū (1204–1206)", "Ken’ei (1206–1207)", "Jōgen (1207–1211)", "Kenryaku (1211–1213)", "Kenpō (1213–1219)", "Jōkyū (1219–1222)", "Jōō (1222–1224)", "Gennin (1224–1225)", "Karoku (1225–1227)", "Antei (1227–1229)", "Kanki (1229–1232)", "Jōei (1232–1233)", "Tenpuku (1233–1234)", "Bunryaku (1234–1235)", "Katei (1235–1238)", "Ryakunin (1238–1239)", "En’ō (1239–1240)", "Ninji (1240–1243)", "Kangen (1243–1247)", "Hōji (1247–1249)", "Kenchō (1249–1256)", "Kōgen (1256–1257)", "Shōka (1257–1259)", "Shōgen (1259–1260)", "Bun’ō (1260–1261)", "Kōchō (1261–1264)", "Bun’ei (1264–1275)", "Kenji (1275–1278)", "Kōan (1278–1288)", "Shōō (1288–1293)", "Einin (1293–1299)", "Shōan (1299–1302)", "Kengen (1302–1303)", "Kagen (1303–1306)", "Tokuji (1306–1308)", "Enkyō (1308–1311)", "Ōchō (1311–1312)", "Shōwa (1312–1317)", "Bunpō (1317–1319)", "Genō (1319–1321)", "Genkō (1321–1324)", "Shōchū (1324–1326)", "Karyaku (1326–1329)", "Gentoku (1329–1331)", "Genkō (1331–1334)", "Kenmu (1334–1336)", "Engen (1336–1340)", "Kōkoku (1340–1346)", "Shōhei (1346–1370)", "Kentoku (1370–1372)", "Bunchū (1372–1375)", "Tenju (1375–1379)", "Kōryaku (1379–1381)", "Kōwa (1381–1384)", "Genchū (1384–1392)", "Meitoku (1384–1387)", "Kakei (1387–1389)", "Kōō (1389–1390)", "Meitoku (1390–1394)", "Ōei (1394–1428)", "Shōchō (1428–1429)", "Eikyō (1429–1441)", "Kakitsu (1441–1444)", "Bun’an (1444–1449)", "Hōtoku (1449–1452)", "Kyōtoku (1452–1455)", "Kōshō (1455–1457)", "Chōroku (1457–1460)", "Kanshō (1460–1466)", "Bunshō (1466–1467)", "Ōnin (1467–1469)", "Bunmei (1469–1487)", "Chōkyō (1487–1489)", "Entoku (1489–1492)", "Meiō (1492–1501)", "Bunki (1501–1504)", "Eishō (1504–1521)", "Taiei (1521–1528)", "Kyōroku (1528–1532)", "Tenbun (1532–1555)", "Kōji (1555–1558)", "Eiroku (1558–1570)", "Genki (1570–1573)", "Tenshō (1573–1592)", "Bunroku (1592–1596)", "Keichō (1596–1615)", "Genna (1615–1624)", "Kan’ei (1624–1644)", "Shōho (1644–1648)", "Keian (1648–1652)", "Jōō (1652–1655)", "Meireki (1655–1658)", "Manji (1658–1661)", "Kanbun (1661–1673)", "Enpō (1673–1681)", "Tenna (1681–1684)", "Jōkyō (1684–1688)", "Genroku (1688–1704)", "Hōei (1704–1711)", "Shōtoku (1711–1716)", "Kyōhō (1716–1736)", "Genbun (1736–1741)", "Kanpō (1741–1744)", "Enkyō (1744–1748)", "Kan’en (1748–1751)", "Hōreki (1751–1764)", "Meiwa (1764–1772)", "An’ei (1772–1781)", "Tenmei (1781–1789)", "Kansei (1789–1801)", "Kyōwa (1801–1804)", "Bunka (1804–1818)", "Bunsei (1818–1830)", "Tenpō (1830–1844)", "Kōka (1844–1848)", "Kaei (1848–1854)", "Ansei (1854–1860)", "Man’en (1860–1861)", "Bunkyū (1861–1864)", "Genji (1864–1865)", "Keiō (1865–1868)", "Meiji", "Taishō", "Shōwa", "Heisei" ], _h7js343 = {
    dayParts: _h7js318,
    days: _h7js340,
    months: _h7js340,
    years: _h7js340,
    zodiacs: _h7js318
}, _h7js344 = {
    dateFormats: _h7js334,
    datePartsOrder: "md",
    dateIntervalFormats: _h7js335,
    dateIntervalFallbackFormat: "{0} – {1}",
    dayNames: _h7js330,
    dayPeriods: _h7js314,
    cyclicNames: _h7js343,
    monthNames: _h7js328,
    monthPatterns: _h7js313,
    quarterNames: _h7js317,
    fields: _h7js337
}, _h7js345 = [ P.territories, _h7js225, _h7js74, _h7js161, _h7js132, _h7js94, _h7js250, _h7js237, _h7js200, _h7js162, _h7js226, _h7js163, _h7js164, _h7js165, _h7js238, _h7js95, _h7js133, _h7js166, _h7js167, _h7js96, _h7js168, _h7js245, _h7js169, _h7js201, _h7js97, _h7js202, _h7js291, _h7js239, _h7js98, _h7js170, _h7js171, _h7js134, _h7js251, _h7js99, _h7js135, _h7js172, _h7js292, _h7js272, _h7js285, _h7js257, _h7js264, _h7js265, _h7js100, _h7js294, _h7js266, _h7js258, _h7js277, _h7js252, _h7js136, _h7js137, _h7js173, _h7js61, _h7js138, _h7js259, _h7js278, _h7js286, _h7js227, _h7js174, _h7js75, _h7js175, _h7js101, _h7js62, _h7js102, _h7js103, _h7js63, _h7js228, _h7js203, _h7js139, _h7js229, _h7js76, _h7js204, _h7js240, _h7js104, _h7js77, _h7js78, _h7js205, _h7js140, _h7js105, _h7js79, _h7js206, _h7js80, _h7js81, _h7js141, _h7js142, _h7js293, _h7js143, _h7js106, _h7js107, _h7js267, _h7js207, _h7js246, _h7js82, _h7js268, _h7js176, _h7js64, _h7js253, _h7js247, _h7js108, _h7js208, _h7js109, _h7js209, _h7js177, _h7js178, _h7js83, _h7js230, _h7js65, _h7js110, _h7js144, _h7js84, _h7js145, _h7js179, _h7js111, _h7js112, _h7js180, _h7js181, _h7js113, _h7js269, _h7js182, _h7js241, _h7js183, _h7js146, _h7js114, _h7js115, _h7js66, _h7js184, _h7js210, _h7js211, _h7js212, _h7js231, _h7js116, _h7js117, _h7js147, _h7js148, _h7js118, _h7js119, _h7js242, _h7js213, _h7js214, _h7js149, _h7js85, _h7js215, _h7js67, _h7js68, _h7js185, _h7js260, _h7js86, _h7js186, _h7js120, _h7js232, _h7js273, _h7js87, _h7js261, _h7js216, _h7js121, _h7js187, _h7js69, _h7js188, _h7js150, _h7js217, _h7js122, _h7js279, _h7js189, _h7js190, _h7js151, _h7js123, _h7js70, _h7js243, _h7js289, _h7js218, _h7js248, _h7js219, _h7js220, _h7js152, _h7js124, _h7js233, _h7js88, _h7js244, _h7js153, _h7js125, _h7js191, _h7js154, _h7js126, _h7js192, _h7js155, _h7js234, _h7js221, _h7js281, _h7js71, _h7js156, _h7js89, _h7js127, _h7js193, _h7js90, _h7js290, _h7js254, _h7js128, _h7js157, _h7js270, _h7js194, _h7js222, _h7js195, _h7js262, _h7js223, _h7js129, _h7js130, _h7js91, _h7js196, _h7js255, _h7js131, _h7js249, _h7js235, _h7js236, _h7js282, _h7js158, _h7js159, _h7js224, _h7js287, _h7js92, _h7js197, _h7js160, _h7js263, _h7js72, _h7js288, _h7js198, _h7js280 ];

inter.id = "yo_bj";

inter.characters = {
    exemplar: {
        "default": [ "a", "á", "à", "b", "d", "e", "é", "è", "ɛ", "{ɛ\\u0301}", "{ɛ\\u0300}", "f", "g", "{gb}", "h", "i", "í", "ì", "j", "k", "l", "m", "n", "o", "ó", "ò", "ɔ", "{ɔ\\u0301}", "{ɔ\\u0300}", "p", "r", "s", "{sh}", "t", "u", "ú", "ù", "w", "y" ],
        auxiliary: [ "c", "q", "v", "x", "z" ],
        index: [ "A", "B", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "R", "S", "T", "U", "W", "Y" ],
        punctuation: [ "\\-", ",", ";", "\\:", "!", "?", ".", "(", ")", "\\[", "\\]", "\\{", "\\}" ]
    },
    ellipsis: {
        "final": "{0}…",
        initial: "…{0}",
        medial: "{0}…{1}",
        "word-final": "{0} …",
        "word-initial": "… {0}",
        "word-medial": "{0} … {1}"
    },
    moreInformation: "?"
};

inter.calendars = {
    buddhist: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _h7js336,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _h7js330,
        dayPeriods: _h7js314,
        monthNames: _h7js333,
        eraNames: {
            wide: [ "BE" ],
            abbreviated: [ "BE" ]
        },
        quarterNames: _h7js317,
        fields: _h7js337
    },
    chinese: _h7js344,
    coptic: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _h7js336,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _h7js330,
        dayPeriods: _h7js314,
        monthNames: {
            format: _h7js322,
            standAlone: _h7js322
        },
        eraNames: _h7js34,
        quarterNames: _h7js317,
        fields: _h7js337
    },
    dangi: _h7js344,
    ethioaa: _h7js338,
    ethiopic: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _h7js336,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _h7js330,
        dayPeriods: _h7js314,
        monthNames: {
            format: _h7js324,
            standAlone: _h7js324
        },
        eraNames: _h7js34,
        quarterNames: _h7js317,
        fields: _h7js337
    },
    gregorian: {
        dateFormats: {
            basic: {
                fullDate: "EEEE, d MMMM y",
                longDate: "d MMMM y",
                mediumDate: "d MMM y",
                shortDate: "dd/MM/y",
                fullTime: "h:mm:ss a zzzz",
                longTime: "h:mm:ss a z",
                mediumTime: "h:mm:ss a",
                shortTime: "h:mm a",
                shortDateTime: "dd/MM/y h:mm a",
                mediumDateTime: "d MMM y h:mm:ss a",
                longDateTime: "d MMMM y h:mm:ss a z",
                fullDateTime: "EEEE, d MMMM y h:mm:ss a zzzz"
            },
            cldr: {
                d: "d",
                hm: "h:mm a",
                Hm: "HH:mm",
                Hms: "HH:mm:ss",
                M: "L",
                Md: "M/d",
                MEd: "E, M/d",
                MMM: "LLL",
                MMMd: "MMM d",
                MMMEd: "E, MMM d",
                MMMMd: "MMMM d",
                MMMMEd: "E, MMMM d",
                ms: "mm:ss",
                y: "y",
                yM: "M/y",
                yMEd: "E, M/d/y",
                yMMM: "MMM y",
                yMMMEd: "E, MMM d, y",
                yMMMM: "MMMM y",
                yQQQ: "QQQ y",
                yQQQQ: "QQQQ y",
                E: "ccc",
                Ed: "d, E",
                Ehm: "E h:mm a",
                EHm: "E HH:mm",
                Ehms: "E h:mm:ss a",
                EHms: "E HH:mm:ss",
                Gy: "G y",
                GyMMM: "G y MMM",
                GyMMMd: "G y MMM d",
                GyMMMEd: "G y MMM d, E",
                h: "h a",
                H: "HH",
                hms: "h:mm:ss a",
                hmsv: "h:mm:ss a v",
                Hmsv: "HH:mm:ss v",
                hmv: "h:mm a v",
                Hmv: "HH:mm v",
                yMd: "y-MM-dd",
                yMMMd: "y MMM d"
            }
        },
        datePartsOrder: "dmy",
        dateIntervalFormats: {
            d: {
                d: "d–d"
            },
            h: _h7js4,
            H: {
                H: "HH–HH"
            },
            hm: _h7js40,
            Hm: _h7js13,
            hmv: _h7js49,
            Hmv: _h7js16,
            hv: _h7js8,
            Hv: {
                H: "HH–HH v"
            },
            M: {
                M: "MM–MM"
            },
            Md: _h7js17,
            MEd: _h7js31,
            MMM: {
                M: "LLL–LLL"
            },
            MMMd: _h7js10,
            MMMEd: _h7js32,
            y: {
                y: "y–y"
            },
            yM: _h7js14,
            yMd: _h7js57,
            yMEd: _h7js276,
            yMMM: {
                M: "y MMM–MMM",
                y: "y MMM – y MMM"
            },
            yMMMd: {
                d: "y MMM d–d",
                M: "y MMM d – MMM d",
                y: "y MMM d – y MMM d"
            },
            yMMMEd: {
                d: "y MMM d, E – MMM d, E",
                M: "y MMM d, E – MMM d, E",
                y: "y MMM d, E – y MMM d, E"
            },
            yMMMM: {
                M: "y MMMM–MMMM",
                y: "y MMMM – y MMMM"
            }
        },
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _h7js330,
        dayPeriods: _h7js314,
        monthNames: _h7js333,
        eraNames: {
            wide: [ "Saju Kristi", "Lehin Kristi" ],
            abbreviated: [ "SK", "LK" ]
        },
        quarterNames: _h7js317,
        fields: _h7js337
    },
    hebrew: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _h7js336,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _h7js330,
        dayPeriods: _h7js314,
        monthNames: {
            format: _h7js320,
            standAlone: _h7js320
        },
        eraNames: {
            wide: [ "AM" ],
            abbreviated: [ "AM" ]
        },
        quarterNames: _h7js317,
        fields: _h7js337
    },
    indian: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _h7js336,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _h7js330,
        dayPeriods: _h7js314,
        monthNames: {
            format: _h7js325,
            standAlone: _h7js325
        },
        eraNames: {
            wide: [ "Saka" ],
            abbreviated: [ "Saka" ]
        },
        quarterNames: _h7js317,
        fields: _h7js337
    },
    islamic: _h7js341,
    "islamic-umalqura": _h7js341,
    "islamic-tbla": _h7js341,
    "islamic-civil": _h7js341,
    "islamic-rgsa": _h7js341,
    iso8601: _h7js338,
    japanese: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _h7js336,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _h7js330,
        dayPeriods: _h7js314,
        monthNames: _h7js333,
        eraNames: {
            wide: _h7js342,
            abbreviated: _h7js342
        },
        quarterNames: _h7js317,
        fields: _h7js337
    },
    persian: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _h7js336,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _h7js330,
        dayPeriods: _h7js314,
        monthNames: {
            format: _h7js321,
            standAlone: _h7js321
        },
        eraNames: {
            wide: [ "AP" ],
            abbreviated: [ "AP" ]
        },
        quarterNames: _h7js317,
        fields: _h7js337
    },
    roc: {
        dateFormats: P.calendars.buddhist.dateFormats,
        datePartsOrder: "dmy",
        dateIntervalFormats: _h7js336,
        dateIntervalFallbackFormat: "{0} – {1}",
        dayNames: _h7js330,
        dayPeriods: _h7js314,
        monthNames: _h7js333,
        eraNames: {
            wide: _h7js6,
            abbreviated: _h7js6
        },
        quarterNames: _h7js317,
        fields: _h7js337
    },
    islamicc: _h7js338
};

inter.pluralRule = function(n) {
    return "other";
};

inter.renderDigitsOrdinal = function(n) {
    if (n < 0) return "−" + this.renderDigitsOrdinal(-n);
    if (n >= 0) return this.renderNumber(n, "#,##0") + ".";
};

inter.renderSpelloutNumbering = function(n) {
    if (n < 0) return "−" + this.renderSpelloutNumbering(-n);
    if (n >= 0) return this.renderNumber(n, "#,##0.#");
};

inter.renderArmenianUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderArmenianUpper(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e4) return this.renderNumber(n, "#,##0");
    if (n >= 9e3) return "Ք" + (n === 9e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 8e3) return "Փ" + (n === 8e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 7e3) return "Ւ" + (n === 7e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 6e3) return "Ց" + (n === 6e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 5e3) return "Ր" + (n === 5e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 4e3) return "Տ" + (n === 4e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 3e3) return "Վ" + (n === 3e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 2e3) return "Ս" + (n === 2e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 1e3) return "Ռ" + (n === 1e3 ? "" : this.renderArmenianUpper(n % 1e3));
    if (n >= 900) return "Ջ" + (n === 900 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 800) return "Պ" + (n === 800 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 700) return "Չ" + (n === 700 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 600) return "Ո" + (n === 600 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 500) return "Շ" + (n === 500 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 400) return "Ն" + (n === 400 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 300) return "Յ" + (n === 300 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 200) return "Մ" + (n === 200 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 100) return "Ճ" + (n === 100 ? "" : this.renderArmenianUpper(n % 100));
    if (n >= 90) return "Ղ" + (n === 90 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 80) return "Ձ" + (n === 80 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 70) return "Հ" + (n === 70 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 60) return "Կ" + (n === 60 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 50) return "Ծ" + (n === 50 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 40) return "Խ" + (n === 40 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 30) return "Լ" + (n === 30 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 20) return "Ի" + (n === 20 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 10) return "Ժ" + (n === 10 ? "" : this.renderArmenianUpper(n % 10));
    if (n >= 9) return "Թ";
    if (n >= 8) return "Ը";
    if (n >= 7) return "Է";
    if (n >= 6) return "Զ";
    if (n >= 5) return "Ե";
    if (n >= 4) return "Դ";
    if (n >= 3) return "Գ";
    if (n >= 2) return "Բ";
    if (n >= 1) return "Ա";
    if (n >= 0) return "0";
};

inter.renderArmenianLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderArmenianLower(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e4) return this.renderNumber(n, "#,##0");
    if (n >= 9e3) return "ք" + (n === 9e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 8e3) return "փ" + (n === 8e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 7e3) return "ւ" + (n === 7e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 6e3) return "ց" + (n === 6e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 5e3) return "ր" + (n === 5e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 4e3) return "տ" + (n === 4e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 3e3) return "վ" + (n === 3e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 2e3) return "ս" + (n === 2e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 1e3) return "ռ" + (n === 1e3 ? "" : this.renderArmenianLower(n % 1e3));
    if (n >= 900) return "ջ" + (n === 900 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 800) return "պ" + (n === 800 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 700) return "չ" + (n === 700 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 600) return "ո" + (n === 600 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 500) return "շ" + (n === 500 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 400) return "ն" + (n === 400 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 300) return "յ" + (n === 300 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 200) return "մ" + (n === 200 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 100) return "ճ" + (n === 100 ? "" : this.renderArmenianLower(n % 100));
    if (n >= 90) return "ղ" + (n === 90 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 80) return "ձ" + (n === 80 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 70) return "հ" + (n === 70 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 60) return "կ" + (n === 60 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 50) return "ծ" + (n === 50 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 40) return "խ" + (n === 40 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 30) return "լ" + (n === 30 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 20) return "ի" + (n === 20 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 10) return "ժ" + (n === 10 ? "" : this.renderArmenianLower(n % 10));
    if (n >= 9) return "թ";
    if (n >= 8) return "ը";
    if (n >= 7) return "է";
    if (n >= 6) return "զ";
    if (n >= 5) return "ե";
    if (n >= 4) return "դ";
    if (n >= 3) return "գ";
    if (n >= 2) return "բ";
    if (n >= 1) return "ա";
    if (n >= 0) return "0";
};

inter.renderCyrillicLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderCyrillicLower(-n);
    if (isFractional && n > 1) return this.renderCyrillicLower(Math.floor(n)) + "." + this.renderCyrillicLower(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e15) return "҂҂҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e15)) + (n === 1e15 ? "" : " " + this.renderCyrillicLower(n % 1e15));
    if (n >= 1e12) return "҂҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e12)) + (n === 1e12 ? "" : " " + this.renderCyrillicLower(n % 1e12));
    if (n >= 1e9) return "҂҂҂" + this.renderCyrillicLower(Math.floor(n / 1e9)) + (n === 1e9 ? "" : " " + this.renderCyrillicLower(n % 1e9));
    if (n >= 1e6) return "҂҂" + this.renderCyrillicLower(Math.floor(n / 1e6)) + (n === 1e6 ? "" : " " + this.renderCyrillicLower(n % 1e6));
    if (n >= 11e3) return this.renderCyrillicLowerThousands(Math.floor(n / 1e4)) + (n === 11e3 ? "" : " " + this.renderCyrillicLower(n % 1e4));
    if (n >= 1e4) return "҂" + this.renderCyrillicLower(Math.floor(n / 1e4)) + (n === 1e4 ? "" : " " + this.renderCyrillicLower(n % 1e4));
    if (n >= 1e3) return "҂" + this.renderCyrillicLower110(Math.floor(n / 1e3)) + this.renderCyrillicLowerPost(n % 1e3);
    if (n >= 900) return "ц" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 801) return "ѿ" + this.renderCyrillicLower(n % 100);
    if (n >= 800) return "ѿ҃";
    if (n >= 700) return "ѱ" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 600) return "х" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 500) return "ф" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 400) return "у" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 300) return "т" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 200) return "с" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 100) return "р" + this.renderCyrillicLowerFinal(n % 100);
    if (n >= 90) return "ч" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 80) return "п" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 70) return "ѻ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 60) return "ѯ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 50) return "н" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 40) return "м" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 30) return "л" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 20) return "к" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 19) return "ѳ҃і";
    if (n >= 18) return "и҃і";
    if (n >= 17) return "з҃і";
    if (n >= 16) return "ѕ҃і";
    if (n >= 15) return "є҃і";
    if (n >= 14) return "д҃і";
    if (n >= 13) return "г҃і";
    if (n >= 12) return "в҃і";
    if (n >= 11) return "а҃і";
    if (n >= 1) return this.renderCyrillicLower110(n) + "҃";
    if (n >= 0) return "0҃";
};

inter.renderEthiopic = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderEthiopic(-n);
    if (isFractional && n > 1) return this.renderEthiopic(Math.floor(n)) + "፡" + this.renderEthiopic(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 1e18) return this.renderNumber(n, "#,###0");
    if (n >= 2e16) return this.renderEthiopic(Math.floor(n / 1e16)) + "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 1e16) return "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 2e12) return this.renderEthiopic(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1e12) return "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 2e8) return this.renderEthiopic(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1e8) return "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 2e4) return this.renderEthiopic(Math.floor(n / 1e4)) + "፼" + (n === 2e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 1e4) return "፼" + (n === 1e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 200) return this.renderEthiopic(Math.floor(n / 100)) + "፻" + (n === 200 ? "" : this.renderEthiopic(n % 100));
    if (n >= 100) return "፻" + (n === 100 ? "" : this.renderEthiopic(n % 100));
    if (n >= 90) return "፺" + (n === 90 ? "" : this.renderEthiopic(n % 10));
    if (n >= 80) return "፹" + (n === 80 ? "" : this.renderEthiopic(n % 10));
    if (n >= 70) return "፸" + (n === 70 ? "" : this.renderEthiopic(n % 10));
    if (n >= 60) return "፷" + (n === 60 ? "" : this.renderEthiopic(n % 10));
    if (n >= 50) return "፶" + (n === 50 ? "" : this.renderEthiopic(n % 10));
    if (n >= 40) return "፵" + (n === 40 ? "" : this.renderEthiopic(n % 10));
    if (n >= 30) return "፴" + (n === 30 ? "" : this.renderEthiopic(n % 10));
    if (n >= 20) return "፳" + (n === 20 ? "" : this.renderEthiopic(n % 10));
    if (n >= 10) return "፲" + (n === 10 ? "" : this.renderEthiopic(n % 10));
    if (n >= 9) return "፱";
    if (n >= 8) return "፰";
    if (n >= 7) return "፯";
    if (n >= 6) return "፮";
    if (n >= 5) return "፭";
    if (n >= 4) return "፬";
    if (n >= 3) return "፫";
    if (n >= 2) return "፪";
    if (n >= 1) return "፩";
    if (n >= 0) return "ባዶ";
};

inter.renderGeorgian = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGeorgian(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 2e4) return this.renderNumber(n, "#,##0");
    if (n >= 1e4) return "ჯ" + (n === 1e4 ? "" : this.renderGeorgian(n % 1e4));
    if (n >= 9e3) return "ჵ" + (n === 9e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 8e3) return "ჴ" + (n === 8e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 7e3) return "ხ" + (n === 7e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 6e3) return "ჭ" + (n === 6e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 5e3) return "წ" + (n === 5e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 4e3) return "ძ" + (n === 4e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 3e3) return "ც" + (n === 3e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 2e3) return "ჩ" + (n === 2e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 1e3) return "შ" + (n === 1e3 ? "" : this.renderGeorgian(n % 1e3));
    if (n >= 900) return "ყ" + (n === 900 ? "" : this.renderGeorgian(n % 100));
    if (n >= 800) return "ღ" + (n === 800 ? "" : this.renderGeorgian(n % 100));
    if (n >= 700) return "ქ" + (n === 700 ? "" : this.renderGeorgian(n % 100));
    if (n >= 600) return "ფ" + (n === 600 ? "" : this.renderGeorgian(n % 100));
    if (n >= 500) return "ჳ" + (n === 500 ? "" : this.renderGeorgian(n % 100));
    if (n >= 400) return "უ" + (n === 400 ? "" : this.renderGeorgian(n % 100));
    if (n >= 300) return "ტ" + (n === 300 ? "" : this.renderGeorgian(n % 100));
    if (n >= 200) return "ს" + (n === 200 ? "" : this.renderGeorgian(n % 100));
    if (n >= 100) return "რ" + (n === 100 ? "" : this.renderGeorgian(n % 100));
    if (n >= 90) return "ჟ" + (n === 90 ? "" : this.renderGeorgian(n % 10));
    if (n >= 80) return "პ" + (n === 80 ? "" : this.renderGeorgian(n % 10));
    if (n >= 70) return "ო" + (n === 70 ? "" : this.renderGeorgian(n % 10));
    if (n >= 60) return "ჲ" + (n === 60 ? "" : this.renderGeorgian(n % 10));
    if (n >= 50) return "ნ" + (n === 50 ? "" : this.renderGeorgian(n % 10));
    if (n >= 40) return "მ" + (n === 40 ? "" : this.renderGeorgian(n % 10));
    if (n >= 30) return "ლ" + (n === 30 ? "" : this.renderGeorgian(n % 10));
    if (n >= 20) return "კ" + (n === 20 ? "" : this.renderGeorgian(n % 10));
    if (n >= 10) return "ი" + (n === 10 ? "" : this.renderGeorgian(n % 10));
    if (n >= 9) return "თ";
    if (n >= 8) return "ჱ";
    if (n >= 7) return "ზ";
    if (n >= 6) return "ვ";
    if (n >= 5) return "ე";
    if (n >= 4) return "დ";
    if (n >= 3) return "გ";
    if (n >= 2) return "ბ";
    if (n >= 1) return "ა";
    if (n >= 0) return this.renderNumber(n, "#,##0");
};

inter.renderGreekUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGreekUpper(-n);
    if (isFractional && n > 1) return this.renderGreekUpper(Math.floor(n)) + "." + this.renderGreekUpper(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 0) return this.renderGreekNumeralMajuscules(n) + "´";
};

inter.renderGreekLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderGreekLower(-n);
    if (isFractional && n > 1) return this.renderGreekLower(Math.floor(n)) + "." + this.renderGreekLower(parseInt(String(n).replace(/\d*\./, ""), 10));
    if (n >= 0) return this.renderGreekNumeralMinuscules(n) + "´";
};

inter.renderSpelloutCardinal = function(n) {
    if (n < 0) return "−" + this.renderSpelloutCardinal(-n);
    if (n >= 0) return this.renderNumber(n, "#,##0.#");
};

inter.renderHebrew = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderHebrew(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1000001) return this.renderNumber(n, "#,##0");
    if (n >= 1e6) return "אלף אלפים";
    if (n >= 3001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 3001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 3e3) return this.renderHebrew(Math.floor(n / 1e3)) + " אלפים";
    if (n >= 2001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 2001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 2e3) return "אלפיים";
    if (n >= 1001) return this.renderHebrewThousands(Math.floor(n / 1e3)) + (n === 1001 ? "" : this.renderHebrew(n % 1e3));
    if (n >= 1e3) return "אלף";
    if (n >= 901) return "תתק" + this.renderHebrew099(n % 100);
    if (n >= 900) return "תת״ק";
    if (n >= 801) return "תת" + this.renderHebrew099(n % 100);
    if (n >= 800) return "ת״ת";
    if (n >= 745) return "תש" + this.renderHebrew099(n % 100);
    if (n >= 744) return "תשד״מ";
    if (n >= 701) return "תש" + this.renderHebrew099(n % 100);
    if (n >= 700) return "ת״ש";
    if (n >= 699) return "תר" + this.renderHebrew099(n % 100);
    if (n >= 698) return "תרח״צ";
    if (n >= 601) return "תר" + this.renderHebrew099(n % 100);
    if (n >= 600) return "ת״ר";
    if (n >= 501) return "תק" + this.renderHebrew099(n % 100);
    if (n >= 500) return "ת״ק";
    if (n >= 400) return "ת" + this.renderHebrew099(n % 100);
    if (n >= 345) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 344) return "שד״מ";
    if (n >= 305) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 304) return "ד״ש";
    if (n >= 300) return "ש" + this.renderHebrew099(n % 100);
    if (n >= 299) return "ר" + this.renderHebrew099(n % 100);
    if (n >= 298) return "רח״צ";
    if (n >= 200) return "ר" + this.renderHebrew099(n % 100);
    if (n >= 100) return "ק" + this.renderHebrew099(n % 100);
    if (n >= 91) return "צ״" + this.renderHebrewItem(n % 10);
    if (n >= 90) return "צ׳";
    if (n >= 81) return "פ״" + this.renderHebrewItem(n % 10);
    if (n >= 80) return "פ׳";
    if (n >= 71) return "ע״" + this.renderHebrewItem(n % 10);
    if (n >= 70) return "ע׳";
    if (n >= 61) return "ס״" + this.renderHebrewItem(n % 10);
    if (n >= 60) return "ס׳";
    if (n >= 51) return "נ״" + this.renderHebrewItem(n % 10);
    if (n >= 50) return "נ׳";
    if (n >= 41) return "מ״" + this.renderHebrewItem(n % 10);
    if (n >= 40) return "מ׳";
    if (n >= 31) return "ל״" + this.renderHebrewItem(n % 10);
    if (n >= 30) return "ל׳";
    if (n >= 21) return "כ״" + this.renderHebrewItem(n % 10);
    if (n >= 20) return "כ׳";
    if (n >= 17) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "ט״ז";
    if (n >= 15) return "ט״ו";
    if (n >= 11) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 0) return this.renderHebrewItem(n) + "׳";
};

inter.renderRomanUpper = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderRomanUpper(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 4e5) return this.renderNumber(n, "#,##0");
    if (n >= 3e5) return "ↈↈↈ" + (n === 3e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 2e5) return "ↈↈ" + (n === 2e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 1e5) return "ↈ" + (n === 1e5 ? "" : this.renderRomanUpper(n % 1e5));
    if (n >= 9e4) return "ↂↈ" + (n === 9e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 8e4) return "ↇↂↂↂ" + (n === 8e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 7e4) return "ↇↂↂ" + (n === 7e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 6e4) return "ↇↂ" + (n === 6e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 5e4) return "ↇ" + (n === 5e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 4e4) return "ↂↇ" + (n === 4e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 3e4) return "ↂↂↂ" + (n === 3e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 2e4) return "ↂↂ" + (n === 2e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 1e4) return "ↂ" + (n === 1e4 ? "" : this.renderRomanUpper(n % 1e4));
    if (n >= 9e3) return "Mↂ" + (n === 9e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 8e3) return "ↁMMM" + (n === 8e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 7e3) return "ↁMM" + (n === 7e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 6e3) return "ↁM" + (n === 6e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 5e3) return "ↁ" + (n === 5e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 4e3) return "Mↁ" + (n === 4e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 3e3) return "MMM" + (n === 3e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 2e3) return "MM" + (n === 2e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 1e3) return "M" + (n === 1e3 ? "" : this.renderRomanUpper(n % 1e3));
    if (n >= 900) return "CM" + (n === 900 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 800) return "DCCC" + (n === 800 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 700) return "DCC" + (n === 700 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 600) return "DC" + (n === 600 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 500) return "D" + (n === 500 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 400) return "CD" + (n === 400 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 300) return "CCC" + (n === 300 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 200) return "CC" + (n === 200 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 100) return "C" + (n === 100 ? "" : this.renderRomanUpper(n % 100));
    if (n >= 90) return "XC" + (n === 90 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 80) return "LXXX" + (n === 80 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 70) return "LXX" + (n === 70 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 60) return "LX" + (n === 60 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 50) return "L" + (n === 50 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 40) return "XL" + (n === 40 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 30) return "XXX" + (n === 30 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 20) return "XX" + (n === 20 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 10) return "X" + (n === 10 ? "" : this.renderRomanUpper(n % 10));
    if (n >= 9) return "IX";
    if (n >= 8) return "VIII";
    if (n >= 7) return "VII";
    if (n >= 6) return "VI";
    if (n >= 5) return "V";
    if (n >= 4) return "IV";
    if (n >= 3) return "III";
    if (n >= 2) return "II";
    if (n >= 1) return "I";
    if (n >= 0) return "N";
};

inter.renderRomanLower = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderRomanLower(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 5e3) return this.renderNumber(n, "#,##0");
    if (n >= 4e3) return "mmmm" + (n === 4e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 3e3) return "mmm" + (n === 3e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 2e3) return "mm" + (n === 2e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 1e3) return "m" + (n === 1e3 ? "" : this.renderRomanLower(n % 1e3));
    if (n >= 900) return "cm" + (n === 900 ? "" : this.renderRomanLower(n % 100));
    if (n >= 800) return "dccc" + (n === 800 ? "" : this.renderRomanLower(n % 100));
    if (n >= 700) return "dcc" + (n === 700 ? "" : this.renderRomanLower(n % 100));
    if (n >= 600) return "dc" + (n === 600 ? "" : this.renderRomanLower(n % 100));
    if (n >= 500) return "d" + (n === 500 ? "" : this.renderRomanLower(n % 100));
    if (n >= 400) return "cd" + (n === 400 ? "" : this.renderRomanLower(n % 100));
    if (n >= 300) return "ccc" + (n === 300 ? "" : this.renderRomanLower(n % 100));
    if (n >= 200) return "cc" + (n === 200 ? "" : this.renderRomanLower(n % 100));
    if (n >= 100) return "c" + (n === 100 ? "" : this.renderRomanLower(n % 100));
    if (n >= 90) return "xc" + (n === 90 ? "" : this.renderRomanLower(n % 10));
    if (n >= 80) return "lxxx" + (n === 80 ? "" : this.renderRomanLower(n % 10));
    if (n >= 70) return "lxx" + (n === 70 ? "" : this.renderRomanLower(n % 10));
    if (n >= 60) return "lx" + (n === 60 ? "" : this.renderRomanLower(n % 10));
    if (n >= 50) return "l" + (n === 50 ? "" : this.renderRomanLower(n % 10));
    if (n >= 40) return "xl" + (n === 40 ? "" : this.renderRomanLower(n % 10));
    if (n >= 30) return "xxx" + (n === 30 ? "" : this.renderRomanLower(n % 10));
    if (n >= 20) return "xx" + (n === 20 ? "" : this.renderRomanLower(n % 10));
    if (n >= 10) return "x" + (n === 10 ? "" : this.renderRomanLower(n % 10));
    if (n >= 9) return "ix";
    if (n >= 8) return "viii";
    if (n >= 7) return "vii";
    if (n >= 6) return "vi";
    if (n >= 5) return "v";
    if (n >= 4) return "iv";
    if (n >= 3) return "iii";
    if (n >= 2) return "ii";
    if (n >= 1) return "i";
    if (n >= 0) return "n";
};

inter.renderTamil = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderTamil(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 1e8) return this.renderNumber(n, "#,##,##0");
    if (n >= 1e6) return this.renderTamil(Math.floor(n / 1e6)) + "௱௲" + (n === 1e6 ? "" : this.renderTamilThousands(n % 1e6));
    if (n >= 2e3) return this.renderTamil(Math.floor(n / 1e3)) + "௲" + (n === 2e3 ? "" : this.renderTamil(n % 1e3));
    if (n >= 1e3) return "௲" + (n === 1e3 ? "" : this.renderTamil(n % 1e3));
    if (n >= 200) return this.renderTamil(Math.floor(n / 100)) + "௱" + (n === 200 ? "" : this.renderTamil(n % 100));
    if (n >= 100) return "௱" + (n === 100 ? "" : this.renderTamil(n % 100));
    if (n >= 20) return this.renderTamil(Math.floor(n / 10)) + "௰" + (n === 20 ? "" : this.renderTamil(n % 10));
    if (n >= 10) return "௰" + (n === 10 ? "" : this.renderTamil(n % 10));
    if (n >= 9) return "௯";
    if (n >= 8) return "௮";
    if (n >= 7) return "௭";
    if (n >= 6) return "௬";
    if (n >= 5) return "௫";
    if (n >= 4) return "௪";
    if (n >= 3) return "௩";
    if (n >= 2) return "௨";
    if (n >= 1) return "௧";
    if (n >= 0) return "௦";
};

inter.renderCyrillicLowerThousands = function(n) {
    if (n >= 900) return "҂ц" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 800) return "҂ѿ" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 700) return "҂ѱ" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 600) return "҂х" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 500) return "҂ф" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 400) return "҂у" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 300) return "҂т" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 200) return "҂с" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 100) return "҂р" + this.renderCyrillicLowerThousands(n % 100);
    if (n >= 90) return "҂ч" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 80) return "҂п" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 70) return "҂ѻ" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 60) return "҂ѯ" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 50) return "҂н" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 40) return "҂м" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 30) return "҂л" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 20) return "҂к" + this.renderCyrillicLowerThousands(n % 10);
    if (n >= 19) return "҂ѳ҃҂і";
    if (n >= 18) return "҂и҃҂і";
    if (n >= 17) return "҂з҃҂і";
    if (n >= 16) return "҂ѕ҃҂і";
    if (n >= 15) return "҂є҃҂і";
    if (n >= 14) return "҂д҃҂і";
    if (n >= 13) return "҂г҃҂і";
    if (n >= 12) return "҂в҃҂і";
    if (n >= 11) return "҂а҃҂і";
    if (n >= 10) return "҃҂і";
    if (n >= 9) return "҃҂ѳ";
    if (n >= 8) return "҃҂и";
    if (n >= 7) return "҃҂з";
    if (n >= 6) return "҃҂ѕ";
    if (n >= 5) return "҃҂є";
    if (n >= 4) return "҃҂д";
    if (n >= 3) return "҃҂г";
    if (n >= 2) return "҃҂в";
    if (n >= 1) return "҃҂а";
    if (n >= 0) return "҃";
};

inter.renderCyrillicLower110 = function(n) {
    if (n >= 10) return "і";
    if (n >= 9) return "ѳ";
    if (n >= 8) return "и";
    if (n >= 7) return "з";
    if (n >= 6) return "ѕ";
    if (n >= 5) return "є";
    if (n >= 4) return "д";
    if (n >= 3) return "г";
    if (n >= 2) return "в";
    if (n >= 1) return "а";
};

inter.renderCyrillicLowerPost = function(n) {
    if (n >= 1) return this.renderCyrillicLower(n);
    if (n >= 0) return "҃";
};

inter.renderCyrillicLowerFinal = function(n) {
    if (n >= 91) return "ч" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 90) return "҃ч";
    if (n >= 81) return "п" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 80) return "҃п";
    if (n >= 71) return "ѻ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 70) return "҃ѻ";
    if (n >= 61) return "ѯ" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 60) return "҃ѯ";
    if (n >= 51) return "н" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 50) return "҃н";
    if (n >= 41) return "м" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 40) return "҃м";
    if (n >= 31) return "л" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 30) return "҃л";
    if (n >= 21) return "к" + this.renderCyrillicLowerFinal(n % 10);
    if (n >= 20) return "҃к";
    if (n >= 19) return "ѳ҃і";
    if (n >= 18) return "и҃і";
    if (n >= 17) return "з҃і";
    if (n >= 16) return "ѕ҃і";
    if (n >= 15) return "є҃і";
    if (n >= 14) return "д҃і";
    if (n >= 13) return "г҃і";
    if (n >= 12) return "в҃і";
    if (n >= 11) return "а҃і";
    if (n >= 1) return "҃" + this.renderCyrillicLower110(n);
    if (n >= 0) return "҃";
};

inter.renderEthiopicP3 = function(n) {
    if (n >= 1e12) return this.renderEthiopic(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1) return "፼፼፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼፼፼";
};

inter.renderEthiopicP2 = function(n) {
    if (n >= 1e8) return this.renderEthiopic(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1) return "፼፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼፼";
};

inter.renderEthiopicP1 = function(n) {
    if (n >= 1e4) return this.renderEthiopic(Math.floor(n / 1e4)) + "፼" + (n === 1e4 ? "" : this.renderEthiopic(n % 1e4));
    if (n >= 1) return "፼" + this.renderEthiopicP(n);
    if (n >= 0) return "፼";
};

inter.renderGreekNumeralMajuscules = function(n) {
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e16) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e16)) + "ΜΜΜΜ" + (n === 1e16 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e16));
    if (n >= 1e12) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e12)) + "ΜΜΜ" + (n === 1e12 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e12));
    if (n >= 1e8) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e8)) + "ΜΜ" + (n === 1e8 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e8));
    if (n >= 1e4) return this.renderGreekNumeralMajuscules(Math.floor(n / 1e4)) + "Μ" + (n === 1e4 ? "" : " " + this.renderGreekNumeralMajuscules(n % 1e4));
    if (n >= 1e3) return "͵" + this.renderGreekNumeralMajuscules(Math.floor(n / 1e3)) + (n === 1e3 ? "" : this.renderGreekNumeralMajuscules(n % 1e3));
    if (n >= 900) return "Ϡ" + (n === 900 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 800) return "Ω" + (n === 800 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 700) return "Ψ" + (n === 700 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 600) return "Χ" + (n === 600 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 500) return "Φ" + (n === 500 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 400) return "Υ" + (n === 400 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 300) return "Τ" + (n === 300 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 200) return "Σ" + (n === 200 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 100) return "Ρ" + (n === 100 ? "" : this.renderGreekNumeralMajuscules(n % 100));
    if (n >= 90) return "Ϟ" + (n === 90 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 80) return "Π" + (n === 80 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 70) return "Ο" + (n === 70 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 60) return "Ξ" + (n === 60 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 50) return "Ν" + (n === 50 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 40) return "Μ" + (n === 40 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 30) return "Λ" + (n === 30 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 20) return "Κ" + (n === 20 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 10) return "Ι" + (n === 10 ? "" : this.renderGreekNumeralMajuscules(n % 10));
    if (n >= 9) return "Θ";
    if (n >= 8) return "Η";
    if (n >= 7) return "Ζ";
    if (n >= 6) return "Ϝ";
    if (n >= 5) return "Ε";
    if (n >= 4) return "Δ";
    if (n >= 3) return "Γ";
    if (n >= 2) return "Β";
    if (n >= 1) return "Α";
    if (n >= 0) return "𐆊";
};

inter.renderGreekNumeralMinuscules = function(n) {
    if (n >= 1e18) return this.renderNumber(n, "#,##0");
    if (n >= 1e16) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e16)) + "μμμμ" + (n === 1e16 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e16));
    if (n >= 1e12) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e12)) + "μμμ" + (n === 1e12 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e12));
    if (n >= 1e8) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e8)) + "μμ" + (n === 1e8 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e8));
    if (n >= 1e4) return this.renderGreekNumeralMinuscules(Math.floor(n / 1e4)) + "μ" + (n === 1e4 ? "" : " " + this.renderGreekNumeralMinuscules(n % 1e4));
    if (n >= 1e3) return "͵" + this.renderGreekNumeralMinuscules(Math.floor(n / 1e3)) + (n === 1e3 ? "" : this.renderGreekNumeralMinuscules(n % 1e3));
    if (n >= 900) return "ϡ" + (n === 900 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 800) return "ω" + (n === 800 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 700) return "ψ" + (n === 700 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 600) return "χ" + (n === 600 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 500) return "φ" + (n === 500 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 400) return "υ" + (n === 400 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 300) return "τ" + (n === 300 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 200) return "σ" + (n === 200 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 100) return "ρ" + (n === 100 ? "" : this.renderGreekNumeralMinuscules(n % 100));
    if (n >= 90) return "ϟ" + (n === 90 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 80) return "π" + (n === 80 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 70) return "ο" + (n === 70 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 60) return "ξ" + (n === 60 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 50) return "ν" + (n === 50 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 40) return "μ" + (n === 40 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 30) return "λ" + (n === 30 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 20) return "κ" + (n === 20 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 10) return "ι" + (n === 10 ? "" : this.renderGreekNumeralMinuscules(n % 10));
    if (n >= 9) return "θ";
    if (n >= 8) return "η";
    if (n >= 7) return "ζ";
    if (n >= 6) return "ϝ";
    if (n >= 5) return "ε";
    if (n >= 4) return "δ";
    if (n >= 3) return "γ";
    if (n >= 2) return "β";
    if (n >= 1) return "α";
    if (n >= 0) return "𐆊";
};

inter.renderHebrewThousands = function(n) {
    if (n >= 401) return this.renderHebrew(n) + "׳";
    if (n >= 100) return this.renderHebrew(n) + (n === 100 ? "" : "׳");
    if (n >= 10) return this.renderHebrew(n) + (n === 10 ? "" : "׳");
    if (n >= 0) return this.renderHebrew(n);
};

inter.renderHebrew099 = function(n) {
    if (n >= 91) return "צ״" + this.renderHebrewItem(n % 10);
    if (n >= 90) return "״צ";
    if (n >= 81) return "פ״" + this.renderHebrewItem(n % 10);
    if (n >= 80) return "״פ";
    if (n >= 71) return "ע״" + this.renderHebrewItem(n % 10);
    if (n >= 70) return "״ע";
    if (n >= 61) return "ס״" + this.renderHebrewItem(n % 10);
    if (n >= 60) return "״ס";
    if (n >= 51) return "נ״" + this.renderHebrewItem(n % 10);
    if (n >= 50) return "״נ";
    if (n >= 41) return "מ״" + this.renderHebrewItem(n % 10);
    if (n >= 40) return "״מ";
    if (n >= 31) return "ל״" + this.renderHebrewItem(n % 10);
    if (n >= 30) return "״ל";
    if (n >= 21) return "כ״" + this.renderHebrewItem(n % 10);
    if (n >= 20) return "״כ";
    if (n >= 17) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "ט״ז";
    if (n >= 15) return "ט״ו";
    if (n >= 11) return "י״" + this.renderHebrewItem(n % 10);
    if (n >= 1) return "״" + this.renderHebrewItem(n);
    if (n >= 0) return "׳";
};

inter.renderHebrewItem = function(n) {
    var isFractional = n !== Math.floor(n);
    if (n < 0) return "−" + this.renderHebrewItem(-n);
    if (isFractional && n > 1) return this.renderNumber(n, "#,##0.00");
    if (n >= 2100) return this.renderNumber(n, "#,##0");
    if (n >= 2e3) return "תתתתת" + (n === 2e3 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1900) return "תתתתש" + (n === 1900 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1800) return "תתתתר" + (n === 1800 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1700) return "תתתתק" + (n === 1700 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1600) return "תתתת" + (n === 1600 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1500) return "תתתש" + (n === 1500 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1400) return "תתתר" + (n === 1400 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1300) return "תתתק" + (n === 1300 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1200) return "תתת" + (n === 1200 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1100) return "תתש" + (n === 1100 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 1e3) return "תתר" + (n === 1e3 ? "" : this.renderHebrewItem(n % 1e3));
    if (n >= 900) return "תתק" + (n === 900 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 800) return "תת" + (n === 800 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 745) return "תש" + this.renderHebrewItem(n % 100);
    if (n >= 744) return "תשדמ";
    if (n >= 700) return "תש" + (n === 700 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 699) return "תר" + this.renderHebrewItem(n % 100);
    if (n >= 698) return "תרחצ";
    if (n >= 600) return "תר" + (n === 600 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 500) return "תק" + (n === 500 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 400) return "ת" + (n === 400 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 345) return "ש" + this.renderHebrewItem(n % 100);
    if (n >= 344) return "שדמ";
    if (n >= 305) return "ש" + this.renderHebrewItem(n % 100);
    if (n >= 304) return "דש";
    if (n >= 300) return "ש" + (n === 300 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 299) return "ר" + this.renderHebrewItem(n % 100);
    if (n >= 298) return "רחצ";
    if (n >= 200) return "ר" + (n === 200 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 100) return "ק" + (n === 100 ? "" : this.renderHebrewItem(n % 100));
    if (n >= 90) return "צ" + (n === 90 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 80) return "פ" + (n === 80 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 70) return "ע" + (n === 70 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 60) return "ס" + (n === 60 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 50) return "נ" + (n === 50 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 40) return "מ" + (n === 40 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 30) return "ל" + (n === 30 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 20) return "כ" + (n === 20 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 17) return "י" + this.renderHebrewItem(n % 10);
    if (n >= 16) return "טז";
    if (n >= 15) return "טו";
    if (n >= 10) return "י" + (n === 10 ? "" : this.renderHebrewItem(n % 10));
    if (n >= 9) return "ט";
    if (n >= 8) return "ח";
    if (n >= 7) return "ז";
    if (n >= 6) return "ו";
    if (n >= 5) return "ה";
    if (n >= 4) return "ד";
    if (n >= 3) return "ג";
    if (n >= 2) return "ב";
    if (n >= 1) return "א";
    if (n >= 0) return "״";
};

inter.renderTamilThousands = function(n) {
    if (n >= 1e3) return this.renderTamilThousands(Math.floor(n / 1e3)) + "௲" + (n === 1e3 ? "" : this.renderTamilThousands(n % 1e3));
    if (n >= 0) return this.renderTamil(n);
};

inter.renderEthiopicP = function(n) {
    if (n >= 1e16) return this.renderEthiopicP(Math.floor(n / 1e16)) + "፼" + this.renderEthiopicP3(n % 1e16);
    if (n >= 1e12) return this.renderEthiopicP(Math.floor(n / 1e12)) + "፼" + this.renderEthiopicP2(n % 1e12);
    if (n >= 1e8) return this.renderEthiopicP(Math.floor(n / 1e8)) + "፼" + this.renderEthiopicP1(n % 1e8);
    if (n >= 1e4) return this.renderEthiopicP(Math.floor(n / 1e4)) + "፼" + (n === 1e4 ? "" : this.renderEthiopicP(n % 1e4));
    if (n >= 1) return this.renderEthiopic(n);
};

inter.territories = _h7js345;

inter.countries = _h7js345;

inter.languages = [].concat(P.languages.slice(0, 8), {
    id: "da",
    displayName: "Èdè Ilɛ̀ Denmark",
    nativeDisplayName: "dansk"
}, {
    id: "de",
    displayName: "Èdè Ilɛ̀ Gemani",
    nativeDisplayName: "Deutsch"
}, P.languages[10], {
    id: "en",
    displayName: "Èdè Gɛ̀ɛ́sì",
    nativeDisplayName: "English"
}, P.languages.slice(12, 26), {
    id: "zu",
    displayName: "Èdè Shulu",
    nativeDisplayName: "isiZulu"
}, P.languages.slice(27, 36), {
    id: "pl",
    displayName: "Èdè Ilɛ̀ Polandi",
    nativeDisplayName: "polski"
}, {
    id: "pt",
    displayName: "Èdè Pɔtugi",
    nativeDisplayName: "português"
}, P.languages.slice(38, 47), {
    id: "tr",
    displayName: "Èdè Tɔɔkisi",
    nativeDisplayName: "Türkçe"
}, P.languages.slice(48, 69), {
    id: "ru",
    displayName: "Èdè ̣Rɔɔsia",
    nativeDisplayName: "русский"
}, P.languages.slice(70, 98));

inter.currencies = [].concat({
    id: "ETB",
    displayName: "Biri ti Orílɛ́ède Eutopia"
}, {
    id: "GMD",
    displayName: "Dalasi ti Orílɛ́ède Gamibia"
}, {
    id: "AED",
    displayName: "Diami ti Awon Orílɛ́ède Arabu"
}, {
    id: "BHD",
    displayName: "Dina ti Orílɛ́ède Báránì"
}, {
    id: "LYD",
    displayName: "Dina ti Orílɛ́ède Libiya"
}, {
    id: "SDG",
    displayName: "Dina ti Orílɛ́ède Sudani"
}, {
    id: "TND",
    displayName: "Dina ti Orílɛ́ède Tunisia"
}, {
    id: "DZD",
    displayName: "Dina ti Orílɛ́ède Àlùgèríánì"
}, {
    id: "MAD",
    displayName: "Dirami ti Orílɛ́ède Moroko"
}, {
    id: "STD",
    displayName: "Dobira ti Orílɛ́ède Sao tome Ati Pirisipe",
    symbol: "Db"
}, {
    id: "USD",
    displayName: "Dɔla ti Orílɛ́ède Amerika",
    symbol: "US$"
}, {
    id: "CAD",
    displayName: "Dɔla ti Orílɛ́ède Kánádà",
    symbol: "CA$"
}, {
    id: "LRD",
    displayName: "Dɔla ti Orílɛ́ède Liberia",
    symbol: "$"
}, {
    id: "NAD",
    displayName: "Dɔla ti Orílɛ́ède Namibia",
    symbol: "$"
}, {
    id: "ZWD",
    displayName: "Dɔla ti Orílɛ́ède Siibabuwe"
}, {
    id: "AUD",
    displayName: "Dɔla ti Orílɛ́ède Ástràlìá",
    symbol: "A$"
}, {
    id: "XAF",
    displayName: "Faransi ti Orílɛ́ède BEKA",
    symbol: "FCFA"
}, {
    id: "XOF",
    displayName: "Faransi ti Orílɛ́ède BIKEAO",
    symbol: "CFA"
}, {
    id: "BIF",
    displayName: "Faransi ti Orílɛ́ède Bùùrúndì"
}, {
    id: "DJF",
    displayName: "Faransi ti Orílɛ́ède Dibouti"
}, {
    id: "GNS",
    displayName: "Faransi ti Orílɛ́ède Gini"
}, {
    id: "CDF",
    displayName: "Faransi ti Orílɛ́ède Kóngò"
}, {
    id: "MGA",
    displayName: "Faransi ti Orílɛ́ède Malagasi",
    symbol: "Ar"
}, {
    id: "RWF",
    displayName: "Faransi ti Orílɛ́ède Ruwanda",
    symbol: "RF"
}, {
    id: "CHF",
    displayName: "Faransi ti Orílɛ́ède Siwisi"
}, {
    id: "KMF",
    displayName: "Faransi ti Orílɛ́ède shomoriani",
    symbol: "CF"
}, {
    id: "CVE",
    displayName: "Kabofediano ti Orílɛ́ède Esuodo"
}, {
    id: "MWK",
    displayName: "Kasha ti Orílɛ́ède Malawi"
}, {
    id: "ZMW",
    displayName: "Kawasha ti Orílɛ́ède Saabia",
    symbol: "ZK"
}, {
    id: "ZMK",
    displayName: "Kawasha ti Orílɛ́ède Saabia (1968–2012)"
}, P.currencies.slice(30, 32), {
    id: "LSL",
    displayName: "Loti ti Orílɛ́ède Lesoto"
}, {
    id: "MZM",
    displayName: "Metika ti Orílɛ́ède Mosamibiki"
}, {
    id: "NGN",
    displayName: "Naira ti Orílɛ́ède Nàìjíríà",
    symbol: "₦"
}, {
    id: "ERN",
    displayName: "Nakifa ti Orílɛ́ède Eriteriani"
}, {
    id: "MRO",
    displayName: "Ouguiya ti Orílɛ́ède Maritania"
}, {
    id: "BWP",
    displayName: "Pula ti Orílɛ́ède Bɔ̀tìsúwánà",
    symbol: "P"
}, {
    id: "GBP",
    displayName: "Pɔɔn ti Orílɛ́ède Bírítísì",
    symbol: "£"
}, {
    id: "SDP",
    displayName: "Pɔɔun ti Orílɛ́ède Sudani"
}, {
    id: "SHP",
    displayName: "Pɔɔun ti Orílɛ́ède ̣Elena",
    symbol: "£"
}, {
    id: "ZAR",
    displayName: "Randi ti Orílɛ́ède Ariwa Afirika",
    symbol: "R"
}, {
    id: "CNY",
    displayName: "Reminibi ti Orílɛ́ède sháínà",
    symbol: "CN¥"
}, {
    id: "SAR",
    displayName: "Riya ti Orílɛ́ède Saudi"
}, {
    id: "INR",
    displayName: "Rupi ti Orílɛ́ède Indina",
    symbol: "₹"
}, {
    id: "MUR",
    displayName: "Rupi ti Orílɛ́ède Maritiusi",
    symbol: "Rs"
}, {
    id: "SCR",
    displayName: "Rupi ti Orílɛ́ède Sayiselesi"
}, {
    id: "UGX",
    displayName: "Siile ti Orílɛ́ède Uganda"
}, {
    id: "SOS",
    displayName: "Sile ti Orílɛ́ède Somali"
}, {
    id: "TZS",
    displayName: "Sile ti Orílɛ́ède Tansania"
}, P.currencies[50], {
    id: "AOA",
    displayName: "Wansa ti Orílɛ́ède Àngólà",
    symbol: "Kz"
}, {
    id: "JPY",
    displayName: "Yeni ti Orílɛ́ède Japani",
    symbol: "JP¥"
}, {
    id: "EGP",
    displayName: "pɔɔn ti Orílɛ́ède Egipiti",
    symbol: "E£"
}, {
    id: "GHC",
    displayName: "shidi ti Orílɛ́ède Gana"
}, {
    id: "KES",
    displayName: "shiili ti Orílɛ́ède Kenya"
});

module.exports = inter;
